--- Makefile.in.orig	2022-08-08 06:23:15.421205746 +0800
+++ Makefile.in	2022-08-08 06:23:33.944565592 +0800
@@ -105,6 +105,9 @@ define GUI_SRCS_LIST_TO_OBJS
 	$(basename $(notdir $(file))))))
 endef
 GUI_OBJS = $(GUI_SRCS_LIST_TO_OBJS)
+ifeq ($(USE_BINCUE),yes)
+GUI_OBJS += bincue.o
+endif
 
 define DYNGENSRCS_LIST_TO_OBJS
 	$(addprefix $(OBJ_DIR)/, $(addsuffix .dgo, $(foreach file, $(DYNGENSRCS), \

