diff -Naur xscreensaver-5.33-orig/driver/xscreensaver-text xscreensaver-5.33/driver/xscreensaver-text
--- xscreensaver-5.33-orig/driver/xscreensaver-text	2015-07-21 21:12:53.283286533 -0400
+++ xscreensaver-5.33/driver/xscreensaver-text	2015-07-21 21:16:48.589580345 -0400
@@ -473,7 +473,7 @@
 
     # Try to omit all tables, since they're impossible to read as text.
     #
-    1 while (s/{{[^{}]*}}/ /gs);		   # {{ ... }}
+    1 while (s/{\{[^{}]*}}/ /gs);		   # {{ ... }}
     1 while (s/{\|.*?\|}/\n\n/gs);		   # {| ... |}
     1 while (s/\|-.*?\|/ /gs);			   # |- ... |  (table cell)
 
