diff --git a/demo/data/org.gnome.Adwaita1.Demo.metainfo.xml.in.in b/demo/data/org.gnome.Adwaita1.Demo.metainfo.xml.in.in
index 19ded561..1e4e9213 100644
--- a/demo/data/org.gnome.Adwaita1.Demo.metainfo.xml.in.in
+++ b/demo/data/org.gnome.Adwaita1.Demo.metainfo.xml.in.in
@@ -4,7 +4,7 @@
   <metadata_license>CC0-1.0</metadata_license>
   <project_license>LGPL-2.1-or-later</project_license>
   <launchable type="desktop-id">org.gnome.Adwaita1.Demo.desktop</launchable>
-  <developer_name>The GNOME Project</developer_name>
+  <developer><name>The GNOME Project</name></developer>
 
   <name>Adwaita Demo</name>
   <summary>Tour of the features in Libadwaita</summary>
