--- src/llvm/tools/llvm-shlib/CMakeLists.txt.orig	2015-09-06 12:31:21.765250429 +0300
+++ src/llvm/tools/llvm-shlib/CMakeLists.txt	2015-09-06 13:17:10.820174432 +0300
@@ -64,7 +64,7 @@
 
   if (LLVM_DYLIB_EXPORT_ALL)
     add_custom_command(OUTPUT ${LLVM_EXPORTED_SYMBOL_FILE}
-      COMMAND echo \"LLVM*\" > ${LLVM_EXPORTED_SYMBOL_FILE} && echo \"_Z*llvm*\" >> ${LLVM_EXPORTED_SYMBOL_FILE}
+      COMMAND echo -e \"LLVM*\\n_Z*llvm*\\nConvertUTF*\\ngetNumBytesForUTF8\\nisLegalUTF8*\" > ${LLVM_EXPORTED_SYMBOL_FILE}
       WORKING_DIRECTORY ${LIB_DIR}
       DEPENDS ${LLVM_DYLIB_REQUIRED_EXPORTS}
       COMMENT "Generating combined export list...")
