--- a/electron/default_app/default_app.js
+++ b/electron/default_app/default_app.js
@@ -27,7 +27,7 @@
   }
 
   if (process.platform === 'linux') {
-    options.icon = path.join(__dirname, 'icon.png')
+    options.icon = '/usr/share/pixmaps/electron.png'
   }
 
   mainWindow = new BrowserWindow(options)
--- a/electron/filenames.gni
+++ b/electron/filenames.gni
@@ -95,7 +95,6 @@
 
   default_app_sources = [
     "default_app/default_app.js",
-    "default_app/icon.png",
     "default_app/index.html",
     "default_app/main.js",
     "default_app/menu.js",
