From 1f043501483beb898f4724220235b3369549fec6 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Tue, 4 Aug 2026 22:52:47 -0400
Subject: [PATCH] Support LLVM 22

---
 deps/LLVMExtra/lib/NewPM.cpp | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/deps/LLVMExtra/lib/NewPM.cpp b/deps/LLVMExtra/lib/NewPM.cpp
index 9d3016cb..155ab14b 100644
--- a/deps/LLVMExtra/lib/NewPM.cpp
+++ b/deps/LLVMExtra/lib/NewPM.cpp
@@ -129,6 +129,16 @@ public:
   }
 #endif
 
+#if LLVM_VERSION_MAJOR >= 22
+  InstructionUniformity getInstructionUniformity(const Value *V) const override {
+    if (Opts.IsSourceOfDivergence &&
+        Opts.IsSourceOfDivergence(wrap(V), Opts.IsSourceOfDivergenceUD) != 0)
+      return InstructionUniformity::NeverUniform;
+    if (Opts.IsAlwaysUniform && Opts.IsAlwaysUniform(wrap(V), Opts.IsAlwaysUniformUD) != 0)
+      return InstructionUniformity::AlwaysUniform;
+    return BaseT::getInstructionUniformity(V);
+  }
+#else
   bool isSourceOfDivergence(const Value *V) const TTI_OVERRIDE {
     if (Opts.IsSourceOfDivergence)
       return Opts.IsSourceOfDivergence(wrap(V),
@@ -141,6 +151,7 @@ public:
       return Opts.IsAlwaysUniform(wrap(V), Opts.IsAlwaysUniformUD) != 0;
     return BaseT::isAlwaysUniform(V);
   }
+#endif
 
   unsigned getAssumedAddrSpace(const Value *V) const TTI_OVERRIDE {
     if (Opts.GetAssumedAddressSpace)
-- 
2.53.0

