From 3ec5e41fb511017f5d6c24a31b3eba2063f25221 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Sat, 4 Jun 2022 11:19:13 -0400
Subject: [PATCH 3/4] Handle ArchLinuxARM server URL in arch-nspawn

---
 arch-nspawn.in | 15 ++++++++++++---
 1 file changed, 12 insertions(+), 3 deletions(-)

diff --git a/arch-nspawn.in b/arch-nspawn.in
index b6c9cb1..810c3e4 100644
--- a/arch-nspawn.in
+++ b/arch-nspawn.in
@@ -54,11 +54,20 @@ if (( ${#cache_dirs[@]} == 0 )); then
 fi
 
 # shellcheck disable=2016
-host_mirrors=($(pacman-conf --repo extra Server 2> /dev/null | sed -r 's#(.*/)extra/os/.*#\1$repo/os/$arch#'))
+pacman_arch=$(pacman-conf Architecture)
+if [[ $pacman_arch = arm ]] || [[ $pacman_arch = armv6h ]] || [[ $pacman_arch = armv7h ]] || [[ $pacman_arch = aarch64 ]]; then
+	host_mirrors=($(pacman-conf --repo extra Server 2> /dev/null | sed -r 's#(.*/)[^/]*/extra#\1$arch/$repo#'))
+	get_file_prefix='s#file://(/.*)/\$arch/\$repo#\1#g'
+	get_url_prefix='s#(.*/)[^/]+/.+#\1#'
+else
+	host_mirrors=($(pacman-conf --repo extra Server 2> /dev/null | sed -r 's#(.*/)extra/os/.*#\1$repo/os/$arch#'))
+	get_file_prefix='s#file://(/.*)/\$repo/os/\$arch#\1#g'
+	get_url_prefix='s#(.*/)[^/]+/os/.+#\1#'
+fi
 
 for host_mirror in "${host_mirrors[@]}"; do
 	if [[ $host_mirror == *file://* ]]; then
-		host_mirror=$(echo "$host_mirror" | sed -r 's#file://(/.*)/\$repo/os/\$arch#\1#g')
+		host_mirror=$(echo "$host_mirror" | sed -r "$get_file_prefix")
 		for m in "$host_mirror"/pool/*/; do
 			in_array "$m" "${cache_dirs[@]}" || cache_dirs+=("$m")
 		done
@@ -67,7 +76,7 @@ done
 
 while read -r line; do
 	mapfile -t lines < <(pacman-conf --config "${pac_conf:-$working_dir/etc/pacman.conf}" \
-		--repo $line Server | sed -r 's#(.*/)[^/]+/os/.+#\1#')
+		--repo $line Server | sed -r "$get_url_prefix")
 	for line in "${lines[@]}"; do
 		if [[ $line = file://* ]]; then
 			line=${line#file://}
-- 
2.36.1

