From 600ca7bc9f6561b1a46bba50e2ebada4bbaf96b0 Mon Sep 17 00:00:00 2001
From: Chih-Hsuan Yen <yan12125@gmail.com>
Date: Sat, 11 Jun 2022 23:36:59 +0800
Subject: [PATCH 5/5] Silent while show errors for curl

---
 makepkg-x86_64.conf | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/makepkg-x86_64.conf b/makepkg-x86_64.conf
index 7725296..e0b09c8 100644
--- a/makepkg-x86_64.conf
+++ b/makepkg-x86_64.conf
@@ -11,10 +11,10 @@
 #
 #-- The download utilities that makepkg should use to acquire sources
 #  Format: 'protocol::agent'
-DLAGENTS=('file::/usr/bin/curl -qgC - -o %o %u'
-          'ftp::/usr/bin/curl -qgfC - --ftp-pasv --retry 3 --retry-delay 3 -o %o %u'
-          'http::/usr/bin/curl -qgb "" -fLC - --retry 3 --retry-delay 3 -o %o %u'
-          'https::/usr/bin/curl -qgb "" -fLC - --retry 3 --retry-delay 3 -o %o %u'
+DLAGENTS=('file::/usr/bin/curl -qSsgC - -o %o %u'
+          'ftp::/usr/bin/curl -qSsgfC - --ftp-pasv --retry 3 --retry-delay 3 -o %o %u'
+          'http::/usr/bin/curl -qSsgb "" -fLC - --retry 3 --retry-delay 3 -o %o %u'
+          'https::/usr/bin/curl -qSsgb "" -fLC - --retry 3 --retry-delay 3 -o %o %u'
           'rsync::/usr/bin/rsync --no-motd -z %u %o'
           'scp::/usr/bin/scp -C %u %o')
 
-- 
2.36.1

