From a087f5431d7111dc44130a23f2ab3ce736e88dce Mon Sep 17 00:00:00 2001
From: kxxt <rsworktech@outlook.com>
Date: Fri, 26 Jul 2024 15:12:30 +0800
Subject: [PATCH 3/4] use QT_VERSION env instead of guessing

---
 scripts/base.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/scripts/base.py b/scripts/base.py
index 4e11fd6..beceb9a 100644
--- a/scripts/base.py
+++ b/scripts/base.py
@@ -681,6 +681,7 @@ def qt_setup(platform):
   return qt_dir  
 
 def qt_version():
+  return get_env("QT_VERSION")
   qt_dir = get_env("QT_DEPLOY")
   qt_dir = qt_dir.split("/")[-3]
   return "".join(i for i in qt_dir if (i.isdigit() or i == "."))
-- 
2.45.2

