From c457eb459b5401e381e97f8ee68b1f275c9b5ee8 Mon Sep 17 00:00:00 2001
Message-ID: <c457eb459b5401e381e97f8ee68b1f275c9b5ee8.1730251944.git.rsworktech@outlook.com>
From: Levi Zim <rsworktech@outlook.com>
Date: Wed, 30 Oct 2024 09:32:13 +0800
Subject: [PATCH] Only build tar

---
 scripts/package_branding.py | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/scripts/package_branding.py b/scripts/package_branding.py
index 5a83895..9a793c2 100644
--- a/scripts/package_branding.py
+++ b/scripts/package_branding.py
@@ -53,21 +53,6 @@ if utils.is_linux():
       "src": "tar/*.tar*",
       "dst": "desktop/linux/generic/"
     },
-    {
-      "make": "deb",
-      "src": "deb/*.deb",
-      "dst": "desktop/linux/debian/"
-    },
-    {
-      "make": "rpm",
-      "src": "rpm/build/RPMS/*/*.rpm",
-      "dst": "desktop/linux/rhel/"
-    },
-    {
-      "make": "rpm-suse",
-      "src": "rpm-suse/build/RPMS/*/*.rpm",
-      "dst": "desktop/linux/suse/"
-    }
   ]
   server_make_targets = [
     {
-- 
2.47.0

