diff --git a/stapbpf/Makefile.am b/stapbpf/Makefile.am
index a303d25f6..703668362 100644
--- a/stapbpf/Makefile.am
+++ b/stapbpf/Makefile.am
@@ -38,14 +38,4 @@ CLEANFILES =
 BUILT_SOURCES += git_version.stamp
 git_version.stamp ../git_version.h:
 	$(MAKE) -C .. $(notdir $@)
-
-# Why the "id -u" condition?  This way, an unprivileged user can run
-# make install, and have "sudo stap ...." or "sudo stapbpf ...." work later.
-install-exec-hook:
-	if [ `id -u` -eq 0 ]; then \
-		getent group stapusr >/dev/null || groupadd -g 156 -r stapusr 2>/dev/null || groupadd -r stapusr; \
-		getent group stapusr >/dev/null \
-		&& chgrp stapusr "$(DESTDIR)$(bindir)/stapbpf" \
-		&& chmod 04110 "$(DESTDIR)$(bindir)/stapbpf"; \
-	fi
 endif
diff --git a/staprun/Makefile.am b/staprun/Makefile.am
index 4073aa01c..978d115f9 100644
--- a/staprun/Makefile.am
+++ b/staprun/Makefile.am
@@ -68,13 +68,3 @@ CLEANFILES =
 BUILT_SOURCES += git_version.stamp
 git_version.stamp ../git_version.h:
 	$(MAKE) -C .. $(notdir $@)
-
-
-# Why the "id -u" condition?  This way, an unprivileged user can run
-# make install, and have "sudo stap ...." or "sudo staprun ...." work later.
-install-exec-hook:
-	if [ `id -u` -eq 0 ]; then \
-		getent group stapusr >/dev/null || groupadd -g 156 -r stapusr 2>/dev/null || groupadd -r stapusr; \
-		getent group stapusr >/dev/null && chgrp stapusr "$(DESTDIR)$(bindir)/staprun"; \
-		chmod 04110 "$(DESTDIR)$(bindir)/staprun"; \
-	fi
