--- ./runtime/stp_utrace.c	2017-02-17 18:37:01.000000000 +0100
+++ /usr/share/systemtap/runtime/stp_utrace.c	2017-08-09 11:59:21.542501780 +0200
@@ -1105,7 +1105,7 @@
 		 */
 		spin_lock_irq(&target->sighand->siglock);
 		if (likely(task_is_stopped(target)))
-			__set_task_state(target, TASK_TRACED);
+			set_current_state(TASK_TRACED);
 		spin_unlock_irq(&target->sighand->siglock);
 	} else if (utrace->resume > UTRACE_REPORT) {
 		utrace->resume = UTRACE_REPORT;
