diff --git a/src/stores/AppStore.js b/src/stores/AppStore.js
index 153fdb2c..c5d03338 100644
--- a/src/stores/AppStore.js
+++ b/src/stores/AppStore.js
@@ -37,7 +37,7 @@ const defaultLocale = DEFAULT_APP_SETTINGS.locale;
 const executablePath = isMac ? remote.process.execPath : process.execPath;
 const autoLauncher = new AutoLaunch({
   name: 'Ferdi',
-  path: executablePath,
+  path: '/usr/bin/ferdi',
 });
 
 const CATALINA_NOTIFICATION_HACK_KEY = '_temp_askedForCatalinaNotificationPermissions';
