diff -aur a52dec-0.7.4.orig/liba52/imdct.c a52dec-0.7.4/liba52/imdct.c
--- a52dec-0.7.4.orig/liba52/imdct.c	2016-06-10 02:39:52.984345055 -0700
+++ a52dec-0.7.4/liba52/imdct.c	2016-06-10 02:41:08.304633486 -0700
@@ -419,13 +419,11 @@
 
 #ifdef LIBA52_DJBFFT
     if (mm_accel & MM_ACCEL_DJBFFT) {
-	fprintf (stderr, "Using djbfft for IMDCT transform\n");
 	ifft128 = (void (*) (complex_t *)) fftc4_un128;
 	ifft64 = (void (*) (complex_t *)) fftc4_un64;
     } else
 #endif
     {
-	fprintf (stderr, "No accelerated IMDCT transform found\n");
 	ifft128 = ifft128_c;
 	ifft64 = ifft64_c;
     }
