From 27f765b5f41638de36de728e9018a1400b0be615 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Tue, 23 Jun 2015 08:40:29 +0200
Subject: [PATCH 1/4] Enable table validation modules

---
 modules.cfg | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules.cfg b/modules.cfg
index 2984e51..a5db21d 100644
--- a/modules.cfg
+++ b/modules.cfg
@@ -110,7 +110,7 @@ RASTER_MODULES += smooth
 AUX_MODULES += cache
 
 # TrueType GX/AAT table validation.  Needs ftgxval.c below.
-# AUX_MODULES += gxvalid
+AUX_MODULES += gxvalid
 
 # Support for streams compressed with gzip (files with suffix .gz).
 #
@@ -129,7 +129,7 @@ AUX_MODULES += bzip2
 
 # OpenType table validation.  Needs ftotval.c below.
 #
-# AUX_MODULES += otvalid
+AUX_MODULES += otvalid
 
 # Auxiliary PostScript driver component to share common code.
 #
-- 
2.6.0

