--- a/build/lib/git-environment.js
+++ b/build/lib/git-environment.js
@@ -24,7 +24,7 @@
         return path.resolve(process.env.LOCAL_GIT_DIRECTORY);
     }
     else {
-        return resolveEmbeddedGitDir();
+        return '/usr';
     }
 }
 /**
@@ -57,7 +57,7 @@
         return path.join(gitDir, 'mingw32', 'libexec', 'git-core');
     }
     else {
-        return path.join(gitDir, 'libexec', 'git-core');
+        return path.join(gitDir, 'lib', 'git-core');
     }
 }
 /**
@@ -108,9 +108,9 @@
         if (!env.GIT_SSL_CAINFO && !env.LOCAL_GIT_DIRECTORY) {
             // use the SSL certificate bundle included in the distribution only
             // when using embedded Git and not providing your own bundle
-            const distDir = resolveEmbeddedGitDir();
-            const sslCABundle = `${distDir}/ssl/cacert.pem`;
-            env.GIT_SSL_CAINFO = sslCABundle;
+            // const distDir = resolveEmbeddedGitDir();
+            // const sslCABundle = `${distDir}/ssl/cacert.pem`;
+            // env.GIT_SSL_CAINFO = sslCABundle;
         }
     }
     return { env, gitLocation };
