From 82e767bfafcc882f3c3811bb9eb008b8f7c2c731 Mon Sep 17 00:00:00 2001
From: Mario Pilz <mario@pilz-at.net>
Date: Fri, 21 Dec 2018 03:48:36 +0100
Subject: [PATCH] Fix tab close on middle click for Electron 3

---
 lib/tab-bar-view.coffee | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/tab-bar-view.coffee b/lib/tab-bar-view.coffee
index 93e958e..2b91ad6 100644
--- a/lib/tab-bar-view.coffee
+++ b/lib/tab-bar-view.coffee
@@ -87,6 +87,7 @@ class TabBarView
 
     @element.addEventListener "mousedown", @onMouseDown.bind(this)
     @element.addEventListener "click", @onClick.bind(this)
+    @element.addEventListener "auxclick", @onClick.bind(this)
     @element.addEventListener "dblclick", @onDoubleClick.bind(this)
 
     @onDropOnOtherWindow = @onDropOnOtherWindow.bind(this)
