--- a/src/main-process/atom-application.js
+++ b/src/main-process/atom-application.js
@@ -430,7 +430,7 @@
     this.openPathOnEvent('application:open-your-keymap', 'atom://.atom/keymap')
     this.openPathOnEvent('application:open-your-snippets', 'atom://.atom/snippets')
     this.openPathOnEvent('application:open-your-stylesheet', 'atom://.atom/stylesheet')
-    this.openPathOnEvent('application:open-license', path.join(process.resourcesPath, 'LICENSE.md'))
+    this.openPathOnEvent('application:open-license', '/usr/share/licenses/atom/LICENSE.md')
 
     this.configFile.onDidChange(settings => {
       for (let window of this.getAllWindows()) {
--- a/src/workspace.js
+++ b/src/workspace.js
@@ -1132,7 +1132,7 @@
 
   // Open Atom's license in the active pane.
   openLicense () {
-    return this.open(path.join(process.resourcesPath, 'LICENSE.md'))
+    return this.open('/usr/share/licenses/atom/LICENSE.md')
   }
 
   // Synchronously open the given URI in the active pane. **Only use this method
