diff --git src/auto_cpufreq-2.4.0/auto_cpufreq/gui/objects.py src.new/auto_cpufreq-2.4.0/auto_cpufreq/gui/objects.py
--- src/auto_cpufreq-2.4.0/auto_cpufreq/gui/objects.py
+++ src.new/auto_cpufreq-2.4.0/auto_cpufreq/gui/objects.py
@@ -193,7 +193,7 @@ class AboutDialog(Gtk.Dialog):
         self.add_button("Close", Gtk.ResponseType.CLOSE)
         self.set_default_size(400, 350)
         img_buffer = GdkPixbuf.Pixbuf.new_from_file_at_scale(
-            filename="/usr/local/share/auto-cpufreq/images/icon.png",
+            filename="/usr/share/pixemaps/auto-cpufreq.png",
             width=150,
             height=150,
             preserve_aspect_ratio=True
