diff -rupN gimp.org/configure.ac gimp/configure.ac
--- gimp.org/configure.ac	2018-05-24 03:36:07.751768955 +0800
+++ gimp/configure.ac	2018-05-24 03:37:45.818613655 +0800
@@ -774,9 +774,9 @@ LIBS=$gimp_save_LIBS
 
 AC_PATH_XTRA
 
-AM_PATH_GTK_3_0(gtk_required_version, :,
-  add_deps_error([gtk+-3.0 >= gtk_required_version],
-                 [Test for GTK+ failed.]))
+PKG_CHECK_MODULES([GTK], [gtk+-3.0],,
+  [add_deps_error([gtk+-3.0],
+                 [Test for GTK+ failed.])])
 
 AC_PATH_PROG(GTK_UPDATE_ICON_CACHE, gtk-update-icon-cache, no)
 if test "x$GTK_UPDATE_ICON_CACHE" = xno; then
