--- src/glmark2-git/wscript.orig        2016-05-17 23:18:02.468475747 +0200
+++ src/glmark2-git/wscript     2016-05-17 23:18:25.791307210 +0200
@@ -90,7 +90,7 @@
         ctx.check_cxx(lib = lib, uselib_store = uselib)
 
     # Check required functions
-    req_funcs = [('memset', 'string.h', []) ,('sqrt', 'math.h', ['m'])]
+    req_funcs = [('memset', 'string.h', [])]
     for func, header, uselib in req_funcs:
         ctx.check_cxx(function_name = func, header_name = header,
                       uselib = uselib, mandatory = True)