diff --git a/dde-osd/common.cpp b/dde-osd/common.cpp
index 0505ac60..ffa55e84 100644
--- a/dde-osd/common.cpp
+++ b/dde-osd/common.cpp
@@ -25,6 +25,7 @@
 
 #include "common.h"
 #include <QSvgRenderer>
+#include <QPainterPath>
 
 void DrawHelper::DrawImage(QPainter *painter, const QStyleOptionViewItem &option, const QString &pix, bool withText, bool withProgress)
 {
diff --git a/dde-osd/notification-center/overlapwidet.cpp b/dde-osd/notification-center/overlapwidet.cpp
index d2aee64e..a89f0bd1 100644
--- a/dde-osd/notification-center/overlapwidet.cpp
+++ b/dde-osd/notification-center/overlapwidet.cpp
@@ -25,6 +25,7 @@
 #include "../notification/constants.h"
 #include "notification/constants.h"
 #include "notifymodel.h"
+#include <QPainterPath>
 
 #include <QTimer>
 #include <QKeyEvent>
