diff --git a/CMakeLists.txt b/CMakeLists.txt
index 27cc41ef..b8fc462a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -116,12 +116,15 @@ function(build_Qt6)
 
         option(WITH_DECORATIONS "Build Klassy window decorations for KWin" ON)
 
+        set(PROJECT_VERSION_SAVED "${PROJECT_VERSION}")
+        set(PROJECT_VERSION "6.2.1")
         find_package(KF6FrameworkIntegration ${KF6_MIN_VERSION} CONFIG )
         set_package_properties(KF${QT_MAJOR_VERSION}FrameworkIntegration PROPERTIES
             DESCRIPTION "KF6 Framework Integration"
             URL "https://projects.kde.org/projects/frameworks/frameworkintegration"
             TYPE OPTIONAL
             PURPOSE "Required to use KStyle convenience functionalities in style")
+        set(PROJECT_VERSION "${PROJECT_VERSION_SAVED}")
 
         find_package(KF6KCMUtils ${KF6_MIN_VERSION})
         set_package_properties(KF6KCMUtils PROPERTIES
