From bbd990f83a23c22da787d10c62bd7ff114d32703 Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Mon, 18 May 2020 08:21:06 +0200
Subject: [PATCH 1/1] work around black screen
---
 src/VBox/Additions/x11/Installer/98vboxadd-xclient | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/VBox/Additions/x11/Installer/98vboxadd-xclient b/src/VBox/Additions/x11/Installer/98vboxadd-xclient
index e7d4e292..7a17a692 100755
--- a/src/VBox/Additions/x11/Installer/98vboxadd-xclient
+++ b/src/VBox/Additions/x11/Installer/98vboxadd-xclient
@@ -34,5 +34,4 @@ elif test -z "${SSH_CONNECTION}"; then
   /usr/bin/VBoxClient --checkhostversion
   /usr/bin/VBoxClient --seamless
   /usr/bin/VBoxClient --draganddrop
-  /usr/bin/VBoxClient --vmsvga  # In case VMSVGA emulation is enabled
 fi
