diff --git a/src/VBox/Frontends/VirtualBox/src/globals/UIImageTools.cpp b/src/VBox/Frontends/VirtualBox/src/globals/UIImageTools.cpp
index b70d289d..9716781e 100644
--- a/src/VBox/Frontends/VirtualBox/src/globals/UIImageTools.cpp
+++ b/src/VBox/Frontends/VirtualBox/src/globals/UIImageTools.cpp
@@ -17,6 +17,7 @@
 
 /* Qt includes: */
 #include <QPainter>
+#include <QPainterPath>
 
 /* GUI include */
 #include "UIImageTools.h"
diff --git a/src/VBox/Frontends/VirtualBox/src/widgets/UIMenuToolBar.cpp b/src/VBox/Frontends/VirtualBox/src/widgets/UIMenuToolBar.cpp
index 4d695f19..406c2dbd 100644
--- a/src/VBox/Frontends/VirtualBox/src/widgets/UIMenuToolBar.cpp
+++ b/src/VBox/Frontends/VirtualBox/src/widgets/UIMenuToolBar.cpp
@@ -19,6 +19,7 @@
 #include <QApplication>
 #include <QHBoxLayout>
 #include <QPainter>
+#include <QPainterPath>
 #include <QStyle>
 #include <QToolButton>
 
diff --git a/src/VBox/Frontends/VirtualBox/src/widgets/UIMiniToolBar.cpp b/src/VBox/Frontends/VirtualBox/src/widgets/UIMiniToolBar.cpp
index 53b545a4..6e38448c 100644
--- a/src/VBox/Frontends/VirtualBox/src/widgets/UIMiniToolBar.cpp
+++ b/src/VBox/Frontends/VirtualBox/src/widgets/UIMiniToolBar.cpp
@@ -21,6 +21,7 @@
 #include <QMenu>
 #include <QMoveEvent>
 #include <QPainter>
+#include <QPainterPath>
 #include <QStateMachine>
 #include <QStyle>
 #include <QTimer>
diff --git a/src/VBox/Frontends/VirtualBox/src/widgets/UIPopupBox.h b/src/VBox/Frontends/VirtualBox/src/widgets/UIPopupBox.h
index 2f153d80..7b18a66b 100644
--- a/src/VBox/Frontends/VirtualBox/src/widgets/UIPopupBox.h
+++ b/src/VBox/Frontends/VirtualBox/src/widgets/UIPopupBox.h
@@ -24,6 +24,7 @@
 /* Qt includes: */
 #include <QIcon>
 #include <QWidget>
+#include <QPainterPath>
 
 /* GUI includes: */
 #include "UILibraryDefs.h"
diff --git a/src/VBox/Frontends/VirtualBox/src/monitor/UIMonitorCommon.cpp b/src/VBox/Frontends/VirtualBox/src/monitor/UIMonitorCommon.cpp
index 2f153d80..7b18a66b 100644
--- a/src/VBox/Frontends/VirtualBox/src/monitor/UIMonitorCommon.cpp	2020-06-14 08:30:24.028884238 +0800
+++ b/src/VBox/Frontends/VirtualBox/src/monitor/UIMonitorCommon.cpp	2020-06-14 08:30:46.948930212 +0800
@@ -17,6 +17,7 @@

 /* Qt includes: */
 #include <QPainter>
+#include <QPainterPath>
 #include <QXmlStreamReader>


