diff --git a/latex-formulae-image.cabal b/latex-formulae-image.cabal
index 22837c7..5204a98 100644
--- a/latex-formulae-image.cabal
+++ b/latex-formulae-image.cabal
@@ -38,7 +38,7 @@ library
      build-depends: base >= 4.7 && < 4.8
                   , transformers-compat >= 0.4 && < 0.5
   else
-     build-depends: base >= 4.8 && < 4.12
+     build-depends: base >= 4.8 && < 5
   build-depends:       JuicyPixels >= 3.2 && < 3.4
                      , errors >= 2.0 && < 2.4
                      , temporary >= 1.2 && < 1.4
