--- a/packages/insomnia-app/scripts/build.js
+++ b/packages/insomnia-app/scripts/build.js
@@ -56,11 +56,6 @@ module.exports.start = async function(forceFromGitRef) {
   console.log(`[build] npm: ${childProcess.spawnSync('npm', ['--version']).stdout}`.trim());
   console.log(`[build] node: ${childProcess.spawnSync('node', ['--version']).stdout}`.trim());
 
-  if (process.version.indexOf('v12.') !== 0) {
-    console.log('[build] Node v12.x.x is required to build');
-    process.exit(1);
-  }
-
   // Remove folders first
   console.log('[build] Removing existing directories');
   await emptyDir(buildFolder);
