--- a/astropy_helpers/commands/build_sphinx.py	2019-11-18 09:57:11.000000000 +0800
+++ b/astropy_helpers/commands/build_sphinx.py	2020-12-05 10:25:49.420269236 +0800
@@ -137,7 +137,7 @@
         build_cmd = self.get_finalized_command('build')
         build_cmd_path = os.path.abspath(build_cmd.build_lib)
 
-        ah_importer = pkgutil.get_importer('astropy_helpers')
+        ah_importer = pkgutil.get_importer('/usr/lib/python3.9/site-packages/astropy_helpers')
         if ah_importer is None:
             ah_path = '.'
         else:
