diff --git a/dde-file-manager-lib/dialogs/openwithdialog.cpp b/dde-file-manager-lib/dialogs/openwithdialog.cpp
index 95d8bb9b..c6cc2ecf 100644
--- a/dde-file-manager-lib/dialogs/openwithdialog.cpp
+++ b/dde-file-manager-lib/dialogs/openwithdialog.cpp
@@ -48,6 +48,7 @@
 #include <QStandardPaths>
 #include <QScroller>
 #include <QCommandLinkButton>
+#include <QPainterPath>
 #include <DHorizontalLine>
 #include <DStyle>
 #include <DIconButton>
diff --git a/dde-file-manager-lib/dialogs/propertydialog.cpp b/dde-file-manager-lib/dialogs/propertydialog.cpp
index 740c1f76..ff822213 100644
--- a/dde-file-manager-lib/dialogs/propertydialog.cpp
+++ b/dde-file-manager-lib/dialogs/propertydialog.cpp
@@ -79,6 +79,7 @@
 #include <QButtonGroup>
 #include <QProgressBar>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPushButton>
 #include <QStackedWidget>
 #include <QStorageInfo>
diff --git a/dde-file-manager-lib/interfaces/dfmglobal.cpp b/dde-file-manager-lib/interfaces/dfmglobal.cpp
index 56b0140a..58be9814 100644
--- a/dde-file-manager-lib/interfaces/dfmglobal.cpp
+++ b/dde-file-manager-lib/interfaces/dfmglobal.cpp
@@ -70,6 +70,7 @@
 #include <QMediaPlayer>
 #include <QDBusObjectPath>
 #include <QRegularExpression>
+#include <QPainterPath>
 
 #include <private/qtextengine_p.h>
 
diff --git a/dde-file-manager-lib/interfaces/dfmstyleditemdelegate.cpp b/dde-file-manager-lib/interfaces/dfmstyleditemdelegate.cpp
index fdf4b2d2..d6fde0e6 100644
--- a/dde-file-manager-lib/interfaces/dfmstyleditemdelegate.cpp
+++ b/dde-file-manager-lib/interfaces/dfmstyleditemdelegate.cpp
@@ -14,6 +14,7 @@
 #include <QDebug>
 #include <QAbstractItemView>
 #include <QPainter>
+#include <QPainterPath>
 #include <QGuiApplication>
 #include <QThreadStorage>
 
diff --git a/dde-file-manager-lib/interfaces/diconitemdelegate.cpp b/dde-file-manager-lib/interfaces/diconitemdelegate.cpp
index 02a91283..d500aac8 100644
--- a/dde-file-manager-lib/interfaces/diconitemdelegate.cpp
+++ b/dde-file-manager-lib/interfaces/diconitemdelegate.cpp
@@ -34,6 +34,7 @@
 
 #include <QLabel>
 #include <QPainter>
+#include <QPainterPath>
 #include <QTextEdit>
 #include <QLineEdit>
 #include <QTextBlock>
diff --git a/dde-file-manager-lib/interfaces/dlistitemdelegate.cpp b/dde-file-manager-lib/interfaces/dlistitemdelegate.cpp
index 3e78dc7a..01392e06 100644
--- a/dde-file-manager-lib/interfaces/dlistitemdelegate.cpp
+++ b/dde-file-manager-lib/interfaces/dlistitemdelegate.cpp
@@ -16,6 +16,7 @@
 
 #include <QLabel>
 #include <QPainter>
+#include <QPainterPath>
 #include <QAbstractItemView>
 #include <QLineEdit>
 #include <QApplication>
diff --git a/dialogs/dfmtaskwidget.cpp b/dialogs/dfmtaskwidget.cpp
index 94069ee8..464c2095 100644
--- a/dialogs/dfmtaskwidget.cpp
+++ b/dialogs/dfmtaskwidget.cpp
@@ -30,6 +30,7 @@
 #include <DGuiApplicationHelper>
 #include <QTimer>
 #include <QException>
+#include <QPainterPath>
 
 #include "app/define.h"
 #include "dfileservices.h"
