From a63cf14859315a9552e0497a6b72263be04ab90e Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Sun, 26 Oct 2014 15:28:52 -0400
Subject: [PATCH] fix path

---
 setuptools/command/easy_install.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/setuptools/command/easy_install.py b/setuptools/command/easy_install.py
index 6854827..8107307 100644
--- a/setuptools/command/easy_install.py
+++ b/setuptools/command/easy_install.py
@@ -1250,7 +1250,7 @@ Please make the appropriate changes for your system and try again."""
 
     INSTALL_SCHEMES = dict(
         posix=dict(
-            install_dir='$base/lib/python$py_version_short/site-packages',
+            install_dir='$base/lib32/python$py_version_short/site-packages',
             script_dir='$base/bin',
         ),
     )
@@ -1296,13 +1296,13 @@ def get_site_dirs():
                 sitedirs.append(os.path.join(prefix, "Lib", "site-packages"))
             elif os.sep == '/':
                 sitedirs.extend([os.path.join(prefix,
-                                              "lib",
+                                              "lib32",
                                               "python" + sys.version[:3],
                                               "site-packages"),
-                                 os.path.join(prefix, "lib", "site-python")])
+                                 os.path.join(prefix, "lib32", "site-python")])
             else:
                 sitedirs.extend(
-                    [prefix, os.path.join(prefix, "lib", "site-packages")]
+                    [prefix, os.path.join(prefix, "lib32", "site-packages")]
                 )
             if sys.platform == 'darwin':
                 # for framework builds *only* we add the standard Apple
-- 
2.1.2

