diff -aur plymouth.src/systemd-units/plymouth-quit.service.in plymouth.patch/systemd-units/plymouth-quit.service.in
--- plymouth.src/systemd-units/plymouth-quit.service.in        2014-02-20 19:38:13.329427597 +1100
+++ plymouth.patch/systemd-units/plymouth-quit.service.in      2014-02-20 19:40:52.619343839 +1100
@@ -1,8 +1,9 @@
 [Unit]
 Description=Terminate Plymouth Boot Screen
 After=rc-local.service plymouth-start.service systemd-user-sessions.service
+Before=getty@tty1.service
 
 [Service]
 ExecStart=-@PLYMOUTH_CLIENT_DIR@/plymouth quit
 Type=oneshot
 TimeoutSec=20
