diff --git a/CMakeLists.txt b/CMakeLists.txt
index 926628f933..1bf9ff4121 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -200,9 +200,23 @@ if (WITH_SYSTEM_PROVIDED_3PARTY)
   find_package(gflags REQUIRED)

   find_package(expat CONFIG REQUIRED)
-  find_package(jansson CONFIG REQUIRED)
+
+  find_package(jansson CONFIG QUIET)
+  if (NOT jansson_FOUND)
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules(PKG_JANSSON REQUIRED IMPORTED_TARGET jansson)
+    if (NOT TARGET jansson::jansson)
+      add_library(jansson::jansson ALIAS PkgConfig::PKG_JANSSON)
+    endif()
+  endif()
+
   find_package(pugixml REQUIRED)
   find_package(utf8cpp REQUIRED)
+  link_libraries(utf8cpp::utf8cpp)
+  find_package(Boost REQUIRED)
+  if (NOT SKIP_TESTS)
+    find_package(GTest REQUIRED)
+  endif()
 endif()

 find_package(ZLIB REQUIRED)
