diff --git a/libs/drape/drape_tests/CMakeLists.txt b/libs/drape/drape_tests/CMakeLists.txt
index 5dc69d71d9..3b3257c310 100644
--- a/libs/drape/drape_tests/CMakeLists.txt
+++ b/libs/drape/drape_tests/CMakeLists.txt
@@ -39,7 +39,7 @@ target_compile_definitions(${PROJECT_NAME}

 if (WITH_SYSTEM_PROVIDED_3PARTY)
   find_package(GTest REQUIRED)
-  target_link_libraries(${PROJECT_NAME} GTest::gtest)
+  target_link_libraries(${PROJECT_NAME} PRIVATE GTest::gtest)
 else()
   add_subdirectory("${CMAKE_CURRENT_LIST_DIR}/../../../3party/googletest" "${CMAKE_CURRENT_BINARY_DIR}/googletest")
   target_include_directories(${PROJECT_NAME} PUBLIC ${OMIM_ROOT}/3party/glm)
