diff --git a/CMakeLists.txt b/CMakeLists.txt
index 926628f933..ccf9f17d2e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -100,29 +100,12 @@ if (${CMAKE_BUILD_TYPE} STREQUAL "Debug")
   endif()
 elseif (${CMAKE_BUILD_TYPE} MATCHES "Rel")
   add_definitions(-DRELEASE)
-  if (NOT MSVC)
-    add_compile_options(-O3)
-  endif()
 else()
   message(FATAL_ERROR "Unknown build type: " ${CMAKE_BUILD_TYPE})
 endif()

 message(STATUS "Build type: " ${CMAKE_BUILD_TYPE})

-if (PLATFORM_LINUX OR PLATFORM_ANDROID)
-  find_program(LLD_FOUND ld.lld)
-  if (LLD_FOUND)
-    message(STATUS "Using ld.lld linker")
-    set(CMAKE_EXE_LINKER_FLAGS "-fuse-ld=lld")
-  else()
-    find_program(GOLD_FOUND ld.gold)
-    if (GOLD_FOUND)
-      message(STATUS "Using ld.gold")
-      set(CMAKE_EXE_LINKER_FLAGS "-fuse-ld=gold")
-    endif()
-  endif()
-endif()
-
 if (NOT SKIP_TESTS)
   enable_testing()
   # Enables ctest -T memcheck with valgrind
