diff --git a/stapbpf/Makefile.am b/stapbpf/Makefile.am
index ba1869765..43dc35ab3 100644
--- a/stapbpf/Makefile.am
+++ b/stapbpf/Makefile.am
@@ -39,12 +39,4 @@ BUILT_SOURCES += git_version.stamp
 git_version.stamp ../git_version.h:
 	$(MAKE) -C .. $(notdir $@)
 
-# Why the "id -u" condition?  This way, an unprivileged user can run
-# make install, and have "sudo stap ...." or "sudo stapbpf ...." work later.
-install-exec-hook:
-	if [ `id -u` -eq 0 ] && (getent group stapusr >/dev/null \
-	                         || groupadd -f -g 156 -r stapusr); then \
-		chgrp stapusr "$(DESTDIR)$(bindir)/stapbpf" \
-		&& chmod 04110 "$(DESTDIR)$(bindir)/stapbpf"; \
-	fi
 endif
diff --git a/stapbpf/Makefile.in b/stapbpf/Makefile.in
index e3a3146e7..efc84185c 100644
--- a/stapbpf/Makefile.in
+++ b/stapbpf/Makefile.in
@@ -747,7 +747,6 @@ maintainer-clean-generic:
 	@echo "This command is intended for maintainers to use"
 	@echo "it deletes files that may require special tools to rebuild."
 	-test -z "$(BUILT_SOURCES)" || rm -f $(BUILT_SOURCES)
-@HAVE_BPF_DECLS_FALSE@install-exec-hook:
 clean: clean-am
 
 clean-am: clean-binPROGRAMS clean-generic mostlyclean-am
@@ -780,7 +779,7 @@ install-dvi-am:
 
 install-exec-am: install-binPROGRAMS
 	@$(NORMAL_INSTALL)
-	$(MAKE) $(AM_MAKEFLAGS) install-exec-hook
+	$(MAKE) $(AM_MAKEFLAGS)
 install-html: install-html-am
 
 install-html-am:
@@ -831,7 +830,7 @@ uninstall-man: uninstall-man8
 	distclean distclean-compile distclean-generic distclean-tags \
 	dvi dvi-am html html-am info info-am install install-am \
 	install-binPROGRAMS install-data install-data-am install-dvi \
-	install-dvi-am install-exec install-exec-am install-exec-hook \
+	install-dvi-am install-exec install-exec-am \
 	install-html install-html-am install-info install-info-am \
 	install-man install-man8 install-pdf install-pdf-am install-ps \
 	install-ps-am install-strip installcheck installcheck-am \
@@ -845,15 +844,6 @@ uninstall-man: uninstall-man8
 @HAVE_BPF_DECLS_TRUE@git_version.stamp ../git_version.h:
 @HAVE_BPF_DECLS_TRUE@	$(MAKE) -C .. $(notdir $@)
 
-# Why the "id -u" condition?  This way, an unprivileged user can run
-# make install, and have "sudo stap ...." or "sudo stapbpf ...." work later.
-@HAVE_BPF_DECLS_TRUE@install-exec-hook:
-@HAVE_BPF_DECLS_TRUE@	if [ `id -u` -eq 0 ] && (getent group stapusr >/dev/null \
-@HAVE_BPF_DECLS_TRUE@	                         || groupadd -f -g 156 -r stapusr); then \
-@HAVE_BPF_DECLS_TRUE@		chgrp stapusr "$(DESTDIR)$(bindir)/stapbpf" \
-@HAVE_BPF_DECLS_TRUE@		&& chmod 04110 "$(DESTDIR)$(bindir)/stapbpf"; \
-@HAVE_BPF_DECLS_TRUE@	fi
-
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
diff --git a/staprun/Makefile.am b/staprun/Makefile.am
index dd1c920b4..d8779f634 100644
--- a/staprun/Makefile.am
+++ b/staprun/Makefile.am
@@ -73,12 +73,3 @@ BUILT_SOURCES += git_version.stamp
 git_version.stamp ../git_version.h:
 	$(MAKE) -C .. $(notdir $@)
 
-
-# Why the "id -u" condition?  This way, an unprivileged user can run
-# make install, and have "sudo stap ...." or "sudo staprun ...." work later.
-install-exec-hook:
-	if [ `id -u` -eq 0 ] && (getent group stapusr >/dev/null \
-	                         || groupadd -f -g 156 -r stapusr); then \
-		chgrp stapusr "$(DESTDIR)$(bindir)/staprun" \
-		&& chmod 04110 "$(DESTDIR)$(bindir)/staprun"; \
-	fi
diff --git a/staprun/Makefile.in b/staprun/Makefile.in
index 35368520e..3585f16a5 100644
--- a/staprun/Makefile.in
+++ b/staprun/Makefile.in
@@ -1075,7 +1075,7 @@ install-dvi-am:
 
 install-exec-am: install-binPROGRAMS install-pkglibexecPROGRAMS
 	@$(NORMAL_INSTALL)
-	$(MAKE) $(AM_MAKEFLAGS) install-exec-hook
+	$(MAKE) $(AM_MAKEFLAGS)
 install-html: install-html-am
 
 install-html-am:
@@ -1143,7 +1143,7 @@ uninstall-man: uninstall-man8
 	distclean distclean-compile distclean-generic distclean-tags \
 	dvi dvi-am html html-am info info-am install install-am \
 	install-binPROGRAMS install-data install-data-am install-dvi \
-	install-dvi-am install-exec install-exec-am install-exec-hook \
+	install-dvi-am install-exec install-exec-am \
 	install-html install-html-am install-info install-info-am \
 	install-man install-man8 install-pdf install-pdf-am \
 	install-pkglibexecPROGRAMS install-ps install-ps-am \
@@ -1158,15 +1158,6 @@ uninstall-man: uninstall-man8
 git_version.stamp ../git_version.h:
 	$(MAKE) -C .. $(notdir $@)
 
-# Why the "id -u" condition?  This way, an unprivileged user can run
-# make install, and have "sudo stap ...." or "sudo staprun ...." work later.
-install-exec-hook:
-	if [ `id -u` -eq 0 ] && (getent group stapusr >/dev/null \
-	                         || groupadd -f -g 156 -r stapusr); then \
-		chgrp stapusr "$(DESTDIR)$(bindir)/staprun" \
-		&& chmod 04110 "$(DESTDIR)$(bindir)/staprun"; \
-	fi
-
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
