From e0581daf3013acfcc2d14e375cf7bc6734f11a9f Mon Sep 17 00:00:00 2001
From: everyx <lunt.luo@gmail.com>
Date: Tue, 24 Dec 2024 13:42:16 +0800
Subject: [PATCH 2/3] download lang packs withou git clone

---
 scripts/download-language-packs.sh | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/scripts/download-language-packs.sh b/scripts/download-language-packs.sh
index 7fa7047..ca8d00e 100644
--- a/scripts/download-language-packs.sh
+++ b/scripts/download-language-packs.sh
@@ -1,16 +1,16 @@
 
-set -ex
+# set -ex
 
 CURRENT_DIR=$(pwd)
 
-git config --global init.defaultBranch main
-git config --global fetch.prune true
+# git config --global init.defaultBranch main
+# git config --global fetch.prune true
 
 cd $CURRENT_DIR
 
-cd ./l10n
-git clone https://github.com/mozilla-l10n/firefox-l10n
-cd $CURRENT_DIR
+# cd ./l10n
+# git clone https://github.com/mozilla-l10n/firefox-l10n
+# cd $CURRENT_DIR
 
 update_language() {
   langId=$1
-- 
2.47.1

