diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index a027c69e0..eb7608518 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -841,7 +841,9 @@ add_library(safeint_interface INTERFACE)
 target_include_directories(safeint_interface INTERFACE ${SAFEINT_INCLUDE_DIR})
 
 if (onnxruntime_PREFER_SYSTEM_LIB)
-  find_package(boost_mp11)
+  # boost on Arch does not support find_package(boost_mp11)
+  add_library(boost_mp11 INTERFACE)
+  add_library(Boost::mp11 ALIAS boost_mp11)
 endif()
 if (NOT TARGET Boost::mp11)
   add_subdirectory(external/mp11 EXCLUDE_FROM_ALL)
@@ -764,10 +765,12 @@ if(onnxruntime_DISABLE_EXCEPTIONS)
 
 set(JSON_BuildTests OFF CACHE INTERNAL "")
 set(JSON_Install OFF CACHE INTERNAL "")
-add_subdirectory(external/json EXCLUDE_FROM_ALL)
+find_package(nlohmann_json REQUIRED)
 
 if (onnxruntime_PREFER_SYSTEM_LIB)
-  find_package(re2)
+  find_package(PkgConfig)
+  pkg_check_modules(RE2 IMPORTED_TARGET re2)
+  add_library(re2::re2 ALIAS PkgConfig::RE2)
 endif()
 if (NOT TARGET re2::re2)
   add_subdirectory(external/re2 EXCLUDE_FROM_ALL)
@@ -1421,7 +1421,7 @@
   find_package(Flatbuffers)
 endif()
 if (Flatbuffers_FOUND)
-  add_library(flatbuffers ALIAS flatbuffers::flatbuffers)
+  add_library(flatbuffers ALIAS flatbuffers::flatbuffers_shared)
 else()
   add_subdirectory(external/flatbuffers EXCLUDE_FROM_ALL)
 endif()
