diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index a027c69e0..eb7608518 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -733,7 +733,7 @@
 
 if (NOT WIN32)
   if (onnxruntime_PREFER_SYSTEM_LIB)
-    find_package(nsync)
+    find_package(nsync_cpp)
   endif()
   if (TARGET nsync_cpp)  # linking error with nsync_FOUND (why?)
     message("Use nsync from preinstalled system lib")
@@ -764,9 +765,11 @@ if(onnxruntime_DISABLE_EXCEPTIONS)
 endif()
 
 if (onnxruntime_PREFER_SYSTEM_LIB)
-  find_package(re2)
+  find_package(PkgConfig)
+  pkg_check_modules(RE2 IMPORTED_TARGET re2)
+  add_library(re2::re2 ALIAS PkgConfig::RE2)
 endif()
-if (re2_FOUND)
+if (TARGET re2::re2)
   message("Use re2 from preinstalled system lib")
 else()
   add_subdirectory(external/re2 EXCLUDE_FROM_ALL)
@@ -1421,7 +1421,7 @@
 endif()
 if (Flatbuffers_FOUND)
   message("Use flatbuffers from preinstalled system lib")
-  add_library(flatbuffers ALIAS flatbuffers::flatbuffers)
+  add_library(flatbuffers ALIAS flatbuffers::flatbuffers_shared)
 else()
   message("Use flatbuffers from submodule")
   # We do not need to build flatc for iOS or Android Cross Compile
