--- lib/Makefile.am.orig	2017-02-17 03:39:51.000000000 +0100
+++ lib/Makefile.am	2017-04-04 23:49:52.849386841 +0200
@@ -7,6 +7,7 @@
 	-I$(top_srcdir)/common \
 	@CCNET_CFLAGS@ \
 	@SEARPC_CFLAGS@ \
+	@SSL_CFLAGS@ \
 	@MSVC_CFLAGS@ \
 	-Wall
 
--- daemon/Makefile.am.orig	2017-04-24 22:57:21.895319096 +0200
+++ daemon/Makefile.am	2017-04-24 22:57:08.558729290 +0200
@@ -8,6 +8,7 @@
 	-I$(top_srcdir)/common \
 	@CCNET_CFLAGS@ \
 	@SEARPC_CFLAGS@ \
+	@SSL_CFLAGS@ \
 	@GLIB2_CFLAGS@ \
 	@MSVC_CFLAGS@ \
 	@CURL_CFLAGS@ \
