From 4d1e01e1df513666642d4ddbb24625c5bcc12d84 Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Thu, 14 Sep 2017 23:14:36 +0200
Subject: [PATCH 1/1] fix compilation with glibc 2.26

Signed-off-by: Christian Hesse <mail@eworm.de>
---
 src/VBox/ExtPacks/VBoxDTrace/include/VBoxDTraceTypes.h | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/VBox/ExtPacks/VBoxDTrace/include/VBoxDTraceTypes.h b/src/VBox/ExtPacks/VBoxDTrace/include/VBoxDTraceTypes.h
index 9032e6d..0d1770e 100644
--- a/src/VBox/ExtPacks/VBoxDTrace/include/VBoxDTraceTypes.h
+++ b/src/VBox/ExtPacks/VBoxDTrace/include/VBoxDTraceTypes.h
@@ -55,11 +55,11 @@ typedef uint64_t                    hrtime_t;
 typedef uint32_t                    id_t;
 typedef uint32_t                    zoneid_t;
 #endif
-#if !defined(NGREG) || !defined(RT_OS_LINUX)
+#if !defined(__NGREG) || !defined(RT_OS_LINUX)
 typedef RTCCINTREG                  greg_t;
 #else
 AssertCompileSize(greg_t, sizeof(RTCCINTREG));
 #endif
 typedef uintptr_t                   pc_t;
 typedef unsigned int                model_t;
 typedef RTCPUID                     processorid_t;
