diff -u -r VirtualBox-5.2.12/src/VBox/Frontends/VirtualBox/src/settings/global/UIGlobalSettingsProxy.cpp VirtualBox-5.2.12-qt/src/VBox/Frontends/VirtualBox/src/settings/global/UIGlobalSettingsProxy.cpp
--- VirtualBox-5.2.12/src/VBox/Frontends/VirtualBox/src/settings/global/UIGlobalSettingsProxy.cpp	2018-05-09 18:38:05.000000000 +0200
+++ VirtualBox-5.2.12-qt/src/VBox/Frontends/VirtualBox/src/settings/global/UIGlobalSettingsProxy.cpp	2018-06-05 21:51:18.401913367 +0200
@@ -21,6 +21,7 @@
 
 /* Qt includes: */
 # include <QRegExpValidator>
+# include <QButtonGroup>
 
 /* GUI includes: */
 # include "QIWidgetValidator.h"
