--- nvidia-texture-tools/src/CMakeLists.txt.old	2010-11-29 20:35:27.000000000 +0000
+++ nvidia-texture-tools/src/CMakeLists.txt	2010-11-29 20:35:44.000000000 +0000
@@ -107,6 +107,7 @@ FIND_PACKAGE(Qt4)
 # Threads
 FIND_PACKAGE(Threads REQUIRED)
 MESSAGE(STATUS "Use thread library: ${CMAKE_THREAD_LIBS_INIT}")
+SET(LIBS ${LIBS} ${CMAKE_THREAD_LIBS_INIT})
 
 # configuration file
 INCLUDE(CheckIncludeFiles)
