diff --git a/ui/style/style_core_font.cpp b/ui/style/style_core_font.cpp
index af0928c..15f982b 100644
--- a/ui/style/style_core_font.cpp
+++ b/ui/style/style_core_font.cpp
@@ -340,14 +340,6 @@ struct Metrics {
 	font.setWeight((flags & (FontFlag::Bold | FontFlag::Semibold))
 		? QFont::DemiBold
 		: QFont::Normal);
-	if (font.bold()) {
-		const auto style = QFontInfo(font).styleName();
-		if (!style.isEmpty() && !style.startsWith(
-				"Semibold",
-				Qt::CaseInsensitive)) {
-			font.setBold(true);
-		}
-	}
 
 	font.setItalic(flags & FontFlag::Italic);
 	font.setUnderline(flags & FontFlag::Underline);
