diff --git a/data/desktop/anbox-android.directory b/data/desktop/anbox-android.directory
index 8af4627..614cde8 100644
--- a/data/desktop/anbox-android.directory
+++ b/data/desktop/anbox-android.directory
@@ -2,4 +2,4 @@
 Type=Directory
 Name=Anbox
 GenericName=Anbox
-Icon=${SNAP}/meta/gui/icon.png
+Icon=anbox
diff --git a/data/desktop/anbox-android.menu b/data/desktop/anbox-android.menu
index ad0af8b..492c78f 100644
--- a/data/desktop/anbox-android.menu
+++ b/data/desktop/anbox-android.menu
@@ -9,7 +9,7 @@
 		<Include>
 			<Or>
 				<Category>Anbox</Category>
-				<Filename>appmgr.desktop</Filename>
+				<Filename>anbox.desktop</Filename>
 			</Or>
 		</Include>
 	</Menu>
