From 2877a352715ce7e30ef364722757c6ec380236b1 Mon Sep 17 00:00:00 2001
Message-Id: <2877a352715ce7e30ef364722757c6ec380236b1.1518037075.git.jan.steffens@gmail.com>
From: Harald van Dijk <harald@gigawatt.nl>
Date: Thu, 18 Jan 2018 15:18:37 -0700
Subject: [PATCH] Bug 1430274 - Define MOZ_ALSA for more source files. r=jld

---
 security/sandbox/linux/moz.build | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/security/sandbox/linux/moz.build b/security/sandbox/linux/moz.build
index 681d658bfeb7..1a49b548af2d 100644
--- a/security/sandbox/linux/moz.build
+++ b/security/sandbox/linux/moz.build
@@ -78,6 +78,9 @@ if CONFIG['MOZ_GMP_SANDBOX']:
         'SandboxOpenedFiles.cpp',
     ]
 
+if CONFIG['MOZ_ALSA']:
+    DEFINES['MOZ_ALSA'] = True
+
 # This copy of SafeSPrintf doesn't need to avoid the Chromium logging
 # dependency like the one in libxul does, but this way the behavior is
 # consistent.  See also the comment in SandboxLogging.h.
-- 
2.16.1

