diff --git i/old-configure.in w/old-configure.in
index 12170a47568f..dbbaa0bc9194 100644
--- i/old-configure.in
+++ w/old-configure.in
@@ -1808,9 +1808,7 @@ if test -n "$_USE_SYSTEM_NSS"; then
     AM_PATH_NSS(3.35, [MOZ_SYSTEM_NSS=1], [AC_MSG_ERROR([you don't have NSS installed or your version is too old])])
 fi
 
-if test -n "$MOZ_SYSTEM_NSS"; then
-   NSS_LIBS="$NSS_LIBS -lcrmf"
-else
+if test -z "$MOZ_SYSTEM_NSS"; then
    NSS_CFLAGS="-I${DIST}/include/nss"
    case "${OS_ARCH}" in
         # Only few platforms have been tested with GYP
