diff -up firefox-33.0/mozilla-release/toolkit/modules/CertUtils.jsm.966424 firefox-33.0/mozilla-release/toolkit/modules/CertUtils.jsm
--- firefox-33.0/mozilla-release/toolkit/modules/CertUtils.jsm.966424	2014-10-14 08:12:14.358697255 +0200
+++ firefox-33.0/mozilla-release/toolkit/modules/CertUtils.jsm	2014-10-14 08:17:59.962181908 +0200
@@ -174,7 +174,9 @@ this.checkCert =
 }
 
 function isBuiltinToken(tokenName) {
-  return tokenName == "Builtin Object Token";
+  return tokenName == "Builtin Object Token" ||
+         tokenName == "Default Trust" ||
+         tokenName == "System Trust";
 }
 
 /**
