--- a/applications/vokoscreen.desktop	2015-05-27 12:13:34.206233690 -0500
+++ b/applications/vokoscreen.desktop	2015-05-27 12:16:02.223851764 -0500
@@ -1,10 +1,10 @@
 [Desktop Entry]
-Comment=screencast
+Comment=An easy to use screencast creator
 Exec=vokoscreen
 Icon=vokoscreen
 Name=vokoscreen
 StartupNotify=false
 Terminal=false
 Type=Application
-Categories=AudioVideo;Recorder;
+Categories=AudioVideo;Recorder;Qt;
 Keywords=Audio;Video;Recorder;Screencast;
