From e5ca712d343956666fa886e244d081d5febe6649 Mon Sep 17 00:00:00 2001
From: gnome-mpv <gnome-mpv@nowhere>
Date: Tue, 13 Jun 2017 17:24:49 +0700
Subject: [PATCH] Bump version number in meson.build

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 5a4f9c4..d69a496 100644
--- a/meson.build
+++ b/meson.build
@@ -1,5 +1,5 @@
 project('gnome-mpv', 'c',
-  version: '0.11',
+  version: '0.12',
   meson_version: '>= 0.37.0',
   default_options: [
     'warning_level=2',
