diff --git a/Makefile.am b/Makefile.am
index ecb94bd..609a125 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -2,7 +2,7 @@ AUTOMAKE_OPTIONS = foreign
 
 SUBDIRS = src data docs po
 
-EXTRA_DIST = README.md AUTHORS meson.build meson_post_install.py po/meson.build
+EXTRA_DIST = README.md AUTHORS meson.build docs/meson.build meson_post_install.py po/meson.build
 
 UPDATE_DESKTOP = update-desktop-database -q || :
 UPDATE_ICON = gtk-update-icon-cache -q $(datadir)/icons/hicolor/ || :
diff --git a/Makefile.in b/Makefile.in
index f0067a6..69f3287 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -168,8 +168,8 @@ CTAGS = ctags
 CSCOPE = cscope
 DIST_SUBDIRS = $(SUBDIRS)
 am__DIST_COMMON = $(srcdir)/Makefile.in ABOUT-NLS AUTHORS COPYING \
-	compile config.guess config.rpath config.sub install-sh \
-	missing
+	compile config.guess config.rpath config.sub depcomp \
+	install-sh missing
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 distdir = $(PACKAGE)-$(VERSION)
 top_distdir = $(distdir)
@@ -341,7 +341,7 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = foreign
 SUBDIRS = src data docs po
-EXTRA_DIST = README.md AUTHORS meson.build meson_post_install.py po/meson.build
+EXTRA_DIST = README.md AUTHORS meson.build docs/meson.build meson_post_install.py po/meson.build
 UPDATE_DESKTOP = update-desktop-database -q || :
 UPDATE_ICON = gtk-update-icon-cache -q $(datadir)/icons/hicolor/ || :
 
diff --git a/data/io.github.GnomeMpv.appdata.xml.in b/data/io.github.GnomeMpv.appdata.xml.in
index c3f5d86..5712047 100644
--- a/data/io.github.GnomeMpv.appdata.xml.in
+++ b/data/io.github.GnomeMpv.appdata.xml.in
@@ -24,7 +24,7 @@
   </ul>
  </description>
  <releases>
-  <release date="2017-06-12" version="0.12"/>
+  <release date="2017-10-13" version="0.13"/>
  </releases>
  <screenshots>
   <screenshot type="default">
diff --git a/docs/meson.build b/docs/meson.build
new file mode 100644
index 0000000..c7aa8ab
--- /dev/null
+++ b/docs/meson.build
@@ -0,0 +1 @@
+install_man('gnome-mpv.1')
