From 0f3c23f0d752af1eff7dec7ca95143e4b4f1eb97 Mon Sep 17 00:00:00 2001
From: gnome-mpv <gnome-mpv@nowhere>
Date: Mon, 23 Oct 2017 09:43:36 -0700
Subject: [PATCH] application: Change the default permission for config dirs to
 755

Fixes #319.
---
 src/gmpv_application.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/gmpv_application.c b/src/gmpv_application.c
index aa90827..d9f425e 100644
--- a/src/gmpv_application.c
+++ b/src/gmpv_application.c
@@ -156,9 +156,9 @@ static void create_dirs()
 	gchar *scripts_dir = get_scripts_dir_path();
 	gchar *watch_dir = get_watch_dir_path();
 
-	g_mkdir_with_parents(config_dir, 0600);
-	g_mkdir_with_parents(scripts_dir, 0700);
-	g_mkdir_with_parents(watch_dir, 0600);
+	g_mkdir_with_parents(config_dir, 0755);
+	g_mkdir_with_parents(scripts_dir, 0755);
+	g_mkdir_with_parents(watch_dir, 0755);
 
 	g_free(config_dir);
 	g_free(scripts_dir);
