diff --git a/CMakeLists.txt b/CMakeLists.txt
index c8753e8..97f8d8a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -244,7 +244,7 @@ function(qt_install_qml_module target_)
   get_target_property(QML_TYEPINFO ${target_} QT_QML_MODULE_TYPEINFO)
   get_target_property(QML_PLUGIN_TARGET ${target_} QT_QML_MODULE_PLUGIN_TARGET)
   set(QML_INSTALL_DIR
-      "${CMAKE_INSTALL_LIBDIR}/qml/${QML_TARGET_PATH}")
+      "${CMAKE_INSTALL_LIBDIR}/qt6/qml/${QML_TARGET_PATH}")
 
   install(TARGETS ${target_})
 
