diff --git a/src/bomi/imports/bomi/qmldir b/src/bomi/imports/bomi/qmldir
index 307cc33c..79d3e8b5 100644
--- a/src/bomi/imports/bomi/qmldir
+++ b/src/bomi/imports/bomi/qmldir
@@ -35,3 +35,6 @@ SimpleBlur			1.0 SimpleBlur.qml
 ToolPlaneStyle		1.0 ToolPlaneStyle.qml
 BarVisualizer		1.0 BarVisualizer.qml
 StateButton			1.0 StateButton.qml
+internal Osd			Osd.qml
+internal PlayInfoText	PlayInfoText.qml
+internal PlayInfoTrack	PlayInfoTrack.qml
