diff --git a/src/mpv/bootstrap.py b/src/mpv/bootstrap.py
index c62914cb..b9b6e9b6 100755
--- a/src/mpv/bootstrap.py
+++ b/src/mpv/bootstrap.py
@@ -5,10 +5,10 @@
 from __future__ import print_function
 import os, sys, stat, hashlib, subprocess
 
-WAFRELEASE = "waf-1.8.4"
+WAFRELEASE = "waf-2.0.9"
 WAFURLS    = ["http://ftp.waf.io/pub/release/" + WAFRELEASE,
               "http://www.freehackers.org/~tnagy/release/" + WAFRELEASE]
-SHA256HASH = "f02035fa5d8814f33f19b2b20d43822ddef6bb39b955ca196c2a247a1f9ffaa8"
+SHA256HASH = "2a8e0816f023995e557f79ea8940d322bec18f286917c8f9a6fa2dc3875dfa48"
 
 if os.path.exists("waf"):
     wafver = subprocess.check_output([sys.executable, './waf', '--version']).decode()
diff --git a/src/mpv/wscript b/src/mpv/wscript
index d28d18fb..e659b43f 100644
--- a/src/mpv/wscript
+++ b/src/mpv/wscript
@@ -811,15 +811,6 @@ standalone_features = [
 ]
 
 _INSTALL_DIRS_LIST = [
-    ('bindir',  '${PREFIX}/bin',      'binary files'),
-    ('libdir',  '${PREFIX}/lib',      'library files'),
-    ('confdir', '${PREFIX}/etc/mpv',  'configuration files'),
-
-    ('incdir',  '${PREFIX}/include',  'include files'),
-
-    ('datadir', '${PREFIX}/share',    'data files'),
-    ('mandir',  '${DATADIR}/man',     'man pages '),
-    ('docdir',  '${DATADIR}/doc/mpv', 'documentation files'),
     ('zshdir',  '${DATADIR}/zsh/site-functions', 'zsh completion functions'),
 ]
 
@@ -827,6 +818,7 @@ def options(opt):
     opt.load('compiler_c')
     opt.load('waf_customizations')
     opt.load('features')
+    opt.load('gnu_dirs')
 
     group = opt.get_option_group("build and install options")
     for ident, default, desc in _INSTALL_DIRS_LIST:
@@ -897,6 +889,7 @@ def configure(ctx):
     ctx.load('dependencies')
     ctx.load('detections.compiler')
     ctx.load('detections.devices')
+    ctx.load('gnu_dirs')
 
     ctx.parse_dependencies(build_options)
     ctx.parse_dependencies(main_dependencies)
diff --git a/src/mpv/wscript_build.py b/src/mpv/wscript_build.py
index 0d0685b0..04d435a5 100644
--- a/src/mpv/wscript_build.py
+++ b/src/mpv/wscript_build.py
@@ -510,14 +510,14 @@ def build(ctx):
             features     = 'subst',
             PREFIX       = ctx.env.PREFIX,
             LIBDIR       = ctx.env.LIBDIR,
-            INCDIR       = ctx.env.INCDIR,
+            INCDIR       = ctx.env.INCLUDEDIR,
             VERSION      = libversion,
             PRIV_LIBS    = get_deps(),
         )
 
         headers = ["client.h", "qthelper.hpp", "opengl_cb.h"]
         for f in headers:
-            ctx.install_as(ctx.env.INCDIR + '/mpv/' + f, 'libmpv/' + f)
+            ctx.install_as(ctx.env.INCLUDEDIR + '/mpv/' + f, 'libmpv/' + f)
 
         ctx.install_as(ctx.env.LIBDIR + '/pkgconfig/mpv.pc', 'libmpv/mpv.pc')
 
