diff -ru aria2-1.37.0.orig/src/LibnettleMessageDigestImpl.cc aria2-1.37.0/src/LibnettleMessageDigestImpl.cc
--- aria2-1.37.0.orig/src/LibnettleMessageDigestImpl.cc	2026-04-30 23:09:31.135493902 +0200
+++ aria2-1.37.0/src/LibnettleMessageDigestImpl.cc	2026-04-30 23:13:51.968572540 +0200
@@ -36,6 +36,7 @@
 #include "MessageDigestImpl.h"
 
 #include <nettle/nettle-meta.h>
+#include <nettle/version.h>
 
 #include "Adler32MessageDigestImpl.h"
 
@@ -69,7 +70,11 @@
   }
   virtual void digest(unsigned char* md) CXX11_OVERRIDE
   {
+#if NETTLE_VERSION_MAJOR > 3
+    hash->digest(ctx_.get(), md);
+#else
     hash->digest(ctx_.get(), getDigestLength(), md);
+#endif
   }
 
 private:
