diff --git a/build.sbt b/build.sbt
index e73ff967..d3e13746 100644
--- a/build.sbt
+++ b/build.sbt
@@ -2,10 +2,7 @@ def localSnapshotVersion = "0.5.1-SNAPSHOT"
 def isCI = System.getenv("CI") != null
 inThisBuild(
   List(
-    version ~= { dynVer =>
-      if (isCI) dynVer
-      else localSnapshotVersion // only for local publishng
-    },
+    version := "0.8.3",
     scalaVersion := V.scala212,
     crossScalaVersions := List(V.scala212),
     scalacOptions ++= List(
