diff --git a/build.sbt b/build.sbt
index 3b3707f81cc..073967402cf 100644
--- a/build.sbt
+++ b/build.sbt
@@ -41,10 +41,7 @@ ThisBuild / semanticdbVersion := V.semanticdb(scalaVersion.value)
 
 inThisBuild(
   List(
-    version ~= { dynVer =>
-      if (isCI && !isTest) dynVer
-      else localSnapshotVersion // only for local publishing
-    },
+    version := "1.5.2",
     scalaVersion := V.scala213,
     crossScalaVersions := List(V.scala213),
     organization := "org.scalameta",
