diff --git a/src/prediction/zero_query_dict.h b/src/prediction/zero_query_dict.h
index c11aa24d..4eece4eb 100644
--- a/src/prediction/zero_query_dict.h
+++ b/src/prediction/zero_query_dict.h
@@ -132,6 +132,17 @@ class ZeroQueryDict {
       return tmp;
     }
 
+    iterator &operator--() {
+      ptr_ -= kTokenByteSize;
+      return *this;
+    }
+
+    iterator operator--(int) {
+      const iterator tmp(ptr_, string_array_);
+      ptr_ -= kTokenByteSize;
+      return tmp;
+    }
+
     iterator &operator+=(ptrdiff_t n) {
       ptr_ += n * kTokenByteSize;
       return *this;
