--- a/modules/demux/avformat/mux.c
+++ b/modules/demux/avformat/mux.c
@@ -73,10 +73,19 @@
 static void DelStream( sout_mux_t *, sout_input_t * );
 static int Mux      ( sout_mux_t * );
 
+#if LIBAVFORMAT_VERSION_CHECK( 61, 01, 100 )
+static int IOWrite( void *opaque, const uint8_t *buf, int buf_size );
+#else
 static int IOWrite( void *opaque, uint8_t *buf, int buf_size );
+#endif
 static int64_t IOSeek( void *opaque, int64_t offset, int whence );
+#if LIBAVFORMAT_VERSION_CHECK( 61, 01, 100 )
+static int IOWriteTyped(void *opaque, const uint8_t *buf, int buf_size,
+                              enum AVIODataMarkerType type, int64_t time);
+#else
 static int IOWriteTyped(void *opaque, uint8_t *buf, int buf_size,
                               enum AVIODataMarkerType type, int64_t time);
+#endif
 
 /*****************************************************************************
  * Open
@@ -416,8 +425,13 @@
     return VLC_SUCCESS;
 }
 
+#if LIBAVFORMAT_VERSION_CHECK( 61, 01, 100 )
+int IOWriteTyped(void *opaque, const uint8_t *buf, int buf_size,
+                              enum AVIODataMarkerType type, int64_t time)
+#else
 int IOWriteTyped(void *opaque, uint8_t *buf, int buf_size,
                               enum AVIODataMarkerType type, int64_t time)
+#endif
 {
     VLC_UNUSED(time);
 
@@ -512,7 +526,11 @@
 /*****************************************************************************
  * I/O wrappers for libavformat
  *****************************************************************************/
+#if LIBAVFORMAT_VERSION_CHECK( 61, 01, 100 )
+static int IOWrite( void *opaque, const uint8_t *buf, int buf_size )
+#else
 static int IOWrite( void *opaque, uint8_t *buf, int buf_size )
+#endif
 {
     sout_mux_t *p_mux = opaque;
     sout_mux_sys_t *p_sys = p_mux->p_sys;
