--- a/modules/gui/qt/maininterface/compositor_platform.cpp
+++ b/modules/gui/qt/maininterface/compositor_platform.cpp
@@ -16,6 +16,11 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston MA 02110-1301, USA.
  *****************************************************************************/
 
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+#include <vlc_window.h>
+
 #include "compositor_platform.hpp"
 
 #include "maininterface/mainctx.hpp"
