--- a/modules/gui/qt/maininterface/compositor_platform.cpp
+++ b/modules/gui/qt/maininterface/compositor_platform.cpp
@@ -165,7 +165,7 @@
         m_parentWindow->update();
     });
 
-    commonGUICreate(m_parentWindow.get(), nullptr, m_quickView, flags);
+    commonGUICreate(m_parentWindow.get(), m_quickView, flags);
 
     // Synchronize parent window size with video surface:
     connect(m_videoSurfaceProvider.get(),
