diff --git a/src/plugins/sqldrivers/mysql/qsql_mysql.cpp b/src/plugins/sqldrivers/mysql/qsql_mysql.cpp
index a641935dc5..19e6a8286d 100644
--- a/src/plugins/sqldrivers/mysql/qsql_mysql.cpp
+++ b/src/plugins/sqldrivers/mysql/qsql_mysql.cpp
@@ -1355,7 +1355,7 @@ bool QMYSQLDriver::open(const QString& db,
     }
 
 #if MYSQL_VERSION_ID >= 50007
-    if (mysql_get_client_version() >= 50503 && mysql_get_server_version(d->mysql) >= 50503) {
+    if (true) {
         // force the communication to be utf8mb4 (only utf8mb4 supports 4-byte characters)
         mysql_set_character_set(d->mysql, "utf8mb4");
 #if QT_CONFIG(textcodec)
@@ -1371,8 +1371,7 @@ bool QMYSQLDriver::open(const QString& db,
     }
 #endif  // MYSQL_VERSION_ID >= 50007
 
-    d->preparedQuerysEnabled = mysql_get_client_version() >= 40108
-                        && mysql_get_server_version(d->mysql) >= 40100;
+    d->preparedQuerysEnabled = true;
 
 #if QT_CONFIG(thread)
     mysql_thread_init();
