--- a/share/applications/baidunetdisk.desktop	2019-06-14 16:29:25.000000000 +0800
+++ b/share/applications/baidunetdisk.desktop	2019-07-23 19:58:15.427322655 +0800
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Name=baidunetdisk
-Exec=/opt/baidunetdisk/baidunetdisk --no-sandbox %U
+Exec=/usr/lib/baidunetdisk/baidunetdisk --no-sandbox %U
 Terminal=false
 Type=Application
 Icon=baidunetdisk
 StartupWMClass=baidunetdisk
