diff -ura usr/share/applications/baidunetdisk.desktop usr.new/share/applications/baidunetdisk.desktop
--- usr/share/applications/baidunetdisk.desktop	2019-06-14 16:29:25.000000000 +0800
+++ usr.new/share/applications/baidunetdisk.desktop	2019-07-21 20:33:47.630360484 +0800
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Name=baidunetdisk
 Comment=百度网盘
-Exec="/opt/baidunetdisk/baidunetdisk" %U
+Exec=env HOME=$HOME/.local/share/ /usr/bin/baidunetdisk %U
 Terminal=false
 Type=Application
 Icon=baidunetdisk
