diff --git a/data/resources/ui/setup_window.blp b/data/resources/ui/setup_window.blp
index 7f5761f..28fee7b 100644
--- a/data/resources/ui/setup_window.blp
+++ b/data/resources/ui/setup_window.blp
@@ -85,8 +85,6 @@ template $FlSetupWindow: Adw.Dialog {
                 label: _("Primary Device");
                 halign: end;
                 hexpand: true;
-                // Primary device currently not supported, turn it off.
-                sensitive: false;
                 clicked => $handle_decision_to_decision_primary() swapped;
               }
