--- wuala~	2013-07-02 17:34:46.000000000 +0800
+++ wuala	2013-08-06 23:26:57.595665974 +0800
@@ -17,7 +17,7 @@
   fi
   echo $JAVA
   pushd "$(dirname $0)" > /dev/null
-  $JAVA -ea -Xmx$MAX_HEAP_SIZE -Djava.net.preferIPv4Stack=true -jar loader3.jar -installed "$@"
+  $JAVA -ea -Xmx$MAX_HEAP_SIZE -Djava.net.preferIPv4Stack=true -jar /usr/share/java/wuala/loader3.jar -installed "$@"
   popd > /dev/null
 }

