--- configure.orig	2014-08-06 01:37:40.425150800 +0700
+++ configure	2014-08-06 01:46:59.393134946 +0700
@@ -14042,20 +14042,20 @@
 
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for indicator >= 0.3.0" >&5
 $as_echo_n "checking for indicator >= 0.3.0... " >&6; }
-  if $PKG_CONFIG "--atleast-version=0.3.0" "indicator" >/dev/null 2>&1; then
-    INDICATOR_VERSION=`$PKG_CONFIG --modversion "indicator"`
+  if $PKG_CONFIG "--atleast-version=0.3.0" "indicator-0.4" >/dev/null 2>&1; then
+    INDICATOR_VERSION=`$PKG_CONFIG --modversion "indicator304"`
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: $INDICATOR_VERSION" >&5
 $as_echo "$INDICATOR_VERSION" >&6; }
 
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking INDICATOR_CFLAGS" >&5
 $as_echo_n "checking INDICATOR_CFLAGS... " >&6; }
-    INDICATOR_CFLAGS=`$PKG_CONFIG --cflags "indicator"`
+    INDICATOR_CFLAGS=`$PKG_CONFIG --cflags "indicator-0.4"`
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: $INDICATOR_CFLAGS" >&5
 $as_echo "$INDICATOR_CFLAGS" >&6; }
 
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking INDICATOR_LIBS" >&5
 $as_echo_n "checking INDICATOR_LIBS... " >&6; }
-    INDICATOR_LIBS=`$PKG_CONFIG --libs "indicator"`
+    INDICATOR_LIBS=`$PKG_CONFIG --libs "indicator-0.4"`
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: $INDICATOR_LIBS" >&5
 $as_echo "$INDICATOR_LIBS" >&6; }
 
