--- Gruntfile.js	2018-09-17 16:49:19.630048059 -0400
+++ Gruntfile.js	2018-09-17 16:43:27.178277541 -0400
@@ -28,8 +28,6 @@
 const INFO_JSON = 'info.json';

 const LINUX_SETTINGS = {
-  afterInstall: 'bin/deb/after-install.tpl',
-  afterRemove: 'bin/deb/after-remove.tpl',
   category: 'Network',
   desktop: {
     Version: '<%= info.version %>.<%= buildNumber %>',
@@ -39,7 +37,7 @@
     Keywords: 'chat;encrypt;e2e;messenger;videocall',
     StartupWMClass: '<%= info.name %>',
   },
-  fpm: ['--name', 'wire-desktop'],
+  executableName: 'wire-desktop'
 };

 module.exports = function(grunt) {
@@ -200,8 +198,7 @@
         options: {
           arch: grunt.option('arch') || process.arch,
           linux: {
-            ...LINUX_SETTINGS,
-            fpm: ['--name', 'wire-desktop'],
+            ...LINUX_SETTINGS
           },
           targets: [grunt.option('target') || 'dir'],
         },
