From 44d7b2ee6148079d1b8c27e6ba02007c31dd14c7 Mon Sep 17 00:00:00 2001
From: Butui Hu <hot123tea123@gmail.com>
Date: Sun, 22 Sep 2024 16:12:40 +0800
Subject: [PATCH] fix extension deps

Signed-off-by: Butui Hu <hot123tea123@gmail.com>
---
 setup.py | 54 +++++++++++++++++++++++++++++++++++++++++++++---------
 1 file changed, 45 insertions(+), 9 deletions(-)

diff --git a/setup.py b/setup.py
index f0e4f7a..3a2eb9e 100644
--- a/setup.py
+++ b/setup.py
@@ -274,25 +274,61 @@ def customize_build_default(EXTENSIONS, OPTIONS):
     # EXTENSIONS['jpeg8']['sources'] = []  # requires libjpeg-turbo v3
 
     # these extensions are required by core dependent libraries
-    keep = {
-        'shared',
-        'imcd',
+    del EXTENSIONS["apng"]
+    keep = [
+        'aec',
+        'avif',
+        'bcn',
+        'bitshuffle',
+        'blosc',
+        'blosc2',
         'bmp',
+        'brotli',
+        'brunsli',
+        'bz2',
         'cms',
         'deflate',
+        'gif',
+        'h5checksum',
+        'heif',
+        'imcd',
+        # 'jetraw',
         'jpeg2k',
-        'jpeg8',
-        # 'jpegxl',  # requires v0.10
+        # 'jpeg8',
+        'jpegls',
+        'jpegsof3',
+        'jpegxl',
         'jpegxr',
-        'lerc',  # requires v4
+        'lerc',
+        'ljpeg',
         'lz4',
+        'lz4f',
+        'lzf',
+        'lzfse',
+        # 'lzham',
         'lzma',
+        # 'lzo',
+        # 'mozjpeg',
+        # 'pcodec',
+        'pglz',
         'png',
-        'tiff',  # requires v4.6
+        'qoi',
+        'quantize',
+        'rcomp',
+        'rgbe',
+        'shared',
+        'snappy',
+        # 'sperr',
+        'spng',
+        'szip',
+        'tiff',
         'webp',
+        'zfp',
         'zlib',
-        'zstd',
-    }
+        'zlibng',
+        'zopfli',
+        'zstd'
+    ]
     for name in tuple(EXTENSIONS.keys()):
         if name not in keep:
             del EXTENSIONS[name]
-- 
2.46.1

