From d02152619bbd4976563dd67da1fefcf6203597a5 Mon Sep 17 00:00:00 2001
From: Butui Hu <hot123tea123@gmail.com>
Date: Mon, 4 Mar 2024 11:20:53 +0800
Subject: [PATCH] fix extension deps

---
 setup.py | 15 +++++----------
 1 file changed, 5 insertions(+), 10 deletions(-)

diff --git a/setup.py b/setup.py
index e527a75..3ea18db 100644
--- a/setup.py
+++ b/setup.py
@@ -233,24 +233,19 @@ def customize_build_default(EXTENSIONS, OPTIONS):
     import platform
 
     del EXTENSIONS['apng']  # apng patch not commonly available
-    del EXTENSIONS['avif']  # libavif library not commonly available
-    del EXTENSIONS['blosc2']  # c-blosc2 library not commonly available
     # del EXTENSIONS['heif']  # LGPL/GPL
     del EXTENSIONS['jetraw']  # commercial
-    del EXTENSIONS['lerc']  # LERC library not commonly available
     del EXTENSIONS['lz4f']  # requires static linking
-    del EXTENSIONS['lzfse']  # lzfse not commonly available
     del EXTENSIONS['lzham']  # lzham not commonly available
     del EXTENSIONS['lzo']  # lzokay not commonly available
     del EXTENSIONS['mozjpeg']  # Win32 only
     del EXTENSIONS['sperr']  # sperr not commonly available
-    del EXTENSIONS['zlibng']  # zlib-ng library not commonly available
 
-    if 'arch' not in platform.platform():
-        del EXTENSIONS['jpegls']  # CharLS 2.1 library not commonly available
-        del EXTENSIONS['jpegxl']  # jpeg-xl library not commonly available
-        del EXTENSIONS['brunsli']  # Brunsli library not commonly available
-        del EXTENSIONS['zfp']  # ZFP library not commonly available
+    # if 'arch' not in platform.platform():
+    #     del EXTENSIONS['jpegls']  # CharLS 2.1 library not commonly available
+    #     del EXTENSIONS['jpegxl']  # jpeg-xl library not commonly available
+    #     del EXTENSIONS['brunsli']  # Brunsli library not commonly available
+    #     del EXTENSIONS['zfp']  # ZFP library not commonly available
 
     if sys.platform == 'win32':
         EXTENSIONS['bz2']['libraries'] = ['libbz2']
-- 
2.44.0

