diff --git a/dialogs/about.wxg b/dialogs/about.wxg
index 90a7688..a99a8d5 100644
--- a/dialogs/about.wxg
+++ b/dialogs/about.wxg
@@ -30,7 +30,6 @@
                         <object class="wxBoxSizer" name="sizer_5" base="EditBoxSizer">
                             <orient>wxHORIZONTAL</orient>
                             <object class="sizeritem">
-                                <flag>wxALIGN_CENTER_HORIZONTAL</flag>
                                 <border>0</border>
                                 <option>1</option>
                                 <object class="wxStaticText" name="AboutProductNameLabel" base="EditStaticText">
@@ -55,7 +54,6 @@
                         <object class="wxBoxSizer" name="AboutProductSizer" base="EditBoxSizer">
                             <orient>wxHORIZONTAL</orient>
                             <object class="sizeritem">
-                                <flag>wxALIGN_CENTER_HORIZONTAL</flag>
                                 <border>0</border>
                                 <option>1</option>
                                 <object class="wxStaticText" name="AboutProductDescriptionLabel" base="EditStaticText">
@@ -80,7 +78,6 @@
                         <object class="wxBoxSizer" name="AboutMessageSizer" base="EditBoxSizer">
                             <orient>wxHORIZONTAL</orient>
                             <object class="sizeritem">
-                                <flag>wxALIGN_CENTER_HORIZONTAL</flag>
                                 <border>0</border>
                                 <option>2</option>
                                 <object class="wxStaticText" name="AboutMessageLabel" base="EditStaticText">
diff --git a/dialogs/audiocontrols.wxg b/dialogs/audiocontrols.wxg
index 92386fd..929cf13 100644
--- a/dialogs/audiocontrols.wxg
+++ b/dialogs/audiocontrols.wxg
@@ -14,7 +14,7 @@
                     <orient>wxVERTICAL</orient>
                     <label>Mixer</label>
                     <object class="sizeritem">
-                        <flag>wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                        <flag>wxEXPAND</flag>
                         <border>0</border>
                         <option>1</option>
                         <object class="wxPanel" name="AudioChannelPanel" base="EditPanel">
@@ -22,7 +22,7 @@
                             <object class="wxBoxSizer" name="AudioChannelSizer" base="EditBoxSizer">
                                 <orient>wxHORIZONTAL</orient>
                                 <object class="sizeritem">
-                                    <flag>wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                    <flag>wxALIGN_CENTER_VERTICAL</flag>
                                     <border>0</border>
                                     <option>1</option>
                                     <object class="wxStaticText" name="NoAudioText" base="EditStaticText">
@@ -58,7 +58,7 @@
                         <object class="wxBoxSizer" name="SoundIOLabelSizer" base="EditBoxSizer">
                             <orient>wxVERTICAL</orient>
                             <object class="sizeritem">
-                                <flag>wxLEFT|wxRIGHT|wxALIGN_CENTER_VERTICAL</flag>
+                                <flag>wxLEFT|wxRIGHT</flag>
                                 <border>5</border>
                                 <option>1</option>
                                 <object class="wxStaticText" name="MidiInLabel" base="EditStaticText">
@@ -68,7 +68,7 @@
                                 </object>
                             </object>
                             <object class="sizeritem">
-                                <flag>wxLEFT|wxRIGHT|wxALIGN_CENTER_VERTICAL</flag>
+                                <flag>wxLEFT|wxRIGHT</flag>
                                 <border>5</border>
                                 <option>1</option>
                                 <object class="wxStaticText" name="MidiOutLabel" base="EditStaticText">
@@ -78,7 +78,7 @@
                                 </object>
                             </object>
                             <object class="sizeritem">
-                                <flag>wxLEFT|wxRIGHT|wxALIGN_CENTER_VERTICAL</flag>
+                                <flag>wxLEFT|wxRIGHT</flag>
                                 <border>5</border>
                                 <option>1</option>
                                 <object class="wxStaticText" name="SoundInLabel" base="EditStaticText">
@@ -151,7 +151,6 @@
                                 <object class="wxBoxSizer" name="sizer_1" base="EditBoxSizer">
                                     <orient>wxVERTICAL</orient>
                                     <object class="sizeritem">
-                                        <flag>wxALIGN_CENTER_VERTICAL</flag>
                                         <border>0</border>
                                         <option>1</option>
                                         <object class="wxStaticText" name="MidiInFileLabel" base="EditStaticText">
@@ -161,7 +160,6 @@
                                         </object>
                                     </object>
                                     <object class="sizeritem">
-                                        <flag>wxALIGN_CENTER_VERTICAL</flag>
                                         <border>0</border>
                                         <option>1</option>
                                         <object class="wxStaticText" name="MidiOutFileLabel" base="EditStaticText">
@@ -171,7 +169,6 @@
                                         </object>
                                     </object>
                                     <object class="sizeritem">
-                                        <flag>wxALIGN_CENTER_VERTICAL</flag>
                                         <border>0</border>
                                         <option>1</option>
                                         <object class="wxStaticText" name="SampleInFileLabel" base="EditStaticText">
diff --git a/dialogs/catapult.wxg b/dialogs/catapult.wxg
index a2fe2cd..f8885b0 100644
--- a/dialogs/catapult.wxg
+++ b/dialogs/catapult.wxg
@@ -8,7 +8,7 @@
         <object class="wxBoxSizer" name="TopSizer" base="EditBoxSizer">
             <orient>wxVERTICAL</orient>
             <object class="sizeritem">
-                <flag>wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                <flag>wxEXPAND</flag>
                 <border>0</border>
                 <option>1</option>
                 <object class="wxPanel" name="MainWindowPanel" base="EditPanel">
@@ -49,7 +49,7 @@
                                     <object class="wxBoxSizer" name="PowerLedSizer" base="EditBoxSizer">
                                         <orient>wxVERTICAL</orient>
                                         <object class="sizeritem">
-                                            <flag>wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                            <flag>wxALIGN_CENTER_HORIZONTAL</flag>
                                             <border>0</border>
                                             <option>0</option>
                                             <object class="wxStaticBitmap" name="PowerLed" base="EditStaticBitmap">
@@ -58,7 +58,7 @@
                                             </object>
                                         </object>
                                         <object class="sizeritem">
-                                            <flag>wxTOP|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                            <flag>wxTOP|wxALIGN_CENTER_HORIZONTAL</flag>
                                             <border>2</border>
                                             <option>0</option>
                                             <object class="wxStaticText" name="PowerLedLabel" base="EditStaticText">
@@ -75,7 +75,7 @@
                                     <object class="wxBoxSizer" name="CapsLedSizer" base="EditBoxSizer">
                                         <orient>wxVERTICAL</orient>
                                         <object class="sizeritem">
-                                            <flag>wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                            <flag>wxALIGN_CENTER_HORIZONTAL</flag>
                                             <border>0</border>
                                             <option>0</option>
                                             <object class="wxStaticBitmap" name="CapsLockLed" base="EditStaticBitmap">
@@ -84,7 +84,7 @@
                                             </object>
                                         </object>
                                         <object class="sizeritem">
-                                            <flag>wxTOP|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                            <flag>wxTOP|wxALIGN_CENTER_HORIZONTAL</flag>
                                             <border>2</border>
                                             <option>0</option>
                                             <object class="wxStaticText" name="CapsLockLedLabel" base="EditStaticText">
@@ -102,7 +102,7 @@
                                     <object class="wxBoxSizer" name="KanaLedSizer" base="EditBoxSizer">
                                         <orient>wxVERTICAL</orient>
                                         <object class="sizeritem">
-                                            <flag>wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                            <flag>wxALIGN_CENTER_HORIZONTAL</flag>
                                             <border>0</border>
                                             <option>0</option>
                                             <object class="wxStaticBitmap" name="KanaLed" base="EditStaticBitmap">
@@ -111,7 +111,7 @@
                                             </object>
                                         </object>
                                         <object class="sizeritem">
-                                            <flag>wxTOP|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                            <flag>wxTOP|wxALIGN_CENTER_HORIZONTAL</flag>
                                             <border>2</border>
                                             <option>0</option>
                                             <object class="wxStaticText" name="KanaLedLabel" base="EditStaticText">
@@ -128,7 +128,7 @@
                                     <object class="wxBoxSizer" name="PauseLedSizer" base="EditBoxSizer">
                                         <orient>wxVERTICAL</orient>
                                         <object class="sizeritem">
-                                            <flag>wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                            <flag>wxALIGN_CENTER_HORIZONTAL</flag>
                                             <border>0</border>
                                             <option>0</option>
                                             <object class="wxStaticBitmap" name="PauseLed" base="EditStaticBitmap">
@@ -137,7 +137,7 @@
                                             </object>
                                         </object>
                                         <object class="sizeritem">
-                                            <flag>wxTOP|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                            <flag>wxTOP|wxALIGN_CENTER_HORIZONTAL</flag>
                                             <border>2</border>
                                             <option>0</option>
                                             <object class="wxStaticText" name="PauseLedLabel" base="EditStaticText">
@@ -154,7 +154,7 @@
                                     <object class="wxBoxSizer" name="TurboLedSizer" base="EditBoxSizer">
                                         <orient>wxVERTICAL</orient>
                                         <object class="sizeritem">
-                                            <flag>wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                            <flag>wxALIGN_CENTER_HORIZONTAL</flag>
                                             <border>0</border>
                                             <option>0</option>
                                             <object class="wxStaticBitmap" name="TurboLed" base="EditStaticBitmap">
@@ -163,7 +163,7 @@
                                             </object>
                                         </object>
                                         <object class="sizeritem">
-                                            <flag>wxTOP|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                            <flag>wxTOP|wxALIGN_CENTER_HORIZONTAL</flag>
                                             <border>2</border>
                                             <option>0</option>
                                             <object class="wxStaticText" name="TurboLedLabel" base="EditStaticText">
@@ -180,7 +180,7 @@
                                     <object class="wxBoxSizer" name="FDDLedSizer" base="EditBoxSizer">
                                         <orient>wxVERTICAL</orient>
                                         <object class="sizeritem">
-                                            <flag>wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                            <flag>wxALIGN_CENTER_HORIZONTAL</flag>
                                             <border>0</border>
                                             <option>0</option>
                                             <object class="wxStaticBitmap" name="FDDLed" base="EditStaticBitmap">
@@ -189,7 +189,7 @@
                                             </object>
                                         </object>
                                         <object class="sizeritem">
-                                            <flag>wxTOP|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                            <flag>wxTOP|wxALIGN_CENTER_HORIZONTAL</flag>
                                             <border>2</border>
                                             <option>0</option>
                                             <object class="wxStaticText" name="FDDLedLabel" base="EditStaticText">
diff --git a/dialogs/checkconfigs.wxg b/dialogs/checkconfigs.wxg
index c3c2672..ed0251b 100644
--- a/dialogs/checkconfigs.wxg
+++ b/dialogs/checkconfigs.wxg
@@ -191,7 +191,7 @@
                 </object>
             </object>
             <object class="sizeritem">
-                <flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                <flag>wxALL|wxALIGN_CENTER_HORIZONTAL</flag>
                 <border>10</border>
                 <option>0</option>
                 <object class="wxBoxSizer" name="ButtonSizer" base="EditBoxSizer">
diff --git a/dialogs/config.wxg b/dialogs/config.wxg
index 94e3a27..f388b17 100644
--- a/dialogs/config.wxg
+++ b/dialogs/config.wxg
@@ -87,7 +87,7 @@
                 </object>
             </object>
             <object class="sizeritem">
-                <flag>wxEXPAND|wxALIGN_RIGHT</flag>
+                <flag>wxEXPAND</flag>
                 <border>0</border>
                 <option>1</option>
                 <object class="wxBoxSizer" name="sizer_9" base="EditBoxSizer">
diff --git a/dialogs/fullscreen.wxg b/dialogs/fullscreen.wxg
index ad57ab9..f2d55f0 100644
--- a/dialogs/fullscreen.wxg
+++ b/dialogs/fullscreen.wxg
@@ -9,13 +9,13 @@
         <object class="wxBoxSizer" name="sizer_1" base="EditBoxSizer">
             <orient>wxVERTICAL</orient>
             <object class="sizeritem">
-                <flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                <flag>wxALL|wxALIGN_CENTER_HORIZONTAL</flag>
                 <border>10</border>
                 <option>1</option>
                 <object class="wxBoxSizer" name="TextSizer" base="EditBoxSizer">
                     <orient>wxHORIZONTAL</orient>
                     <object class="sizeritem">
-                        <flag>wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                        <flag>wxALIGN_CENTER_VERTICAL</flag>
                         <border>0</border>
                         <option>1</option>
                         <object class="wxStaticText" name="label_1" base="EditStaticText">
@@ -34,7 +34,7 @@
                 </object>
             </object>
             <object class="sizeritem">
-                <flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                <flag>wxALL|wxALIGN_CENTER_HORIZONTAL</flag>
                 <border>10</border>
                 <option>0</option>
                 <object class="wxBoxSizer" name="ButtonSizer" base="EditBoxSizer">
diff --git a/dialogs/ipsselect.wxg b/dialogs/ipsselect.wxg
index c15692f..9c4f387 100644
--- a/dialogs/ipsselect.wxg
+++ b/dialogs/ipsselect.wxg
@@ -69,7 +69,7 @@
                 </object>
             </object>
             <object class="sizeritem">
-                <flag>wxRIGHT|wxBOTTOM|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                <flag>wxRIGHT|wxBOTTOM|wxALIGN_CENTER_HORIZONTAL</flag>
                 <border>5</border>
                 <option>0</option>
                 <object class="wxBoxSizer" name="ButtonSizer" base="EditBoxSizer">
diff --git a/dialogs/misccontrols.wxg b/dialogs/misccontrols.wxg
index d08a3f8..b6f3132 100644
--- a/dialogs/misccontrols.wxg
+++ b/dialogs/misccontrols.wxg
@@ -309,7 +309,7 @@
                 </object>
             </object>
             <object class="sizeritem">
-                <flag>wxALL|wxEXPAND|wxALIGN_RIGHT|wxALIGN_CENTER_HORIZONTAL</flag>
+                <flag>wxALL|wxEXPAND</flag>
                 <border>5</border>
                 <option>0</option>
                 <object class="wxStaticBoxSizer" name="sizer_2" base="EditStaticBoxSizer">
diff --git a/dialogs/romtype.wxg b/dialogs/romtype.wxg
index 4ce4e18..9e65ed9 100644
--- a/dialogs/romtype.wxg
+++ b/dialogs/romtype.wxg
@@ -28,7 +28,7 @@
                 </object>
             </object>
             <object class="sizeritem">
-                <flag>wxRIGHT|wxBOTTOM|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                <flag>wxRIGHT|wxBOTTOM|wxALIGN_CENTER_HORIZONTAL</flag>
                 <border>5</border>
                 <option>0</option>
                 <object class="wxBoxSizer" name="ButtonSizer" base="EditBoxSizer">
diff --git a/dialogs/screenshot.wxg b/dialogs/screenshot.wxg
index b9e8c61..bbae564 100644
--- a/dialogs/screenshot.wxg
+++ b/dialogs/screenshot.wxg
@@ -9,13 +9,13 @@
         <object class="wxBoxSizer" name="sizer_1" base="EditBoxSizer">
             <orient>wxVERTICAL</orient>
             <object class="sizeritem">
-                <flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                <flag>wxALL|wxALIGN_CENTER_HORIZONTAL</flag>
                 <border>10</border>
                 <option>0</option>
                 <object class="wxBoxSizer" name="TextSizer" base="EditBoxSizer">
                     <orient>wxHORIZONTAL</orient>
                     <object class="sizeritem">
-                        <flag>wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                        <flag>wxALIGN_CENTER_VERTICAL</flag>
                         <border>0</border>
                         <option>1</option>
                         <object class="wxStaticText" name="label_1" base="EditStaticText">
@@ -34,7 +34,7 @@
                 </object>
             </object>
             <object class="sizeritem">
-                <flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                <flag>wxALL|wxALIGN_CENTER_HORIZONTAL</flag>
                 <border>10</border>
                 <option>0</option>
                 <object class="wxBoxSizer" name="ButtonSizer" base="EditBoxSizer">
diff --git a/dialogs/session.wxg b/dialogs/session.wxg
index f3e9c48..d30074e 100644
--- a/dialogs/session.wxg
+++ b/dialogs/session.wxg
@@ -371,7 +371,7 @@
                                                         </object>
                                                     </object>
                                                     <object class="sizeritem">
-                                                        <flag>wxLEFT|wxRIGHT|wxALIGN_CENTER_HORIZONTAL</flag>
+                                                        <flag>wxLEFT|wxRIGHT</flag>
                                                         <border>3</border>
                                                         <option>0</option>
                                                         <object class="wxButton" name="RewindButton" base="EditButton">
diff --git a/dialogs/videocontrols.wxg b/dialogs/videocontrols.wxg
index 4b9a57a..01b998b 100644
--- a/dialogs/videocontrols.wxg
+++ b/dialogs/videocontrols.wxg
@@ -20,7 +20,7 @@
                         <object class="wxBoxSizer" name="VideoSwitchSizer" base="EditBoxSizer">
                             <orient>wxHORIZONTAL</orient>
                             <object class="sizeritem">
-                                <flag>wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                <flag>wxEXPAND</flag>
                                 <border>0</border>
                                 <option>0</option>
                                 <object class="wxBoxSizer" name="sizer_18" base="EditBoxSizer">
@@ -71,7 +71,7 @@
                                                 </object>
                                             </object>
                                             <object class="sizeritem">
-                                                <flag>wxTOP|wxBOTTOM|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                                <flag>wxTOP|wxBOTTOM|wxALIGN_CENTER_VERTICAL</flag>
                                                 <border>5</border>
                                                 <option>0</option>
                                                 <object class="wxToggleButton" name="DeInterlaceButton" base="EditToggleButton">
@@ -84,7 +84,7 @@
                                 </object>
                             </object>
                             <object class="sizeritem">
-                                <flag>wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                <flag>wxEXPAND</flag>
                                 <border>0</border>
                                 <option>0</option>
                                 <object class="wxBoxSizer" name="sizer_22" base="EditBoxSizer">
@@ -105,13 +105,13 @@
                                                 </object>
                                             </object>
                                             <object class="sizeritem">
-                                                <flag>wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                                <flag>wxEXPAND</flag>
                                                 <border>5</border>
                                                 <option>1</option>
                                                 <object class="wxBoxSizer" name="ScalerComboSizer" base="EditBoxSizer">
                                                     <orient>wxHORIZONTAL</orient>
                                                     <object class="sizeritem">
-                                                        <flag>wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                                        <flag>wxEXPAND</flag>
                                                         <border>0</border>
                                                         <option>1</option>
                                                         <object class="wxBoxSizer" name="sizer_8" base="EditBoxSizer">
@@ -173,7 +173,7 @@
                                                 </object>
                                             </object>
                                             <object class="sizeritem">
-                                                <flag>wxTOP|wxBOTTOM|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                                <flag>wxTOP|wxBOTTOM|wxALIGN_CENTER_VERTICAL</flag>
                                                 <border>5</border>
                                                 <option>0</option>
                                                 <object class="wxToggleButton" name="LimitSpriteButton" base="EditToggleButton">
@@ -186,7 +186,7 @@
                                 </object>
                             </object>
                             <object class="sizeritem">
-                                <flag>wxEXPAND|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                <flag>wxEXPAND</flag>
                                 <border>0</border>
                                 <option>0</option>
                                 <object class="wxBoxSizer" name="sizer_27" base="EditBoxSizer">
@@ -233,7 +233,7 @@
                                                 </object>
                                             </object>
                                             <object class="sizeritem">
-                                                <flag>wxTOP|wxBOTTOM|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                                <flag>wxTOP|wxBOTTOM|wxALIGN_CENTER_VERTICAL</flag>
                                                 <border>5</border>
                                                 <option>0</option>
                                                 <object class="wxToggleButton" name="FullScreenButton" base="EditToggleButton">
@@ -246,19 +246,18 @@
                                 </object>
                             </object>
                             <object class="sizeritem">
-                                <flag>wxALIGN_CENTER_HORIZONTAL</flag>
                                 <border>0</border>
                                 <option>0</option>
                                 <object class="wxBoxSizer" name="sizer_6" base="EditBoxSizer">
                                     <orient>wxVERTICAL</orient>
                                     <object class="sizeritem">
-                                        <flag>wxALIGN_BOTTOM|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                        <flag>wxALIGN_CENTER_HORIZONTAL</flag>
                                         <border>0</border>
                                         <option>1</option>
                                         <object class="wxBoxSizer" name="sizer_28_copy" base="EditBoxSizer">
                                             <orient>wxHORIZONTAL</orient>
                                             <object class="sizeritem">
-                                                <flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                                <flag>wxALL|wxALIGN_CENTER_VERTICAL</flag>
                                                 <border>5</border>
                                                 <option>0</option>
                                                 <object class="wxStaticText" name="VideoSourceLabel" base="EditStaticText">
@@ -297,7 +296,7 @@
                                                 </object>
                                             </object>
                                             <object class="sizeritem">
-                                                <flag>wxTOP|wxBOTTOM|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+                                                <flag>wxTOP|wxBOTTOM|wxALIGN_CENTER_VERTICAL</flag>
                                                 <border>5</border>
                                                 <option>0</option>
                                                 <object class="wxToggleButton" name="DisableSpritesButton" base="EditToggleButton">

