diff -ura a/browser/lib/newNote.js b/browser/lib/newNote.js
--- a/browser/lib/newNote.js	2018-12-17 06:56:16.000000000 +0100
+++ b/browser/lib/newNote.js	2018-12-18 09:14:38.440189997 +0100
@@ -1,11 +1,8 @@
 import { hashHistory } from 'react-router'
 import dataApi from 'browser/main/lib/dataApi'
 import ee from 'browser/main/lib/eventEmitter'
-import AwsMobileAnalyticsConfig from 'browser/main/lib/AwsMobileAnalyticsConfig'
 
 export function createMarkdownNote (storage, folder, dispatch, location, params, config) {
-  AwsMobileAnalyticsConfig.recordDynamicCustomEvent('ADD_MARKDOWN')
-  AwsMobileAnalyticsConfig.recordDynamicCustomEvent('ADD_ALLNOTE')
 
   let tags = []
   if (config.ui.tagNewNoteWithFilteringTags && location.pathname.match(/\/tags/)) {
@@ -37,8 +34,6 @@
 }
 
 export function createSnippetNote (storage, folder, dispatch, location, params, config) {
-  AwsMobileAnalyticsConfig.recordDynamicCustomEvent('ADD_SNIPPET')
-  AwsMobileAnalyticsConfig.recordDynamicCustomEvent('ADD_ALLNOTE')
 
   let tags = []
   if (config.ui.tagNewNoteWithFilteringTags && location.pathname.match(/\/tags/)) {
diff -ura a/browser/main/Detail/MarkdownNoteDetail.js b/browser/main/Detail/MarkdownNoteDetail.js
--- a/browser/main/Detail/MarkdownNoteDetail.js	2018-12-17 06:56:16.000000000 +0100
+++ b/browser/main/Detail/MarkdownNoteDetail.js	2018-12-18 09:15:07.053824657 +0100
@@ -15,7 +15,6 @@
 import StatusBar from '../StatusBar'
 import _ from 'lodash'
 import { findNoteTitle } from 'browser/lib/findNoteTitle'
-import AwsMobileAnalyticsConfig from 'browser/main/lib/AwsMobileAnalyticsConfig'
 import ConfigManager from 'browser/main/lib/ConfigManager'
 import TrashButton from './TrashButton'
 import FullscreenButton from './FullscreenButton'
@@ -124,7 +123,6 @@
           type: 'UPDATE_NOTE',
           note: note
         })
-        AwsMobileAnalyticsConfig.recordDynamicCustomEvent('EDIT_NOTE')
       })
   }
 
@@ -163,7 +161,6 @@
 
   handleStarButtonClick (e) {
     const { note } = this.state
-    if (!note.isStarred) AwsMobileAnalyticsConfig.recordDynamicCustomEvent('ADD_STAR')
 
     note.isStarred = !note.isStarred
 
diff -ura a/browser/main/Detail/SnippetNoteDetail.js b/browser/main/Detail/SnippetNoteDetail.js
--- a/browser/main/Detail/SnippetNoteDetail.js	2018-12-17 06:56:16.000000000 +0100
+++ b/browser/main/Detail/SnippetNoteDetail.js	2018-12-18 09:15:46.964251990 +0100
@@ -19,7 +19,6 @@
 import _ from 'lodash'
 import {findNoteTitle} from 'browser/lib/findNoteTitle'
 import convertModeName from 'browser/lib/convertModeName'
-import AwsMobileAnalyticsConfig from 'browser/main/lib/AwsMobileAnalyticsConfig'
 import TrashButton from './TrashButton'
 import RestoreButton from './RestoreButton'
 import PermanentDeleteButton from './PermanentDeleteButton'
@@ -140,7 +139,6 @@
           type: 'UPDATE_NOTE',
           note: note
         })
-        AwsMobileAnalyticsConfig.recordDynamicCustomEvent('EDIT_NOTE')
       })
   }
 
@@ -179,7 +177,6 @@
 
   handleStarButtonClick (e) {
     const { note } = this.state
-    if (!note.isStarred) AwsMobileAnalyticsConfig.recordDynamicCustomEvent('ADD_STAR')
 
     note.isStarred = !note.isStarred
 
@@ -375,9 +372,6 @@
     const mode = syntax != null ? syntax.name : null
     if (mode != null) {
       snippets[index].mode = mode
-      AwsMobileAnalyticsConfig.recordDynamicCustomEvent('SNIPPET_LANG', {
-        name: mode
-      })
     }
     this.setState(state => ({note: Object.assign(state.note, {snippets: snippets})}))
 
@@ -399,10 +393,6 @@
       }), () => {
         this.save()
       })
-
-      AwsMobileAnalyticsConfig.recordDynamicCustomEvent('SELECT_LANG', {
-        name
-      })
     }
   }
 
diff -ura a/browser/main/Detail/TagSelect.js b/browser/main/Detail/TagSelect.js
--- a/browser/main/Detail/TagSelect.js	2018-12-17 06:56:16.000000000 +0100
+++ b/browser/main/Detail/TagSelect.js	2018-12-18 09:16:00.454354015 +0100
@@ -3,7 +3,6 @@
 import CSSModules from 'browser/lib/CSSModules'
 import styles from './TagSelect.styl'
 import _ from 'lodash'
-import AwsMobileAnalyticsConfig from 'browser/main/lib/AwsMobileAnalyticsConfig'
 import i18n from 'browser/lib/i18n'
 import ee from 'browser/main/lib/eventEmitter'
 import Autosuggest from 'react-autosuggest'
@@ -27,7 +26,6 @@
   }
 
   addNewTag (newTag) {
-    AwsMobileAnalyticsConfig.recordDynamicCustomEvent('ADD_TAG')
 
     newTag = newTag.trim().replace(/ +/g, '_')
     if (newTag.charAt(0) === '#') {
diff -ura a/browser/main/Main.js b/browser/main/Main.js
--- a/browser/main/Main.js	2018-12-17 06:56:16.000000000 +0100
+++ b/browser/main/Main.js	2018-12-18 09:16:21.980893940 +0100
@@ -10,7 +10,6 @@
 import dataApi from 'browser/main/lib/dataApi'
 import _ from 'lodash'
 import ConfigManager from 'browser/main/lib/ConfigManager'
-import mobileAnalytics from 'browser/main/lib/AwsMobileAnalyticsConfig'
 import eventEmitter from 'browser/main/lib/eventEmitter'
 import { hashHistory } from 'react-router'
 import store from 'browser/main/store'
@@ -25,10 +24,6 @@
   constructor (props) {
     super(props)
 
-    if (process.env.NODE_ENV === 'production') {
-      mobileAnalytics.initAwsMobileAnalytics()
-    }
-
     const { config } = props
 
     this.state = {
diff -ura a/browser/main/modals/CreateFolderModal.js b/browser/main/modals/CreateFolderModal.js
--- a/browser/main/modals/CreateFolderModal.js	2018-12-17 06:56:16.000000000 +0100
+++ b/browser/main/modals/CreateFolderModal.js	2018-12-18 09:16:35.717493677 +0100
@@ -6,7 +6,6 @@
 import store from 'browser/main/store'
 import consts from 'browser/lib/consts'
 import ModalEscButton from 'browser/components/ModalEscButton'
-import AwsMobileAnalyticsConfig from 'browser/main/lib/AwsMobileAnalyticsConfig'
 import i18n from 'browser/lib/i18n'
 
 class CreateFolderModal extends React.Component {
@@ -51,7 +50,6 @@
   }
 
   confirm () {
-    AwsMobileAnalyticsConfig.recordDynamicCustomEvent('ADD_FOLDER')
     if (this.state.name.trim().length > 0) {
       const { storage } = this.props
       const input = {
diff -ura a/browser/main/modals/PreferencesModal/InfoTab.js b/browser/main/modals/PreferencesModal/InfoTab.js
--- a/browser/main/modals/PreferencesModal/InfoTab.js	2018-12-17 06:56:16.000000000 +0100
+++ b/browser/main/modals/PreferencesModal/InfoTab.js	2018-12-18 09:17:14.424024793 +0100
@@ -3,7 +3,6 @@
 import styles from './InfoTab.styl'
 import ConfigManager from 'browser/main/lib/ConfigManager'
 import store from 'browser/main/store'
-import AwsMobileAnalyticsConfig from 'browser/main/lib/AwsMobileAnalyticsConfig'
 import _ from 'lodash'
 import i18n from 'browser/lib/i18n'
 
@@ -25,47 +24,6 @@
     e.preventDefault()
   }
 
-  handleConfigChange (e) {
-    const newConfig = { amaEnabled: this.refs.amaEnabled.checked }
-
-    this.setState({ config: newConfig })
-  }
-
-  handleSaveButtonClick (e) {
-    const newConfig = {
-      amaEnabled: this.state.config.amaEnabled
-    }
-
-    if (!newConfig.amaEnabled) {
-      AwsMobileAnalyticsConfig.recordDynamicCustomEvent('DISABLE_AMA')
-      this.setState({
-        amaMessage: i18n.__('We hope we will gain your trust')
-      })
-    } else {
-      this.setState({
-        amaMessage: i18n.__('Thank\'s for trusting us')
-      })
-    }
-
-    _.debounce(() => {
-      this.setState({
-        amaMessage: ''
-      })
-    }, 3000)()
-
-    ConfigManager.set(newConfig)
-
-    store.dispatch({
-      type: 'SET_CONFIG',
-      config: newConfig
-    })
-  }
-
-  infoMessage () {
-    const { amaMessage } = this.state
-    return amaMessage ? <p styleName='policy-confirm'>{amaMessage}</p> : null
-  }
-
   render () {
     return (
       <div styleName='root'>
@@ -135,23 +93,6 @@
             {i18n.__('License: GPL v3')}
           </li>
         </ul>
-
-        <hr styleName='separate-line' />
-
-        <div styleName='policy'>{i18n.__('Analytics')}</div>
-        <div>{i18n.__('Boostnote collects anonymous data for the sole purpose of improving the application, and strictly does not collect any personal information such the contents of your notes.')}</div>
-        <div>{i18n.__('You can see how it works on ')}<a href='https://github.com/BoostIO/Boostnote' onClick={(e) => this.handleLinkClick(e)}>GitHub</a>.</div>
-        <br />
-        <div>{i18n.__('You can choose to enable or disable this option.')}</div>
-        <input onChange={(e) => this.handleConfigChange(e)}
-          checked={this.state.config.amaEnabled}
-          ref='amaEnabled'
-          type='checkbox'
-        />
-        {i18n.__('Enable analytics to help improve Boostnote')}<br />
-        <button styleName='policy-submit' onClick={(e) => this.handleSaveButtonClick(e)}>{i18n.__('Save')}</button>
-        <br />
-        {this.infoMessage()}
       </div>
     )
   }
diff -ura a/browser/main/NoteList/index.js b/browser/main/NoteList/index.js
--- a/browser/main/NoteList/index.js	2018-12-17 06:56:16.000000000 +0100
+++ b/browser/main/NoteList/index.js	2018-12-18 09:17:36.037313386 +0100
@@ -17,7 +17,6 @@
 import path from 'path'
 import { hashHistory } from 'react-router'
 import copy from 'copy-to-clipboard'
-import AwsMobileAnalyticsConfig from 'browser/main/lib/AwsMobileAnalyticsConfig'
 import Markdown from '../../lib/markdown'
 import i18n from 'browser/lib/i18n'
 import { confirmDeleteNote } from 'browser/lib/confirmDeleteNote'
@@ -686,7 +685,6 @@
             note: newNote
           })
         })
-        AwsMobileAnalyticsConfig.recordDynamicCustomEvent('EDIT_NOTE')
       })
       .catch((err) => {
         console.error('Notes could not go to trash: ' + err)
@@ -704,8 +702,6 @@
     const firstNote = selectedNotes[0]
     const eventName = firstNote.type === 'MARKDOWN_NOTE' ? 'ADD_MARKDOWN' : 'ADD_SNIPPET'
 
-    AwsMobileAnalyticsConfig.recordDynamicCustomEvent(eventName)
-    AwsMobileAnalyticsConfig.recordDynamicCustomEvent('ADD_ALLNOTE')
     dataApi
       .createNote(storage.key, {
         type: firstNote.type,
diff -ura a/package.json b/package.json
--- a/package.json	2018-12-17 06:56:16.000000000 +0100
+++ b/package.json	2018-12-18 09:17:47.943964022 +0100
@@ -52,8 +52,6 @@
     "@rokt33r/markdown-it-math": "^4.0.1",
     "@rokt33r/season": "^5.3.0",
     "@susisu/mte-kernel": "^2.0.0",
-    "aws-sdk": "^2.48.0",
-    "aws-sdk-mobile-analytics": "^0.9.2",
     "chart.js": "^2.7.2",
     "codemirror": "^5.40.2",
     "codemirror-mode-elixir": "^1.1.1",
