diff --unified --recursive --text hyprland-test/src/hyprland/subprojects/wlroots/render/gles2/renderer.c hyprland-git/src/hyprland/subprojects/wlroots/render/gles2/renderer.c
--- render/gles2/renderer.c	2022-10-23 11:18:56.598893482 +1000
+++ render/gles2/renderer.c	2022-10-23 11:16:25.923036282 +1000
@@ -159,7 +159,7 @@
 		assert(wlr_egl_is_current(renderer->egl));
 
 		push_gles2_debug(renderer);
-		glFlush();
+		glFinish();
 		glBindFramebuffer(GL_FRAMEBUFFER, 0);
 		pop_gles2_debug(renderer);
