--- ./packages/flutter_tools/gradle/build.gradle.kts.orig
+++ ./packages/flutter_tools/gradle/build.gradle.kts
@@ -7,6 +7,8 @@
     `groovy`
 }

+project.buildDir = file(project.gradle.gradleUserHomeDir.path + "/flutter/build")
+
 group = "dev.flutter.plugin"
 version = "1.0.0"

--- ./packages/flutter_tools/lib/src/android/gradle.dart.orig
+++ ./packages/flutter_tools/lib/src/android/gradle.dart
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.

+import 'dart:io' as io show Platform;
 import 'dart:math';

 import 'package:crypto/crypto.dart';
@@ -361,6 +362,10 @@
       "Running Gradle task '$assembleTask'...",
     );

+
+    final gradleUserHome = io.Platform.environment['GRADLE_USER_HOME'] ??
+        io.Platform.environment['HOME']! + '/.gradle';
+    command.add('--project-cache-dir=$gradleUserHome/flutter/.gradle');
     if (_logger.isVerbose) {
       command.add('--full-stacktrace');
       command.add('--info');
