diff --git a/src/ui/events-list-view.cpp b/src/ui/events-list-view.cpp
index b1d04abb..09536b9e 100644
--- a/src/ui/events-list-view.cpp
+++ b/src/ui/events-list-view.cpp
@@ -1,4 +1,5 @@
 #include <QPainter>
+#include <QPainterPath>
 #include <QApplication>
 #include <QPixmap>
 #include <QToolTip>
diff --git a/src/ui/private-share-dialog.cpp b/src/ui/private-share-dialog.cpp
index 511d476b..284cc043 100644
--- a/src/ui/private-share-dialog.cpp
+++ b/src/ui/private-share-dialog.cpp
@@ -2,6 +2,7 @@
 #include <QCompleter>
 #include <QLineEdit>
 #include <QPainter>
+#include <QPainterPath>
 #include <QResizeEvent>
 #include <QStringList>
 #include <QStringListModel>
