=== modified file 'usr/share/byobu/profiles/bashrc'
--- usr/share/byobu/profiles/bashrc	2018-08-12 16:32:34 +0000
+++ usr/share/byobu/profiles/bashrc	2019-01-23 22:09:04 +0000
@@ -54,7 +54,7 @@
 			;;
 			*)
 				# Use Googley colors (blue / red / yellow / blue / green / red )
-                                PS1="${debian_chroot:+($debian_chroot)}\[\e[31m\]\$(byobu_prompt_status)\[\e[38;5;69m\]\u\[\e[38;5;214m\]@\[\e[38;5;167m\]\h\[\e[38;5;214m\]:\[\e[38;5;71m\]\w\[\e[38;5;214m\]\$(byobu_prompt_symbol)\e[00m\] "
+				PS1="${debian_chroot:+($debian_chroot)}\[\e[31m\]\$(byobu_prompt_status)\[\e[38;5;69m\]\u\[\e[38;5;214m\]@\[\e[38;5;167m\]\h\[\e[38;5;214m\]:\[\e[38;5;71m\]\w\[\e[38;5;214m\]\$(byobu_prompt_symbol)\[\e[00m\] "
 			;;
 		esac
 	fi

