--- dropbox.py	2011-04-04 20:32:01.000000000 +0200
+++ dropbox.py	2011-04-28 22:55:17.976623103 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright 2008 Evenflow, Inc.
 #
@@ -610,6 +610,8 @@
 
 def start_dropbox():
     db_path = os.path.expanduser(u"~/.dropbox-dist/dropboxd").encode(sys.getfilesystemencoding())
+    if not os.path.exists(db_path):
+        db_path = u"/usr/bin/dropbox"
     if os.access(db_path, os.X_OK):
         f = open("/dev/null", "w")
         # we don't reap the child because we're gonna die anyway, let init do it
