diff --git a/vpython b/vpython
index 9befaa690..03ebe525b 100755
--- a/vpython
+++ b/vpython
@@ -4,7 +4,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-if [[ $VPYTHON_BYPASS == "manually managed python not supported by chrome operations" ]]
+if [[ 1 ]]
 then
   NEWARGS=()
   while [[ $# -gt 0 ]]
@@ -30,7 +30,7 @@ then
         ;;
     esac
   done
-  exec "python" "${NEWARGS[@]}"
+  exec "python2" "${NEWARGS[@]}"
 fi
 
 
