diff -ura applet-window-buttons-0.8.0/libappletdecoration/decorationpalette.cpp applet-window-buttons-0.8.0.new/libappletdecoration/decorationpalette.cpp
--- applet-window-buttons-0.8.0/libappletdecoration/decorationpalette.cpp	2020-01-23 23:52:27.000000000 +0800
+++ applet-window-buttons-0.8.0.new/libappletdecoration/decorationpalette.cpp	2020-02-12 16:55:11.292318038 +0800
@@ -136,7 +136,13 @@
     m_activeTitleBarColor     = wmConfig.readEntry("activeBackground", m_palette.color(QPalette::Active, QPalette::Highlight));
     m_inactiveTitleBarColor   = wmConfig.readEntry("inactiveBackground", m_inactiveFrameColor);
     m_activeForegroundColor   = wmConfig.readEntry("activeForeground", m_palette.color(QPalette::Active, QPalette::HighlightedText));
+    
+#if QT_VERSION >= QT_VERSION_CHECK(5,13,0)
+    m_inactiveForegroundColor = wmConfig.readEntry("inactiveForeground", m_activeForegroundColor.darker());
+#else
     m_inactiveForegroundColor = wmConfig.readEntry("inactiveForeground", m_activeForegroundColor.dark());
+    m_inactiveForegroundColor = wmConfig.readEntry("inactiveForeground", m_activeForegroundColor.dark());
+#endif
 
     KConfigGroup windowColorsConfig(config, QStringLiteral("Colors:Window"));
     m_warningForegroundColor = windowColorsConfig.readEntry("ForegroundNegative", QColor(237, 21, 2));
diff -ura applet-window-buttons-0.8.0/libappletdecoration/previewclient.cpp applet-window-buttons-0.8.0.new/libappletdecoration/previewclient.cpp
--- applet-window-buttons-0.8.0/libappletdecoration/previewclient.cpp	2020-01-23 23:52:27.000000000 +0800
+++ applet-window-buttons-0.8.0.new/libappletdecoration/previewclient.cpp	2020-02-12 16:55:50.035131369 +0800
@@ -396,6 +396,12 @@
 }
 #endif
 
+#if KDECORATION2_VERSION_MINOR >= 18
+QSize PreviewClient::size() const
+{
+    return {m_width, m_height};
+}
+#endif
 
 void PreviewClient::requestClose()
 {
diff -ura applet-window-buttons-0.8.0/libappletdecoration/previewclient.h applet-window-buttons-0.8.0.new/libappletdecoration/previewclient.h
--- applet-window-buttons-0.8.0/libappletdecoration/previewclient.h	2020-01-23 23:52:27.000000000 +0800
+++ applet-window-buttons-0.8.0.new/libappletdecoration/previewclient.h	2020-02-12 16:56:14.078144742 +0800
@@ -107,6 +107,10 @@
     void requestHideToolTip() override;
 #endif
 
+#if KDECORATION2_VERSION_MINOR >= 18
+    QSize size() const override;
+#endif
+
     void requestClose() override;
     void requestContextHelp() override;
     void requestToggleMaximization(Qt::MouseButtons buttons) override;
