--- a/plugins/Satellites/src/Satellite.cpp	2022-09-21 20:07:05.000000000 +0800
+++ b/plugins/Satellites/src/Satellite.cpp	2022-10-02 11:31:58.215784717 +0800
@@ -251,8 +251,8 @@
 	map["stdMag"] = stdMag;
 	map["rcs"] = RCS;
 	map["status"] = status;
-	map["tle1"] = tleElements.first.data();
-	map["tle2"] = tleElements.second.data();
+	map["tle1"] = QString::fromUtf8(tleElements.first.data());
+	map["tle2"] = QString::fromUtf8(tleElements.second.data());
 
 	if (!description.isEmpty())
 		map["description"] = description;
