From 972c6ba72f575964fbf2049a22d51b4d1fd3983c Mon Sep 17 00:00:00 2001
From: "Alexander V. Wolf" <alex.v.wolf@gmail.com>
Date: Sun, 23 Mar 2025 22:40:28 +0700
Subject: use system md4c

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fef6b6e07ea48..0c569f7b7843f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,12 +2,7 @@ CMAKE_MINIMUM_REQUIRED(VERSION 3.16)
 # CMake 3.20+ is required if x-compiling for Windows on ARM
 MESSAGE(STATUS "Found CMake ${CMAKE_VERSION}")
 
+CMAKE_POLICY(VERSION 3.16)
-# Fix behavior of CMAKE_CXX_STANDARD and CMAKE_C_STANDARD when targeting macOS.
-CMAKE_POLICY(SET CMP0025 NEW)
-# Potential dangerous comparison of variables. Details: https://cmake.org/cmake/help/v3.1/policy/CMP0054.html
-CMAKE_POLICY(SET CMP0054 NEW)
-# Let AUTOMOC and AUTOUIC process GENERATED files.
-CMAKE_POLICY(SET CMP0071 NEW)
 # CMake Policies introduced after version CMAKE_MINIMUM_REQUIRED
 IF(POLICY CMP0175)
      CMAKE_POLICY(SET CMP0175 OLD)
@@ -827,31 +827,18 @@ ELSE()
 ENDIF()
 
 IF(ENABLE_XLSX)
-     # TODO: switch to something like vcpkg or conan, and plain find_package,
-     # without the fallback of cmake downloading the dep itself.
-     # But some libraries are not available there yet.
-
-     # Version 1.4.4 exports "QXlsx", but future (unreleased as of 10 Nov 2022)
-     # version exports "QXlsxQt5" (or ...Qt6) instead. Try both, but
-     # download newer one, because 1.4.4 doesn't link on windows when
-     # added via add_subdirectory().
-     FIND_PACKAGE(QXlsx NAMES QXlsxQt${QT_VERSION_MAJOR} QXlsx)
-     IF(QXlsx_FOUND)
-         MESSAGE(STATUS "Using system-provided QXlsx ${QXlsx_VERSION}")
-     ELSE()
-         CPMAddPackage(NAME QXlsxQt${QT_VERSION_MAJOR}
-             URL https://github.com/QtExcel/QXlsx/archive/refs/tags/v1.5.0.tar.gz
-             URL_HASH SHA256=4461bb0fc9b88b6a970520e8b32eb7d2fc391a1a5c326b139a0af836278cdb09
-             EXCLUDE_FROM_ALL ON
-             SOURCE_SUBDIR QXlsx)
-     ENDIF()
+     CPMFindPackage(NAME QXlsxQt${QT_VERSION_MAJOR}
+          URL https://github.com/QtExcel/QXlsx/archive/refs/tags/v1.5.0.tar.gz
+          URL_HASH SHA256=4461bb0fc9b88b6a970520e8b32eb7d2fc391a1a5c326b139a0af836278cdb09
+          EXCLUDE_FROM_ALL ON
+          SOURCE_SUBDIR QXlsx)
 ENDIF()
 
-CPMAddPackage(NAME md4c
-              URL https://github.com/mity/md4c/archive/refs/tags/release-0.4.8.tar.gz
-              URL_HASH SHA256=4a457df853425b6bb6e3457aa1d1a13bccec587a04c38c622b1013a0da41439f
-              EXCLUDE_FROM_ALL yes
-              OPTIONS "BUILD_MD2HTML_EXECUTABLE OFF")
+CPMFindPackage(NAME md4c
+     URL https://github.com/mity/md4c/archive/refs/tags/release-0.4.8.tar.gz
+     URL_HASH SHA256=4a457df853425b6bb6e3457aa1d1a13bccec587a04c38c622b1013a0da41439f
+     EXCLUDE_FROM_ALL yes
+     OPTIONS "BUILD_MD2HTML_EXECUTABLE OFF")
 
 ########### Set some global variables ###########
 IF(UNIX AND NOT WIN32)

