diff -ruN GDCM-3.0.24/CMakeLists.txt GDCM-3.0.24.new/CMakeLists.txt
--- GDCM-3.0.24/CMakeLists.txt	2024-05-03 20:12:32.000000000 +0800
+++ GDCM-3.0.24.new/CMakeLists.txt	2024-12-26 09:44:02.983130719 +0800
@@ -702,6 +702,7 @@
         LICENSE_DESTINATION   "${GDCM_INSTALL_DATA_DIR}/vtkgdcm-${GDCM_SHORT_VERSION}"
         HIERARCHY_DESTINATION "${GDCM_INSTALL_LIB_DIR}/vtk${vtk_version_suffix}/hierarchy/vtkgdcm"
         LIBRARY_NAME_SUFFIX   "${vtkgdcm_library_suffix}"
+	SPDX_DESTINATION      "${GDCM_INSTALL_DATA_DIR}/doc/${CMAKE_PROJECT_NAME}/spdx"
         VERSION               "${GDCM_VERSION}"
         SOVERSION             "1"
         # TODO: these are probably not set as they should be
@@ -724,6 +725,7 @@
           MODULE_DESTINATION  "${GDCM_VTK_INSTALL_PYTHONMODULE_DIR}"
           SOABI               "${Python${VTK_PYTHON_VERSION}_SOABI}"
           BUILD_STATIC        OFF
+	  INSTALL_HEADERS     OFF
         )
 
         file(GENERATE
