commit 06947f61cbf3b65f62187e139f068cc077c38fec
Author: Butui Hu <hot123tea123@gmail.com>
Date:   Mon Apr 27 01:46:10 2020 +0800

    fix building with poppler 1.83 or newer

diff --git a/Applications/Cxx/gdcminfo.cxx b/Applications/Cxx/gdcminfo.cxx
index bef80248e..64af39966 100644
--- a/Applications/Cxx/gdcminfo.cxx
+++ b/Applications/Cxx/gdcminfo.cxx
@@ -511,9 +511,17 @@ static int ProcessOneFile( std::string const & filename, gdcm::Defs const & defs
 
     UnicodeMap *uMap;
 #ifdef LIBPOPPLER_GLOBALPARAMS_CSTOR_HAS_PARAM
-    globalParams = new GlobalParams(0);
+  #if POPPLER_MAJOR_VERSION >= 0 || POPPLER_MINOR_VERSION >= 83
+    globalParams.reset(new GlobalParams(0));
+  #else
+      globalParams = new GlobalParams(0);
+  #endif
 #else
+  #if POPPLER_MAJOR_VERSION >= 0 || POPPLER_MINOR_VERSION >= 83
+    globalParams.reset(new GlobalParams(0));
+  #else
     globalParams = new GlobalParams();
+  #endif
 #endif
     uMap = globalParams->getTextEncoding();
 
diff --git a/Applications/Cxx/gdcmpdf.cxx b/Applications/Cxx/gdcmpdf.cxx
index 4d97d81e0..203b998f3 100644
--- a/Applications/Cxx/gdcmpdf.cxx
+++ b/Applications/Cxx/gdcmpdf.cxx
@@ -337,9 +337,17 @@ int main (int argc, char *argv[])
   ownerPW = NULL;
   userPW = NULL;
 #ifdef LIBPOPPLER_GLOBALPARAMS_CSTOR_HAS_PARAM
-  globalParams = new GlobalParams(0);
+#if POPPLER_MAJOR_VERSION >= 0 || POPPLER_MINOR_VERSION >= 83
+globalParams.reset(new GlobalParams(0));
+#else
+globalParams = new GlobalParams(0);
+#endif
+#else
+#if POPPLER_MAJOR_VERSION >= 0 || POPPLER_MINOR_VERSION >= 83
+  globalParams.reset(new GlobalParams());
 #else
   globalParams = new GlobalParams();
+#endif
 #endif
   uMap = globalParams->getTextEncoding();
 
