diff -ruN GDCM-3.0.0/CMake/FindCharLS.cmake GDCM-3.0.0-fix/CMake/FindCharLS.cmake
--- GDCM-3.0.0/CMake/FindCharLS.cmake	2019-05-03 21:21:40.000000000 +0800
+++ GDCM-3.0.0-fix/CMake/FindCharLS.cmake	2019-07-01 20:57:42.805615640 +0800
@@ -6,7 +6,7 @@
 #  For details see the accompanying COPYING-CMAKE-SCRIPTS file.
 #
 
-find_path(CHARLS_INCLUDE_DIR CharLS/interface.h
+find_path(CHARLS_INCLUDE_DIR CharLS/charls.h
 /usr/local/include
 /usr/include
 )
diff -ruN GDCM-3.0.0/Source/MediaStorageAndFileFormat/gdcmJPEGLSCodec.cxx GDCM-3.0.0-fix/Source/MediaStorageAndFileFormat/gdcmJPEGLSCodec.cxx
--- GDCM-3.0.0/Source/MediaStorageAndFileFormat/gdcmJPEGLSCodec.cxx	2019-05-03 21:21:40.000000000 +0800
+++ GDCM-3.0.0-fix/Source/MediaStorageAndFileFormat/gdcmJPEGLSCodec.cxx	2019-07-01 20:59:54.907104444 +0800
@@ -22,6 +22,11 @@
 // CharLS includes
 #include "gdcm_charls.h"
 
+#include <cstring>
+
+using BYTE=unsigned char;
+#define ILV_LINE charls::InterleaveMode::Line;
+#define ILV_NONE charls::InterleaveMode::None;
 #if defined(__GNUC__) && GCC_VERSION < 50101
 #pragma GCC diagnostic ignored "-Wmissing-field-initializers"
 #endif
diff -ruN GDCM-3.0.0/Utilities/gdcm_charls.h GDCM-3.0.0-fix/Utilities/gdcm_charls.h
--- GDCM-3.0.0/Utilities/gdcm_charls.h	2019-05-03 21:21:40.000000000 +0800
+++ GDCM-3.0.0-fix/Utilities/gdcm_charls.h	2019-07-01 20:54:25.531690271 +0800
@@ -19,20 +19,8 @@
 #ifdef GDCM_USE_SYSTEM_CHARLS
 // It is expected that version 2.0.0 is used
 # include <CharLS/charls.h>
-# include <CharLS/publictypes.h>
-# include <CharLS/util.h>
-# include <CharLS/defaulttraits.h>
-# include <CharLS/losslesstraits.h>
-# include <CharLS/colortransform.h>
-# include <CharLS/processline.h>
 #else
 #include "gdcmcharls/charls.h"
-#include "gdcmcharls/publictypes.h"
-#include "gdcmcharls/util.h"
-#include "gdcmcharls/defaulttraits.h"
-#include "gdcmcharls/losslesstraits.h"
-#include "gdcmcharls/colortransform.h"
-#include "gdcmcharls/processline.h"
 #endif
 
 #endif
