--- winusb-1.0.11/src/findFile.cpp.orig	2014-01-21 15:11:07.698652278 +0100
+++ winusb-1.0.11/src/findFile.cpp	2014-01-21 15:11:29.388444590 +0100
@@ -37,14 +37,14 @@
     }
 
     #if defined(__UNIX__)
-    testPath = wxString(wxStandardPaths().GetInstallPrefix() + _T("/share/") + _T(PACKAGE) + _T("/") + str);
+    testPath = wxString(wxStandardPaths::Get().GetInstallPrefix() + _T("/share/") + _T(PACKAGE) + _T("/") + str);
     if(wxFileExists(testPath) || wxDirExists(testPath)) // Si c'est dans le dossier /usr/.../share/nomprog
     {
         return testPath;
     }
     #endif // defined(__UNIX__)
 
-    testPath = wxString(wxStandardPaths().GetDataDir() + _T("/") + str);
+    testPath = wxString(wxStandardPaths::Get().GetDataDir() + _T("/") + str);
     if(wxFileExists(testPath) || wxDirExists(testPath)) // Si c'est dans le dossier /usr/.../share/nomPaquet
     {
         return testPath;
