diff --git a/scripts/addons_core/io_scene_gltf2/io/com/draco.py b/scripts/addons_core/io_scene_gltf2/io/com/draco.py
index 75e23162c67..a386463a2a9 100644
--- a/scripts/addons_core/io_scene_gltf2/io/com/draco.py
+++ b/scripts/addons_core/io_scene_gltf2/io/com/draco.py
@@ -6,6 +6,7 @@ import os
 import sys
 from pathlib import Path
 import bpy
+import site
 
 
 def find_draco_dll_in_module(library_name: str) -> Path:
@@ -35,7 +36,7 @@ def dll_path() -> Path:
     python_lib = Path('{v[0]}.{v[1]}/python/lib'.format(v=bpy.app.version))
     python_version = 'python{v[0]}.{v[1]}'.format(v=sys.version_info)
 
-    path = os.environ.get('BLENDER_EXTERN_DRACO_LIBRARY_PATH')
+    path = f"{site.getsitepackages()[0]}/libextern_draco.so"
     if path is not None:
         return Path(path)
 
