From 57b7ad2e6a642cd425c5ca548a7ad2487bcde356 Mon Sep 17 00:00:00 2001
From: Butui Hu <hot123tea123@gmail.com>
Date: Mon, 25 Mar 2024 21:35:23 +0800
Subject: [PATCH] fix building with zlib 1.3.1

---
 pyreadr/zlib.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/pyreadr/zlib.h b/pyreadr/zlib.h
index 3e0c767..b2a8702 100644
--- a/pyreadr/zlib.h
+++ b/pyreadr/zlib.h
@@ -1346,7 +1346,7 @@ ZEXTERN int ZEXPORT gzwrite OF((gzFile file,
    error.
 */
 
-ZEXTERN int ZEXPORTVA gzprintf Z_ARG((gzFile file, const char *format, ...));
+ZEXTERN int ZEXPORTVA gzprintf(gzFile file, const char *format, ...);
 /*
      Converts, formats, and writes the arguments to the compressed file under
    control of the format string, as in fprintf.  gzprintf returns the number of
@@ -1755,9 +1755,9 @@ ZEXTERN gzFile         ZEXPORT gzopen_w OF((const wchar_t *path,
 #endif
 #if defined(STDC) || defined(Z_HAVE_STDARG_H)
 #  ifndef Z_SOLO
-ZEXTERN int            ZEXPORTVA gzvprintf Z_ARG((gzFile file,
+ZEXTERN int            ZEXPORTVA gzvprintf(gzFile file,
                                                   const char *format,
-                                                  va_list va));
+                                                  va_list va);
 #  endif
 #endif
 
-- 
2.44.0

