diff --git a/misc/dde-daemon/gesture.json b/misc/dde-daemon/gesture.json
index 73a3d7ee..fbcc8fd8 100644
--- a/misc/dde-daemon/gesture.json
+++ b/misc/dde-daemon/gesture.json
@@ -53,15 +53,6 @@
             "Action": "ShowWindow"
         }
     },
-    {
-        "Name": "tap",
-        "Direction": "none",
-        "Fingers": 3,
-        "Action": {
-            "Type": "built-in",
-            "Action": "MoveWindow"
-        }
-    },
     {
         "Name": "swipe",
         "Direction": "up",
diff --git a/system/gesture/core.c b/system/gesture/core.c
index f953cc1c..4a49749e 100644
--- a/system/gesture/core.c
+++ b/system/gesture/core.c
@@ -305,7 +305,7 @@ handle_gesture_events(struct libinput_event *ev, int type)
         }
         raw_event_reset(raw);
         break;
-    case LIBINPUT_EVENT_GESTURE_TAP_BEGIN:
+    /*case LIBINPUT_EVENT_GESTURE_TAP_BEGIN:
         break;
     case LIBINPUT_EVENT_GESTURE_TAP_END:
         if (libinput_event_gesture_get_cancelled(gesture)) {
@@ -314,7 +314,7 @@ handle_gesture_events(struct libinput_event *ev, int type)
         raw->fingers = libinput_event_gesture_get_finger_count(gesture);
         g_debug("[Tap] fingers: %d", raw->fingers);
         handleGestureEvent(GESTURE_TYPE_TAP, GESTURE_DIRECTION_NONE, raw->fingers);
-        break;
+        break;*/
     }
 }
 
@@ -433,9 +433,9 @@ handle_events(struct libinput *li)
         case LIBINPUT_EVENT_GESTURE_SWIPE_BEGIN:
         case LIBINPUT_EVENT_GESTURE_SWIPE_UPDATE:
         case LIBINPUT_EVENT_GESTURE_SWIPE_END:
-        case LIBINPUT_EVENT_GESTURE_TAP_BEGIN:
+        /*case LIBINPUT_EVENT_GESTURE_TAP_BEGIN:
         case LIBINPUT_EVENT_GESTURE_TAP_UPDATE:
-        case LIBINPUT_EVENT_GESTURE_TAP_END:{
+        case LIBINPUT_EVENT_GESTURE_TAP_END:*/{
             handle_gesture_events(ev, type);
             break;
         }
