From 28cd8826346cd7554a13cc0166844b8d9fbdd598 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Wed, 20 Jan 2016 07:45:16 -0500
Subject: [PATCH] Do not set stack size on linux

---
 driver/others/memory.c | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/driver/others/memory.c b/driver/others/memory.c
index af7bd81..655e578 100644
--- a/driver/others/memory.c
+++ b/driver/others/memory.c
@@ -1363,18 +1363,6 @@ void CONSTRUCTOR gotoblas_init(void) {
    gotoblas_memory_init();
 #endif
 
-#if defined(OS_LINUX)
-   struct rlimit curlimit;
-   if ( getrlimit(RLIMIT_STACK, &curlimit ) == 0 )
-   {
-	if ( curlimit.rlim_cur != curlimit.rlim_max )
-	{
-		curlimit.rlim_cur = curlimit.rlim_max;
-		setrlimit(RLIMIT_STACK, &curlimit);
-	}
-   }
-#endif
-
 #ifdef SMP
   if (blas_cpu_number == 0) blas_get_cpu_number();
 #ifdef SMP_SERVER
-- 
2.6.4

