From 0f213788aff74b0ae55a8689ea3ee18be64e798e Mon Sep 17 00:00:00 2001
From: edward-p <edward@edward-p.xyz>
Date: Mon, 12 Dec 2022 16:57:00 +0800
Subject: [PATCH] fix build error on linux-6.1

---
 xt_FULLCONENAT.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/xt_FULLCONENAT.c b/xt_FULLCONENAT.c
index f96cfd8..4f6e24b 100644
--- a/xt_FULLCONENAT.c
+++ b/xt_FULLCONENAT.c
@@ -325,7 +325,7 @@ static uint16_t find_appropriate_port6(struct net *net, const u16 zone, const ui
     /* for now we do the same thing for both --random and --random-fully */
 
     /* select a random starting point */
-    start = (uint16_t)(prandom_u32() % (u32)range_size);
+    start = (uint16_t)(get_random_u32() % (u32)range_size);
   } else {
 
     if ((original_port >= min && original_port <= min + range_size - 1)
@@ -995,7 +995,7 @@ static uint16_t find_appropriate_port(struct net *net, const u16 zone, const uin
     /* for now we do the same thing for both --random and --random-fully */
 
     /* select a random starting point */
-    start = (uint16_t)(prandom_u32() % (u32)range_size);
+    start = (uint16_t)(get_random_u32() % (u32)range_size);
   } else {
 
     if ((original_port >= min && original_port <= min + range_size - 1)
-- 
2.38.1

