--- a/ntfs_fs.h
+++ b/ntfs_fs.h
@@ -836,7 +836,7 @@
 
 /* globals from xattr.c */
 #ifdef CONFIG_NTFS3_FS_POSIX_ACL
-struct posix_acl *ntfs_get_acl(struct inode *inode, int type, bool rcu);
+struct posix_acl *ntfs_get_acl(struct inode *inode, int type);
 int ntfs_set_acl(struct user_namespace *mnt_userns, struct inode *inode,
 		 struct posix_acl *acl, int type);
 int ntfs_init_acl(struct user_namespace *mnt_userns, struct inode *inode,

--- a/xattr.c
+++ b/xattr.c
@@ -530,11 +530,8 @@
 /*
  * ntfs_get_acl - inode_operations::get_acl
  */
-struct posix_acl *ntfs_get_acl(struct inode *inode, int type, bool rcu)
+struct posix_acl *ntfs_get_acl(struct inode *inode, int type)
 {
-	if (rcu)
-		return ERR_PTR(-ECHILD);
-
 	/* TODO: init_user_ns? */
 	return ntfs_get_acl_ex(&init_user_ns, inode, type, 0);
 }
