diff --git a/librawstudio/rawstudio.h b/librawstudio/rawstudio.h
index 14dd42fe..4b58d9f6 100644
--- a/librawstudio/rawstudio.h
+++ b/librawstudio/rawstudio.h
@@ -20,10 +20,6 @@
 #ifndef RAWSTUDIO_H
 #define RAWSTUDIO_H
 
-#ifdef  __cplusplus
-extern "C" {
-#endif
-
 #include <sys/types.h>
 #include "rs-types.h"
 
@@ -78,8 +74,4 @@ extern "C" {
 
 #include "x86-cpu.h"
 
-#ifdef  __cplusplus
-} /* extern "c" */
-#endif
-
 #endif /* RAWSTUDIO_H */
