diff --git a/PKGBUILD b/PKGBUILD
index bef1907..56f82db 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -3,11 +3,11 @@
 
 pkgbase=gtk3
 pkgname=(
-  gtk3
-  gtk3-demos
-  gtk3-docs
+  gtk3-lily
+  # gtk3-demos
+  # gtk3-docs
 )
-pkgdesc="GObject-based multi-platform GUI toolkit"
+pkgdesc="GObject-based multi-platform GUI toolkit, with lilydjwg's patches"
 url="https://www.gtk.org/"
 arch=(x86_64)
 license=(LGPL-2.0-only)
@@ -62,11 +62,13 @@ makedepends=(
 _commit=f926323d04db220ba2ba9a9026a5bf54e81fbb39  # tags/3.24.40^0
 source=(
   "git+https://gitlab.gnome.org/GNOME/gtk.git#commit=$_commit"
+  0001-wayland-im-notify-wayland-after-set_cursor_location.patch
   gtk-query-immodules-3.0.hook
   6786.patch
 )
 b2sums=('SKIP'
         '8e6a3906126749c6d853f582e3802254cdbba099c6af7190ad576eff6ea5425404a72b1b36950a87e3afdac82295cfe246003172c3e0341a73bd931a36f3b407'
+        '2718db7adc07ee2df86b6dacbfb7638217b2cb76c5f5acde4ce9bd7a4617989744326069670b6cc4e5aff59fac410db9b60db7f07dd2a2e36a40b7a3b0d2d8b9'
         '137f4c03dd88c02f6171ab40e0c040d9c08aa5c2ee76260742ded966962b64ef4988ccb46a8261178ca1c88e0a5c2815e319d9f8e3af5c6b0a7d22eb3243b322')
 
 pkgver() {
@@ -80,6 +82,7 @@ prepare() {
   # https://gitlab.archlinux.org/archlinux/packaging/packages/gtk3/-/issues/5
   # https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/6786
   git apply -3 ../6786.patch
+  patch -Np1 < ../0001-wayland-im-notify-wayland-after-set_cursor_location.patch
 }
 
 build() {
@@ -108,16 +111,17 @@ _pick() {
   done
 }
 
-package_gtk3() {
+package_gtk3-lily() {
   depends+=(gtk-update-icon-cache)
   optdepends=('evince: Default print preview command')
   provides=(
+    gtk3=$pkgver
     gtk3-print-backends
     libgailutil-3.so
     libgdk-3.so
     libgtk-3.so
   )
-  conflicts=(gtk3-print-backends)
+  conflicts=(gtk3-print-backends gtk3)
   replaces=("gtk3-print-backends<=3.22.26-1")
   install=gtk3.install
 
