--- plasma-workspace/xembed-sni-proxy/CMakeLists.txt	2017-06-07 15:40:05.139090630 -0700
+++ plasma-workspace.new/xembed-sni-proxy/CMakeLists.txt	2017-06-07 15:38:23.591731403 -0700
@@ -1,3 +1,37 @@
+cmake_minimum_required(VERSION 2.8.12 FATAL_ERROR)
+
+project(xembedsniproxy)
+
+set(QT_MIN_VERSION "5.7.0")
+set(KF5_MIN_VERSION "5.32.0")
+find_package(ECM 1.8.0 REQUIRED NO_MODULE)
+include(FeatureSummary)
+include(WriteBasicConfigVersionFile)
+include(GenerateExportHeader)
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${ECM_MODULE_PATH} ${ECM_KDE_MODULE_DIR})
+
+find_package (X11 REQUIRED)
+
+find_package(Qt5 ${QT_MIN_VERSION} CONFIG REQUIRED COMPONENTS
+    Core
+    DBus
+    Gui
+    X11Extras
+)
+
+include(KDEInstallDirs)
+include(KDECMakeSettings)
+include(KDECompilerSettings NO_POLICY_SCOPE)
+
+include(ECMInstallIcons)
+include(ECMOptionalAddSubdirectory)
+include(ECMQtDeclareLoggingCategory)
+
+# required frameworks by Core
+find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS
+    WindowSystem
+)
+
 add_definitions(-DQT_NO_CAST_TO_ASCII
 -DQT_NO_CAST_FROM_ASCII
 -DQT_NO_URL_CAST_FROM_STRING
