--- snx-rs.orig/assets/snx-rs-gui.desktop
+++ snx-rs/assets/snx-rs-gui.desktop
@@ -51,7 +51,7 @@ Comment[ru_RU]=Клиент VPN Check Point
 Comment[sk_SK]=Klient VPN Check Point
 Comment[sv_SE]=Check Point VPN-klient
 Categories=Network;
-Exec=/opt/snx-rs/snx-rs-gui
+Exec=/usr/bin/snx-rs-gui
 Icon=network-vpn
 Keywords=network;vpn;
 Keywords[cs_CZ]=síť;vpn;
@@ -70,7 +70,7 @@ Keywords[pt_PT]=rede;vpn;
 Keywords[ru_RU]=сеть;vpn;
 Keywords[sk_SK]=sieť;vpn;
 Keywords[sv_SE]=nätverk;vpn;
-TryExec=/opt/snx-rs/snx-rs-gui
+TryExec=/usr/bin/snx-rs-gui
 Type=Application
 Actions=connect;disconnect;settings;status

@@ -92,7 +92,7 @@ Name[pt_PT]=Ligar
 Name[ru_RU]=Подключиться
 Name[sk_SK]=Pripojiť
 Name[sv_SE]=Anslut
-Exec=/opt/snx-rs/snx-rs-gui -m connect
+Exec=/usr/bin/snx-rs-gui -m connect

 [Desktop Action disconnect]
 Name=Disconnect
@@ -112,7 +112,7 @@ Name[pt_PT]=Desligar
 Name[ru_RU]=Отключиться
 Name[sk_SK]=Odpojiť
 Name[sv_SE]=Koppla från
-Exec=/opt/snx-rs/snx-rs-gui -m disconnect
+Exec=/usr/bin/snx-rs-gui -m disconnect

 [Desktop Action settings]
 Name=Settings
@@ -132,7 +132,7 @@ Name[pt_PT]=Definições
 Name[ru_RU]=Настройки
 Name[sk_SK]=Nastavenia
 Name[sv_SE]=Inställningar
-Exec=/opt/snx-rs/snx-rs-gui -m settings
+Exec=/usr/bin/snx-rs-gui -m settings

 [Desktop Action status]
 Name=Connection information
@@ -152,4 +152,4 @@ Name[pt_PT]=Informações de ligação
 Name[ru_RU]=Информация о соединении
 Name[sk_SK]=Informácie o pripojení
 Name[sv_SE]=Anslutningsinformation
-Exec=/opt/snx-rs/snx-rs-gui -m status
+Exec=/usr/bin/snx-rs-gui -m status
