diff --git a/src/player/PlayerComponent.cpp b/src/player/PlayerComponent.cpp
index bf55aab..bb0d4a3 100644
--- a/src/player/PlayerComponent.cpp
+++ b/src/player/PlayerComponent.cpp
@@ -318,6 +318,10 @@ void PlayerComponent::queueMedia(const QString& url, const QVariantMap& options,
   command << "loadfile" << qurl.toString(QUrl::FullyEncoded);
   command << "append-play"; // if nothing is playing, play it now, otherwise just enqueue it
 
+#if MPV_CLIENT_API_VERSION >= MPV_MAKE_VERSION(2, 3)
+  command << -1; // insert_at_idx
+#endif
+
   QVariantMap extraArgs;
 
   quint64 startMilliseconds = options["startMilliseconds"].toLongLong();
