--- a/toolkit/moz.configure	2019-03-13 22:33:54.000000000 +0100
+++ b/toolkit/moz.configure	2019-03-19 10:18:15.769460797 +0100
@@ -856,9 +856,9 @@
     # x86 and x86_64. It's meant to work on 32-bit ARM, too,
     # but currently does not.
     if target.cpu in ('aarch64', 'x86', 'x86_64') and value:
-        if rustc_info and rustc_info.version >= Version('1.33.0'):
-            die('--enable-rust-simd does not work with Rust 1.33 or later. '
-                'See https://bugzilla.mozilla.org/show_bug.cgi?id=1521249 .')
+        #if rustc_info and rustc_info.version >= Version('1.33.0'):
+        #    die('--enable-rust-simd does not work with Rust 1.33 or later. '
+        #        'See https://bugzilla.mozilla.org/show_bug.cgi?id=1521249 .')
         return True
 
 set_config('MOZ_RUST_SIMD', rust_simd)
