--- a/src-tauri/src/main.rs
+++ b/src-tauri/src/main.rs
@@ -106,10 +106,6 @@
         tauri::RunEvent::ExitRequested { api, .. } => {
             api.prevent_exit();
         }
-        tauri::RunEvent::Updater(tauri::UpdaterEvent::Downloaded) => {
-            resolve::resolve_reset();
-            api::process::kill_children();
-        }
         tauri::RunEvent::WindowEvent { label, event, .. } => {
             if label == "main" {
                 match event {
