diff --git a/src/hardwareintegration/compositor/wayland-egl/waylandeglclientbufferintegration.cpp b/src/hardwareintegration/compositor/wayland-egl/waylandeglclientbufferintegration.cpp
index c8819ba1..33395b4e 100644
--- a/src/hardwareintegration/compositor/wayland-egl/waylandeglclientbufferintegration.cpp
+++ b/src/hardwareintegration/compositor/wayland-egl/waylandeglclientbufferintegration.cpp
@@ -169,14 +169,10 @@ public:
 
     QEGLStreamConvenience *funcs = nullptr;
     static WaylandEglClientBufferIntegrationPrivate *get(WaylandEglClientBufferIntegration *integration) {
-        return shuttingDown ? nullptr : integration->d_ptr.data();
+        return integration ? integration->d_ptr.data() : nullptr;
     }
-
-    static bool shuttingDown;
 };
 
-bool WaylandEglClientBufferIntegrationPrivate::shuttingDown = false;
-
 WaylandEglClientBufferIntegrationPrivate::WaylandEglClientBufferIntegrationPrivate()
 {
 }
@@ -417,7 +413,6 @@ WaylandEglClientBufferIntegration::WaylandEglClientBufferIntegration()
 
 WaylandEglClientBufferIntegration::~WaylandEglClientBufferIntegration()
 {
-    WaylandEglClientBufferIntegrationPrivate::shuttingDown = true;
     Q_D(WaylandEglClientBufferIntegration);
     if (d->egl_unbind_wayland_display && d->display_bound) {
         Q_ASSERT(d->wlDisplay);
diff --git a/src/hardwareintegration/compositor/wayland-egl/waylandeglclientbufferintegration_p.h b/src/hardwareintegration/compositor/wayland-egl/waylandeglclientbufferintegration_p.h
index ca4dd373..9c5fe8f6 100644
--- a/src/hardwareintegration/compositor/wayland-egl/waylandeglclientbufferintegration_p.h
+++ b/src/hardwareintegration/compositor/wayland-egl/waylandeglclientbufferintegration_p.h
@@ -17,13 +17,14 @@
 
 #include <QtWaylandCompositor/private/qwlclientbufferintegration_p.h>
 #include <QtCore/QScopedPointer>
+#include <QtCore/QPointer>
 #include <QtWaylandCompositor/private/qwlclientbuffer_p.h>
 
 QT_BEGIN_NAMESPACE
 
 class WaylandEglClientBufferIntegrationPrivate;
 
-class Q_WAYLANDCOMPOSITOR_EXPORT WaylandEglClientBufferIntegration : public QtWayland::ClientBufferIntegration
+class Q_WAYLANDCOMPOSITOR_EXPORT WaylandEglClientBufferIntegration : public QObject, public QtWayland::ClientBufferIntegration
 {
     Q_DECLARE_PRIVATE(WaylandEglClientBufferIntegration)
 public:
@@ -61,7 +62,7 @@ private:
     friend class WaylandEglClientBufferIntegrationPrivate;
 
     BufferState *d = nullptr;
-    WaylandEglClientBufferIntegration *m_integration = nullptr;
+    QPointer<WaylandEglClientBufferIntegration> m_integration;
 };
 
 QT_END_NAMESPACE
