From dc26645316a1996a10995d9f5fae53ca1caddade Mon Sep 17 00:00:00 2001
From: Matt Dallmeyer <mattdallmeyer@gmail.com>
Date: Fri, 29 Sep 2023 14:23:15 -0700
Subject: [PATCH] Update .clang-format

---
 .clang-format | 1 -
 1 file changed, 1 deletion(-)

diff --git a/.clang-format b/.clang-format
index 1be83906..e8a5e85a 100644
--- a/.clang-format
+++ b/.clang-format
@@ -34,7 +34,6 @@ DerivePointerAlignment: false
 DisableFormat: false
 FixNamespaceComments: true
 ForEachMacros: []
-IndentCaseLabels: false
 IncludeCategories:
   - Regex: '^("|<)stdafx\.h(pp)?("|>)'
     Priority: -1
