diff --git a/pypy/goal/targetpypystandalone.py b/pypy/goal/targetpypystandalone.py
index 77b714d..0be54de 100644
--- a/pypy/goal/targetpypystandalone.py
+++ b/pypy/goal/targetpypystandalone.py
@@ -57,7 +57,7 @@ def create_entry_point(space, w_dict):
                 space.call_function(w_run_toplevel, w_call_startup_gateway)
                 if rlocale.HAVE_LANGINFO:
                     try:
-                        rlocale.setlocale(rlocale.LC_ALL, '')
+                        rlocale.setlocale(rlocale.LC_CTYPE, '')
                     except rlocale.LocaleError:
                         pass
                 w_executable = space.fsdecode(space.wrapbytes(argv[0]))
