From b1f35de35e1e73f84550697be079c30f4caed9ab Mon Sep 17 00:00:00 2001
From: Edward Pacman <edward@edward-p.xyz>
Date: Tue, 24 Oct 2023 16:39:46 +0800
Subject: [PATCH] fix AppGuard on qt-6.6

Signed-off-by: Edward Pacman <edward@edward-p.xyz>
---
 core/AppGuard.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/core/AppGuard.cpp b/core/AppGuard.cpp
index 26ab9c6..bfa4f24 100644
--- a/core/AppGuard.cpp
+++ b/core/AppGuard.cpp
@@ -55,11 +55,13 @@ bool AppGuard::isAnotherRunning(void)
         return false;
 
     memLock.acquire();
-    const bool isRunning = sharedMem.attach();
+
+    QSharedMemory _sharedMem_1( sharedmemKey );
+    const bool isRunning = _sharedMem_1.attach();
 
     if ( isRunning )
     {
-        sharedMem.detach();
+        _sharedMem_1.detach();
     }
 
     memLock.release();
-- 
2.42.0

