From d76bc9a2efd17d574f8c707203e0ed21b1f267b2 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Tue, 16 Sep 2025 23:20:38 -0400
Subject: [PATCH] Fix compilation

---
 src/jlgen.jl | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/jlgen.jl b/src/jlgen.jl
index 1bff8a0..96a3301 100644
--- a/src/jlgen.jl
+++ b/src/jlgen.jl
@@ -589,7 +589,11 @@ function ci_cache_populate(interp, cache, mi, min_world, max_world)
             # now make sure everything has source code, if desired
             mi = CC.get_ci_mi(callee)
             if CC.use_const_api(callee)
-                src = CC.codeinfo_for_const(interp, mi, ci.rettype_const)
+                @static if VERSION >= v"1.13.0-DEV.1131"
+                    src = CC.codeinfo_for_const(interp, mi, CC.WorldRange(min_world, max_world), Core.svec(), ci.rettype_const)
+                else
+                    src = CC.codeinfo_for_const(interp, mi, ci.rettype_const)
+                end
             else
                 # TODO: typeinf_code could return something with different edges/ages/owner/abi (needing an update to callee), which we don't handle here
                 src = CC.typeinf_code(interp, mi, true)
-- 
2.51.0

