From 198dd985fd0003f1e918bea81a8859afbb0f6c1e Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Tue, 9 Dec 2025 09:34:07 -0500
Subject: [PATCH] Fix loading

---
 src/jlgen.jl | 18 ++++++++++++++++--
 1 file changed, 16 insertions(+), 2 deletions(-)

diff --git a/src/jlgen.jl b/src/jlgen.jl
index 330cf7f..db831fe 100644
--- a/src/jlgen.jl
+++ b/src/jlgen.jl
@@ -523,7 +523,11 @@ end
 
 
 ## world view of the cache
-using Core.Compiler: WorldView
+if !isdefined(CC, :WorldView)
+    using Core.Compiler: InternalCodeCache
+else
+    using Core.Compiler: WorldView
+end
 
 if !HAS_INTEGRATED_CACHE
 
@@ -645,7 +649,17 @@ function ci_cache_populate(interp, cache, mi, min_world, max_world)
 end
 
 function ci_cache_lookup(cache, mi, min_world, max_world)
-    wvc = WorldView(cache, min_world, max_world)
+    @static if !isdefined(CC, :WorldView)
+        # Use the overlay cache if possible
+        globalcache = cache.globalcache
+        cacheworlds = globalcache.worlds
+        if cacheworlds.min_world == min_world && cacheworlds.min_world == max_world
+            return CC.get(cache, mi, nothing)
+        end
+        wvc = InternalCodeCache(globalcache.owner, min_world, max_world)
+    else
+        wvc = WorldView(cache, min_world, max_world)
+    end
     ci = CC.get(wvc, mi, nothing)
     if VERSION < v"1.12.0-DEV.1434" && ci !== nothing && ci.inferred === nothing
         # if for some reason we did end up with a codeinfo without inferred source, e.g.,
-- 
2.51.2

