From 715d2247a969e18bf1346fd2b575b0a2d5bfe792 Mon Sep 17 00:00:00 2001
From: Nicholas F <nfrederick01@protonmail.com>
Date: Tue, 28 Mar 2023 22:01:57 -0400
Subject: [PATCH] Allow user to enter fullscreen with F11

---
 src/gui/MainWindow.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gui/MainWindow.cpp b/src/gui/MainWindow.cpp
index 342a1adc3..40830547f 100644
--- a/src/gui/MainWindow.cpp
+++ b/src/gui/MainWindow.cpp
@@ -1418,7 +1418,7 @@ void MainWindow::OnKeyUp(wxKeyEvent& event)
 	const auto code = event.GetKeyCode();
 	if (code == WXK_ESCAPE)
 		SetFullScreen(false);
-	else if (code == WXK_RETURN && event.AltDown())
+	else if (code == WXK_RETURN && event.AltDown() || code == WXK_F11)
 		SetFullScreen(!IsFullScreen());
 	else if (code == WXK_F12)
 		g_window_info.has_screenshot_request = true; // async screenshot request
