diff -Naur gcc-4.9-20140604-old/gcc/graphite-clast-to-gimple.c gcc-4.9-20140604/gcc/graphite-clast-to-gimple.c
--- gcc-4.9-20140604-old/gcc/graphite-clast-to-gimple.c	2014-03-03 21:39:22.000000000 +1000
+++ gcc-4.9-20140604/gcc/graphite-clast-to-gimple.c	2014-06-25 15:07:57.958697105 +1000
@@ -28,6 +28,8 @@
 #include <isl/constraint.h>
 #include <isl/ilp.h>
 #include <isl/aff.h>
+#include <isl/deprecated/int.h>
+#include <isl/deprecated/ilp_int.h>
 #include <cloog/cloog.h>
 #include <cloog/isl/domain.h>
 #endif
diff -Naur gcc-4.9-20140604-old/gcc/graphite-interchange.c gcc-4.9-20140604/gcc/graphite-interchange.c
--- gcc-4.9-20140604-old/gcc/graphite-interchange.c	2014-01-03 08:23:26.000000000 +1000
+++ gcc-4.9-20140604/gcc/graphite-interchange.c	2014-06-25 15:10:06.882899243 +1000
@@ -29,6 +29,9 @@
 #include <isl/map.h>
 #include <isl/union_map.h>
 #include <isl/ilp.h>
+#include <isl/deprecated/int.h>
+#include <isl/deprecated/ilp_int.h>
+#include <isl/deprecated/constraint_int.h>
 #include <cloog/cloog.h>
 #include <cloog/isl/domain.h>
 #endif
diff -Naur gcc-4.9-20140604-old/gcc/graphite-optimize-isl.c gcc-4.9-20140604/gcc/graphite-optimize-isl.c
--- gcc-4.9-20140604-old/gcc/graphite-optimize-isl.c	2014-01-03 08:23:26.000000000 +1000
+++ gcc-4.9-20140604/gcc/graphite-optimize-isl.c	2014-06-25 15:16:57.038386166 +1000
@@ -28,6 +28,8 @@
 #include <isl/band.h>
 #include <isl/aff.h>
 #include <isl/options.h>
+#include <isl/deprecated/int.h>
+#include <isl/deprecated/aff_int.h>
 #endif
 
 #include "system.h"
@@ -373,7 +375,7 @@
 	{
 	  for (i = ScheduleDimensions - 1 ;  i >= 0 ; i--)
 	    {
-	      if (isl_band_member_is_zero_distance (Band, i))
+	      if (isl_band_member_is_coincident (Band, i))
 		{
 		  isl_map *TileMap;
 		  isl_union_map *TileUMap;
diff -Naur gcc-4.9-20140604-old/gcc/graphite-poly.c gcc-4.9-20140604/gcc/graphite-poly.c
--- gcc-4.9-20140604-old/gcc/graphite-poly.c	2014-01-03 08:23:26.000000000 +1000
+++ gcc-4.9-20140604/gcc/graphite-poly.c	2014-06-25 15:18:01.207157796 +1000
@@ -28,6 +28,8 @@
 #include <isl/constraint.h>
 #include <isl/ilp.h>
 #include <isl/aff.h>
+#include <isl/deprecated/int.h>
+#include <isl/deprecated/ilp_int.h>
 #include <cloog/cloog.h>
 #include <cloog/isl/domain.h>
 #endif
diff -Naur gcc-4.9-20140604-old/gcc/graphite-sese-to-poly.c gcc-4.9-20140604/gcc/graphite-sese-to-poly.c
--- gcc-4.9-20140604-old/gcc/graphite-sese-to-poly.c	2014-04-08 20:59:40.000000000 +1000
+++ gcc-4.9-20140604/gcc/graphite-sese-to-poly.c	2014-06-25 15:19:46.575140398 +1000
@@ -26,6 +26,9 @@
 #include <isl/union_map.h>
 #include <isl/constraint.h>
 #include <isl/aff.h>
+#include <isl/deprecated/int.h>
+#include <isl/deprecated/aff_int.h>
+#include <isl/deprecated/constraint_int.h>
 #include <cloog/cloog.h>
 #include <cloog/cloog.h>
 #include <cloog/isl/domain.h>
