--- i/src/anbox/network/base_socket_messenger.cpp
+++ w/src/anbox/network/base_socket_messenger.cpp
@@ -138,7 +138,7 @@ unsigned short BaseSocketMessenger<stream_protocol>::local_port() const {
 
 template <typename stream_protocol>
 void BaseSocketMessenger<stream_protocol>::set_no_delay() {
-  const auto fd = socket->native();
+  const auto fd = socket->native_handle();
   int flag = 1;
   const auto ret =
       ::setsockopt(fd, IPPROTO_TCP, TCP_NODELAY,
--- i/src/anbox/runtime.cpp
+++ w/src/anbox/runtime.cpp
@@ -53,7 +53,7 @@ std::shared_ptr<Runtime> Runtime::create(std::uint32_t pool_size) {
 
 Runtime::Runtime(std::uint32_t pool_size)
     : pool_size_{pool_size},
-      service_{pool_size_},
+      service_{static_cast<int>(pool_size_)},
       strand_{service_},
       keep_alive_{service_} {}
