From 61e1a639f4700dcd3f006a31ea8e0102b8015b0f Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Tue, 20 Feb 2024 00:57:56 -0500
Subject: [PATCH] Use pointer instead of unsafe_convert

---
 src/utils/simd.jl | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/utils/simd.jl b/src/utils/simd.jl
index dbb0d57b..2a0fbc8f 100644
--- a/src/utils/simd.jl
+++ b/src/utils/simd.jl
@@ -135,8 +135,8 @@ end
                 ($textir, "entry"),
                 Bool,
                 Tuple{Ptr{T}, Ptr{T}, Int64},
-                Base.unsafe_convert(Ptr{T}, a) + sizeof(T) * offset,
-                Base.unsafe_convert(Ptr{T}, b) + sizeof(T) * offset,
+                pointer(a) + sizeof(T) * offset,
+                pointer(b) + sizeof(T) * offset,
                 length(a) - offset
             )
         end
@@ -235,8 +235,8 @@ end
                 ($textir, "entry"),
                 Bool,
                 Tuple{Ptr{T}, Ptr{T}, Int64},
-                Base.unsafe_convert(Ptr{T}, a) + sizeof(T) * offset,
-                Base.unsafe_convert(Ptr{T}, b) + sizeof(T) * offset,
+                pointer(a) + sizeof(T) * offset,
+                pointer(b) + sizeof(T) * offset,
                 length(a) - offset
             )
         end
@@ -349,8 +349,8 @@ end
                 ($textir, "entry"),
                 Bool,
                 Tuple{Ptr{T}, Ptr{T}, Int64},
-                Base.unsafe_convert(Ptr{T}, a) + sizeof(T) * offset,
-                Base.unsafe_convert(Ptr{T}, b) + sizeof(T) * offset,
+                pointer(a) + sizeof(T) * offset,
+                pointer(b) + sizeof(T) * offset,
                 length(a) - offset
             )
         end
@@ -467,8 +467,8 @@ end
                 ($textir, "entry"),
                 Bool,
                 Tuple{Ptr{T}, Ptr{T}, Int64},
-                Base.unsafe_convert(Ptr{T}, a) + sizeof(T) * offset,
-                Base.unsafe_convert(Ptr{T}, b) + sizeof(T) * offset,
+                pointer(a) + sizeof(T) * offset,
+                pointer(b) + sizeof(T) * offset,
                 length(a) - offset
             )
         end
-- 
2.42.1

