From e03fa158a09e3d70401f3be76a0c0e8cda52d18c Mon Sep 17 00:00:00 2001
From: bgme <i@bgme.me>
Date: Tue, 31 Dec 2024 13:15:54 +0800
Subject: [PATCH] fix build

---
 functions.mk | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/functions.mk b/functions.mk
index 67ac96c..447a084 100644
--- a/functions.mk
+++ b/functions.mk
@@ -49,7 +49,7 @@ define gobuild
 	CGO_CFLAGS='-O2 -g -gdwarf-4 -I$(CURDIR)/lib/libpcap/' \
 	CGO_LDFLAGS='-O2 -g -L$(CURDIR)/lib/libpcap/ -lpcap -static' \
 	GOOS=linux GOARCH=$(GOARCH) CC=$(CMD_CC_PREFIX)$(CMD_CC) \
-	$(CMD_GO) build -tags '$(TARGET_TAG),netgo' -ldflags "-w -s -X 'github.com/gojue/ecapture/cli/cmd.GitVersion=$(TARGET_TAG)_$(GOARCH):$(VERSION_NUM):$(VERSION_FLAG)' -X 'github.com/gojue/ecapture/cli/cmd.ByteCodeFiles=$(BYTECODE_FILES)' -linkmode=external -extldflags -static " -o $(OUT_BIN)
+	$(CMD_GO) build -trimpath -buildmode=pie -mod=readonly -modcacherw -tags '$(TARGET_TAG),netgo' -ldflags "-w -X 'github.com/gojue/ecapture/cli/cmd.GitVersion=$(TARGET_TAG)_$(GOARCH):$(VERSION_NUM):$(VERSION_FLAG)' -X 'github.com/gojue/ecapture/cli/cmd.ByteCodeFiles=$(BYTECODE_FILES)' -linkmode=external -extldflags -static " -o $(OUT_BIN)
 	$(CMD_FILE) $(OUT_BIN)
 endef
 
@@ -73,4 +73,4 @@ define release_tar
 	$(CMD_CP) README_CN.md $(TAR_DIR)/README_CN.md
 	$(CMD_CP) $(OUTPUT_DIR)/ecapture $(TAR_DIR)/ecapture
 	$(CMD_TAR) -czf $(OUT_ARCHIVE) $(TAR_DIR)
-endef
\ No newline at end of file
+endef
-- 
2.47.1

