diff --git a/tests/test_stdin.py b/tests/test_stdin.py
index 212b309..c765a00 100644
--- a/tests/test_stdin.py
+++ b/tests/test_stdin.py
@@ -1,7 +1,15 @@
 """Tests for polyfill's stdin monkey patching."""
 import flake8
-import pep8
-import pycodestyle
+try:
+    import pep8
+except ImportError:
+    pep8 = None
+
+try:
+    import pycodestyle
+except ImportError:
+    pycodestyle = None
+
 import pytest
 
 from flake8_polyfill import stdin
