From 0f544debdf6255e542ab422e9355db4405e97a8a Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Sat, 20 Sep 2014 12:13:37 -0400
Subject: [PATCH 1/2] fix for pch

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 94b0fc4be..88f0f77b1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -598,6 +598,7 @@ execute_process(
     COMMAND ${CMAKE_COMMAND} -E copy ${PROJECT_SOURCE_DIR}/all.h ${PROJECT_BINARY_DIR}/all.h
     WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
     )
+set(BUILD_PCH false)
 
 precompiled_header(QT_INCLUDES all ${BUILD_PCH})
 
-- 
2.16.1

