From 06503911daadca6c5b02a56b30e067ffce2d2e6e Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Fri, 19 Apr 2013 22:29:13 -0400
Subject: [PATCH 1/2] fix for pch

---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0b7463b..dc892e9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -297,9 +297,11 @@ add_custom_command(
     WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
     )
 
-precompiled_header(QT_INCLUDES all)
+# precompiled_header(QT_INCLUDES all)
 ADD_CUSTOM_TARGET(mops1 DEPENDS ${PROJECT_BINARY_DIR}/all.h)
 ADD_CUSTOM_TARGET(mops2 DEPENDS ${PCH})
+set(PCH_HEADER "${PROJECT_BINARY_DIR}/all.h")
+set(PCH_INCLUDE "-include ${PCH_HEADER}")
 
 ##
 ## add subdirs
-- 
1.8.2.1

