From c878fe8a211f8ef8f2d8145edb02cf7fe5453cd0 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Sat, 20 Apr 2013 00:12:03 -0400
Subject: [PATCH 2/2] fix for manual install

---
 CMakeLists.txt | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index dc892e9..4ce4523 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -380,8 +380,10 @@ add_custom_command(OUTPUT ${PROJECT_BINARY_DIR}/plugins
    )
 
 add_custom_target(install_plugins_manual
-   COMMAND ${PROJECT_BINARY_DIR}/manual/${CMAKE_CFG_INTDIR}/genManual ${PROJECT_SOURCE_DIR} ${CMAKE_INSTALL_PREFIX}/${Mscore_SHARE_NAME}${Mscore_INSTALL_NAME}manual
-   )
+  COMMAND ${CMAKE_EXECUTABLE} -E make_directory
+  "${PROJECT_BINARY_DIR}/manual/plugins_manual/"
+  COMMAND ${PROJECT_BINARY_DIR}/manual/${CMAKE_CFG_INTDIR}/genManual ${PROJECT_SOURCE_DIR} "${PROJECT_BINARY_DIR}/manual/plugins_manual/"
+  )
 
 if (MINGW)
    install(CODE "execute_process(COMMAND cmd /c \"mingw32-make install_plugins_manual\")")
-- 
1.8.2.1

