From 049e51165b427ab14012fb9fdf3da16b2449082e Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Sat, 20 Sep 2014 12:29:23 -0400
Subject: [PATCH 2/2] add back native style

---
 mscore/preferences.h  | 3 ++-
 mscore/prefsdialog.ui | 5 +++++
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/mscore/preferences.h b/mscore/preferences.h
index 75393e8b9..aea8b0dc5 100644
--- a/mscore/preferences.h
+++ b/mscore/preferences.h
@@ -65,7 +65,8 @@ enum {
 
 enum class MuseScoreStyleType : char {
       DARK_FUSION = 0,
-      LIGHT_FUSION
+      LIGHT_FUSION,
+      NATIVE
       };
 
 // MusicXML export break values
diff --git a/mscore/prefsdialog.ui b/mscore/prefsdialog.ui
index f8b8931d8..fbacc1cf0 100644
--- a/mscore/prefsdialog.ui
+++ b/mscore/prefsdialog.ui
@@ -715,6 +715,11 @@
               <string>Light</string>
              </property>
             </item>
+            <item>
+             <property name="text">
+              <string>native</string>
+             </property>
+            </item>
            </widget>
           </item>
           <item row="1" column="1">
-- 
2.16.1

