--- ergo.service.orig	2022-02-27 20:40:51.259454330 +0000
+++ ergo.service	2022-02-27 20:40:51.379453700 +0000
@@ -1,5 +1,5 @@
 [Unit]
-Description=ergo
+Description=ergochat
 After=network.target
 # If you are using MySQL for history storage, comment out the above line
 # and uncomment these two instead (you must independently install and configure
@@ -9,9 +9,9 @@
 
 [Service]
 Type=notify
-User=ergo
-WorkingDirectory=/home/ergo
-ExecStart=/home/ergo/ergo run --conf /home/ergo/ircd.yaml
+User=ergochat
+WorkingDirectory=/var/lib/ergochat
+ExecStart=/usr/bin/ergochat run --conf /etc/ergochat/ircd.yaml
 ExecReload=/bin/kill -HUP $MAINPID
 Restart=on-failure
 LimitNOFILE=1048576
