diff -ru mjpegtools-2.1.0/y4mdenoise/Makefile.am new/y4mdenoise/Makefile.am
--- mjpegtools-2.1.0/y4mdenoise/Makefile.am	2012-04-01 10:43:55.000000000 +0200
+++ new/y4mdenoise/Makefile.am	2018-09-04 02:00:06.296975920 +0200
@@ -4,8 +4,8 @@
 
 EXTRA_DIST = implementation.html
 
-AM_CFLAGS = -DNDEBUG -finline-functions @PROGRAM_NOPIC@
-AM_CXXFLAGS = -DNDEBUG -finline-functions @PROGRAM_NOPIC@
+AM_CFLAGS = -DNDEBUG -finline-functions 
+AM_CXXFLAGS = -DNDEBUG -finline-functions
 
 INCLUDES =  -I$(top_srcdir) -I$(top_srcdir)/utils
 
diff -ru mjpegtools-2.1.0/y4munsharp/Makefile.am new/y4munsharp/Makefile.am
--- mjpegtools-2.1.0/y4munsharp/Makefile.am	2005-06-16 19:09:38.000000000 +0200
+++ new/y4munsharp/Makefile.am	2018-09-04 01:59:37.330309169 +0200
@@ -4,7 +4,7 @@
 
 INCLUDES = -I $(top_srcdir)/utils -I $(top_srcdir)
 
-y4munharp_CFLAGS=@PROGRAM_NOPIC@
+y4munharp_CFLAGS=
 
 bin_PROGRAMS = y4munsharp
 
diff -ru mjpegtools-2.1.0/y4mutils/Makefile.am new/y4mutils/Makefile.am
--- mjpegtools-2.1.0/y4mutils/Makefile.am	2012-04-01 10:43:35.000000000 +0200
+++ new/y4mutils/Makefile.am	2018-09-04 01:58:58.273642386 +0200
@@ -1,8 +1,7 @@
 # Makefile for y4mutils
 
 MAINTAINERCLEANFILES = Makefile.in
-
-AM_CFLAGS   = @PROGRAM_NOPIC@
+AM_CFLAGS=
 AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/utils
 
 LIBMJPEGUTILS = $(top_builddir)/utils/libmjpegutils.la
diff -ru mjpegtools-2.1.0/yuvscaler/Makefile.am new/yuvscaler/Makefile.am
--- mjpegtools-2.1.0/yuvscaler/Makefile.am	2012-04-01 10:44:27.000000000 +0200
+++ new/yuvscaler/Makefile.am	2018-09-04 02:00:06.296975920 +0200
@@ -18,6 +18,6 @@
 
 EXTRA_DIST = yuvscaler_implementation.txt
 
-yuvscaler_CFLAGS=@PROGRAM_NOPIC@
+yuvscaler_CFLAGS=
 yuvscaler_SOURCES = yuvscaler.c yuvscaler_resample.c yuvscaler_bicubic.c
 yuvscaler_LDADD = $(LIBMJPEGUTILS) $(LIBM_LIBS)
