Index: ImageMagick/branches/ImageMagick-6/coders/rle.c
===================================================================
--- ImageMagick/branches/ImageMagick-6/coders/rle.c	(revision 17434)
+++ ImageMagick/branches/ImageMagick-6/coders/rle.c	(revision 17516)
@@ -457,5 +457,5 @@
                 MagickFalse)
               break;
-            *p=colormap[index];
+            *p=colormap[(ssize_t) index];
             p++;
           }
@@ -468,5 +468,5 @@
                     (*p & mask)),&index,exception) == MagickFalse)
                   break;
-                *p=colormap[index];
+                *p=colormap[(ssize_t) index];
                 p++;
               }
