Index: ImageMagick/branches/ImageMagick-6/coders/palm.c
===================================================================
--- ImageMagick/branches/ImageMagick-6/coders/palm.c	(revision 17434)
+++ ImageMagick/branches/ImageMagick-6/coders/palm.c	(revision 17515)
@@ -511,5 +511,5 @@
         IndexPacket index=ConstrainColormapIndex(image,(mask-transparentIndex));
         if (bits_per_pixel != 16)
-          SetMagickPixelPacket(image,image->colormap+index,
+          SetMagickPixelPacket(image,image->colormap+(ssize_t) index,
             (const IndexPacket *) NULL,&transpix);
         (void) TransparentPaintImage(image,&transpix,(Quantum)
