diff --git a/libfakeroot.c b/libfakeroot.c
index f867758..7ef6e47 100644
--- a/libfakeroot.c
+++ b/libfakeroot.c
@@ -256,10 +256,12 @@ void load_library_symbols(void){
  /* clear dlerror() just in case dlsym() legitimately returns NULL */
     msg = dlerror();
     *(next_wrap[i].doit)=dlsym(get_libc(), next_wrap[i].name);
+#ifdef LIBFAKEROOT_DEBUGGING
     if ( (msg = dlerror()) != NULL){
       fprintf (stderr, "dlsym(%s): %s\n", next_wrap[i].name, msg);
 /*    abort ();*/
     }
+#endif /* LIBFAKEROOT_DEBUGGING */
   }
 }
 
