From ecaa3b7b305974fd79d0ea9fedd8fa4b6eab1730 Mon Sep 17 00:00:00 2001
From: Paul A McAuley <kde@paulmcauley.com>
Date: Thu, 26 May 2022 20:27:39 +0100
Subject: [PATCH] Fix incomplete type compile error in GCC12 by including
 <array>

---
 kstyle/breezestyle.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kstyle/breezestyle.cpp b/kstyle/breezestyle.cpp
index bec6bd36a..d00f8f519 100644
--- a/kstyle/breezestyle.cpp
+++ b/kstyle/breezestyle.cpp
@@ -54,6 +54,7 @@
 #include <QDialogButtonBox>
 #include <QGraphicsItem>
 #include <QGraphicsProxyWidget>
+#include <array>
 
 #if BREEZE_HAVE_QTQUICK
 #include <QQuickWindow>
