diff --git a/src/msg.c b/src/msg.c
index 53f8a5d..94ff13d 100644
--- a/src/msg.c
+++ b/src/msg.c
@@ -36,7 +36,7 @@ zenity_msg_construct_question_dialog (GtkWidget *dialog, ZenityMsgData *msg_data
   cancel_button = gtk_dialog_add_button (GTK_DIALOG (dialog), GTK_STOCK_NO, GTK_RESPONSE_CANCEL);
   ok_button = gtk_dialog_add_button (GTK_DIALOG (dialog), GTK_STOCK_YES, GTK_RESPONSE_OK);
 
-  gtk_widget_grab_focus (ok_button);
+  gtk_widget_grab_focus (data->default_no ? cancel_button : ok_button);
 
   if (data->cancel_label) {
     gtk_button_set_label (GTK_BUTTON (cancel_button), data->cancel_label);
diff --git a/src/option.c b/src/option.c
index 7658e8b..1c3d78a 100644
--- a/src/option.c
+++ b/src/option.c
@@ -47,6 +47,7 @@ static gchar   *zenity_general_ok_button;
 static gchar   *zenity_general_cancel_button;
 static gboolean zenity_general_modal;
 static gint     zenity_general_attach;
+static gboolean zenity_general_default_no;
 
 /* Calendar Dialog Options */
 static gboolean zenity_calendar_active;
@@ -762,6 +763,15 @@ static GOptionEntry question_options[] = {
     NULL
   },
   {
+    "default-no",
+    '\0',
+    G_OPTION_FLAG_NOALIAS,
+    G_OPTION_ARG_NONE,
+    &zenity_general_default_no,
+    N_("Give Cancel button focus by default"),
+    NULL
+  },
+  {
     "text",
     '\0',
     G_OPTION_FLAG_NOALIAS,
@@ -1890,6 +1900,7 @@ zenity_question_post_callback (GOptionContext *context,
     results->msg_data->mode = ZENITY_MSG_QUESTION;
     results->msg_data->no_wrap = zenity_general_dialog_no_wrap;
     results->msg_data->no_markup = zenity_general_dialog_no_markup;
+    results->data->default_no = zenity_general_default_no;
   }
 
   return TRUE;
diff --git a/src/zenity.h b/src/zenity.h
index 501109a..874beb6 100644
--- a/src/zenity.h
+++ b/src/zenity.h
@@ -34,6 +34,7 @@ typedef struct {
   gint   timeout_delay;
   gboolean modal;
   gint   attach;
+  gboolean default_no;
 } ZenityData;
 
 typedef enum {
