diff --unified --recursive --text pynput-1.7.3-a/setup.py pynput-1.7.3-b/setup.py
--- pynput-1.7.3-a/setup.py	2021-02-10 09:50:04.000000000 -0700
+++ pynput-1.7.3-b/setup.py	2021-09-01 08:15:42.243368000 -0600
@@ -21,11 +21,6 @@
 RUNTIME_PACKAGES = [
     'six']
 
-#: Additional requirements used during setup
-SETUP_PACKAGES = [
-    'setuptools-lint >=0.5',
-    'sphinx >=1.3.1']
-
 #: Packages requires for different environments
 EXTRA_PACKAGES = {
     ':sys_platform == "darwin"': [
@@ -90,7 +85,6 @@
     long_description=README + '\n\n' + CHANGES,
 
     install_requires=RUNTIME_PACKAGES,
-    setup_requires=RUNTIME_PACKAGES + SETUP_PACKAGES,
     extras_require=EXTRA_PACKAGES,
 
     author=INFO['author'],
