*** zen/scope_guard.h_old	2015-12-12 12:03:45.649292864 +0100
--- zen/scope_guard.h	2015-12-12 12:04:29.992359871 +0100
***************
*** 20,26 ****
  #elif defined ZEN_LINUX || defined ZEN_MAC
  //std::uncaught_exceptions() currently unsupported on GCC and Clang => clean up ASAP
  #ifdef ZEN_LINUX
!     static_assert(__GNUC__ < 5 || (__GNUC__ == 5 && (__GNUC_MINOR__ < 2 || (__GNUC_MINOR__ == 2 && __GNUC_PATCHLEVEL__ <= 1))), "check std::uncaught_exceptions support");
  #else
      static_assert(__clang_major__ < 7 || (__clang_major__ == 7 && __clang_minor__ <= 0), "check std::uncaught_exceptions support");
  #endif
--- 20,26 ----
  #elif defined ZEN_LINUX || defined ZEN_MAC
  //std::uncaught_exceptions() currently unsupported on GCC and Clang => clean up ASAP
  #ifdef ZEN_LINUX
!     static_assert(__GNUC__ < 5 || (__GNUC__ == 5 && (__GNUC_MINOR__ < 3 || (__GNUC_MINOR__ == 3 && __GNUC_PATCHLEVEL__ <= 1))), "check std::uncaught_exceptions support");
  #else
      static_assert(__clang_major__ < 7 || (__clang_major__ == 7 && __clang_minor__ <= 0), "check std::uncaught_exceptions support");
  #endif
