diff -ru a/src/main.c b/src/main.c
--- a/src/main.c	2015-07-22 22:02:17.451773876 -0400
+++ b/src/main.c	2015-07-22 22:02:34.618281726 -0400
@@ -765,6 +765,7 @@
         ret = yad_print_run ();
         break;
 
+#ifdef HAVE_HTML
       case YAD_MODE_HTML:
         /* Webkit doesn't handle focus for child dialogs when gtk_dialog_run() is used */
         /* FIXME: maybe this solution must be expanded to all dialogs */
@@ -775,6 +776,7 @@
 
         gtk_main ();
         break;
+#endif
 
       default:
         dialog = create_dialog ();
