diff --git a/Telegram/gyp/telegram_linux.gypi b/Telegram/gyp/telegram_linux.gypi
index 29386a9b..06dd1abc 100644
--- a/Telegram/gyp/telegram_linux.gypi
+++ b/Telegram/gyp/telegram_linux.gypi
@@ -68,9 +68,8 @@
 #      '<!(pkg-config 2> /dev/null --libs <@(pkgconfig_libs))',
     ],
     'cflags_cc': [
-      '<!(pkg-config 2> /dev/null --cflags appindicator-0.1)',
-      '<!(pkg-config 2> /dev/null --cflags gtk+-2.0)',
-      '<!(pkg-config 2> /dev/null --cflags glib-2.0)',
+      '<!(pkg-config 2> /dev/null --cflags appindicator3-0.1)',
+      '<!(pkg-config 2> /dev/null --cflags gtk+-3.0)',
       '<!(pkg-config 2> /dev/null --cflags dee-1.0)',
     ],
     'configurations': {
