--- a/usr/share/applications/franz.desktop	2018-09-12 16:04:43.010695442 +0200
+++ b/usr/share/applications/franz.desktop	2018-09-12 16:05:54.771682179 +0200
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Name=Franz
 Comment=Franz is your messaging app / former Emperor of Austria and combines chat & messaging services into one application. Franz currently supports Slack, WhatsApp, WeChat, HipChat, Facebook Messenger, Telegram, Google Hangouts, GroupMe, Skype and many more. You can download Franz for free for Mac & Windows.
-Exec="/opt/Franz/franz" %U
+Exec=/usr/bin/franz %U
 Terminal=false
 Type=Application
 Icon=franz
