diff --git a/PKGBUILD b/PKGBUILD
index bbebd5b..e3d67be 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -55,15 +55,19 @@ b2sums=('e32d607f2732c5ad719aa2f31c7f42f6243d7e9d52447dc923563d2982a68a3bed836ac
         '410dc8911051905c5c01b47890eeff817fc180434372864cfa9ee0d77e0ff43571b9fcc3c193d562c4dcd49511edf7c6c01dde12dd0778845d1868dc435531ea'
         'c3bc7e3ea2b943c2dcd5d2169eae63e90436963eb547c2301a3449169cdc8da13500c88acbfb4cc033ab78aa72950cf88e9fce00bc86d7a0af68b649761c2829')
 export KBUILD_BUILD_HOST=archlinux
+pkgbase=linux-lts-amdfixes
 export KBUILD_BUILD_USER=$pkgbase
 export KBUILD_BUILD_TIMESTAMP="$(date -Ru${SOURCE_DATE_EPOCH:+d @$SOURCE_DATE_EPOCH})"
+source+=(0004-amdgpu-fix-edp-brightness-control-on-HP-laptops.patch)
+sha256sums+=('13464deccb3d30ff4d2b731f6110befc3a794fcf538fa4313720c8f8c93c9d3b')
+b2sums+=('2fb944d6a277806fa113f86566b5b28814e2f43b85b52fbd791cdcce76a22e7aae3e143cbabb17b096b9e39fa9f36aa9b014224b871b4e0a8c2e4e8aa90cb9b5')
 
 prepare() {
   cd $_srcname
 
   echo "Setting version..."
   echo "-$pkgrel" > localversion.10-pkgrel
-  echo "${pkgbase#linux}" > localversion.20-pkgname
+  echo "-lts" > localversion.20-pkgname
 
   local src
   for src in "${source[@]}"; do
@@ -96,6 +100,7 @@ build() {
 
 _package() {
   pkgdesc="The $pkgdesc kernel and modules"
+  conflicts=('linux-lts')
   license=(
     'Apache-2.0 OR MIT'
 
@@ -155,6 +160,7 @@ _package() {
     VIRTUALBOX-GUEST-MODULES
     WIREGUARD-MODULE
   )
+  provides+=('linux-lts' "linux-lts=${pkgver}")
   replaces=(
     wireguard-lts
   )
@@ -168,7 +174,7 @@ _package() {
   install -Dm644 "$(make -s image_name)" "$modulesdir/vmlinuz"
 
   # Used by mkinitcpio to name the kernel
-  echo "$pkgbase" | install -Dm644 /dev/stdin "$modulesdir/pkgbase"
+  echo "linux-lts" | install -Dm644 /dev/stdin "$modulesdir/pkgbase"
 
   echo "Installing modules..."
   ZSTD_CLEVEL=19 make INSTALL_MOD_PATH="$pkgdir/usr" INSTALL_MOD_STRIP=1 \
@@ -185,6 +191,8 @@ _package() {
 
 _package-headers() {
   pkgdesc="Headers and scripts for building modules for the $pkgdesc kernel"
+  provides=('linux-lts-headers' "linux-lts-headers=${pkgver}")
+  conflicts=('linux-lts-headers')
   license=(
     BSD-3-Clause
     'BSD-3-Clause OR GPL-2.0-only'
@@ -325,6 +333,8 @@ _package-headers() {
 
 _package-docs() {
   pkgdesc="Documentation for the $pkgdesc kernel"
+  provides=('linux-lts-docs' "linux-lts-docs=${pkgver}")
+  conflicts=('linux-lts-docs')
   license=(
     BSD-3-Clause
 
