From b609203df7333beea20dbfd604262a9486f01497 Mon Sep 17 00:00:00 2001
Message-Id: <b609203df7333beea20dbfd604262a9486f01497.1552648361.git.jan.steffens@gmail.com>
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Tue, 23 Jun 2015 08:40:29 +0200
Subject: [PATCH 1/4] Enable table validation modules

---
 modules.cfg | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules.cfg b/modules.cfg
index dc6c8d42d..2ab145016 100644
--- a/modules.cfg
+++ b/modules.cfg
@@ -111,26 +111,26 @@ AUX_MODULES += cache

 # TrueType GX/AAT table validation.  Needs `ftgxval.c' below.
 #
-# AUX_MODULES += gxvalid
+AUX_MODULES += gxvalid

 # Support for streams compressed with gzip (files with suffix .gz).
 #
 # See include/freetype/ftgzip.h for the API.
 AUX_MODULES += gzip

 # Support for streams compressed with LZW (files with suffix .Z).
 #
 # See include/freetype/ftlzw.h for the API.
 AUX_MODULES += lzw

 # Support for streams compressed with bzip2 (files with suffix .bz2).
 #
 # See include/freetype/ftbzip2.h for the API.
 AUX_MODULES += bzip2

 # OpenType table validation.  Needs `ftotval.c' below.
 #
-# AUX_MODULES += otvalid
+AUX_MODULES += otvalid

 # Auxiliary PostScript driver component to share common code.
 #
--
2.21.0
