--- gitea.service
+++ gitea.service	2018-01-07 04:29:41.890000000 +0300
@@ -18,10 +18,10 @@
 Type=simple
 User=git
 Group=git
-WorkingDirectory=/home/git/gitea
-ExecStart=/home/git/gitea/gitea web
+WorkingDirectory=/var/lib/gitea
+ExecStart=/usr/bin/gitea web --config /etc/gitea/app.ini
 Restart=always
-Environment=USER=git HOME=/home/git
+Environment=USER=git HOME=/var/lib/gitea
 
 [Install]
 WantedBy=multi-user.target
