From 6ec4db701be89b95218be1f64a04892ed6b906bd Mon Sep 17 00:00:00 2001
From: Stelios Tsampas <loathingkernel@gmail.com>
Date: Sun, 19 Mar 2023 10:56:59 +0200
Subject: [PATCH 2/2] AUR Do not update cargo crates

---
 make/rules-cargo.mk | 2 --
 1 file changed, 2 deletions(-)

diff --git a/make/rules-cargo.mk b/make/rules-cargo.mk
index 5ebcf009..4f5c3162 100644
--- a/make/rules-cargo.mk
+++ b/make/rules-cargo.mk
@@ -14,8 +14,6 @@ $$(OBJ)/.$(1)-configure$(3):
 $$(OBJ)/.$(1)-build$(3):
 	@echo ":: building $(3)bit $(1)..." >&2
 	cd $$($(2)_SRC) && env $$($(2)_ENV$(3)) \
-	cargo update --locked --offline --verbose --workspace
-	cd $$($(2)_SRC) && env $$($(2)_ENV$(3)) \
 	cargo build \
 	      -j$$(SUBJOBS) --locked --offline --verbose \
 	      --target "$$(CARGO_TARGET_$(3))" \
-- 
2.40.0

