From fe1e557f5cbbd6091e47978a18ce4d182dfd7ed4 Mon Sep 17 00:00:00 2001
From: Stelios Tsampas <loathingkernel@gmail.com>
Date: Sun, 19 Mar 2023 10:56:59 +0200
Subject: [PATCH 2/3] AUR Do not update cargo crates

---
 make/rules-cargo.mk | 2 --
 1 file changed, 2 deletions(-)

diff --git a/make/rules-cargo.mk b/make/rules-cargo.mk
index 45bd7381..11e36c1d 100644
--- a/make/rules-cargo.mk
+++ b/make/rules-cargo.mk
@@ -13,8 +13,6 @@ $$(OBJ)/.$(1)-configure$(3):
 $$(OBJ)/.$(1)-build$(3):
 	@echo ":: building $(3)bit $(1)..." >&2
 	cd $$($(2)_SRC) && env $$($(2)_ENV$(3)) \
-	cargo update --locked --offline --verbose --workspace
-	cd $$($(2)_SRC) && env $$($(2)_ENV$(3)) \
 	cargo build \
 	      -j$$(SUBJOBS) --locked --offline --verbose \
 	      --target "$$(CARGO_TARGET_$(3))" \
-- 
2.41.0

