--- a/compatibilitytool.vdf	2021-10-19 15:49:39.298953579 +0300
+++ b/compatibilitytool.vdf	2021-10-19 15:49:03.259011088 +0300
@@ -19,5 +19,12 @@
       "from_oslist"  "windows"
       "to_oslist"    "linux"
     }
+    "proton-native" // Internal name of this tool
+    {
+      "install_path" "."
+      "display_name" "proton-native (deprecated entry, use 'proton' above)"
+      "from_oslist"  "windows"
+      "to_oslist"    "linux"
+    }
   }
 }
