--- obfsproxy-0.2.2/setup.py	2013-04-16 01:33:31.000000000 +0800
+++ obfsproxy-0.2.2/setup.py	2013-04-16 01:33:31.000000000 +0800
@@ -10,6 +10,11 @@
 versioneer.tag_prefix = 'obfsproxy-' # tags are like 1.2.0
 versioneer.parentdir_prefix = 'obfsproxy-' # dirname like 'myproject-1.2.0'
 
+cond_requires = []
+##  argparse became built-in from python 2.7 and 3.2
+if sys.version_info[:2] < (2, 7):
+    cond_requires.append('argparse')
+
 setup(
     name = "obfsproxy",
     author = "asn",
@@ -32,7 +37,6 @@
         'setuptools',
         'PyCrypto',
         'Twisted',
-        'argparse',
         'pyptlib'
-        ],
+	] + cond_requires,
 )
