diff -Naur webkitgtk-2.40.0/Source/bmalloc/libpas/src/chaos/Chaos.cpp webkitgtk-2.40.0-enlarged/Source/bmalloc/libpas/src/chaos/Chaos.cpp
--- webkitgtk-2.40.0/Source/bmalloc/libpas/src/chaos/Chaos.cpp	2023-02-20 17:22:22.429757600 +0800
+++ webkitgtk-2.40.0-enlarged/Source/bmalloc/libpas/src/chaos/Chaos.cpp	2023-03-19 16:16:45.337609472 +0800
@@ -51,7 +51,7 @@
     macro(uintptr_t,     seed,                           666,             "%lu") \
     macro(uintptr_t,     seedAdvance,                    1,               "%lu") \
     macro(uintptr_t,     maxSmallObjectSize,             1000,            "%lu") \
-    macro(uintptr_t,     maxObjectSize,                  65536,           "%lu") \
+    macro(uintptr_t,     maxObjectSize,                  16777216,        "%lu") \
     macro(double,        smallObjectProbability,         .95,             "%lf") \
     macro(double,        memalignProbability,            .05,             "%lf") \
     macro(uintptr_t,     maxMemalignShift,               14,              "%lu") \
