diff --git a/tests/test_backend.py b/tests/test_backend.py
index 36e27d1..8881715 100644
--- a/tests/test_backend.py
+++ b/tests/test_backend.py
@@ -432,7 +432,7 @@ class BackendTests(Tf2OnnxBackendTestBase):
         strides = [1, 2, 2, 1]
         x_val = np.arange(1, 1 + np.prod(x_shape)).astype("float32").reshape(x_shape)
         kernel_val = np.arange(1, 1 + np.prod(kernel_shape)).astype("float32").reshape(kernel_shape)
-        self._conv_test(x_val, kernel_val, strides=strides, padding="VALID", rtol=1e-05)
+        self._conv_test(x_val, kernel_val, strides=strides, padding="VALID", rtol=1.1e-05)
 
     @check_tf_min_version("1.14", "tf 1.14 needed for explicit padding")
     def test_conv2d_explicit_padding(self):
@@ -442,7 +442,7 @@ class BackendTests(Tf2OnnxBackendTestBase):
         strides = [1, 1, 1, 1]
         x_val = np.arange(1, 1 + np.prod(x_shape)).astype("float32").reshape(x_shape)
         kernel_val = np.arange(1, 1 + np.prod(kernel_shape)).astype("float32").reshape(kernel_shape)
-        self._conv_test(x_val, kernel_val, strides=strides, padding=pads, rtol=1e-05)
+        self._conv_test(x_val, kernel_val, strides=strides, padding=pads, rtol=1.1e-05)
 
     def test_conv2d_dilation_same(self):
         x_shape = [1, 35, 35, 288]  # NHWC
@@ -451,7 +451,7 @@ class BackendTests(Tf2OnnxBackendTestBase):
         dilations = [1, 3, 1, 1]  # NHWC
         x_val = np.arange(1, 1 + np.prod(x_shape)).astype("float32").reshape(x_shape)
         kernel_val = np.arange(1, 1 + np.prod(kernel_shape)).astype("float32").reshape(kernel_shape)
-        self._conv_test(x_val, kernel_val, strides=strides, padding="SAME", dilations=dilations, rtol=1e-05)
+        self._conv_test(x_val, kernel_val, strides=strides, padding="SAME", dilations=dilations, rtol=1.1e-05)
 
     def test_conv2d_dilation_strides_same(self):
         x_shape = [1, 35, 35, 288]  # NHWC
