From e5f66a896442b0b3daff252886a91b0cb2e09d31 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Mon, 30 Oct 2023 17:17:47 -0400
Subject: [PATCH] Patch for ArchLinux package

---
 src/C/context.jl | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/src/C/context.jl b/src/C/context.jl
index 57c7d3ad..ba8bb1bc 100644
--- a/src/C/context.jl
+++ b/src/C/context.jl
@@ -50,7 +50,7 @@ function init_context()
         end
     else
         # Find Python executable
-        exe_path = get(ENV, "JULIA_PYTHONCALL_EXE", "")
+        exe_path = get(ENV, "JULIA_PYTHONCALL_EXE", "/usr/bin/python")
         if exe_path == "" || exe_path == "@CondaPkg"
             if CondaPkg.backend() == :Null
                 exe_path = Sys.which("python")
@@ -83,12 +83,6 @@ function init_context()
             CTX.which = :unknown
         end
 
-        # Ensure Python is runnable
-        try
-            run(pipeline(`$exe_path --version`, stdout = devnull, stderr = devnull))
-        catch
-            error("Python executable $(repr(exe_path)) is not executable.")
-        end
         CTX.exe_path = exe_path
 
         # For calling Python with UTF-8 IO
-- 
2.45.2

