From 597e86b9d0cadc0ace47b0dc4967e702fa6cfe4b Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Mon, 30 Oct 2023 17:17:47 -0400
Subject: [PATCH] Patch for ArchLinux package

---
 src/C/context.jl | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/src/C/context.jl b/src/C/context.jl
index 8d2714c5..5da0cdfb 100644
--- a/src/C/context.jl
+++ b/src/C/context.jl
@@ -49,7 +49,7 @@ function init_context()
         end
     else
         # Find Python executable
-        exe_path = get(ENV, "JULIA_PYTHONCALL_EXE", "")
+        exe_path = get(ENV, "JULIA_PYTHONCALL_EXE", "/usr/bin/python")
         if exe_path == "" || exe_path == "@CondaPkg"
             if CondaPkg.backend() == :Null
                 exe_path = Sys.which("python")
@@ -103,15 +103,6 @@ function init_context()
             end
         end
 
-        # Ensure Python is runnable
-        if !ispath(exe_path)
-            error("Python executable $(repr(exe_path)) does not exist.")
-        end
-        try
-            run(pipeline(`$exe_path --version`, stdout = devnull, stderr = devnull))
-        catch
-            error("Python executable $(repr(exe_path)) is not executable.")
-        end
         CTX.exe_path = exe_path
 
         # For calling Python with UTF-8 IO
-- 
2.51.0

