From bd7101de1537eba1705f517ec68a77230a00e843 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Fri, 24 Oct 2025 15:50:44 -0400
Subject: [PATCH 3/3] Restored API from locally patched llvm

---
 lib/21/libLLVM.jl | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/lib/21/libLLVM.jl b/lib/21/libLLVM.jl
index 5fc3b595..c19e8219 100644
--- a/lib/21/libLLVM.jl
+++ b/lib/21/libLLVM.jl
@@ -11083,6 +11083,15 @@ function LLVMOrcCreateNewThreadSafeContextFromLLVMContext(Ctx)
     ccall((:LLVMOrcCreateNewThreadSafeContextFromLLVMContext, libllvm), LLVMOrcThreadSafeContextRef, (LLVMContextRef,), Ctx)
 end
 
+"""
+    LLVMOrcThreadSafeContextGetContext(TSCtx)
+
+Get a reference to the wrapped LLVMContext.
+"""
+function LLVMOrcThreadSafeContextGetContext(TSCtx)
+    ccall((:LLVMOrcThreadSafeContextGetContext, libllvm), LLVMContextRef, (LLVMOrcThreadSafeContextRef,), TSCtx)
+end
+
 """
     LLVMOrcDisposeThreadSafeContext(TSCtx)
 
-- 
2.51.0

