--- lbfgsb.f	2012-07-19 06:29:28.000000000 +0800
+++ lbfgsb.f	2020-02-23 11:19:30.000000000 +0800
@@ -3279,8 +3279,10 @@
  55   continue
       if ( dd_p .gt.zero ) then
          call dcopy( n, xp, 1, x, 1 )
-         write(6,*) ' Positive dir derivative in projection '
-         write(6,*) ' Using the backtracking step '
+         if ( iprint .ge. 0 ) then
+             write(6,*) ' Positive dir derivative in projection '
+             write(6,*) ' Using the backtracking step '
+         endif
       else
          go to 911
       endif
