PR: https://github.com/openzfs/zfs/pull/16861
--- config/zfs-build.m4	2024-12-12 07:17:34.170795755 +0800
+++ config/zfs-build.m4.new	2024-12-13 13:57:27.117288089 +0800
@@ -629,6 +629,7 @@
 		debian)     bashcompletiondir=/usr/share/bash-completion/completions   ;;
 		freebsd)    bashcompletiondir=$sysconfdir/bash_completion.d;;
 		gentoo)     bashcompletiondir=/usr/share/bash-completion/completions   ;;
+		arch)       bashcompletiondir=/usr/share/bash-completion/completions   ;;
 		*)          bashcompletiondir=/etc/bash_completion.d   ;;
 	esac
 	AC_MSG_RESULT([$bashcompletiondir])
