diff -Naur ./nemo-dropbox/configure.ac ./nemo-dropbox/configure.ac
--- ./nemo-dropbox/configure.ac	2017-08-25 13:02:43.639767896 -0400
+++ ./nemo-dropbox/configure.ac	2017-08-25 13:05:51.487464850 -0400
@@ -29,36 +29,6 @@
 PKG_CHECK_MODULES(NEMO, libnemo-extension >= $NEMO_REQUIRED)
 PKG_CHECK_MODULES(GLIB, glib-2.0 >= $GLIB_REQUIRED)

-AC_PATH_PROG([PYTHON], [python])
-
-AC_PATH_PROG([RST2MAN], [rst2man], [python rst2man.py])
-AC_SUBST(RST2MAN)
-
-# define module checking macro
-AC_DEFUN([PYTHON_CHECK_MODULE], [
-AC_MSG_CHECKING([for $1])
-
-cat <<EOF | python
-try:
- import $2
-except:
- exit(1)
-else:
- exit(0)
-EOF
-
-if test $? -ne 0; then
-   AC_MSG_RESULT([no])
-   AC_MSG_ERROR([couldn't find $1])
-else
-   AC_MSG_RESULT([yes])
-fi
-])
-
-PYTHON_CHECK_MODULE(pygtk, gtk)
-PYTHON_CHECK_MODULE(gobject, gobject)
-PYTHON_CHECK_MODULE(docutils, docutils)
-
 # Make dependency CFLAGS and LIBS available
 AC_SUBST(NEMO_CFLAGS)
 AC_SUBST(NEMO_LIBS)
diff -Naur ./nemo-dropbox/data/Makefile.am ./nemo-dropbox/data/Makefile.am
--- ./nemo-dropbox/data/Makefile.am	2017-08-25 13:02:43.639767896 -0400
+++ ./nemo-dropbox/data/Makefile.am	2017-08-25 13:04:11.388692078 -0400
@@ -1,6 +1 @@
-applicationdir = $(datadir)/applications
-application_DATA = dropbox.desktop
-
-EXTRA_DIST = $(application_DATA)
-
-SUBDIRS = icons emblems
+SUBDIRS = emblems
