From 6c476233f1e356ed410375e94eab84445bbde9f4 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Fri, 3 Feb 2023 15:06:11 -0500
Subject: [PATCH] Support suitesparse version 7.0

---
 src/solvers/cholmod.jl | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/src/solvers/cholmod.jl b/src/solvers/cholmod.jl
index 1f838f5..2590b86 100644
--- a/src/solvers/cholmod.jl
+++ b/src/solvers/cholmod.jl
@@ -231,7 +231,16 @@ function __init__()
         end
 
         # Register gc tracked allocator if CHOLMOD is new enough
-        if current_version >= v"3.0.0"
+        if current_version >= v"7.0.0"
+            @ccall libcholmod.SuiteSparse_config_malloc_func_set(
+                cglobal(:jl_malloc, Ptr{Cvoid})::Ptr{Cvoid})::Cvoid
+            @ccall libcholmod.SuiteSparse_config_calloc_func_set(
+                cglobal(:jl_calloc, Ptr{Cvoid})::Ptr{Cvoid})::Cvoid
+            @ccall libcholmod.SuiteSparse_config_realloc_func_set(
+                cglobal(:jl_realloc, Ptr{Cvoid})::Ptr{Cvoid})::Cvoid
+            @ccall libcholmod.SuiteSparse_config_free_func_set(
+                cglobal(:jl_free, Ptr{Cvoid})::Ptr{Cvoid})::Cvoid
+        elseif current_version >= v"3.0.0"
             cnfg = cglobal((:SuiteSparse_config, :libsuitesparseconfig), Ptr{Cvoid})
             unsafe_store!(cnfg, cglobal(:jl_malloc, Ptr{Cvoid}), 1)
             unsafe_store!(cnfg, cglobal(:jl_calloc, Ptr{Cvoid}), 2)
-- 
2.39.0

