From 02f3b242263d8e1fb6a40a4a0f9254ccb445fb27 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Sat, 3 Oct 2020 08:45:51 -0400
Subject: [PATCH 1/8] Force enable GDB debug info registration

---
 src/codegen.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/codegen.cpp b/src/codegen.cpp
index 71dd815548..aefe075d3b 100644
--- a/src/codegen.cpp
+++ b/src/codegen.cpp
@@ -10148,7 +10148,7 @@ extern "C" void jl_init_llvm(void)
 
     bool jl_using_gdb_jitevents = false;
     // Register GDB event listener
-#if defined(JL_DEBUG_BUILD)
+#if defined(JL_DEBUG_BUILD) || 1
     jl_using_gdb_jitevents = true;
 #endif
     const char *jit_gdb = getenv("ENABLE_GDBLISTENER");
-- 
2.51.0

