From 8a91fdb0370853fb490691270d0936745c5fae58 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Fri, 30 Dec 2022 16:15:59 -0500
Subject: [PATCH 4/7] Use libc for libssp

---
 .../src/CompilerSupportLibraries_jll.jl                      | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/stdlib/CompilerSupportLibraries_jll/src/CompilerSupportLibraries_jll.jl b/stdlib/CompilerSupportLibraries_jll/src/CompilerSupportLibraries_jll.jl
index e386a6e19d..357dafdcc9 100644
--- a/stdlib/CompilerSupportLibraries_jll/src/CompilerSupportLibraries_jll.jl
+++ b/stdlib/CompilerSupportLibraries_jll/src/CompilerSupportLibraries_jll.jl
@@ -49,10 +49,7 @@ else
     const _libstdcxx_path = BundledLazyLibraryPath("libstdc++.so.6")
     const _libgomp_path = BundledLazyLibraryPath("libgomp.so.1")
     if libc(HostPlatform()) != "musl"
-        const _libssp_path = BundledLazyLibraryPath("libssp.so.0")
-    end
-    if arch(HostPlatform()) ∈ ("x86_64", "i686")
-        const _libquadmath_path = BundledLazyLibraryPath("libquadmath.so.0")
+        const _libssp_path = "libc.so.6"
     end
 end
 
-- 
2.48.1

