From 3c9035f581ac46ae77697eebd7d0ca2ed67d154e Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Fri, 30 Dec 2022 16:15:59 -0500
Subject: [PATCH 4/7] Use libc for libssp

---
 .../src/CompilerSupportLibraries_jll.jl                         | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/stdlib/CompilerSupportLibraries_jll/src/CompilerSupportLibraries_jll.jl b/stdlib/CompilerSupportLibraries_jll/src/CompilerSupportLibraries_jll.jl
index a57b275b18..c9a2b65bf1 100644
--- a/stdlib/CompilerSupportLibraries_jll/src/CompilerSupportLibraries_jll.jl
+++ b/stdlib/CompilerSupportLibraries_jll/src/CompilerSupportLibraries_jll.jl
@@ -47,7 +47,7 @@ else
     const libstdcxx = "libstdc++.so.6"
     const libgomp = "libgomp.so.1"
     if libc(HostPlatform()) != "musl"
-        const libssp = "libssp.so.0"
+        const libssp = "libc.so.6"
     end
 end
 
-- 
2.44.0

