From 9035799e26612c51979bc592115fc1c07d18718d Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Mon, 26 May 2025 23:20:16 -0400
Subject: [PATCH 6/9] Disable UCX signal handler

---
 src/init.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/init.c b/src/init.c
index f79a2695f5..4867593783 100644
--- a/src/init.c
+++ b/src/init.c
@@ -668,6 +668,8 @@ static void init_global_mutexes(void) {
 
 JL_DLLEXPORT void jl_init_(jl_image_buf_t sysimage)
 {
+    putenv("UCX_HANDLE_ERRORS=none"); // disable openucx signal handler
+
     // initialize many things, in no particular order
     // but generally running from simple platform things to optional
     // configuration features
-- 
2.51.0

