From 9887055655bcbc856ac3cf38c1662f22a0ad4600 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Sun, 24 Aug 2025 19:28:22 -0400
Subject: [PATCH 08/17] Reduce likelyhood of symbol conflict

---
 base/docs/Docs.jl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/base/docs/Docs.jl b/base/docs/Docs.jl
index 8817d82a6a..06642c5700 100644
--- a/base/docs/Docs.jl
+++ b/base/docs/Docs.jl
@@ -75,7 +75,7 @@ export doc, hasdoc, undocumented_names
 # Basic API / Storage
 
 const modules = Module[]
-const META    = gensym(:meta)
+const META    = gensym("#Base#Docs#meta#")
 const METAType = IdDict{Any,Any}
 
 function meta(m::Module; autoinit::Bool=true)
-- 
2.51.2

