From debc46f76b4a88f80cc07aa274309d3e43e48208 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Fri, 24 Oct 2025 10:50:40 -0400
Subject: [PATCH 10/10] Disable custom ehframe registrar on llvm21 for now

---
 src/jitlayers.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/jitlayers.cpp b/src/jitlayers.cpp
index 37200fc372..15ea8584f7 100644
--- a/src/jitlayers.cpp
+++ b/src/jitlayers.cpp
@@ -1224,6 +1224,7 @@ std::unique_ptr<jitlink::JITLinkMemoryManager> createJITLinkMemoryManager() JL_N
 #endif
 }
 
+#if JL_LLVM_VERSION < 210000
 class JLEHFrameRegistrar final : public jitlink::EHFrameRegistrar {
 public:
     Error registerEHFrames(orc::ExecutorAddrRange EHFrameSection) override {
@@ -1236,6 +1237,7 @@ public:
         return Error::success();
     }
 };
+#endif
 
 RTDyldMemoryManager *createRTDyldMemoryManager(void) JL_NOTSAFEPOINT;
 
@@ -1948,6 +1950,7 @@ JuliaOJIT::JuliaOJIT()
     OptSelLayer(ES, OptimizeLayer, static_cast<orc::ThreadSafeModule (*)(orc::ThreadSafeModule, orc::MaterializationResponsibility&)>(selectOptLevel))
 {
 #ifdef JL_USE_JITLINK
+#if JL_LLVM_VERSION < 210000
 # if defined(LLVM_SHLIB)
     // When dynamically linking against LLVM, use our custom EH frame registration code
     // also used with RTDyld to inform both our and the libc copy of libunwind.
@@ -1957,6 +1960,9 @@ JuliaOJIT::JuliaOJIT()
 # endif
     ObjectLayer.addPlugin(std::make_unique<EHFrameRegistrationPlugin>(
         ES, std::move(ehRegistrar)));
+#else
+    ObjectLayer.addPlugin(std::make_unique<EHFrameRegistrationPlugin>(ES));
+#endif
 
     ObjectLayer.addPlugin(std::make_unique<JLDebuginfoPlugin>());
     ObjectLayer.addPlugin(std::make_unique<JLMemoryUsagePlugin>(&jit_bytes_size));
-- 
2.51.0

