--- a/src/cpp/session/CMakeLists.txt	2024-05-07 06:31:08.000000000 +1200
+++ b/src/cpp/session/CMakeLists.txt	2024-05-30 15:19:47.978691253 +1200
@@ -699,12 +699,6 @@ if(NOT RSTUDIO_SESSION_WIN32 AND NOT RSE
    install(DIRECTORY "${RSTUDIO_DEPENDENCIES_MATHJAX_DIR}"
            DESTINATION "${RSTUDIO_INSTALL_SUPPORTING}/resources")
 
-   # install node
-   install(
-      DIRECTORY "${RSTUDIO_DEPENDENCIES_DIR}/common/node/${RSTUDIO_INSTALLED_NODE_VERSION}-patched/"
-      DESTINATION "${RSTUDIO_INSTALL_BIN}/node"
-      USE_SOURCE_PERMISSIONS)
-
    # install quarto (or pandoc if quarto disabled)
    if(QUARTO_ENABLED)
       # install some quarto folders into Resources, as needed
