diff -up src/xrdp-0.9.9/instfiles/xrdp.service.in.orig src/xrdp-0.9.9/instfiles/xrdp.service.in
--- src/xrdp-0.9.9/instfiles/xrdp.service.in.orig	2018-10-19 01:09:19.000000000 -0600
+++ src/xrdp-0.9.9/instfiles/xrdp.service.in	2019-01-13 06:49:55.000000000 -0700
@@ -6,11 +6,9 @@ After=network.target xrdp-sesman.service
 
 [Service]
 Type=forking
-PIDFile=@localstatedir@/run/xrdp.pid
-EnvironmentFile=-@sysconfdir@/sysconfig/xrdp
-EnvironmentFile=-@sysconfdir@/default/xrdp
-ExecStart=@sbindir@/xrdp $XRDP_OPTIONS
-ExecStop=@sbindir@/xrdp $XRDP_OPTIONS --kill
+PIDFile=/run/xrdp.pid
+ExecStart=@sbindir@/xrdp
+ExecStop=@sbindir@/xrdp --kill
 
 [Install]
 WantedBy=multi-user.target
diff -up src/xrdp-0.9.9/instfiles/xrdp-sesman.service.in.orig src/xrdp-0.9.9/instfiles/xrdp-sesman.service.in
--- src/xrdp-0.9.9/instfiles/xrdp-sesman.service.in.orig	2018-10-19 01:09:19.000000000 -0600
+++ src/xrdp-0.9.9/instfiles/xrdp-sesman.service.in	2019-01-13 06:49:55.000000000 -0700
@@ -7,11 +7,9 @@ BindsTo=xrdp.service
 
 [Service]
 Type=forking
-PIDFile=@localstatedir@/run/xrdp-sesman.pid
-EnvironmentFile=-@sysconfdir@/sysconfig/xrdp
-EnvironmentFile=-@sysconfdir@/default/xrdp
-ExecStart=@sbindir@/xrdp-sesman $SESMAN_OPTIONS
-ExecStop=@sbindir@/xrdp-sesman $SESMAN_OPTIONS --kill
+PIDFile=/run/xrdp-sesman.pid
+ExecStart=@sbindir@/xrdp-sesman
+ExecStop=@sbindir@/xrdp-sesman --kill
 
 [Install]
 WantedBy=multi-user.target
diff -up src/xrdp-0.9.9/sesman/startwm.sh.orig src/xrdp-0.9.9/sesman/startwm.sh
--- src/xrdp-0.9.9/sesman/startwm.sh.orig	2018-10-25 00:20:43.000000000 -0600
+++ src/xrdp-0.9.9/sesman/startwm.sh	2019-01-13 06:52:12.000000000 -0700
@@ -58,11 +58,22 @@ post_start()
 #start the window manager
 wm_start()
 {
-  if [ -r /etc/default/locale ]; then
-    . /etc/default/locale
+  if [ -r /etc/locale.conf ]; then
+    . /etc/locale.conf
     export LANG LANGUAGE
   fi
 
+  # arch user
+  if [ -r ~/.xinitrc ]; then
+    . ~/.xinitrc
+    exit 0
+  fi
+  # arch
+  if [ -r /etc/X11/xinit/xinitrc ]; then
+    . /etc/X11/xinit/xinitrc
+    exit 0
+  fi
+
   # debian
   if [ -r /etc/X11/Xsession ]; then
     pre_start
