diff -up src/xrdp-0.9.4/instfiles/xrdp.service.in.orig src/xrdp-0.9.4/instfiles/xrdp.service.in
--- src/xrdp-0.9.4/instfiles/xrdp.service.in.orig	2017-07-18 22:23:49.000000000 -0600
+++ src/xrdp-0.9.4/instfiles/xrdp.service.in	2017-11-18 23:48:10.000000000 -0700
@@ -6,11 +6,9 @@ After=network.target xrdp-sesman.service
 
 [Service]
 Type=forking
-PIDFile=@localstatedir@/run/xrdp.pid
-EnvironmentFile=-@sysconfdir@/sysconfig/xrdp
-EnvironmentFile=-@sysconfdir@/default/xrdp
-ExecStart=@sbindir@/xrdp $XRDP_OPTIONS
-ExecStop=@sbindir@/xrdp $XRDP_OPTIONS --kill
+PIDFile=/run/xrdp.pid
+ExecStart=@sbindir@/xrdp
+ExecStop=@sbindir@/xrdp --kill
 
 [Install]
 WantedBy=multi-user.target
diff -up src/xrdp-0.9.4/instfiles/xrdp-sesman.service.in.orig src/xrdp-0.9.4/instfiles/xrdp-sesman.service.in
--- src/xrdp-0.9.4/instfiles/xrdp-sesman.service.in.orig	2017-07-18 22:23:49.000000000 -0600
+++ src/xrdp-0.9.4/instfiles/xrdp-sesman.service.in	2017-11-18 23:47:37.000000000 -0700
@@ -7,11 +7,9 @@ BindsTo=xrdp.service
 
 [Service]
 Type=forking
-PIDFile=@localstatedir@/run/xrdp-sesman.pid
-EnvironmentFile=-@sysconfdir@/sysconfig/xrdp
-EnvironmentFile=-@sysconfdir@/default/xrdp
-ExecStart=@sbindir@/xrdp-sesman $SESMAN_OPTIONS
-ExecStop=@sbindir@/xrdp-sesman $SESMAN_OPTIONS --kill
+PIDFile=/run/xrdp-sesman.pid
+ExecStart=@sbindir@/xrdp-sesman
+ExecStop=@sbindir@/xrdp-sesman --kill
 
 [Install]
 WantedBy=multi-user.target
diff -up src/xrdp-0.9.4/instfiles/xrdp.sh.orig src/xrdp-0.9.4/instfiles/xrdp.sh
--- src/xrdp-0.9.4/instfiles/xrdp.sh.orig	2017-06-26 07:09:39.000000000 -0600
+++ src/xrdp-0.9.4/instfiles/xrdp.sh	2017-11-18 23:44:01.000000000 -0700
@@ -17,7 +17,7 @@
 # Description: starts xrdp
 ### END INIT INFO
 
-SBINDIR=/usr/local/sbin
+SBINDIR=/usr/bin
 LOG=/dev/null
 CFGDIR=/etc/xrdp
 
diff -up src/xrdp-0.9.4/sesman/startwm.sh.orig src/xrdp-0.9.4/sesman/startwm.sh
--- src/xrdp-0.9.4/sesman/startwm.sh.orig	2017-07-18 22:23:49.000000000 -0600
+++ src/xrdp-0.9.4/sesman/startwm.sh	2017-11-18 23:42:35.000000000 -0700
@@ -56,6 +56,17 @@ wm_start()
     export LANG LANGUAGE
   fi
 
+  # arch user
+  if [ -r ~/.xinitrc ]; then
+    . ~/.xinitrc
+    exit 0
+  fi
+  # arch
+  if [ -r /etc/X11/xinit/xinitrc ]; then
+    . /etc/X11/xinit/xinitrc
+    exit 0
+  fi
+
   # debian
   if [ -r /etc/X11/Xsession ]; then
     pre_start
