diff --git a/ui/style/style_core_custom_font.cpp b/ui/style/style_core_custom_font.cpp
index 825726f1af..8525c701c0 100644
--- a/ui/style/style_core_custom_font.cpp
+++ b/ui/style/style_core_custom_font.cpp
@@ -30,40 +30,18 @@ QFont ResolveFont(uint32 flags, int size) {
 	static auto Database = QFontDatabase();
 
 	const auto bold = ((flags & FontBold) || (flags & FontSemibold));
-	const auto italic = (flags & FontItalic);
-	const auto &custom = bold ? BoldFont : RegularFont;
-	const auto useCustom = !custom.family.isEmpty();
+   const QString FontFamilyName = qEnvironmentVariable("TELEGRAM_DESKTOP_UI_FONT");
+   const QString FontFamilyNameMono = qEnvironmentVariable("TELEGRAM_DESKTOP_UI_FONT_MONO");
 
-	auto result = QFont(QGuiApplication::font().family());
+	auto result = QFont(FontFamilyName.isEmpty() ? QGuiApplication::font().family() : FontFamilyName);
 	if (flags & FontMonospace) {
-		result.setFamily(MonospaceFont());
-	} else if (useCustom) {
-		const auto sizes = Database.smoothSizes(custom.family, custom.style);
-		const auto good = sizes.isEmpty()
-			? Database.pointSizes(custom.family, custom.style)
-			: sizes;
-		const auto point = good.isEmpty() ? size : good.front();
-		result = Database.font(custom.family, custom.style, point);
-	} else {
-		result.setFamily(GetFontOverride(flags));
-		if (bold) {
-#ifdef DESKTOP_APP_USE_PACKAGED_FONTS
-			result.setWeight(QFont::DemiBold);
-#else // DESKTOP_APP_USE_PACKAGED_FONTS
-			result.setBold(true);
-#endif // !DESKTOP_APP_USE_PACKAGED_FONTS
-
-			if (flags & FontItalic) {
-				result.setStyleName("Semibold Italic");
-			} else {
-				result.setStyleName("Semibold");
-			}
-		}
+       result.setFamily(FontFamilyNameMono.isEmpty() ? MonospaceFont() : FontFamilyNameMono);
 	}
-	if (italic) {
-		result.setItalic(true);
+	if (bold) {
+           result.setWeight(QFont::Medium);
 	}
 
+	result.setItalic(flags & FontItalic);
 	result.setUnderline(flags & FontUnderline);
 	result.setStrikeOut(flags & FontStrikeOut);
 	result.setPixelSize(size);
