From 3042e88eec32d01ad7e624cb7cefdc65c33f2fd8 Mon Sep 17 00:00:00 2001
From: Megumi_fox <i@megumifox.com>
Date: Fri, 21 Aug 2020 13:33:31 +0800
Subject: [PATCH] fix webrtc integration build missing libx11

---
 Telegram/CMakeLists.txt | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/Telegram/CMakeLists.txt b/Telegram/CMakeLists.txt
index 89e073424..dc05cb718 100644
--- a/Telegram/CMakeLists.txt
+++ b/Telegram/CMakeLists.txt
@@ -183,6 +183,15 @@ if (LINUX)
             target_link_libraries(Telegram PRIVATE X11)
         endif()
     endif()
+
+    if (DESKTOP_APP_DISABLE_GTK_INTEGRATION AND NOT DESKTOP_APP_DISABLE_X11_INTEGRATION)
+        find_package(PkgConfig REQUIRED)
+        pkg_check_modules(X11 REQUIRED IMPORTED_TARGET x11)
+        target_link_libraries(Telegram
+        PRIVATE
+            PkgConfig::X11
+        )
+    endif()
 endif()
 
 target_precompile_headers(Telegram PRIVATE ${src_loc}/stdafx.h)
-- 
2.28.0

