--- Telegram/lib_ui/ui/widgets/input_fields.cpp~	2021-09-02 00:08:07.000000000 +0800
+++ Telegram/lib_ui/ui/widgets/input_fields.cpp	2021-09-17 21:07:48.331415832 +0800
@@ -933,12 +933,6 @@
 const InstantReplaces &InstantReplaces::TextOnly() {
 	static const auto result = [] {
 		auto result = InstantReplaces();
-		result.add("--", QString(1, QChar(8212)));
-		result.add("<<", QString(1, QChar(171)));
-		result.add(">>", QString(1, QChar(187)));
-		result.add(
-			":shrug:",
-			QChar(175) + QString("\\_(") + QChar(12484) + ")_/" + QChar(175));
 		return result;
 	}();
 	return result;
