--- config.h	2012-08-01 12:56:47.000000000 +0200
+++ config.h	2012-08-03 09:53:05.025039199 +0200
@@ -27,11 +27,11 @@
  * Default settings for Linux - leave these last.
  */
 #ifndef DEFAULT_MAIL_COMMAND
-#define DEFAULT_MAIL_COMMAND "/bin/mail"
+#define DEFAULT_MAIL_COMMAND "/usr/bin/mail"
 #endif
 
 #ifndef COMPRESS_COMMAND
-#define COMPRESS_COMMAND "/bin/gzip"
+#define COMPRESS_COMMAND "/usr/bin/gzip"
 #endif
 
 #ifndef COMPRESS_EXT
@@ -39,7 +39,7 @@
 #endif
 
 #ifndef UNCOMPRESS_COMMAND
-#define UNCOMPRESS_COMMAND "/bin/gunzip"
+#define UNCOMPRESS_COMMAND "/usr/bin/gunzip"
 #endif
 
 #ifndef STATEFILE
