--- webkit-qtwebkit-23/Source/JavaScriptCore/DerivedSources.pri~	2013-01-09 11:29:16.452266719 +0000
+++ webkit-qtwebkit-23/Source/JavaScriptCore/DerivedSources.pri	2013-01-09 11:30:18.188538725 +0000
@@ -85,14 +85,14 @@
 retgen.output = RegExpJitTables.h
 retgen.script = $$PWD/create_regex_tables
 retgen.input = retgen.script
-retgen.commands = python $$retgen.script > ${QMAKE_FILE_OUT}
+retgen.commands = python2 $$retgen.script > ${QMAKE_FILE_OUT}
 GENERATORS += retgen
 
 #GENERATOR: "KeywordLookup.h": decision tree used by the lexer
 klgen.output = KeywordLookup.h
 klgen.script = $$PWD/KeywordLookupGenerator.py
 klgen.input = KEYWORDLUT_FILES
-klgen.commands = python $$klgen.script ${QMAKE_FILE_NAME} > ${QMAKE_FILE_OUT}
+klgen.commands = python2 $$klgen.script ${QMAKE_FILE_NAME} > ${QMAKE_FILE_OUT}
 GENERATORS += klgen
 
 EXTRACTOR_BINARY = LLIntOffsetsExtractor$$BIN_EXTENSION
@@ -117,7 +117,7 @@
     disassembler.input = DISASSEMBLER_FILES
     disassembler.script = $$PWD/disassembler/udis86/itab.py
     disassembler.depends = $$DISASSEMBLER_DEPENDENCY
-    disassembler.commands = python $$disassembler.script ${QMAKE_FILE_NAME} --outputDir ${QMAKE_FUNC_FILE_OUT_PATH}
+    disassembler.commands = python2 $$disassembler.script ${QMAKE_FILE_NAME} --outputDir ${QMAKE_FUNC_FILE_OUT_PATH}
     disassembler.CONFIG += no_link
     GENERATORS += disassembler
 }
--- webkit-qtwebkit-23/Source/WebCore/DerivedSources.pri~	2013-01-09 11:46:01.165612894 +0000
+++ webkit-qtwebkit-23/Source/WebCore/DerivedSources.pri	2013-01-09 11:46:27.465404725 +0000
@@ -779,7 +779,7 @@
 inspectorValidate.output = InspectorProtocolVersion.h
 inspectorValidate.input = INSPECTOR_JSON
 inspectorValidate.script = $$PWD/inspector/generate-inspector-protocol-version
-inspectorValidate.commands = python $$inspectorValidate.script -o ${QMAKE_FILE_OUT} ${QMAKE_FILE_IN}
+inspectorValidate.commands = python2 $$inspectorValidate.script -o ${QMAKE_FILE_OUT} ${QMAKE_FILE_IN}
 inspectorValidate.depends = $$PWD/inspector/generate-inspector-protocol-version
 inspectorValidate.add_output_to_sources = false
 GENERATORS += inspectorValidate
@@ -787,7 +787,7 @@
 inspectorJSON.output = InspectorFrontend.cpp InspectorBackendDispatcher.cpp InspectorTypeBuilder.cpp
 inspectorJSON.input = INSPECTOR_JSON
 inspectorJSON.script = $$PWD/inspector/CodeGeneratorInspector.py
-inspectorJSON.commands = python $$inspectorJSON.script $$PWD/inspector/Inspector.json --output_h_dir ${QMAKE_FUNC_FILE_OUT_PATH} --output_cpp_dir ${QMAKE_FUNC_FILE_OUT_PATH}
+inspectorJSON.commands = python2 $$inspectorJSON.script $$PWD/inspector/Inspector.json --output_h_dir ${QMAKE_FUNC_FILE_OUT_PATH} --output_cpp_dir ${QMAKE_FUNC_FILE_OUT_PATH}
 inspectorJSON.depends = $$inspectorJSON.script
 GENERATORS += inspectorJSON
 
@@ -908,7 +908,7 @@
 entities.output = HTMLEntityTable.cpp
 entities.input = HTML_ENTITIES
 entities.script = $$PWD/html/parser/create-html-entity-table
-entities.commands = python $$entities.script -o ${QMAKE_FILE_OUT} $$HTML_ENTITIES
+entities.commands = python2 $$entities.script -o ${QMAKE_FILE_OUT} $$HTML_ENTITIES
 entities.clean = ${QMAKE_FILE_OUT}
 entities.depends = $$PWD/html/parser/create-html-entity-table
 GENERATORS += entities
