--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1517,7 +1517,7 @@ config("default_warnings") {
           "-Wno-deprecated-copy",
 
           # TODO(https://crbug.com/1050281): Clean up, enable.
-          "-Wno-non-c-typedef-for-linkage",
+          # "-Wno-non-c-typedef-for-linkage",
 
           # TODO(https://crbug.com/1059231): Clean up, enable.
           "-Wno-pointer-to-int-cast",
