diff --git a/dropbox-cli.py b/dropbox-cli.py
index 3f301c0..1cd865c 100644
--- a/dropbox-cli.py
+++ b/dropbox-cli.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!/usr/bin/python2
 #
 # Copyright (c) Dropbox, Inc.
 #
@@ -787,6 +787,8 @@ def requires_dropbox_running(meth):
 
 def start_dropbox():
     db_path = DROPBOXD_PATH.encode(sys.getfilesystemencoding())
+    if not os.path.exists(db_path):
+        db_path = u"/usr/bin/dropbox"
     if os.access(db_path, os.X_OK):
         f = open("/dev/null", "w")
         # we don't reap the child because we're gonna die anyway, let init do it
