From 6f1ce9575735a9dc6c571191dda4eee67816cc75 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Sun, 7 Mar 2021 06:59:46 -0500
Subject: [PATCH] fix

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c56cb7f..a8e57a5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -87,7 +87,7 @@ find_package(OpenSSL REQUIRED)
 find_package(Sanitizers REQUIRED)
 find_package(Threads REQUIRED)
 find_package(sodium REQUIRED)
-set(Protobuf_USE_STATIC_LIBS ON)
+# set(Protobuf_USE_STATIC_LIBS ON)
 set(protobuf_MSVC_STATIC_RUNTIME ON)
 find_package(Protobuf REQUIRED)
 find_package(ZLIB REQUIRED)
-- 
2.30.0

