From 358587b8a881bca4d933b7a1dd9471322b9f4bd7 Mon Sep 17 00:00:00 2001
From: Greg Nietsky <gregory@distrotech.co.za>
Date: Thu, 21 Nov 2013 12:05:25 +0200
Subject: [PATCH 1/2] Add support for x32 yasm

---
 configure | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/configure b/configure
index 2026833..e7c1ce3 100755
--- a/configure
+++ b/configure
@@ -155,6 +155,7 @@ host_cpu="${host%%-*}"
 host="${host#*-}"
 host_vendor="${host%%-*}"
 host_os="${host#*-}"
+host_ver="${host_os#*-}"
 
 case $host_os in
   beos*)
@@ -191,6 +192,9 @@ case $host_os in
     SYS="LINUX"
     CFLAGS="$CFLAGS -DHAVE_MALLOC_H"
     LDFLAGS="$LDFLAGS -lm"
+    if [ "${host_ver}" == "gnux32" ];then
+      ABI="X32";
+    fi;
     ;;
   cygwin*)
     SYS="MINGW"
@@ -242,8 +246,10 @@ case $host_cpu in
       LDFLAGS="$LDFLAGS -arch x86_64"
     elif [ "$SYS" = MINGW ]; then
       ASFLAGS="$ASFLAGS -f win32 -m amd64 -DPREFIX"
+    elif [ "$ABI" = X32 ]; then
+      ASFLAGS="$ASFLAGS -f elfx32 -m amd64"
     else
-      ASFLAGS="$ASFLAGS -f elf -m amd64"
+      ASFLAGS="$ASFLAGS -f elf64 -m amd64"
     fi
     ;;
   powerpc|powerpc64)
-- 
2.6.2

