diff -ura a/browser/main/lib/ConfigManager.js b/browser/main/lib/ConfigManager.js
--- a/browser/main/lib/ConfigManager.js	2017-07-29 08:04:53.000000000 +0200
+++ b/browser/main/lib/ConfigManager.js	2017-07-29 08:48:35.279227114 +0200
@@ -15,7 +15,7 @@
   navWidth: 200,
   sortBy: 'UPDATED_AT', // 'CREATED_AT', 'UPDATED_AT', 'APLHABETICAL'
   listStyle: 'DEFAULT', // 'DEFAULT', 'SMALL'
-  amaEnabled: true,
+  amaEnabled: false,
   hotkey: {
     toggleFinder: OSX ? 'Cmd + Alt + S' : 'Super + Alt + S',
     toggleMain: OSX ? 'Cmd + Alt + L' : 'Super + Alt + E'
