diff -ura a/gruntfile.js b/gruntfile.js
--- a/gruntfile.js	2018-09-04 09:11:27.000000000 +0200
+++ b/gruntfile.js	2018-09-04 12:29:45.072029833 +0200
@@ -6,6 +6,7 @@
 
 module.exports = function (grunt) {
   var authCode
+  /*
   try {
     authCode = grunt.file.readJSON('secret/auth_code.json')
   } catch (e) {
@@ -13,6 +14,7 @@
       console.warn('secret/auth_code.json is not found. CodeSigning is not available.')
     }
   }
+  */
   const OSX_COMMON_NAME = authCode != null ? authCode.OSX_COMMON_NAME : ''
   const WIN_CERT_PASSWORD = authCode != null ? authCode.WIN_CERT_PASSWORD : ''
 
@@ -74,10 +76,12 @@
 
   grunt.initConfig(initConfig)
   grunt.loadNpmTasks('grunt-electron-installer')
+  /*
   if (!WIN) {
     grunt.loadNpmTasks('grunt-electron-installer-debian')
     grunt.loadNpmTasks('grunt-electron-installer-redhat')
   }
+  */
 
   grunt.registerTask('compile', function () {
     var done = this.async()
diff -ura a/package.json b/package.json
--- a/package.json	2018-09-04 09:11:27.000000000 +0200
+++ b/package.json	2018-09-04 12:30:40.252950385 +0200
@@ -126,7 +126,7 @@
     "eslint-plugin-react": "^7.8.2",
     "eslint-plugin-standard": "^3.0.1",
     "faker": "^3.1.0",
-    "grunt": "^0.4.5",
+    "grunt": "^1.0.3",
     "grunt-electron-installer": "2.1.0",
     "history": "^1.17.0",
     "identity-obj-proxy": "^3.0.0",
@@ -146,7 +146,7 @@
     "signale": "^1.2.1",
     "standard": "^8.4.0",
     "style-loader": "^0.12.4",
-    "stylus": "^0.52.4",
+    "stylus": "^0.54.5",
     "stylus-loader": "^2.3.1",
     "webpack": "^1.12.2",
     "webpack-dev-server": "^1.12.0"
@@ -177,4 +177,4 @@
       "jest-localstorage-mock"
     ]
   }
-}
+}
\ Geen regeleindeteken (LF) aan einde van bestand
