--- a/Anaconda3-2.3.0-Linux-x86_64.sh	2015-07-03 10:44:50.662500605 +0200
+++ b/Anaconda3-2.3.0-Linux-x86_64.sh	2015-07-03 10:47:37.900536717 +0200
@@ -70,9 +70,9 @@
     fi
 fi
 # verify the size of the installer
-wc -c "$THIS_PATH" | grep 353018424 >/dev/null
+wc -c "$THIS_PATH" | grep 353018431 >/dev/null
 if (( $? )); then
-    echo "ERROR: size of $THIS_FILE should be 353018424 bytes" >&2
+    echo "ERROR: size of $THIS_FILE should be 353018431 bytes" >&2
     exit 1
 fi
 
@@ -385,7 +385,7 @@
 extract_dist _cache-0.0-x0
 
 mkdir $PREFIX/envs
-mkdir $HOME/.continuum 2>/dev/null
+#mkdir $HOME/.continuum 2>/dev/null
 
 PYTHON="$PREFIX/pkgs/python-3.4.3-1/bin/python -E"
 $PYTHON -V
@@ -395,7 +395,7 @@
     uname -a
     exit 1
 fi
-
+exit 0
 echo "creating default environment..."
 CONDA_INSTALL="$PREFIX/pkgs/conda-3.14.1-py34_0/lib/python3.4/site-packages/conda/install.py"
 $PYTHON $CONDA_INSTALL --prefix=$PREFIX --pkgs-dir=$PREFIX/pkgs --link-all || exit 1
