diff --git a/hydrus/core/HydrusConstants.py b/hydrus/core/HydrusConstants.py
index d59bf1aa..8cbf8fee 100644
--- a/hydrus/core/HydrusConstants.py
+++ b/hydrus/core/HydrusConstants.py
@@ -11,34 +11,9 @@ import yaml
 RUNNING_CLIENT = False
 RUNNING_SERVER = False
 
-RUNNING_FROM_FROZEN_BUILD = getattr( sys, 'frozen', False )
-
-if RUNNING_FROM_FROZEN_BUILD:
-    
-    real_exe_path = os.path.realpath( sys.executable )
-    
-    BASE_DIR = os.path.dirname( real_exe_path )
-    
-else:
-    
-    try:
-        
-        hc_realpath_dir = os.path.dirname( os.path.realpath( __file__ ) )
-        
-        HYDRUS_MODULE_DIR = os.path.split( hc_realpath_dir )[0]
-        
-        BASE_DIR = os.path.split( HYDRUS_MODULE_DIR )[0]
-        
-    except NameError: # if __file__ is not defined due to some weird OS
-        
-        BASE_DIR = os.path.realpath( sys.path[0] )
-        
-    
-    if BASE_DIR == '':
-        
-        BASE_DIR = os.getcwd()
-        
-    
+RUNNING_FROM_FROZEN_BUILD = False
+HYDRUS_MODULE_DIR = '/opt/hydrus/hydrus'
+BASE_DIR = os.path.expanduser("~/.local/share/hydrus")
 
 muh_platform = sys.platform.lower()
 
@@ -69,34 +44,18 @@ elif RUNNING_FROM_FROZEN_BUILD:
 elif RUNNING_FROM_MACOS_APP:
     NICE_RUNNING_AS_STRING = 'from App'
 
-BIN_DIR = os.path.join( BASE_DIR, 'bin' )
-HELP_DIR = os.path.join( BASE_DIR, 'help' )
-INCLUDE_DIR = os.path.join( BASE_DIR, 'include' )
-STATIC_DIR = os.path.join( BASE_DIR, 'static' )
+BIN_DIR = "/opt/hydrus/bin"
+HELP_DIR = "/opt/hydrus/help"
+INCLUDE_DIR = "/opt/hydrus/include"
+STATIC_DIR = "/opt/hydrus/static"
 
 DEFAULT_DB_DIR = os.path.join( BASE_DIR, 'db' )
 
-if PLATFORM_MACOS:
-    
-    desired_userpath_db_dir = os.path.join( '~', 'Library', 'Hydrus' )
-    
-else:
-    
-    desired_userpath_db_dir = os.path.join( '~', 'Hydrus' )
-    
-
-USERPATH_DB_DIR = os.path.expanduser( desired_userpath_db_dir )
-
-if USERPATH_DB_DIR == desired_userpath_db_dir:
-    
-    # could not figure it out, probably a crazy user situation atm
-    
-    USERPATH_DB_DIR = None
-    
+USERPATH_DB_DIR = DEFAULT_DB_DIR
 
 WE_SWITCHED_TO_USERPATH = False
 
-LICENSE_PATH = os.path.join( BASE_DIR, 'license.txt' )
+LICENSE_PATH = "/usr/share/licenses/hydrus/license.txt"
 
 #
 
