Index: fvwm/menus.c
===================================================================
--- fvwm/menus.c	(revision 16)
+++ fvwm/menus.c	(revision 17)
@@ -3340,6 +3340,13 @@
 	}
 	context = (*pexc)->w.wcontext;
 
+	/* Popup menu with first item (not title) under pointer */
+	if(MI_IS_TITLE(MR_FIRST_ITEM(mr)))
+	{
+		y-=MI_HEIGHT(MR_FIRST_ITEM(mr));
+		do_warp_to_item = True;
+	}
+
 	/*
 	 * Create a new menu instance (if necessary)
 	 */
