From cfffe17c04ceef2ef76415ea8342d5b5fed734dc Mon Sep 17 00:00:00 2001
From: Robert Coleman <github@robert.net.nz>
Date: Sun, 9 Jun 2019 09:04:07 +1200
Subject: [PATCH] add feature flag for single instance (see #358)

---
 app/index.js | 22 +++++++++++++++++++---
 1 file changed, 19 insertions(+), 3 deletions(-)

diff --git a/app/index.js b/app/index.js
index d02ccd3..8dd68da 100644
--- a/app/index.js
+++ b/app/index.js
@@ -152,13 +152,29 @@ app.on('activate', function() {
   updateManager.checkForUpdate();
 });
 
-const isSecondInstance = app.makeSingleInstance((commandLine, workingDirectory) => {
-  // Someone tried to run a second instance, we should focus our window.
+// feature flag: https://github.com/electron/electron/blob/master/docs/api/breaking-changes.md#appmakesingleinstance
+const hasRequestSingleInstanceLock = app.requestSingleInstanceLock ? true : false;
+let isSecondInstance = null;
+
+// Someone tried to run a second instance, we should focus our window.
+const handleSecondInstance = (argv, cwd) => {
   if (win) {
     if (win.isMinimized()) win.restore()
     win.focus()
   }
-})
+}
+
+if (hasRequestSingleInstanceLock) {
+  isSecondInstance = !app.requestSingleInstanceLock()
+
+  app.on('second-instance', (event, argv, cwd) => {
+    handleSecondInstance(argv, cwd)
+  })
+} else {
+  isSecondInstance = app.makeSingleInstance((argv, cwd) => {
+    handleSecondInstance(argv, cwd)
+  })
+}
 
 app.on('ready', function(){
 
