--- a/usr/share/applications/slack.desktop	2019-07-18 08:39:46.981221305 +0200
+++ b/usr/share/applications/slack.desktop	2019-07-18 08:40:45.194552861 +0200
@@ -4,7 +4,7 @@
 Comment=Slack Desktop
 GenericName=Slack Client for Linux
-Exec=/usr/bin/slack %U
+Exec=/usr/bin/slack -s %U
-Icon=/usr/share/pixmaps/slack.png
+Icon=slack
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Network;InstantMessaging;
