diff --git a/kernel/nvidia/nv-dma.c b/kernel/nvidia/nv-dma.c
index 73cdeed..e691014 100644
--- a/kernel/nvidia/nv-dma.c
+++ b/kernel/nvidia/nv-dma.c
@@ -606,7 +606,7 @@ static NvBool nv_dma_is_map_resource_implemented
 #if defined(NV_DMA_MAP_RESOURCE_PRESENT)
     const struct dma_map_ops *ops = get_dma_ops(&nvl->dev->dev);
 
-    return (ops->map_resource != NULL);
+    return (ops && ops->map_resource);
 #else
     return NV_FALSE;
 #endif
