diff --git a/Telegram/SourceFiles/qt_static_plugins.cpp b/Telegram/SourceFiles/qt_static_plugins.cpp
index b975c7a8..5ad461a4 100644
--- a/Telegram/SourceFiles/qt_static_plugins.cpp
+++ b/Telegram/SourceFiles/qt_static_plugins.cpp
@@ -30,6 +30,7 @@ Q_IMPORT_PLUGIN(QGenericEnginePlugin)
 #elif defined Q_OS_LINUX // Q_OS_WIN | Q_OS_MAC
 Q_IMPORT_PLUGIN(QWebpPlugin)
 Q_IMPORT_PLUGIN(QXcbIntegrationPlugin)
+Q_IMPORT_PLUGIN(QWaylandIntegrationPlugin)
 Q_IMPORT_PLUGIN(QConnmanEnginePlugin)
 Q_IMPORT_PLUGIN(QGenericEnginePlugin)
 Q_IMPORT_PLUGIN(QNetworkManagerEnginePlugin)
diff --git a/Telegram/gyp/Telegram.gyp b/Telegram/gyp/Telegram.gyp
index 31480945..193690e8 100644
--- a/Telegram/gyp/Telegram.gyp
+++ b/Telegram/gyp/Telegram.gyp
@@ -87,25 +87,21 @@
     ],
 
     'defines': [
-      'AL_LIBTYPE_STATIC',
       'AL_ALEXT_PROTOTYPES',
       'TGVOIP_USE_CXX11_LIB',
       '<!@(python -c "for s in \'<(build_defines)\'.split(\',\'): print(s)")',
+      '__STDC_FORMAT_MACROS',
+      'TDESKTOP_DISABLE_AUTOUPDATE',
+      'TDESKTOP_DISABLE_REGISTER_CUSTOM_SCHEME',
     ],
 
     'include_dirs': [
       '<(src_loc)',
       '<(SHARED_INTERMEDIATE_DIR)',
-      '<(libs_loc)/breakpad/src',
-      '<(libs_loc)/lzma/C',
-      '<(libs_loc)/zlib',
-      '<(libs_loc)/ffmpeg',
-      '<(libs_loc)/openal-soft/include',
-      '<(libs_loc)/opus/include',
-      '<(libs_loc)/range-v3/include',
       '<(minizip_loc)',
       '<(sp_media_key_tap_loc)',
       '<(emoji_suggestions_loc)',
+      '/usr/include/breakpad',
       '<(submodules_loc)/GSL/include',
       '<(submodules_loc)/variant/include',
     ],
diff --git a/Telegram/gyp/linux_glibc_wraps.gyp b/Telegram/gyp/linux_glibc_wraps.gyp
index 07a8276f..99ac4cac 100644
--- a/Telegram/gyp/linux_glibc_wraps.gyp
+++ b/Telegram/gyp/linux_glibc_wraps.gyp
@@ -27,7 +27,7 @@
     'sources': [
       '../SourceFiles/platform/linux/linux_glibc_wraps.c',
     ],
-    'conditions': [[ '"<!(uname -p)" == "x86_64"', {
+    'conditions': [[ '"<!(uname -m)" == "x86_64"', {
       'sources': [
         '../SourceFiles/platform/linux/linux_glibc_wraps_64.c',
       ],
diff --git a/Telegram/gyp/qt.gypi b/Telegram/gyp/qt.gypi
index dea127a3..57106533 100644
--- a/Telegram/gyp/qt.gypi
+++ b/Telegram/gyp/qt.gypi
@@ -38,7 +38,6 @@
             'Qt5Network',
             'Qt5Widgets',
             'Qt5Gui',
-            'qtharfbuzzng',
           ],
           'qt_version%': '<(qt_version)',
           'conditions': [
@@ -97,20 +96,9 @@
               '<@(qt_libs)',
               'Qt5DBus',
               'Qt5Core',
-              'qtpcre',
-              'Xi',
-              'Xext',
-              'Xfixes',
-              'SM',
-              'ICE',
-              'fontconfig',
-              'expat',
-              'freetype',
-              'z',
-              'xcb-shm',
-              'xcb-xfixes',
-              'xcb-render',
-              'xcb-static',
+              'qwayland-generic',
+              'qwayland-egl',
+              'Qt5WaylandClient',
             ],
           }],
         ],
@@ -201,6 +189,7 @@
     '<(qt_loc)/include/QtGui/<(qt_version)',
     '<(qt_loc)/include/QtCore/<(qt_version)/QtCore',
     '<(qt_loc)/include/QtGui/<(qt_version)/QtGui',
+    '/usr/include/openssl-1.0',
   ],
   'library_dirs': [
     '<(qt_loc)/lib',
@@ -225,7 +214,6 @@
       ],
       'libraries': [
         '<(PRODUCT_DIR)/obj.target/liblinux_glibc_wraps.a',
-        '<(linux_path_xkbcommon)/lib/libxkbcommon.a',
         '<@(qt_libs_release)',
         '<(linux_lib_ssl)',
         '<(linux_lib_crypto)',
@@ -238,12 +226,32 @@
         '-lgthread-2.0',
         '-lglib-2.0',
         '-lpthread',
+
+        '-lXi',
+        '-lXext',
+        '-lXfixes',
+        '-lSM',
+        '-lICE',
+        '-lfontconfig',
+        '-lexpat',
+        '-lfreetype',
+
+        '<!(pkg-config 2> /dev/null --libs xkbcommon xkbcommon-x11)',
+        '<!(pkg-config 2> /dev/null --libs harfbuzz)',
+        '<!(pkg-config 2> /dev/null --libs libpcre16)',
+        '<!(pkg-config 2> /dev/null --libs zlib)',
+        '<!(pkg-config 2> /dev/null --libs libpng16)',
+        '<!(pkg-config 2> /dev/null --libs libjpeg)',
+        '<!(pkg-config 2> /dev/null --libs libwebp)',
+        '<!(pkg-config 2> /dev/null --libs xcb-shm xcb-xfixes xcb-render xcb-renderutil xcb-sync xcb-randr xcb-xinerama xcb-xkb xcb-icccm xcb-image xcb-shape xcb-keysyms xcb-util)',
+        '<!(pkg-config 2> /dev/null --libs wayland-client wayland-cursor)',
+        '-lproxy',
+        '-lGL',
       ],
       'include_dirs': [
         '<(qt_loc)/mkspecs/linux-g++',
       ],
       'ldflags': [
-        '-static-libstdc++',
         '-pthread',
         '-rdynamic',
       ],
diff --git a/Telegram/gyp/settings_linux.gypi b/Telegram/gyp/settings_linux.gypi
index b93bace4..0f14aa1f 100644
--- a/Telegram/gyp/settings_linux.gypi
+++ b/Telegram/gyp/settings_linux.gypi
@@ -24,7 +24,6 @@
         'linux_common_flags': [
           '-pipe',
           '-Wall',
-          '-Werror',
           '-W',
           '-fPIC',
           '-Wno-unused-variable',
diff --git a/Telegram/gyp/telegram_linux.gypi b/Telegram/gyp/telegram_linux.gypi
index f6918c22..36ef6ad3 100644
--- a/Telegram/gyp/telegram_linux.gypi
+++ b/Telegram/gyp/telegram_linux.gypi
@@ -22,11 +22,20 @@
     'variables': {
       'not_need_gtk%': '<!(python -c "print(\'TDESKTOP_DISABLE_GTK_INTEGRATION\' in \'<(build_defines)\')")',
       'pkgconfig_libs': [
-# In order to work libxkbcommon must be linked statically,
-# PKGCONFIG links it like "-L/usr/local/lib -lxkbcommon"
-# which makes a dynamic link which leads to segfault in
-# QApplication() -> createPlatformIntegration -> QXcbIntegrationPlugin::create
-        #'xkbcommon',
+        'liblzma',
+        'openal',
+        'libavformat',
+        'libavcodec',
+        'libswresample',
+        'libswscale',
+        'libavutil',
+        'opus',
+        'libva-x11',
+        'libva-drm',
+        'libva',
+        'vdpau',
+        'libdrm',
+        'zlib',
       ],
       'linux_path_ffmpeg%': '/usr/local',
       'linux_path_openal%': '/usr/local',
@@ -37,20 +46,11 @@
       'linux_path_range%': '/usr/local',
     },
     'include_dirs': [
-      '/usr/local/include',
-      '<(linux_path_ffmpeg)/include',
-      '<(linux_path_openal)/include',
-      '<(linux_path_breakpad)/include/breakpad',
       '<(linux_path_opus_include)',
-      '<(linux_path_range)/include',
+      '/usr/include',
     ],
     'library_dirs': [
-      '/usr/local/lib',
-      '<(linux_path_ffmpeg)/lib',
-      '<(linux_path_openal)/lib',
-      '<(linux_path_va)/lib',
-      '<(linux_path_vdpau)/lib',
-      '<(linux_path_breakpad)/lib',
+      '/usr/lib',
     ],
     'libraries': [
       'breakpad_client',
@@ -58,23 +58,14 @@
       'ibusplatforminputcontextplugin',
       'fcitxplatforminputcontextplugin',
       'himeplatforminputcontextplugin',
-      'liblzma.a',
-      'libopenal.a',
-      'libavformat.a',
-      'libavcodec.a',
-      'libswresample.a',
-      'libswscale.a',
-      'libavutil.a',
-      'libopus.a',
-      'libva-x11.a',
-      'libva-drm.a',
-      'libva.a',
-      'libvdpau.a',
-      'libdrm.a',
-      'libz.a',
-#      '<!(pkg-config 2> /dev/null --libs <@(pkgconfig_libs))',
+      '<!(pkg-config 2> /dev/null --libs <@(pkgconfig_libs))',
     ],
     'cflags_cc': [
+      '<!(pkg-config 2> /dev/null --cflags appindicator-0.1)',
+      '<!(pkg-config 2> /dev/null --cflags gtk+-2.0)',
+      '<!(pkg-config 2> /dev/null --cflags glib-2.0)',
+      '<!(pkg-config 2> /dev/null --cflags dee-1.0)',
+      '<!(pkg-config 2> /dev/null --cflags <@(pkgconfig_libs))',
       '-Wno-strict-overflow',
     ],
     'ldflags': [
