diff -up thunderbird-60.8.0/servo/components/style_traits/values.rs.rust-1.33-build thunderbird-60.8.0/servo/components/style_traits/values.rs
--- thunderbird-60.8.0/servo/components/style_traits/values.rs.rust-1.33-build	2019-07-03 17:25:28.000000000 +0200
+++ thunderbird-60.8.0/servo/components/style_traits/values.rs	2019-07-11 13:38:53.687318154 +0200
@@ -135,6 +135,7 @@ where
     }
 }
 
+/// Some comment
 #[macro_export]
 macro_rules! serialize_function {
     ($dest: expr, $name: ident($( $arg: expr, )+)) => {
@@ -404,6 +405,7 @@ impl_to_css_for_predefined_type!(::csspa
 impl_to_css_for_predefined_type!(::cssparser::Color);
 impl_to_css_for_predefined_type!(::cssparser::UnicodeRange);
 
+/// Some comment
 #[macro_export]
 macro_rules! define_css_keyword_enum {
     (pub enum $name:ident { $($variant:ident = $css:expr,)+ }) => {
diff -up thunderbird-60.8.0/servo/components/style/properties/properties.mako.rs.old thunderbird-60.8.0/servo/components/style/properties/properties.mako.rs
--- thunderbird-60.8.0/servo/components/style/properties/properties.mako.rs.old	2019-07-11 14:22:51.393784701 +0200
+++ thunderbird-60.8.0/servo/components/style/properties/properties.mako.rs	2019-07-11 14:24:03.182578100 +0200
@@ -55,6 +55,7 @@ use style_adjuster::StyleAdjuster;
 
 pub use self::declaration_block::*;
 
+/// Neco
 #[cfg(feature = "gecko")]
 #[macro_export]
 macro_rules! property_name {
@@ -3852,7 +3853,7 @@ impl fmt::Debug for AliasId {
     }
 }
 
-// NOTE(emilio): Callers are responsible to deal with prefs.
+/// NOTE(emilio): Callers are responsible to deal with prefs.
 #[macro_export]
 macro_rules! css_properties_accessors {
     ($macro_name: ident) => {
@@ -3875,6 +3876,7 @@ macro_rules! css_properties_accessors {
     }
 }
 
+/// Neco
 #[macro_export]
 macro_rules! longhand_properties_idents {
     ($macro_name: ident) => {
