--- src/share/applications/mellowplayer.desktop	2017-12-18 19:35:26.705475288 +0100
+++ src/share/applications/mellowplayer.desktop	2017-12-18 21:19:48.807747384 +0100
@@ -3,7 +3,7 @@
 Version=1.0
 Type=Application
 Name=MellowPlayer
-Exec=MellowPlayer
+Exec=MellowPlayer --ppapi-widevine-path=/usr/lib/qt/plugins/ppapi/libwidevinecdmadapter.so
 TryExec=MellowPlayer
 Icon=mellowplayer
 Terminal=false
