From 103dd4e11325ff845af7dda79c6f2a715b3e2e98 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Gee=20Hjwi=C4=87?= <ghj096@gmail.com>
Date: Fri, 27 Jan 2023 16:06:13 +0300
Subject: [PATCH] Fix typo in PyJsUint32Array creation

---
 js2py/base.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js2py/base.py b/js2py/base.py
index f4ee721c..5423e384 100644
--- a/js2py/base.py
+++ b/js2py/base.py
@@ -203,7 +203,7 @@ def Js(val, Clamped=False):
         elif val.dtype == numpy.int32:
             return PyJsInt32Array(val, Int32ArrayPrototype)
         elif val.dtype == numpy.uint32:
-            return PyJsUint16Array(val, Uint32ArrayPrototype)
+            return PyJsUint32Array(val, Uint32ArrayPrototype)
 
         elif val.dtype == numpy.float32:
             return PyJsFloat32Array(val, Float32ArrayPrototype)
