diff --git a/src/bomi/imports/bomi/MessageBox.qml b/src/bomi/imports/bomi/MessageBox.qml
index 96ef2500..6c7df178 100644
--- a/src/bomi/imports/bomi/MessageBox.qml
+++ b/src/bomi/imports/bomi/MessageBox.qml
@@ -84,7 +84,7 @@ Rectangle {
             buttons: [B.ButtonBox.Cancel, B.ButtonBox.Ok]
             visible: buttons.length > 0
             width: 150; height: 25
-            anchors.horizontalCenter: parent.horizontalCenter
+            Layout.alignment: Qt.AlignHCenter
             source: Rectangle {
                 color: mouseArea.pressed ? Qt.rgba(1.0, 1.0, 1.0, 1.0)
                                          : Qt.rgba(0.3, 0.3, 0.3, 1.0)
diff --git a/src/bomi/skins/Tethys/bomi.qml b/src/bomi/skins/Tethys/bomi.qml
index 5afdd641..8c34ed90 100644
--- a/src/bomi/skins/Tethys/bomi.qml
+++ b/src/bomi/skins/Tethys/bomi.qml
@@ -259,13 +259,13 @@ B.AppWithDock {
                 anchors.rightMargin: anchors.leftMargin
                 Loader {
                     readonly property int bigSize: 32
-                    anchors.verticalCenter: parent.verticalCenter
+                    Layout.alignment: Qt.AlignVCenter
                     sourceComponent: mediaButtonComponent
                 }
                 B.TimeDuration {
                     id: td; spacing: 2
                     height: parent.height; width: contentWidth
-                    anchors.verticalCenter: parent.verticalCenter
+                    Layout.alignment: Qt.AlignVCenter
                     textStyle {
                         color: "white"; style: Text.Raised; styleColor: "black"
                         font.pixelSize: 9;
@@ -277,7 +277,7 @@ B.AppWithDock {
 
                 Loader {
                     readonly property var timeDuration: td
-                    anchors.verticalCenter: parent.verticalCenter
+                    Layout.alignment: Qt.AlignVCenter
                     width: 100; height: 14
                     sourceComponent: timeslider
                     Layout.fillWidth: true
@@ -285,13 +285,13 @@ B.AppWithDock {
 
                 Loader {
                     sourceComponent: volumeIcon
-                    anchors.verticalCenter: parent.verticalCenter
+                    Layout.alignment: Qt.AlignVCenter
                 }
 
                 B.VolumeSlider {
                     id: volumeCompact; style: sliders
                     width: 70; height: 14
-                    anchors.verticalCenter: parent.verticalCenter
+                    Layout.alignment: Qt.AlignVCenter
                     visible: app.width > 500
                 }
 
diff --git a/src/bomi/skins/metro/bomi.qml b/src/bomi/skins/metro/bomi.qml
index bdafb308..d364f7b9 100644
--- a/src/bomi/skins/metro/bomi.qml
+++ b/src/bomi/skins/metro/bomi.qml
@@ -137,7 +137,7 @@ B.AppWithDock {
                 MetroButton { prefix: "next"; action: "play/next" }
                 Item {
                     height: 30; Layout.fillWidth: true
-                    anchors.verticalCenter: parent.verticalCenter
+                    Layout.alignment: Qt.AlignVCenter
                     MouseArea {
                         id: volumeArea
                         width: volume.width; height: 30
