diff -Naur foxitreader.orig/FoxitReader.desktop foxitreader/FoxitReader.desktop
--- foxitreader.orig/FoxitReader.desktop	2018-02-13 18:54:39.000000000 +0100
+++ foxitreader/FoxitReader.desktop	2018-10-28 16:55:01.954090619 +0100
@@ -1,10 +1,12 @@
 [Desktop Entry]
 Name=Foxit Reader
-Comment=View pdf documents
+Comment=View PDF documents
 Keywords=pdf;ppdf;
 StartupNotify=true
 Terminal=false
 Type=Application
 Categories=Application;Office;Viewer;X-Red-Hat-Base;
 MimeType=application/pdf;application/ppdf;
-Icon=FoxitReader
+Icon=foxitreader
+Exec=foxitreader %F
+
diff -Naur foxitreader.orig/FoxitReader.sh foxitreader/FoxitReader.sh
--- foxitreader.orig/FoxitReader.sh	2018-02-13 16:57:48.000000000 +0100
+++ foxitreader/FoxitReader.sh	2018-10-28 16:54:35.792773802 +0100
@@ -1,3 +1,5 @@
 #!/bin/sh
- appname="FoxitReader"
-
+appname="FoxitReader"
+selfpath="$(dirname $(readlink -f "$0"))"
+export LD_LIBRARY_PATH="${selfpath}/lib:${selfpath}/platforms:${selfpath}/printsupport:${selfpath}/rmssdk:${selfpath}/sensors:${selfpath}/imageformats:${selfpath}/platforminputcontexts"
+exec "${selfpath}/${appname}" "$@"
