diff -Naur foxitreader.orig/usr/lib/foxitreader/FoxitReader.desktop foxitreader/usr/lib/foxitreader/FoxitReader.desktop
--- foxitreader.orig/usr/lib/foxitreader/FoxitReader.desktop	2016-02-16 15:23:41.000000000 +0100
+++ foxitreader/usr/lib/foxitreader/FoxitReader.desktop	2016-03-06 02:41:40.882259342 +0100
@@ -1,10 +1,11 @@
 [Desktop Entry]
 Name=Foxit Reader
-Comment=View pdf documents
+Comment=View PDF documents
 Keywords=pdf;octet-stream;
 StartupNotify=true
 Terminal=false
 Type=Application
 Categories=Application;Office;Viewer;X-Red-Hat-Base;
 MimeType=application/pdf;application/octet-stream;
-Icon=FoxitReader
+Icon=foxitreader
+Exec=foxitreader %F
