diff --git a/ycmd/completers/go/go_completer.py b/ycmd/completers/go/go_completer.py
index 6e26aa19..da3befc4 100644
--- a/ycmd/completers/go/go_completer.py
+++ b/ycmd/completers/go/go_completer.py
@@ -31,7 +31,7 @@ import threading
 
 from ycmd import responses
 from ycmd import utils
-from ycmd.utils import ToBytes, ToUnicode, ExecutableName
+from ycmd.utils import ToBytes, ToUnicode
 from ycmd.completers.completer import Completer
 
 BINARY_NOT_FOUND_MESSAGE = ( '{0} binary not found. Did you build it? '
@@ -48,14 +48,9 @@ GOCODE_NO_COMPLETIONS_MESSAGE = 'No completions found.'
 GOCODE_PANIC_MESSAGE = ( 'Gocode panicked trying to find completions, '
                          'you likely have a syntax error.' )
 
-GO_DIR = os.path.abspath(
-  os.path.join( os.path.dirname( __file__ ), '..', '..', '..', 'third_party',
-                'go', 'src', 'github.com' ) )
 GO_BINARIES = dict( {
-  'gocode': os.path.join( GO_DIR, 'mdempsky', 'gocode',
-                          ExecutableName( 'gocode' ) ),
-  'godef': os.path.join( GO_DIR, 'rogpeppe', 'godef',
-                         ExecutableName( 'godef' ) )
+  'gocode': '/usr/bin/gocode',
+  'godef': '/usr/bin/godef',
 } )
 
 LOGFILE_FORMAT = 'gocode_{port}_{std}_'
diff --git a/ycmd/completers/rust/rust_completer.py b/ycmd/completers/rust/rust_completer.py
index 80376782..34b2f915 100644
--- a/ycmd/completers/rust/rust_completer.py
+++ b/ycmd/completers/rust/rust_completer.py
@@ -42,14 +42,8 @@ from os import path as p
 
 _logger = logging.getLogger( __name__ )
 
-DIR_OF_THIRD_PARTY = p.abspath(
-  p.join( p.dirname( __file__ ), '..', '..', '..', 'third_party' ) )
-
-RACERD_BINARY_NAME = 'racerd' + ( '.exe' if utils.OnWindows() else '' )
-RACERD_BINARY_RELEASE = p.join( DIR_OF_THIRD_PARTY, 'racerd', 'target',
-                        'release', RACERD_BINARY_NAME )
-RACERD_BINARY_DEBUG = p.join( DIR_OF_THIRD_PARTY, 'racerd', 'target',
-                        'debug', RACERD_BINARY_NAME )
+RACERD_BINARY_RELEASE = '/usr/bin/racerd'
+RACERD_BINARY_DEBUG = ''
 
 RACERD_HMAC_HEADER = 'x-racerd-hmac'
 HMAC_SECRET_LENGTH = 16
