From 8c6b93c352a2628c7de81f54d7409a9292de67dc Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Sat, 3 Nov 2018 07:51:00 -0400
Subject: [PATCH] Fix webengine installation

---
 mscore/CMakeLists.txt | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/mscore/CMakeLists.txt b/mscore/CMakeLists.txt
index 9888d094d..2ae9e8bf3 100644
--- a/mscore/CMakeLists.txt
+++ b/mscore/CMakeLists.txt
@@ -649,20 +649,6 @@ else (MINGW)
 
    if ( NOT MSVC )
 ## install qwebengine core
-      if (NOT APPLE AND USE_WEBENGINE)
-         install(FILES
-            ${QT_INSTALL_PREFIX}/libexec/QtWebEngineProcess
-            DESTINATION bin
-            )
-         install(DIRECTORY
-            ${QT_INSTALL_PREFIX}/resources
-            DESTINATION lib/qt5
-            )
-         install(DIRECTORY
-            ${QT_INSTALL_PREFIX}/translations/qtwebengine_locales
-            DESTINATION lib/qt5/translations
-            )
-      endif(NOT APPLE AND USE_WEBENGINE)
 
       target_link_libraries(mscore
          ${ALSA_LIB}
-- 
2.19.1

