From 2abfb48ef7f38cc5bc2fce02950007117fed4ae3 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Sat, 20 Sep 2014 12:13:37 -0400
Subject: [PATCH 1/2] fix for pch

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index bf12993a1..c97afd993 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -730,6 +730,7 @@ if (NOT MSVC)
       WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
       )
 endif (NOT MSVC)
+set(BUILD_PCH false)
 
 precompiled_header(QT_INCLUDES all ${BUILD_PCH})
 
-- 
2.26.2

