--- SConscript	2014-08-31 04:33:31.000000000 +0300
+++ SConscript.arch	2014-08-31 23:31:50.732594365 +0300
@@ -248,13 +248,8 @@
 
 	if not GetOption('nolua') and not GetOption('renderer'):
 		#Look for Lua
-		luaver = "lua5.1"
-		if not conf.CheckLib(['lua5.1', 'lua-5.1', 'lua51', 'lua']):
-			if conf.CheckLib(['lua5.2', 'lua-5.2', 'lua52']):
-				env.Append(CPPDEFINES=["LUA_COMPAT_ALL"])
-				luaver = "lua5.2"
-			elif platform != "Darwin" or not conf.CheckFramework("Lua"):
-				FatalError("lua5.1 development library not found or not installed")
+		env.Append(CPPDEFINES=["LUA_COMPAT_ALL"])
+		luaver = "lua52"
 		if platform == "Linux":
 			try:
 				env.ParseConfig("pkg-config --cflags {0}".format(luaver))
