diff -ruN a/tests/Makefile.am b/tests/Makefile.am
--- a/tests/Makefile.am	2013-12-08 02:57:05.000000000 +0100
+++ b/tests/Makefile.am	2022-02-19 16:47:41.562672003 +0100
@@ -37,7 +37,7 @@
 # Used to make N-triples output consistent
 BASE_URI=http://librdf.org/raptor/tests/
 
-SUBDIRS = rdfxml ntriples ntriples-2013 nquads-2013 turtle turtle-2013 trig grddl rdfa rdfa11 json feeds
+SUBDIRS = ntriples ntriples-2013 nquads-2013 turtle turtle-2013 trig grddl rdfa rdfa11 json
 
 
 $(top_builddir)/src/libraptor2.la:
diff -ruN a/tests/Makefile.in b/tests/Makefile.in
--- a/tests/Makefile.in	2014-11-02 07:04:38.000000000 +0100
+++ b/tests/Makefile.in	2022-02-19 16:47:41.569338695 +0100
@@ -338,7 +338,7 @@
 
 # Used to make N-triples output consistent
 BASE_URI = http://librdf.org/raptor/tests/
-SUBDIRS = rdfxml ntriples ntriples-2013 nquads-2013 turtle turtle-2013 trig grddl rdfa rdfa11 json feeds
+SUBDIRS = ntriples ntriples-2013 nquads-2013 turtle turtle-2013 trig grddl rdfa rdfa11 json
 all: all-recursive
 
 .SUFFIXES:
