--- a/usr/share/applications/wps-office-et.desktop	2019-09-10 17:59:37.000000000 +0800
+++ b/usr/share/applications/wps-office-et.desktop	2019-10-29 20:33:33.331491047 +0800
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Comment=Use WPS Spreadsheets to analyze manage data.
 Comment[zh_CN]=使用WPS表格分析、管理数据
-Exec=/usr/bin/et %f
+Exec=/usr/bin/et %F
 GenericName=WPS Spreadsheets
 GenericName[zh_CN]=WPS 表格
 MimeType=application/wps-office.et;application/wps-office.ett;application/wps-office.ets;application/wps-office.eto;application/wps-office.xls;application/wps-office.xlt;application/vnd.ms-excel;application/msexcel;application/x-msexcel;application/wps-office.xlsx;application/wps-office.xltx;application/vnd.openxmlformats-officedocument.spreadsheetml.sheet
--- a/usr/share/applications/wps-office-pdf.desktop	2019-09-10 17:59:37.000000000 +0800
+++ b/usr/share/applications/wps-office-pdf.desktop	2019-10-29 20:33:45.314824380 +0800
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Comment=Kingsoft Pdf Reader
 Comment[zh_CN]=金山PDF阅读器
-Exec=/usr/bin/wpspdf %f
+Exec=/usr/bin/wpspdf %F
 GenericName=Kingsoft Pdf Reader
 GenericName[zh_CN]=金山 PDF
 MimeType=application/wps-office.pdf;application/pdf
--- a/usr/share/applications/wps-office-wpp.desktop	2019-09-10 17:59:37.000000000 +0800
+++ b/usr/share/applications/wps-office-wpp.desktop	2019-10-29 20:33:58.498157713 +0800
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Comment=Use WPS Presentation to edit and play presentations.
 Comment[zh_CN]=使用 WPS 演示编辑、播放演示文稿
-Exec=/usr/bin/wpp %f
+Exec=/usr/bin/wpp %F
 GenericName=WPS Presentation
 GenericName[zh_CN]=WPS 演示
 MimeType=application/wps-office.dps;application/wps-office.dpt;application/wps-office.dpss;application/wps-office.dpso;application/wps-office.ppt;application/wps-office.pot;application/vnd.ms-powerpoint;application/vnd.mspowerpoint;application/mspowerpoint;application/powerpoint;application/x-mspowerpoint;application/wps-office.pptx;application/wps-office.potx;application/vnd.openxmlformats-officedocument.presentationml.presentation;application/vnd.openxmlformats-officedocument.presentationml.slideshow
--- a/usr/share/applications/wps-office-wps.desktop	2019-09-10 17:59:37.000000000 +0800
+++ b/usr/share/applications/wps-office-wps.desktop	2019-10-29 20:33:25.658157711 +0800
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Comment=Use WPS Writer to edit articles and reports.
 Comment[zh_CN]=使用 WPS 文字编写报告，排版文章
-Exec=/usr/bin/wps %f
+Exec=/usr/bin/wps %F
 GenericName=WPS Writer
 GenericName[zh_CN]=WPS 文字
 MimeType=application/wps-office.wps;application/wps-office.wpt;application/wps-office.wpso;application/wps-office.wpss;application/wps-office.doc;application/wps-office.dot;application/vnd.ms-word;application/msword;application/x-msword;application/msword-template;application/wps-office.docx;application/wps-office.dotx;application/rtf;application/vnd.ms-word.document.macroEnabled.12;application/vnd.openxmlformats-officedocument.wordprocessingml.document
