diff --unified --recursive bsnes_v108-source.orig/bsnes/GNUmakefile bsnes_v108-source.new/bsnes/GNUmakefile
--- bsnes_v108-source.orig/bsnes/GNUmakefile.orig	2020-03-03 18:56:48.000000000 +0800
+++ bsnes_v108-source.new/bsnes/GNUmakefile	2021-04-18 17:45:28.350429325 +0800
@@ -1,8 +1,11 @@
+flags = $(CFLAGS)
+options = $(LDFLAGS)
+
 target := bsnes
 binary := application
-build := performance
+build := release
 openmp := true
-local := true
+local := false
 flags += -I. -I..
 
 # in order for this to work, obj/lzma.o must be omitted or bsnes will hang on startup.

diff --unified --recursive bsnes_v113-source.orig/bsnes/target-bsnes/GNUmakefile bsnes_v113-source.new/bsnes/target-bsnes/GNUmakefile
--- bsnes_v113-source.orig/bsnes/target-bsnes/GNUmakefile       2019-12-11 12:19:18.782618740 -0700
+++ bsnes_v113-source.new/bsnes/target-bsnes/GNUmakefile        2019-12-11 12:20:36.969034855 -0700
@@ -40,8 +40,6 @@

 install: all
 ifeq ($(platform),windows)
-else ifeq ($(shell id -un),root)
-       $(error "make install should not be run as root")
 else ifeq ($(platform),macos)
        mkdir -p ~/Library/Application\ Support/$(name)/
        mkdir -p ~/Library/Application\ Support/$(name)/Database/
