From ecead5a765ddcbee09a2d90daae8e9ab609dc193 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Sun, 17 Sep 2017 07:27:24 -0400
Subject: [PATCH] Add gc_preserve_* as valid Expr head.

---
 base/codevalidation.jl | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/base/codevalidation.jl b/base/codevalidation.jl
index 433eaa7bab..84283bd2c9 100644
--- a/base/codevalidation.jl
+++ b/base/codevalidation.jl
@@ -24,7 +24,9 @@ const VALID_EXPR_HEADS = ObjectIdDict(
     :global => 1:1,
     :foreigncall => 3:typemax(Int),
     :isdefined => 1:1,
-    :simdloop => 0:0
+    :simdloop => 0:0,
+    :gc_preserve_begin => 0:typemax(Int),
+    :gc_preserve_end => 0:typemax(Int)
 )
 
 # @enum isn't defined yet, otherwise I'd use it for this
-- 
2.14.1

