From 0903653415304d5c975e5c11baac1d82b39ac0ca Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Wed, 23 Sep 2015 13:36:33 -0400
Subject: [PATCH] Use tbaa_immut for getfield of immutable types

---
 src/cgutils.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cgutils.cpp b/src/cgutils.cpp
index ee71e8b..3ddcf7a 100644
--- a/src/cgutils.cpp
+++ b/src/cgutils.cpp
@@ -1361,7 +1361,7 @@ static jl_cgval_t emit_getfield_knownidx(const jl_cgval_t &strct, unsigned idx,
             LLVM37_param(cast<PointerType>(strct.V->getType()->getScalarType())->getElementType())
             strct.V, 0, idx);
         assert(!jt->mutabl);
-        return typed_load(addr, NULL, jfty, ctx, NULL);
+        return typed_load(addr, NULL, jfty, ctx, tbaa_immut);
     }
     else {
         assert(strct.V->getType()->isVectorTy());
-- 
2.5.1

