From 9ecfc2dce03c2b7af887ecff093bf00e4b89ce6f Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Sat, 14 Jan 2017 01:09:49 -0500
Subject: [PATCH] Workaround gdb jit debugging bug

---
 src/jitlayers.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/jitlayers.cpp b/src/jitlayers.cpp
index 0c4ba6a..fb53337 100644
--- a/src/jitlayers.cpp
+++ b/src/jitlayers.cpp
@@ -282,9 +282,9 @@ extern "C" {
   // We put information about the JITed function in this global, which the
   // debugger reads.  Make sure to specify the version statically, because the
   // debugger checks the version before we can set it during runtime.
-  extern struct jit_descriptor __jit_debug_descriptor;
+  JL_DLLEXPORT struct jit_descriptor __jit_debug_descriptor;
 
-  LLVM_ATTRIBUTE_NOINLINE extern void __jit_debug_register_code();
+  JL_DLLEXPORT LLVM_ATTRIBUTE_NOINLINE void __jit_debug_register_code() {}
 }
 
 namespace {
-- 
2.10.2

