From 3a0abed35320b0e9b57ceb9cbc31b044bb2f809c Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Sun, 16 Apr 2017 15:27:37 -0400
Subject: [PATCH 1/5] Optimize `Threads.threadid()`

---
 src/ccall.cpp | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/src/ccall.cpp b/src/ccall.cpp
index f881411d39..4691aeb4a6 100644
--- a/src/ccall.cpp
+++ b/src/ccall.cpp
@@ -1676,6 +1676,20 @@ static jl_cgval_t emit_ccall(jl_value_t **args, size_t nargs, jl_codectx_t *ctx)
             emit_bitcast(ctx->ptlsStates, lrt),
             retboxed, rt, unionall, static_rt, ctx);
     }
+    if (fptr == (void(*)(void))&jl_threadid ||
+        ((!f_lib || (intptr_t)f_lib == 2) && f_name &&
+         strcmp(f_name, "jl_threadid") == 0)) {
+        assert(lrt == T_int16);
+        assert(!isVa && !llvmcall);
+        assert(nargt == 0);
+        JL_GC_POP();
+        Value *ptls_i16 = emit_bitcast(ctx->ptlsStates, T_pint16);
+        const int tid_offset = offsetof(jl_tls_states_t, tid);
+        Value *ptid = builder.CreateGEP(ptls_i16, ConstantInt::get(T_size, tid_offset / 2));
+        return mark_or_box_ccall_result(
+            tbaa_decorate(tbaa_const, builder.CreateLoad(ptid)),
+            retboxed, rt, unionall, static_rt, ctx);
+    }
     if (fptr == &jl_sigatomic_begin ||
         ((!f_lib || (intptr_t)f_lib == 2) && f_name &&
          strcmp(f_name, "jl_sigatomic_begin") == 0)) {
-- 
2.12.2

