From 3f6abadc9d12bb9d59ec775db4277b3841395d0b Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Wed, 13 Jan 2016 08:58:38 -0500
Subject: [PATCH] Fix compilation on llvm-svn due to function renaming

---
 src/intrinsics.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/intrinsics.cpp b/src/intrinsics.cpp
index 4433c0c..7b8bf24 100644
--- a/src/intrinsics.cpp
+++ b/src/intrinsics.cpp
@@ -863,12 +863,20 @@ struct math_builder {
              jl_options.fast_math == JL_OPTIONS_FAST_MATH_ON)) {
             FastMathFlags fmf;
             fmf.setUnsafeAlgebra();
+#ifdef LLVM38
+            builder.setFastMathFlags(fmf);
+#else
             builder.SetFastMathFlags(fmf);
+#endif
         }
     }
     IRBuilder<>& operator()() const { return builder; }
     ~math_builder() {
+#ifdef LLVM38
+        builder.setFastMathFlags(old_fmf);
+#else
         builder.SetFastMathFlags(old_fmf);
+#endif
     }
 };
 
-- 
2.6.4

