From 2b72fbb0c2bea51ad9e2a5e93ff3326bdc8e0e8f Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Wed, 19 Apr 2017 23:04:20 -0400
Subject: [PATCH 2/5] Allow armv8 and armv8-a as CPU targets on aarch64

[ci skip]
---
 src/codegen.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/codegen.cpp b/src/codegen.cpp
index c694299a88..a42f4923ff 100644
--- a/src/codegen.cpp
+++ b/src/codegen.cpp
@@ -7111,7 +7111,11 @@ static inline void checkARMArchFeature(std::string &cpu,
         HostFeatures[cpu] = true;
     }
 #else
-    HostFeatures[cpu.substr(3)] = true;
+    // These two are allowed on 32bit. Allow them on 64bits too for consistency since
+    // they basically mean "generic" on aarch64.
+    // In particular, "armv8-a" is the generic value for the GCC `-march` option.
+    if (cpu != "armv8" && cpu != "armv8-a")
+        HostFeatures[cpu.substr(3)] = true;
 #endif
     cpu = "generic";
 }
-- 
2.12.2

