diff --git a/src/MEGASync/control/CrashHandler.cpp b/src/MEGASync/control/CrashHandler.cpp
index 2040102f..bd8b2592 100644
--- a/src/MEGASync/control/CrashHandler.cpp
+++ b/src/MEGASync/control/CrashHandler.cpp
@@ -209,7 +209,7 @@ string getDistroVersion()
         oss << "Error info:\n";
         if (info)
         {
-            oss << sys_siglist[sig] << " (" << sig << ") at address " << std::showbase << std::hex << info->si_addr << std::dec << "\n";
+            oss << strsignal(sig) << " (" << sig << ") at address " << std::showbase << std::hex << info->si_addr << std::dec << "\n";
         }
         else
         {
