From 499aa81a68fb4c8aac1c80f0d6a4cce05941c4cc Mon Sep 17 00:00:00 2001
From: Bernd Kuhls <>
Date: Wed, 26 Nov 2014 19:25:28 +0200
Subject: [PATCH] Build: Support libpng 1.6

---
 wscript | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wscript b/wscript
index 50dc5ee..0354210 100644
--- a/wscript
+++ b/wscript
@@ -94,7 +94,7 @@ def configure(ctx):
                       uselib = uselib, mandatory = True)
 
     # Check for a supported version of libpng
-    supp_png_pkgs = (('libpng12', '1.2'), ('libpng15', '1.5'),)
+    supp_png_pkgs = (('libpng12', '1.2'), ('libpng15', '1.5'), ('libpng16', '1.6'),)
     have_png = False
     for (pkg, atleast) in supp_png_pkgs:
         try:
