diff --git a/pcsxr/data/pcsxr.desktop b/pcsxr/data/pcsxr.desktop
index c284f003..36940bc1 100644
--- a/pcsxr/data/pcsxr.desktop
+++ b/pcsxr/data/pcsxr.desktop
@@ -3,13 +3,13 @@ Version=1.0
 Type=Application
 Name=PCSXR
 GenericName=PlayStation Emulator
-GenericName[ru_RU]=Эмулятор PlayStation
+GenericName[ru]=Эмулятор PlayStation
 GenericName[zh_CN]=PlayStation 模拟器
 GenericName[zh_TW]=PlayStation 模擬器
 Comment=Sony PlayStation emulator
-Comment[ru_RU]=Эмулятор Sony PlayStation
+Comment[ru]=Эмулятор Sony PlayStation
 Comment[zh_CN]=Sony PlayStation 模拟器
 Comment[zh_TW]=Sony PlayStation 模擬器
 Exec=pcsxr
 Icon=pcsxr-icon
-Categories=Game;
+Categories=Game;Emulator;
