From b833229d41c9f18fb88c1b0c830037991da88ad2 Mon Sep 17 00:00:00 2001
From: Hu Butui <hot123tea123@gmail.com>
Date: Wed, 10 Dec 2025 09:59:33 +0800
Subject: [PATCH] COMP: fix QT_PLUGINS_DIR detection using qmake

Signed-off-by: Hu Butui <hot123tea123@gmail.com>
---
 CMake/SlicerBlockFindQtAndCheckVersion.cmake | 15 ++++++++++++---
 1 file changed, 12 insertions(+), 3 deletions(-)

diff --git a/CMake/SlicerBlockFindQtAndCheckVersion.cmake b/CMake/SlicerBlockFindQtAndCheckVersion.cmake
index 86cd7f02d0..5f1dbf1cc7 100644
--- a/CMake/SlicerBlockFindQtAndCheckVersion.cmake
+++ b/CMake/SlicerBlockFindQtAndCheckVersion.cmake
@@ -87,14 +87,23 @@ message(STATUS "Configuring ${_project_name} with Qt ${_major}.${_minor}.${_patc
 
 # Since neither Qt5 or Qt6 set CMake variables for plugins and binary directories,
 # we explicitly set them here.
-set(QT_PLUGINS_DIR "${Qt${_major}_DIR}/../../../plugins")
-get_filename_component(QT_PLUGINS_DIR ${QT_PLUGINS_DIR} ABSOLUTE)
-message(STATUS "Setting QT_PLUGINS_DIR: ${QT_PLUGINS_DIR}")
 
 set(QT_BINARY_DIR "${Qt${_major}_DIR}/../../../bin")
 get_filename_component(QT_BINARY_DIR ${QT_BINARY_DIR} ABSOLUTE)
 message(STATUS "Setting QT_BINARY_DIR: ${QT_BINARY_DIR}")
 
+find_program(QMAKE_EXECUTABLE NAMES qmake-qt${_major} qmake${_major} qmake PATHS ${QT_BINARY_DIR} NO_DEFAULT_PATH)
+execute_process(
+    COMMAND ${QMAKE_EXECUTABLE} -query QT_INSTALL_PLUGINS
+    OUTPUT_VARIABLE QT_PLUGINS_DIR
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    RESULT_VARIABLE qmake_result
+)
+if(NOT qmake_result EQUAL 0 OR NOT EXISTS "${QT_PLUGINS_DIR}")
+    message(FATAL_ERROR "Failed to get Qt plugins directory using qmake")
+endif()
+message(STATUS "Setting QT_PLUGINS_DIR: ${QT_PLUGINS_DIR}")
+
 set(QT_LIBRARY_DIR "${Qt${_major}_DIR}/../../../lib")
 get_filename_component(QT_LIBRARY_DIR ${QT_LIBRARY_DIR} ABSOLUTE)
 message(STATUS "Setting QT_LIBRARY_DIR: ${QT_LIBRARY_DIR}")
-- 
2.52.0

