From 5599e51522a95b49ed2571260d8fbca4cab6c596 Mon Sep 17 00:00:00 2001
From: Hu Butui <hot123tea123@gmail.com>
Date: Mon, 29 Nov 2021 22:13:00 +0800
Subject: [PATCH] fix vtk limits

---
 SuperBuild/External_VTK.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SuperBuild/External_VTK.cmake b/SuperBuild/External_VTK.cmake
index cc095bc40..4c43e83e1 100644
--- a/SuperBuild/External_VTK.cmake
+++ b/SuperBuild/External_VTK.cmake
@@ -226,6 +226,7 @@ if((NOT DEFINED VTK_DIR OR NOT DEFINED VTK_SOURCE_DIR) AND NOT Slicer_USE_SYSTEM
     INSTALL_COMMAND ""
     DEPENDS
       ${${proj}_DEPENDENCIES}
+    PATCH_COMMAND bash -c "grep -lR 'std::numeric_limits'|xargs sed -i '1 i\#include <limits>'"
     )
 
   if(Slicer_USE_PYTHONQT AND NOT Slicer_USE_SYSTEM_python)
-- 
2.32.0

