diff --git a/utils/test/test-aa-notify.py b/utils/test/test-aa-notify.py
index d706edf2..a8f1fa86 100644
--- a/utils/test/test-aa-notify.py
+++ b/utils/test/test-aa-notify.py
@@ -198,14 +198,13 @@ Display AppArmor notifications or messages for DENIED entries.
   -p, --poll            poll AppArmor logs and display notifications
   --display DISPLAY     set the DISPLAY environment variable (might be needed if
                         sudo resets $DISPLAY)
-  -f FILE, --file FILE  search FILE for AppArmor messages
+  -f, --file FILE       search FILE for AppArmor messages
   -l, --since-last      display stats since last login
-  -s NUM, --since-days NUM
-                        show stats for last NUM days (can be used alone or with
+  -s, --since-days NUM  show stats for last NUM days (can be used alone or with
                         -p)
   -v, --verbose         show messages with stats
-  -u USER, --user USER  user to drop privileges to when not using sudo
-  -w NUM, --wait NUM    wait NUM seconds before displaying notifications (with
+  -u, --user USER       user to drop privileges to when not using sudo
+  -w, --wait NUM        wait NUM seconds before displaying notifications (with
                         -p)
   --prompt-filter PF    kind of operations which display a popup prompt
   --debug               debug mode
