From 9eb1fefff61bb66413dfdf63847968e257d6cb0b Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Sun, 18 Feb 2024 18:43:54 -0500
Subject: [PATCH] fix gsl64 patching script

---
 enzyme/BCLoad/fixgsl64.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/enzyme/BCLoad/fixgsl64.sh b/enzyme/BCLoad/fixgsl64.sh
index 06fd4d6b..82a6c3b2 100755
--- a/enzyme/BCLoad/fixgsl64.sh
+++ b/enzyme/BCLoad/fixgsl64.sh
@@ -15,7 +15,7 @@ case $OS in
   sed -i.bu "1s/^/#include <stdint.h>\n/" cblas/*.c
   ;;
 *)
-  sed "s/int/int64_t/g" -i cblas/*.c
+  sed "s/\bint\b/int64_t/g" -i cblas/*.c
   sed "s/gsl\//gsl64\//g" -i cblas/*.c cblas/*.h gsl_math.h gsl_precision.h gsl_pow_int.h gsl_minmax.h
   sed "s/cblas_\(.*\) (/cblas_\164_ (/g" -i cblas/*.c
   sed "1s/^/#include <stdint.h>\n/" -i cblas/*.c
-- 
2.55.0

