diff --git a/src/3rdParty/salomesmesh/inc/SMESH_Algo.hxx b/src/3rdParty/salomesmesh/inc/SMESH_Algo.hxx
index a362ed290..7e23608f5 100644
--- a/src/3rdParty/salomesmesh/inc/SMESH_Algo.hxx
+++ b/src/3rdParty/salomesmesh/inc/SMESH_Algo.hxx
@@ -30,6 +30,7 @@
 
 #include "SMESH_SMESH.hxx"
 
+#include "SMDS_MeshNode.hxx"
 #include "SMDSAbs_ElementType.hxx"
 #include "SMESH_Comment.hxx"
 #include "SMESH_ComputeError.hxx"
@@ -45,7 +46,6 @@
 #include <map>
 #include <set>
 
-class SMDS_MeshNode;
 class SMESHDS_Mesh;
 class SMESHDS_SubMesh;
 class SMESH_Gen;
