--- third_party/webrtc/modules/desktop_capture/linux/window_capturer_x11.cc.orig	2019-01-20 10:57:09.166224517 +0000
+++ third_party/webrtc/modules/desktop_capture/linux/window_capturer_x11.cc	2019-01-20 06:01:05.901545307 +0000
@@ -14,6 +14,7 @@
 #include <X11/extensions/Xcomposite.h>
 #include <X11/extensions/Xrender.h>
 
+#include <string.h>
 #include <memory>
 #include <string>
 #include <utility>
