--- a/media/gpu/vaapi/vaapi_video_decode_accelerator.cc	2020-06-15 11:58:38.000000000 -0700
+++ b/media/gpu/vaapi/vaapi_video_decode_accelerator.cc	2020-06-23 17:18:20.219862776 -0700
@@ -1163,6 +1163,9 @@
   // of frames needed by the client pipeline (see b/133733739).
   // TODO(crbug.com/911754): Enable for VP9 Profile 2.
   if (IsGeminiLakeOrLater() &&
+#if defined(OS_LINUX)
+      false &&
+#endif
       (profile_ == VP9PROFILE_PROFILE0 || profile_ == VP8PROFILE_ANY)) {
     // Add one to the reference frames for the one being currently egressed, and
     // an extra allocation for both |client_| and |decoder_|, see
