--- a/pkgsrc/pkgtools/mtree/files-build/pack_dev.c	2018-08-12 00:20:05.968656198 -0400
+++ b/pkgsrc/pkgtools/mtree/files-build/pack_dev.c	2018-08-12 00:22:26.052941801 -0400
@@ -57,6 +57,11 @@
 #if HAVE_SYS_STAT_H
 #include <sys/stat.h>
 #endif
+/* Arch AUR packages presume glibc installed.
+   'major' and 'minor' have been moved to
+   sysmacros.h in some recent version of glibc.
+*/
+#include <sys/sysmacros.h>
 
 #if HAVE_LIMITS_H
 #include <limits.h>

