--- linux-3.10/drivers/cpufreq/intel_pstate.c	2013-06-30 18:13:29.000000000 -0400
+++ linux-3.10.mod/drivers/cpufreq/intel_pstate.c	2013-07-05 03:10:36.164568840 -0400
@@ -522,6 +522,11 @@
 	ICPU(0x2a, default_policy),
 	ICPU(0x2d, default_policy),
 	ICPU(0x3a, default_policy),
+	ICPU(0x3a, default_policy),
+	ICPU(0x3c, default_policy),
+	ICPU(0x3f, default_policy),
+	ICPU(0x45, default_policy),
+	ICPU(0x46, default_policy),
 	{}
 };
 MODULE_DEVICE_TABLE(x86cpu, intel_pstate_cpu_ids);
--- linux-3.10/drivers/cpufreq/Kconfig.x86	2013-06-30 18:13:29.000000000 -0400
+++ linux-3.10.mod/drivers/cpufreq/Kconfig.x86	2013-07-05 03:13:22.823827792 -0400
@@ -6,12 +6,12 @@
        bool "Intel P state control"
        depends on X86
        help
-          This driver provides a P state for Intel core processors.
+          This driver provides a P state for Intel Core processors.
 	  The driver implements an internal governor and will become
-          the scaling driver and governor for Sandy bridge processors.
+          the scaling driver and governor for Sandy/Ivy Bridge and Haswell processors.
 
 	  When this driver is enabled it will become the perferred
-          scaling driver for Sandy bridge processors.
+          scaling driver for Sandy/Ivy Bridge and Haswell processors.
 
 	  If in doubt, say N.
 
