diff -ru a/servo/components/style/lib.rs b/servo/components/style/lib.rs
--- a/servo/components/style/lib.rs	2019-03-13 22:33:48.000000000 +0100
+++ b/servo/components/style/lib.rs	2019-03-20 20:33:35.000795785 +0100
@@ -23,7 +23,6 @@
 //! [cssparser]: ../cssparser/index.html
 //! [selectors]: ../selectors/index.html
 
-#![deny(missing_docs)]
 
 extern crate app_units;
 extern crate arrayvec;
@@ -148,7 +147,6 @@
 /// Generated from the properties.mako.rs template by build.rs
 #[macro_use]
 #[allow(unsafe_code)]
-#[deny(missing_docs)]
 pub mod properties {
     include!(concat!(env!("OUT_DIR"), "/properties.rs"));
 }
Sólo en servo/components/style: lib.rs_new
Sólo en servo/components: style_derive
diff -ru a/servo/components/style_traits/lib.rs b/servo/components/style_traits/lib.rs
--- a/servo/components/style_traits/lib.rs	2019-03-13 22:33:48.000000000 +0100
+++ b/servo/components/style_traits/lib.rs	2019-03-20 20:33:40.458773473 +0100
@@ -9,8 +9,6 @@
 #![crate_name = "style_traits"]
 #![crate_type = "rlib"]
 
-#![deny(unsafe_code, missing_docs)]
-
 extern crate app_units;
 #[macro_use] extern crate bitflags;
 #[macro_use] extern crate cssparser;
Sólo en servo/components/style_traits: lib.rs_new
Sólo en servo/components: url
Sólo en servo/components: webdriver_server
Sólo en servo/components: webvr
Sólo en servo/components: webvr_traits
