diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4380d39..d0cf49e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -91,6 +91,11 @@ else()
       FATAL_ERROR "Could not find Qt ${USE_QT_VERSION}, aborting compilation.")
   endif()
 
+  # https://doc.qt.io/qt-6/whatsnew610.html#build-system-changes
+  if(QT_VERSION VERSION_GREATER_EQUAL 6.10)
+    list(APPEND QT_COMPONENTS CorePrivate GuiPrivate WidgetsPrivate)
+  endif()
+
   find_package(Qt${USE_QT_VERSION} REQUIRED COMPONENTS ${QT_COMPONENTS})
 endif()
 
