From 988a82dfb61b9f2fb31785a5335225e00c8bd2f0 Mon Sep 17 00:00:00 2001
From: axionl <i@axionl.me>
Date: Fri, 7 Dec 2018 11:29:17 +0900
Subject: [PATCH 2/2] fixed_libdrm_not_found

---
 src/login/logind-session-device.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/login/logind-session-device.c b/src/login/logind-session-device.c
index 0fcbe98c2a..601d96129c 100644
--- a/src/login/logind-session-device.c
+++ b/src/login/logind-session-device.c
@@ -8,7 +8,7 @@
 #include <sys/types.h>
 
 /* Old drm.h may needs to be included after stdint.h and sys/types.h */
-#include <drm/drm.h>
+#include <libdrm/drm.h>
 
 #include "sd-device.h"
 
-- 
2.19.2

