diff -upr xrdp-git/instfiles/Makefile.am xrdp-git-new/instfiles/Makefile.am
--- xrdp-git/instfiles/Makefile.am	2015-04-24 16:38:02.092787754 -0700
+++ xrdp-git-new/instfiles/Makefile.am	2015-04-24 16:47:21.346145399 -0700
@@ -4,14 +4,16 @@ xrdp.service
 
 SUBDIRS = \
   pam.d \
-  init.d \
-  default \
   pulse
 
 if HAVE_SYSTEMD
 systemdsystemunit_DATA = \
         xrdp-sesman.service \
         xrdp.service
+else
+SUBDIRS+= \
+  init.d \
+  default
 endif
 
 startscriptdir=$(sysconfdir)/xrdp
@@ -35,5 +37,7 @@ startscript_DATA = \
 # must be tab below
 install-data-hook:
 	chmod 755 $(DESTDIR)$(sysconfdir)/xrdp/xrdp.sh
-	chmod 755 $(DESTDIR)$(sysconfdir)/init.d/xrdp
-	sed -i 's|__BASE__|$(prefix)|' $(DESTDIR)$(sysconfdir)/init.d/xrdp;
+	if [ -f $(DESTDIR)$(sysconfdir)/init.d/xrdp ]; then \
+	  chmod 755 $(DESTDIR)$(sysconfdir)/init.d/xrdp \
+	  sed -i 's|__BASE__|$(prefix)|' $(DESTDIR)$(sysconfdir)/init.d/xrdp; \
+	fi
diff -upr xrdp-git/instfiles/pam.d/xrdp-sesman xrdp-git-new/instfiles/pam.d/xrdp-sesman
--- xrdp-git/instfiles/pam.d/xrdp-sesman	2015-04-24 16:38:02.096121088 -0700
+++ xrdp-git-new/instfiles/pam.d/xrdp-sesman	2015-04-24 16:44:57.272805802 -0700
@@ -1,5 +1,9 @@
 #%PAM-1.0
-@include common-auth
-@include common-account
-@include common-session
-@include common-password
+#@include common-auth
+#@include common-account
+#@include common-session
+#@include common-password
+auth include system-auth
+account include system-auth
+password include system-auth
+session include system-auth
diff -upr xrdp-git/sesman/startwm.sh xrdp-git-new/sesman/startwm.sh
--- xrdp-git/sesman/startwm.sh	2015-04-24 16:38:02.116121089 -0700
+++ xrdp-git-new/sesman/startwm.sh	2015-04-24 16:43:36.872802307 -0700
@@ -8,6 +8,17 @@ wm_start()
     export LANG LANGUAGE
   fi
 
+  # arch user
+  if [ -r ~/.xinitrc ]; then
+    . ~/.xinitrc
+    exit 0
+  fi
+  # arch
+  if [ -r /etc/X11/xinit/xinitrc ]; then
+    . /etc/X11/xinit/xinitrc
+    exit 0
+  fi
+
   # debian
   if [ -r /etc/X11/Xsession ]; then
     . /etc/X11/Xsession
diff -upr xrdp-git/xrdp/xrdp.ini xrdp-git-new/xrdp/xrdp.ini
--- xrdp-git/xrdp/xrdp.ini	2015-04-24 16:38:02.152787757 -0700
+++ xrdp-git-new/xrdp/xrdp.ini	2015-04-24 16:46:19.786142723 -0700
@@ -51,7 +51,7 @@ grey=dedede
 # require_credentials=yes

 # set a default entry for autorun if the client send login and pass directly
-autorun=xrdp1
+autorun=xrdp2

 bulk_compression=yes

@@ -139,6 +139,15 @@ tcutils=true
 # for debugging xrdp, add following line to section xrdp1
 # chansrvport=/tmp/.xrdp/xrdp_chansrv_socket_7210

+[xrdp0]
+name=sesman-Xvnc-delay
+lib=libvnc.so
+username=ask
+password=ask
+ip=127.0.0.1
+port=-1
+delay_ms=2000
+
 [xrdp1]
 name=sesman-X11rdp
 lib=libxup.so
