diff -up src/xrdp-0.8.0/instfiles/Makefile.am.orig src/xrdp-0.8.0/instfiles/Makefile.am
--- src/xrdp-0.8.0/instfiles/Makefile.am.orig	2014-05-19 23:41:42.000000000 -0600
+++ src/xrdp-0.8.0/instfiles/Makefile.am	2015-12-04 06:39:06.000000000 -0700
@@ -28,5 +28,7 @@ startscript_DATA = \
 # must be tab below
 install-data-hook:
 	chmod 755 $(DESTDIR)$(sysconfdir)/xrdp/xrdp.sh
-	chmod 755 $(DESTDIR)$(sysconfdir)/init.d/xrdp
-	sed -i 's|__BASE__|$(prefix)|' $(DESTDIR)$(sysconfdir)/init.d/xrdp;
+	if [ -f $(DESTDIR)$(sysconfdir)/init.d/xrdp ]; then \
+	  chmod 755 $(DESTDIR)$(sysconfdir)/init.d/xrdp; \
+	  sed -i 's|__BASE__|$(prefix)|' $(DESTDIR)$(sysconfdir)/init.d/xrdp; \
+	fi
diff -up src/xrdp-0.8.0/instfiles/pam.d/xrdp-sesman.orig src/xrdp-0.8.0/instfiles/pam.d/xrdp-sesman
--- src/xrdp-0.8.0/instfiles/pam.d/xrdp-sesman.orig	2014-05-19 23:41:42.000000000 -0600
+++ src/xrdp-0.8.0/instfiles/pam.d/xrdp-sesman	2015-12-04 06:39:06.000000000 -0700
@@ -1,5 +1,9 @@
 #%PAM-1.0
-@include common-auth
-@include common-account
-@include common-session
-@include common-password
+#@include common-auth
+#@include common-account
+#@include common-session
+#@include common-password
+auth include system-auth
+account include system-auth
+password include system-auth
+session include system-auth
diff -up src/xrdp-0.8.0/instfiles/xrdp.service.orig src/xrdp-0.8.0/instfiles/xrdp.service
--- src/xrdp-0.8.0/instfiles/xrdp.service.orig	2015-12-04 06:39:06.000000000 -0700
+++ src/xrdp-0.8.0/instfiles/xrdp.service	2015-12-04 06:40:54.000000000 -0700
@@ -6,9 +6,8 @@ After=syslog.target network.target xrdp-
 [Service]
 Type=forking
 PIDFile=/var/run/xrdp.pid
-EnvironmentFile=/etc/sysconfig/xrdp
-ExecStart=/usr/sbin/xrdp $XRDP_OPTIONS
-ExecStop=/usr/sbin/xrdp $XRDP_OPTIONS --kill
+ExecStart=/usr/bin/xrdp
+ExecStop=/usr/bin/xrdp --kill
 
 [Install]
 WantedBy=multi-user.target
diff -up src/xrdp-0.8.0/instfiles/xrdp-sesman.service.orig src/xrdp-0.8.0/instfiles/xrdp-sesman.service
--- src/xrdp-0.8.0/instfiles/xrdp-sesman.service.orig	2015-12-04 06:39:06.000000000 -0700
+++ src/xrdp-0.8.0/instfiles/xrdp-sesman.service	2015-12-04 06:41:09.000000000 -0700
@@ -6,9 +6,8 @@ StopWhenUnneeded=true
 [Service]
 Type=forking
 PIDFile=/var/run/xrdp-sesman.pid
-EnvironmentFile=/etc/sysconfig/xrdp
-ExecStart=/usr/sbin/xrdp-sesman $SESMAN_OPTIONS
-ExecStop=/usr/sbin/xrdp-sesman $SESMAN_OPTIONS --kill
+ExecStart=/usr/bin/xrdp-sesman
+ExecStop=/usr/bin/xrdp-sesman --kill
 
 [Install]
 WantedBy=multi-user.target
diff -up src/xrdp-0.8.0/instfiles/xrdp.sh.orig src/xrdp-0.8.0/instfiles/xrdp.sh
--- src/xrdp-0.8.0/instfiles/xrdp.sh.orig	2015-12-04 06:39:06.000000000 -0700
+++ src/xrdp-0.8.0/instfiles/xrdp.sh	2015-12-04 06:39:06.000000000 -0700
@@ -17,7 +17,7 @@
 # Description: starts xrdp
 ### END INIT INFO
 
-SBINDIR=/usr/local/sbin
+SBINDIR=/usr/bin
 LOG=/dev/null
 CFGDIR=/etc/xrdp
 
diff -up src/xrdp-0.8.0/sesman/startwm.sh.orig src/xrdp-0.8.0/sesman/startwm.sh
--- src/xrdp-0.8.0/sesman/startwm.sh.orig	2014-05-19 23:41:42.000000000 -0600
+++ src/xrdp-0.8.0/sesman/startwm.sh	2015-12-04 06:39:06.000000000 -0700
@@ -8,6 +8,17 @@ wm_start()
     export LANG LANGUAGE
   fi
 
+  # arch user
+  if [ -r ~/.xinitrc ]; then
+    . ~/.xinitrc
+    exit 0
+  fi
+  # arch
+  if [ -r /etc/X11/xinit/xinitrc ]; then
+    . /etc/X11/xinit/xinitrc
+    exit 0
+  fi
+
   # debian
   if [ -r /etc/X11/Xsession ]; then
     . /etc/X11/Xsession
diff -up src/xrdp-0.8.0/xrdp/xrdp.ini.orig src/xrdp-0.8.0/xrdp/xrdp.ini
--- src/xrdp-0.8.0/xrdp/xrdp.ini.orig	2014-05-19 23:41:42.000000000 -0600
+++ src/xrdp-0.8.0/xrdp/xrdp.ini	2015-12-04 06:39:06.000000000 -0700
@@ -36,7 +36,7 @@ grey=dedede
 #red=ff0000
 #green=00ff00
 #background=626c72
-#autorun=xrdp1
+autorun=xrdp2
 #hidelogwindow=yes
 
 # when true, userid/password *must* be passed on cmd line
@@ -119,6 +119,15 @@ xrdpvr=true
 # for debugging xrdp, add following line to section xrdp1
 # chansrvport=/tmp/.xrdp/xrdp_chansrv_socket_7210
 
+[xrdp0]
+name=sesman-Xvnc-delay
+lib=libvnc.so
+username=ask
+password=ask
+ip=127.0.0.1
+port=-1
+delay_ms=2000
+
 [xrdp1]
 name=sesman-X11rdp
 lib=libxup.so
