diff -up src/ocserv-0.3.2/configure.ac.orig src/ocserv-0.3.2/configure.ac
--- src/ocserv-0.3.2/configure.ac.orig	2014-03-30 06:46:56.000000000 -0600
+++ src/ocserv-0.3.2/configure.ac	2014-03-30 06:56:15.000000000 -0600
@@ -65,9 +65,11 @@ AM_CONDITIONAL(HAVE_DBUS, test "x$dbus_e
 
 if test "x$dbus_enabled" != xno;then
 dbus_conf_dir=$($PKG_CONFIG dbus-1 --variable=sysconfdir)
-AC_SUBST(DBUS_CONFIG_DIR, [$dbus_conf_dir/etc/dbus-1/system.d/])
+AC_SUBST(dbusconfdir, [$dbus_conf_dir/dbus-1/system.d/])
 fi
 
+AC_SUBST(systemddir, ['${libdir}/systemd/system'])
+
 AC_ARG_WITH(protobuf,
   AS_HELP_STRING([--without-protobuf], [use the included protobuf library]),
   test_for_protobuf=$withval,
diff -up src/ocserv-0.3.2/doc/Makefile.am.orig src/ocserv-0.3.2/doc/Makefile.am
--- src/ocserv-0.3.2/doc/Makefile.am.orig	2014-03-30 06:47:13.000000000 -0600
+++ src/ocserv-0.3.2/doc/Makefile.am	2014-03-30 06:56:08.000000000 -0600
@@ -20,6 +20,9 @@ ocpasswd.8: ../src/ocpasswd-args.def
 	@AUTOGEN@ -L../src -DMAN_SECTION=8 -Tagman-cmd.tpl "$<".tmp && \
 	rm -f "$<".tmp
 
-install-data-local:
-	-test -d "$(DBUS_CONFIG_DIR)" && ! test -f $(DBUS_CONFIG_DIR)/org.infradead.ocserv.conf && $(INSTALL_DATA) dbus/org.infradead.ocserv.conf $(DBUS_CONFIG_DIR)
-	-test -d "/usr/lib/systemd/system/" && ! test -f /usr/lib/systemd/system/ocserv.service && $(INSTALL_DATA) systemd/standalone/ocserv.service /usr/lib/systemd/system/
+dist_dbusconf_DATA = \
+	dbus/org.infradead.ocserv.conf
+
+dist_systemd_DATA = \
+	systemd/socket-activated/ocserv.service \
+	systemd/socket-activated/ocserv.socket
