From 544c8cc8a4bf39a79a80c15a7890ff43c4fac02d Mon Sep 17 00:00:00 2001
From: Nicolas Werner <nicolas.werner@hotmail.de>
Date: Thu, 16 Sep 2021 15:51:33 +0200
Subject: [PATCH] Fix compilaton with Olm 3.2.5

---
 lib/crypto/client.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/crypto/client.cpp b/lib/crypto/client.cpp
index 95cfb35..5017fbb 100644
--- a/lib/crypto/client.cpp
+++ b/lib/crypto/client.cpp
@@ -37,15 +37,15 @@ static const std::array olmErrorStrings{
 
 };
 
-OlmErrorCode
+mtx::crypto::OlmErrorCode
 olm_exception::ec_from_string(std::string_view error)
 {
         for (size_t i = 0; i < olmErrorStrings.size(); i++) {
                 if (olmErrorStrings[i] == error)
-                        return static_cast<OlmErrorCode>(i);
+                        return static_cast<mtx::crypto::OlmErrorCode>(i);
         }
 
-        return OlmErrorCode::UNKNOWN_ERROR;
+        return mtx::crypto::OlmErrorCode::UNKNOWN_ERROR;
 }
 
 void
-- 
2.33.0

