diff --git a/alvr/server/build.rs b/alvr/server/build.rs
index e336d4ad..d0a458dc 100644
--- a/alvr/server/build.rs
+++ b/alvr/server/build.rs
@@ -75,43 +75,15 @@ fn main() {
 
     let use_ffmpeg = cfg!(feature = "gpl") || cfg!(target_os = "linux");
 
-    if use_ffmpeg {
-        let ffmpeg_path = get_ffmpeg_path();
-
-        assert!(ffmpeg_path.join("include").exists());
-        build.include(ffmpeg_path.join("include"));
-    }
-
     #[cfg(feature = "gpl")]
     build.define("ALVR_GPL", None);
 
     build.compile("bindings");
 
     if use_ffmpeg {
-        let ffmpeg_path = get_ffmpeg_path();
-        let ffmpeg_lib_path = ffmpeg_path.join("lib");
-
-        assert!(ffmpeg_lib_path.exists());
-
-        println!(
-            "cargo:rustc-link-search=native={}",
-            ffmpeg_lib_path.to_string_lossy()
-        );
-
         #[cfg(target_os = "linux")]
         {
-            let ffmpeg_pkg_path = ffmpeg_lib_path.join("pkgconfig");
-            assert!(ffmpeg_pkg_path.exists());
-
-            let ffmpeg_pkg_path = ffmpeg_pkg_path.to_string_lossy().to_string();
-            env::set_var(
-                "PKG_CONFIG_PATH",
-                env::var("PKG_CONFIG_PATH").map_or(ffmpeg_pkg_path.clone(), |old| {
-                    format!("{ffmpeg_pkg_path}:{old}")
-                }),
-            );
-
-            let pkg = pkg_config::Config::new().statik(true).to_owned();
+            let pkg = pkg_config::Config::new().to_owned();
 
             for lib in ["libavutil", "libavfilter", "libavcodec"] {
                 pkg.probe(lib).unwrap();
