diff -Naur installer-framework-4.8.1.orig/src/libs/installer/binaryformatengine.cpp installer-framework-4.8.1/src/libs/installer/binaryformatengine.cpp
--- installer-framework-4.8.1.orig/src/libs/installer/binaryformatengine.cpp	2025-01-26 21:10:21.904198528 +0100
+++ installer-framework-4.8.1/src/libs/installer/binaryformatengine.cpp	2025-02-02 14:48:43.663444647 +0100
@@ -266,14 +266,6 @@
 */
 #if QT_VERSION >= QT_VERSION_CHECK(6, 8, 0)
 QAbstractFileEngine::IteratorUniquePtr
-BinaryFormatEngine::beginEntryList(const QString &path, QDir::Filters filters,
-                                    const QStringList &filterNames)
-{
-    const QStringList entries = entryList(filters, filterNames);
-    return std::make_unique<StringListIterator>(path, entries, filters, filterNames);
-}
-
-QAbstractFileEngine::IteratorUniquePtr
 BinaryFormatEngine::beginEntryList(const QString &path, QDirListing::IteratorFlags filters, const QStringList &filterNames)
 {
     const QStringList entries = entryList(filters, filterNames);
diff -Naur installer-framework-4.8.1.orig/src/libs/installer/binaryformatengine.h installer-framework-4.8.1/src/libs/installer/binaryformatengine.h
--- installer-framework-4.8.1.orig/src/libs/installer/binaryformatengine.h	2025-01-26 21:10:21.904198528 +0100
+++ installer-framework-4.8.1/src/libs/installer/binaryformatengine.h	2025-02-02 14:32:33.436791815 +0100
@@ -62,7 +62,6 @@
     FileFlags fileFlags(FileFlags type = FileInfoAll) const override;
 
 #if QT_VERSION >= QT_VERSION_CHECK(6, 8, 0)
-    IteratorUniquePtr beginEntryList(const QString &path, QDir::Filters filters, const QStringList &filterNames) override;
     IteratorUniquePtr beginEntryList(const QString &path, QDirListing::IteratorFlags filters, const QStringList &filterNames) override;
     QStringList entryList(QDirListing::IteratorFlags filters, const QStringList &filterNames) const override;
 #else
diff -Naur installer-framework-4.8.1.orig/src/libs/installer/remotefileengine.cpp installer-framework-4.8.1/src/libs/installer/remotefileengine.cpp
--- installer-framework-4.8.1.orig/src/libs/installer/remotefileengine.cpp	2025-01-26 21:10:21.904198528 +0100
+++ installer-framework-4.8.1/src/libs/installer/remotefileengine.cpp	2025-02-02 14:35:52.237030579 +0100
@@ -80,7 +80,7 @@
 class RemoteFileEngineIterator : public QAbstractFileEngineIterator
 {
 public:
-    RemoteFileEngineIterator(const QString &path, QDir::Filters filters, const QStringList &nameFilters,
+    RemoteFileEngineIterator(const QString &path, QDirListing::IteratorFlags filters, const QStringList &nameFilters,
             const QStringList &files)
 #if QT_VERSION >= QT_VERSION_CHECK(6, 8, 0)
         : QAbstractFileEngineIterator(path, filters, nameFilters)
@@ -177,8 +177,7 @@
 */
 #if QT_VERSION >= QT_VERSION_CHECK(6, 8, 0)
 QAbstractFileEngine::IteratorUniquePtr
-RemoteFileEngine::beginEntryList(const QString &path, QDir::Filters filters,
-                                   const QStringList &filterNames)
+RemoteFileEngine::beginEntryList(const QString &path, QDirListing::IteratorFlags filters, const QStringList &filterNames)
 #else
 QAbstractFileEngine::Iterator* RemoteFileEngine::beginEntryList(QDir::Filters filters,
     const QStringList &filterNames)
@@ -246,6 +245,16 @@
 {
     if ((const_cast<RemoteFileEngine *>(this))->connectToServer()) {
         return callRemoteMethod<QStringList>
+            (QString::fromLatin1(Protocol::QAbstractFileEngineEntryList),
+            static_cast<qint32>(filters), filterNames);
+    }
+    return m_fileEngine.entryList(filters, filterNames);
+}
+
+QStringList RemoteFileEngine::entryList(QDirListing::IteratorFlags filters, const QStringList &filterNames) const
+{
+    if ((const_cast<RemoteFileEngine *>(this))->connectToServer()) {
+        return callRemoteMethod<QStringList>
             (QString::fromLatin1(Protocol::QAbstractFileEngineEntryList),
             static_cast<qint32>(filters), filterNames);
     }
diff -Naur installer-framework-4.8.1.orig/src/libs/installer/remotefileengine.h installer-framework-4.8.1/src/libs/installer/remotefileengine.h
--- installer-framework-4.8.1.orig/src/libs/installer/remotefileengine.h	2025-01-26 21:10:21.904198528 +0100
+++ installer-framework-4.8.1/src/libs/installer/remotefileengine.h	2025-02-02 14:35:46.713115235 +0100
@@ -86,6 +86,7 @@
     bool caseSensitive() const override;
     bool isRelativePath() const override;
     QStringList entryList(QDir::Filters filters, const QStringList &filterNames) const override;
+    QStringList entryList(QDirListing::IteratorFlags filters, const QStringList &filterNames) const override;
     FileFlags fileFlags(FileFlags type = FileInfoAll) const override;
     bool setPermissions(uint perms) override;
     QString fileName(FileName file = DefaultName) const override;
@@ -102,8 +103,9 @@
     uchar *map(qint64, qint64, QFile::MemoryMapFlags) { return 0; }
     bool unmap(uchar *) { return true; }
 #if QT_VERSION >= QT_VERSION_CHECK(6, 8, 0)
-    IteratorUniquePtr beginEntryList(const QString &path, QDir::Filters filters, const QStringList &filterNames) override;
+    IteratorUniquePtr beginEntryList(const QString &path, QDirListing::IteratorFlags filters, const QStringList &filterNames) override;
     IteratorUniquePtr endEntryList() override;
+
 #else
     Iterator *beginEntryList(QDir::Filters filters, const QStringList &filterNames) override;
     Iterator *endEntryList() override;
