diff --git a/apfsck/Makefile b/apfsck/Makefile
index e6d6f84..6f65668 100644
--- a/apfsck/Makefile
+++ b/apfsck/Makefile
@@ -15,7 +15,7 @@ override CFLAGS += -Wall -I$(CURDIR)/../include
 
 apfsck: $(OBJS) $(LIBRARY)
 	@echo '  Linking...'
-	@gcc $(CFLAGS) -o apfsck $(OBJS) $(LIBRARY)
+	@gcc $(CFLAGS) $(LDFLAGS) -o apfsck $(OBJS) $(LIBRARY)
 	@echo '  Build complete'
 
 # Build the common libraries
diff --git a/mkapfs/Makefile b/mkapfs/Makefile
index 67b39ab..2280615 100644
--- a/mkapfs/Makefile
+++ b/mkapfs/Makefile
@@ -14,7 +14,7 @@ override CFLAGS += -Wall -I$(CURDIR)/../include
 
 mkapfs: $(OBJS) $(LIBRARY)
 	@echo '  Linking...'
-	@gcc $(CFLAGS) -o mkapfs $(OBJS) $(LIBRARY)
+	@gcc $(CFLAGS) $(LDFLAGS) -o mkapfs $(OBJS) $(LIBRARY)
 	@echo '  Build complete'
 
 # Build the common libraries
