diff --git a/apfsck/Makefile b/apfsck/Makefile
index e6d6f84..3636b2b 100644
--- a/apfsck/Makefile
+++ b/apfsck/Makefile
@@ -37,9 +37,9 @@ endif
 clean:
 	rm -f $(OBJS) $(DEPS) apfsck
 install:
-	install -d $(BINDIR)
-	install -t $(BINDIR) apfsck
-	ln -fs -T $(BINDIR)/apfsck $(BINDIR)/fsck.apfs
-	install -d $(MANDIR)
-	install -m 644 -t $(MANDIR) apfsck.8
-	ln -fs -T $(MANDIR)/apfsck.8 $(MANDIR)/fsck.apfs.8
+	install -d $(DESTDIR)$(BINDIR)
+	install -t $(DESTDIR)$(BINDIR) apfsck
+	ln -fs -T $(BINDIR)/apfsck $(DESTDIR)$(BINDIR)/fsck.apfs
+	install -d $(DESTDIR)$(MANDIR)
+	install -m 644 -t $(DESTDIR)$(MANDIR) apfsck.8
+	ln -fs -T $(MANDIR)/apfsck.8 $(DESTDIR)$(MANDIR)/fsck.apfs.8
diff --git a/mkapfs/Makefile b/mkapfs/Makefile
index 67b39ab..bb66dd1 100644
--- a/mkapfs/Makefile
+++ b/mkapfs/Makefile
@@ -36,9 +36,9 @@ endif
 clean:
 	rm -f $(OBJS) $(DEPS) mkapfs
 install:
-	install -d $(BINDIR)
-	install -t $(BINDIR) mkapfs
-	ln -fs -T $(BINDIR)/mkapfs $(BINDIR)/mkfs.apfs
-	install -d $(MANDIR)
-	install -m 644 -t $(MANDIR) mkapfs.8
-	ln -fs -T $(MANDIR)/mkapfs.8 $(MANDIR)/mkfs.apfs.8
+	install -d $(DESTDIR)$(BINDIR)
+	install -t $(DESTDIR)$(BINDIR) mkapfs
+	ln -fs -T $(BINDIR)/mkapfs $(DESTDIR)$(BINDIR)/mkfs.apfs
+	install -d $(DESTDIR)$(MANDIR)
+	install -m 644 -t $(DESTDIR)$(MANDIR) mkapfs.8
+	ln -fs -T $(MANDIR)/mkapfs.8 $(DESTDIR)$(MANDIR)/mkfs.apfs.8
