--- a/packages/flutter_tools/lib/src/cache.dart
+++ b/packages/flutter_tools/lib/src/cache.dart
@@ -1010,31 +1010,6 @@
   }
 
   Future<bool> checkForArtifacts(String? engineVersion) async {
-    engineVersion ??= version;
-    final String url = '${cache.storageBaseUrl}/flutter_infra_release/flutter/$engineVersion/';
-
-    bool exists = false;
-    for (final String pkgName in getPackageDirs()) {
-      exists = await cache.doesRemoteExist(
-        'Checking package $pkgName is available...',
-        Uri.parse('$url$pkgName.zip'),
-      );
-      if (!exists) {
-        return false;
-      }
-    }
-
-    for (final List<String> toolsDir in getBinaryDirs()) {
-      final String cacheDir = toolsDir[0];
-      final String urlPath = toolsDir[1];
-      exists = await cache.doesRemoteExist(
-        'Checking $cacheDir tools are available...',
-        Uri.parse(url + urlPath),
-      );
-      if (!exists) {
-        return false;
-      }
-    }
     return true;
   }
 
@@ -1036,6 +1016,8 @@
     Directory location,
     void Function(File, Directory) extractor,
   ) async {
+	throwToolExit('Tried to download $url. Report on aur.');
+
     final String downloadPath = flattenNameSubdirs(url, _fileSystem);
     final File tempFile = _createDownloadFile(downloadPath);
     Status status;
