diff -rupN rpcs3_orig/Utilities/sysinfo.cpp rpcs3/Utilities/sysinfo.cpp
--- rpcs3_orig/Utilities/sysinfo.cpp	2018-02-16 04:15:45.755844762 +0800
+++ rpcs3/Utilities/sysinfo.cpp	2018-02-16 04:17:38.716276945 +0800
@@ -21,28 +21,32 @@ bool utils::has_avx()
 
 bool utils::has_avx2()
 {
-	static const bool g_value = get_cpuid(0, 0)[0] >= 0x7 && get_cpuid(7, 0)[1] & 0x20;
-	return g_value;
+	// static const bool g_value = get_cpuid(0, 0)[0] >= 0x7 && get_cpuid(7, 0)[1] & 0x20;
+	// return g_value;
+  return 0;
 }
 
 bool utils::has_rtm()
 {
 	// Check RTM and MPX extensions in order to filter out TSX on Haswell CPUs
-	static const bool g_value = get_cpuid(0, 0)[0] >= 0x7 && (get_cpuid(7, 0)[1] & 0x4800) == 0x4800;
-	return g_value;
+	// static const bool g_value = get_cpuid(0, 0)[0] >= 0x7 && (get_cpuid(7, 0)[1] & 0x4800) == 0x4800;
+	// return g_value;
+  return 0;
 }
 
 bool utils::has_512()
 {
 	// Check AVX512F, AVX512CD, AVX512DQ, AVX512BW, AVX512VL extensions (Skylake-X level support)
-	static const bool g_value = get_cpuid(0, 0)[0] >= 0x7 && (get_cpuid(7, 0)[1] & 0xd0030000) == 0xd0030000;
-	return g_value;
+	// static const bool g_value = get_cpuid(0, 0)[0] >= 0x7 && (get_cpuid(7, 0)[1] & 0xd0030000) == 0xd0030000;
+	// return g_value;
+  return 0;
 }
 
 bool utils::has_xop()
 {
-	static const bool g_value = has_avx() && get_cpuid(0x80000001, 0)[2] & 0x800;
-	return g_value;
+	// static const bool g_value = has_avx() && get_cpuid(0x80000001, 0)[2] & 0x800;
+	// return g_value;
+  return 0;
 }
 
 std::string utils::get_system_info()
