diff --git i/ext/wpe/meson.build w/ext/wpe/meson.build
index 9381e9b08..e94965c02 100644
--- i/ext/wpe/meson.build
+++ w/ext/wpe/meson.build
@@ -6,7 +6,7 @@ if not gstgl_dep.found()
   subdir_done()
 endif
 
-wpe_dep = dependency('wpe-webkit-1.0', version : '>= 2.24', required : get_option('wpe'))
+wpe_dep = dependency('wpe-webkit-1.1', version : '>= 2.24', required : get_option('wpe'))
 wpe_fdo_dep = dependency('wpebackend-fdo-1.0', required : get_option('wpe'))
 egl_dep = dependency('egl', required : get_option('wpe'))
 xkbcommon_dep = dependency('xkbcommon', version : '>= 0.8', required : get_option('wpe'))
