diff -rupN RetroArch.orig/retroarch.cfg RetroArch/retroarch.cfg
--- RetroArch.orig/retroarch.cfg	2018-11-14 10:16:14.586946273 +0100
+++ RetroArch/retroarch.cfg	2018-11-14 10:15:09.879807008 +0100
@@ -625,7 +625,7 @@
 # menu_show_online_updater = true
 
 # If disabled, will hide the ability to update cores (and core info files) inside the menu.
-# menu_show_core_updater = true
+menu_show_core_updater = false
 
 # If disabled, the libretro core will keep running in the background when we
 # are in the menu.
@@ -754,7 +754,7 @@
 
 # Assets directory. This location is queried by default when menu interfaces try to look for
 # loadable assets, etc.
-# assets_directory =
+assets_directory = /usr/share/retroarch/assets
 
 # Dynamic wallpapers directory. The place to store the wallpapers dynamically
 # loaded by the menu depending on context.
@@ -767,10 +767,10 @@
 # rgui_browser_directory =
 
 # Core directory for libretro core implementations.
-# libretro_directory =
+libretro_directory = /usr/lib/libretro
 
 # Core info directory for libretro core information.
-# libretro_info_path =
+libretro_info_path = /usr/share/libretro/info
 
 # Path to content database directory.
 # content_database_path =
@@ -782,13 +782,13 @@
 # cheat_database_path =
 
 # Defines a directory where CPU-based video filters are kept.
-# video_filter_dir =
+video_filter_dir = /usr/lib/retroarch/filters/video
 
 # Directory where DSP plugins are kept.
-# audio_filter_dir =
+audio_filter_dir = /usr/lib/retroarch/filters/audio
 
 # Defines a directory where shaders (Cg, CGP, GLSL) are kept for easy access.
-# video_shader_dir =
+video_shader_dir = /usr/share/libretro/shaders
 
 # Recording output directory. Where recordings are saved.
 # recording_output_directory =
