--- request.py	2017-06-18 18:17:20.000000000 +0800
+++ request.py	2017-06-30 16:37:24.024653648 +0800
@@ -27,12 +27,12 @@
 except ImportError:
     import json
 
-import certifi
+#import certifi
 try:
-    import telegram.vendor.ptb_urllib3.urllib3 as urllib3
-    import telegram.vendor.ptb_urllib3.urllib3.contrib.appengine as appengine
-    from telegram.vendor.ptb_urllib3.urllib3.connection import HTTPConnection
-    from telegram.vendor.ptb_urllib3.urllib3.util.timeout import Timeout
+    import urllib3 as urllib3
+    import urllib3.contrib.appengine as appengine
+    from urllib3.connection import HTTPConnection
+    from urllib3.util.timeout import Timeout
 except ImportError:
     warnings.warn("python-telegram-bot wasn't properly installed. Please refer to README.rst on "
                   "how to properly install.")
@@ -79,7 +79,7 @@
         kwargs = dict(
             maxsize=con_pool_size,
             cert_reqs='CERT_REQUIRED',
-            ca_certs=certifi.where(),
+            ca_certs="/etc/ssl/certs/ca-certificates.crt",
             socket_options=HTTPConnection.default_socket_options + [
                 (socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1),
             ],
