From 7ec194d090db53ca049341920ba91276159e4264 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Sat, 19 Sep 2020 12:47:35 -0400
Subject: [PATCH] Add cmdline option to open with window closed

---
 src/index.js | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/index.js b/src/index.js
index 0e11f13..fd519bf 100644
--- a/src/index.js
+++ b/src/index.js
@@ -140,6 +140,8 @@ if (!settings.get('enableGPUAcceleration')) {
   app.disableHardwareAcceleration();
 }
 
+const start_closed = process.argv.includes('--start-closed');
+
 const createWindow = () => {
   // Remember window size
   const mainWindowState = windowStateKeeper({
@@ -286,6 +288,8 @@ const createWindow = () => {
     debug('Skip taskbar: false');
     mainWindow.setSkipTaskbar(false);
   });
+  if (start_closed)
+    mainWindow.close();
 
   app.mainWindow = mainWindow;
   app.isMaximized = mainWindow.isMaximized();
-- 
2.28.0

