diff -uprNEBZ --suppress-blank-empty a/src/mainwindow.cpp b/src/mainwindow.cpp
--- a/src/mainwindow.cpp	2018-12-21 17:01:47.452059818 -0600
+++ b/src/mainwindow.cpp	2018-12-21 17:02:54.862133874 -0600
@@ -122,7 +122,7 @@ MainWindow::MainWindow(QWidget *parent)
     vk_setCornerWidget( ui->tabWidgetScreencast );

     // need a move
-    move( 0, 0 );
+    //move( 0, 0 );

     makeAndSetValidIconForSideBar( ui->tabWidgetSideBar->indexOf( ui->tabSidebarScreencast ), QIcon::fromTheme( "video-display", QIcon( ":/pictures/video-display.svg" ) ) );
     makeAndSetValidIconForSideBar( ui->tabWidgetSideBar->indexOf( ui->tabSidebarScreenshot ), QIcon::fromTheme( "computer",      QIcon( ":/pictures/computer.svg" ) ) );
