diff --git a/main.c b/main.c
index 0eba280..666c785 100644
--- a/main.c
+++ b/main.c
@@ -11,6 +11,7 @@
  */
 
 #include "defs.h"
+#include <sys/prctl.h>
 
 Prototype short DebugOpt;
 Prototype short LogLevel;
@@ -289,6 +290,7 @@ main(int ac, char **av)
 	 *             of 1 second.
 	 */
 
+        prctl(PR_SET_CHILD_SUBREAPER, 1);
 	printlogf(LOG_NOTICE,"%s " VERSION " dillon's cron daemon, started with loglevel %s\n", av[0], LevelAry[LogLevel]);
 	SynchronizeDir(CDir, NULL, 1);
 	SynchronizeDir(SCDir, "root", 1);
diff --git a/subs.c b/subs.c
index b6217e8..993cf52 100644
--- a/subs.c
+++ b/subs.c
@@ -130,7 +130,8 @@ void waitmailjob(int sig) {
 	 */
 	pid_t child;
 	do {
-		child = waitpid(-DaemonPid, NULL, WNOHANG);
+                /* subreaper should wait for processes not in its process group too */
+		child = waitpid(-1, NULL, WNOHANG);
 		/* call was interrupted, try again: won't happen because we use SA_RESTART */
 		/* if (child == (pid_t)-1 && errno == EINTR) continue; */
 	} while (child > (pid_t) 0);
