diff --git a/source/renderer-gi-gtk.lisp b/source/renderer-gi-gtk.lisp
index 5684866a..f01097df 100644
--- a/source/renderer-gi-gtk.lisp
+++ b/source/renderer-gi-gtk.lisp
@@ -24,8 +24,8 @@
                         (lambda ()
                           (with-protect ("Error on GTK thread: ~a" :condition)
                             (glib:g-set-prgname "nyxt")
-                            (gdk:gdk-set-program-class "Nyxt")
-                            (gir:invoke ((gir:ffi "Gtk") 'main))))
+                            ;;(gdk:gdk-set-program-class "Nyxt")
+                            (gir:invoke ((gir:ffi "Gtk" "3.0") 'main))))
                         :name "main thread")))
       (finalize browser urls startup-timestamp)
       (unless *run-from-repl-p*
