diff -Naur Bottles-52.0-orig/bottles/frontend/meson.build Bottles-52.0/bottles/frontend/meson.build
--- Bottles-52.0-orig/bottles/frontend/meson.build	2025-11-03 02:20:26
+++ Bottles-52.0/bottles/frontend/meson.build	2025-11-03 02:14:48
@@ -23,11 +23,6 @@
     configuration: conf
 )

-fs = import('fs')
-
-if not fs.is_file('/' + '.flatpak-info')
-  error('file does not exist')
-endif

 bottles_sources = [
   '__init__.py',
diff -Naur Bottles-52.0-orig/bottles/frontend/views/bottle_details.py Bottles-52.0/bottles/frontend/views/bottle_details.py
--- Bottles-52.0-orig/bottles/frontend/views/bottle_details.py	2025-11-03 02:20:26
+++ Bottles-52.0/bottles/frontend/views/bottle_details.py	2025-11-03 02:16:03
@@ -21,7 +21,7 @@
 from gettext import gettext as _
 from typing import List, Optional

-from gi.repository import Gtk, Gio, Adw, Gdk, GLib, Xdp
+from gi.repository import Gtk, Gio, Adw, Gdk, GLib

 from bottles.backend.managers.backup import BackupManager
 from bottles.backend.models.config import BottleConfig
@@ -436,20 +436,19 @@
             dialog.connect("response", execute)
             dialog.show()

-        if Xdp.Portal.running_under_sandbox():
-            if self.window.settings.get_boolean("show-sandbox-warning"):
-                dialog = Adw.MessageDialog.new(
-                    self.window,
-                    _("Be Aware of Sandbox"),
-                    _(
-                        "Bottles is running in a sandbox, a restricted permission environment needed to keep you safe. If the program won't run, consider moving inside the bottle (3 dots icon on the top), then launch from there."
-                    ),
-                )
-                dialog.add_response("dismiss", _("_Dismiss"))
-                dialog.connect("response", show_chooser)
-                dialog.present()
-            else:
-                show_chooser()
+        if self.window.settings.get_boolean("show-sandbox-warning"):
+            dialog = Adw.MessageDialog.new(
+                self.window,
+                _("Be Aware of Sandbox"),
+                _(
+                    "Bottles is running in a sandbox, a restricted permission environment needed to keep you safe. If the program won't run, consider moving inside the bottle (3 dots icon on the top), then launch from there."
+                ),
+            )
+            dialog.add_response("dismiss", _("_Dismiss"))
+            dialog.connect("response", show_chooser)
+            dialog.present()
+        else:
+            show_chooser()

     def __backup(self, widget, backup_type):
         """
diff -Naur Bottles-52.0-orig/bottles/frontend/views/bottle_preferences.py Bottles-52.0/bottles/frontend/views/bottle_preferences.py
--- Bottles-52.0-orig/bottles/frontend/views/bottle_preferences.py	2025-11-03 02:20:26
+++ Bottles-52.0/bottles/frontend/views/bottle_preferences.py	2025-11-03 02:16:30
@@ -20,7 +20,7 @@
 import re
 from gettext import gettext as _

-from gi.repository import Gtk, Adw, Xdp
+from gi.repository import Gtk, Adw

 from bottles.backend.globals import (
     gamemode_available,
@@ -139,7 +139,7 @@
         self.queue = details.queue
         self.details = details

-        if not gamemode_available or not Xdp.Portal.running_under_sandbox():
+        if not gamemode_available:
             return

         _not_available = _("This feature is unavailable on your system.")
diff -Naur Bottles-52.0-orig/bottles/frontend/views/list.py Bottles-52.0/bottles/frontend/views/list.py
--- Bottles-52.0-orig/bottles/frontend/views/list.py	2025-11-03 02:20:26
+++ Bottles-52.0/bottles/frontend/views/list.py	2025-11-03 02:16:57
@@ -18,7 +18,7 @@
 from datetime import datetime
 from gettext import gettext as _

-from gi.repository import Gtk, GLib, Adw, Xdp
+from gi.repository import Gtk, GLib, Adw

 from bottles.backend.models.config import BottleConfig
 from bottles.backend.models.result import Result
@@ -82,9 +82,6 @@

     def run_executable(self, *_args):
         """Display file dialog for executable"""
-        if not Xdp.Portal.running_under_sandbox():
-            return
-
         def set_path(_dialog, response):
             if response != Gtk.ResponseType.ACCEPT:
                 return
diff -Naur Bottles-52.0-orig/bottles/frontend/views/new_bottle_dialog.py Bottles-52.0/bottles/frontend/views/new_bottle_dialog.py
--- Bottles-52.0-orig/bottles/frontend/views/new_bottle_dialog.py	2025-11-03 02:20:26
+++ Bottles-52.0/bottles/frontend/views/new_bottle_dialog.py	2025-11-03 02:17:29
@@ -21,7 +21,7 @@
 from typing import Any, Optional

 from bottles.backend.state import Task, TaskManager
-from gi.repository import Gtk, Adw, Pango, Gio, Xdp, GObject, GLib
+from gi.repository import Gtk, Adw, Pango, Gio, GObject, GLib

 from bottles.backend.models.config import BottleConfig
 from bottles.backend.utils.threading import RunAsync
@@ -86,7 +86,7 @@
         super().__init__(**kwargs)
         # common variables and references
         self.window = GtkUtils.get_parent_window()
-        if not self.window or not Xdp.Portal.running_under_sandbox():
+        if not self.window:
             return

         self.app = self.window.get_application()
diff -Naur Bottles-52.0-orig/bottles/frontend/windows/window.py Bottles-52.0/bottles/frontend/windows/window.py
--- Bottles-52.0-orig/bottles/frontend/windows/window.py	2025-11-03 02:20:26
+++ Bottles-52.0/bottles/frontend/windows/window.py	2025-11-03 02:18:08
@@ -21,7 +21,7 @@
 from gettext import gettext as _
 from typing import Optional

-from gi.repository import Gtk, GLib, Gio, Adw, GObject, Gdk, Xdp
+from gi.repository import Gtk, GLib, Gio, Adw, GObject, Gdk

 from bottles.backend.globals import Paths
 from bottles.backend.health import HealthChecker
@@ -110,35 +110,6 @@
         if self.settings.get_boolean("dark-theme"):
             manager = Adw.StyleManager.get_default()
             manager.set_color_scheme(Adw.ColorScheme.FORCE_DARK)
-
-        # Be VERY explicit that non-sandboxed environments are unsupported
-        if not Xdp.Portal.running_under_sandbox():
-
-            def response(dialog, response, *args):
-                if response == "close":
-                    quit(1)
-
-            body = _(
-                "Bottles is only supported within a sandboxed environment. Official sources of Bottles are available at"
-            )
-            download_url = "usebottles.com/download"
-
-            error_dialog = Adw.AlertDialog.new(
-                _("Unsupported Environment"),
-                f"{body} <a href='https://{download_url}' title='https://{download_url}'>{download_url}.</a>",
-            )
-
-            error_dialog.add_response("close", _("Close"))
-            error_dialog.set_body_use_markup(True)
-            error_dialog.connect("response", response)
-            error_dialog.present(self)
-            logging.error(
-                _(
-                    "Bottles is only supported within a sandboxed format. Official sources of Bottles are available at:"
-                )
-            )
-            logging.error("https://usebottles.com/download/")
-            return

         # Loading view
         self.page_loading = LoadingView()