*** main.rs.orig	2018-03-15 16:00:01.114533700 +0530
--- main.rs	2018-03-15 16:00:14.814699952 +0530
***************
*** 299,305 ****
  
  fn cmd_install(serial: Option<&String>) -> Result<(), CommandExecutionError> {
      info!(target: TAG, "Installing gnirehtet client...");
!     exec_adb(serial, vec!["install", "-r", "gnirehtet.apk"])
  }
  
  fn cmd_uninstall(serial: Option<&String>) -> Result<(), CommandExecutionError> {
--- 299,305 ----
  
  fn cmd_install(serial: Option<&String>) -> Result<(), CommandExecutionError> {
      info!(target: TAG, "Installing gnirehtet client...");
!     exec_adb(serial, vec!["install", "-r", "/usr/share/gnirehtet/gnirehtet.apk"])
  }
  
  fn cmd_uninstall(serial: Option<&String>) -> Result<(), CommandExecutionError> {
