From 3c2e772aa7f4846ca4cabac6852d0aafdd35d7fd Mon Sep 17 00:00:00 2001
From: c0re100 <corehusky@gmail.com>
Date: Thu, 8 Dec 2022 00:54:33 +0800
Subject: [PATCH] Fix ApplicationName

---
 Telegram/SourceFiles/core/launcher.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Telegram/SourceFiles/core/launcher.cpp b/Telegram/SourceFiles/core/launcher.cpp
index 5fbe22421..5b2043e4d 100644
--- a/Telegram/SourceFiles/core/launcher.cpp
+++ b/Telegram/SourceFiles/core/launcher.cpp
@@ -293,7 +293,7 @@ void Launcher::init() {
 	prepareSettings();
 	initQtMessageLogging();
 
-	QApplication::setApplicationName(u"64GramDesktop"_q);
+	QApplication::setApplicationName(u"64Gram"_q);
 	QApplication::setAttribute(Qt::AA_DisableHighDpiScaling, true);
 	QApplication::setHighDpiScaleFactorRoundingPolicy(
 		Qt::HighDpiScaleFactorRoundingPolicy::Floor);
