diff --git i/tools/repository.cpp w/tools/repository.cpp
index 5b5c737..9140851 100644
--- i/tools/repository.cpp
+++ w/tools/repository.cpp
@@ -10,7 +10,8 @@ static std::set<std::string> basic_types{"gchar", "guchar", "gshort", "gushort",
     "gint", "guint", "glong", "gulong", "gssize", "gsize", "gintptr",
     "guintptr", "gpointer", "gconstpointer", "gboolean", "gint8", "gint16",
     "guint8", "guint16", "gint32", "guint32", "gint64", "guint64", "gfloat",
-    "gdouble", "GType", "utf8", "filename", "gi::cstring", "gunichar"};
+    "gdouble", "GType", "utf8", "filename", "gi::cstring", "gunichar",
+    "dev_t", "gid_t", "pid_t", "socklen_t", "uid_t"};

 class RepositoryPriv : public Repository
 {
