--- qtwebkit-opensource-src-5.5.1/Source/JavaScriptCore/API/JSStringRef.h.orig	2015-10-13 06:37:10.000000000 +0200
+++ qtwebkit-opensource-src-5.5.1/Source/JavaScriptCore/API/JSStringRef.h	2017-04-24 12:26:42.495345570 +0200
@@ -32,6 +32,7 @@
 #include <stdbool.h>
 #endif
 #include <stddef.h> /* for size_t */
+#include <uchar.h>
 
 #ifdef __cplusplus
 extern "C" {
@@ -43,7 +44,7 @@
 @typedef JSChar
 @abstract A Unicode character.
 */
-    typedef unsigned short JSChar;
+    typedef char16_t JSChar;
 #else
     typedef wchar_t JSChar;
 #endif
--- qtwebkit-opensource-src-5.5.1/Source/WebKit2/Shared/API/c/WKString.h.orig	2015-10-13 06:37:12.000000000 +0200
+++ qtwebkit-opensource-src-5.5.1/Source/WebKit2/Shared/API/c/WKString.h	2017-04-24 12:27:33.432011867 +0200
@@ -31,6 +31,7 @@
 #ifndef __cplusplus
 #include <stdbool.h>
 #endif
+#include <uchar.h>
 
 #ifdef __cplusplus
 extern "C" {
@@ -38,7 +39,7 @@
 
 #if !defined(WIN32) && !defined(_WIN32) \
     && !((defined(__CC_ARM) || defined(__ARMCC__)) && !defined(__linux__)) /* RVCT */
-    typedef unsigned short WKChar;
+    typedef char16_t WKChar;
 #else
     typedef wchar_t WKChar;
 #endif
