diff --git a/src/meson.build b/src/meson.build
index 900f3fa..e08b062 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -149,19 +149,3 @@ test('Unit tests',
   timeout: 180,
   env: env
 )
-
-test('Clippy',
-  cargo,
-  args: [
-    'clippy',
-    '--manifest-path', cargo_manifest,
-    '--target-dir', meson.project_build_root() / 'src',
-    '--',
-    '-D', 'warnings',
-    '-A', 'clippy::module_inception',
-    '-A', 'clippy::new_without_default',
-    '-A', 'clippy::enum-variant-names'
-  ],
-  timeout: 180,
-  env: env
-)
