--- meson-a.build	2024-06-01 14:49:01.000000000 -0600
+++ meson-b.build	2024-06-02 11:10:31.249288754 -0600
@@ -30,16 +30,9 @@
 prefix = get_option('prefix')
 bindir = prefix / get_option('bindir')
 localedir = prefix / get_option('localedir')
-
-# Why don't we use localdir? Using localdir works perfectly when
-# developing locally. However, the Flathub build servers put locale files
-# into separate flatpak packages which is breaking my Fluent localisation
-# setup. Fotema will fail starting up saying it canot find the localisation
-# files.
-i18ndir = prefix / 'i18n'
-
 datadir = prefix / get_option('datadir')
 pkgdatadir = datadir / meson.project_name()
+i18ndir = pkgdatadir / 'i18n'
 iconsdir = datadir / 'icons'
 podir = meson.project_source_root() / 'po'
 gettext_package = meson.project_name()
