diff --git a/configure b/configure
index c660bcfbe4..68f0f3c6c8 100755
--- a/configure
+++ b/configure
@@ -524,7 +524,6 @@ case "$cpu" in
     cpu="x86_64"
     host_arch=x86_64
     linux_arch=x86
-    CPU_CFLAGS="-m64"
     ;;
 esac
 
diff --git a/meson.build b/meson.build
index 6d6aadd0d7..3cd9c1e970 100644
--- a/meson.build
+++ b/meson.build
@@ -2302,6 +2302,7 @@ if host_os == 'windows'
       ])
 elif host_os == 'linux'
   vulkan = dependency('vulkan')
+  libglslang = dependency('glslang')
 endif
 
 if vulkan.found()
@@ -3533,11 +3534,9 @@ if have_libvduse
   libvduse = libvduse_proj.get_variable('libvduse_dep')
 endif
 
-tomlplusplus_proj = subproject('tomlplusplus', default_options: ['default_library=static'])
-tomlplusplus = tomlplusplus_proj.get_variable('tomlplusplus_dep')
+tomlplusplus = dependency('tomlplusplus')
 
-xxhash_proj = subproject('xxhash', default_options: ['default_library=static'])
-xxhash = xxhash_proj.get_variable('xxhash_dep')
+xxhash = dependency('libxxhash')
 
 #####################
 # Generated sources #
diff --git a/ui/thirdparty/meson.build b/ui/thirdparty/meson.build
index 2684979282..57799e0e33 100644
--- a/ui/thirdparty/meson.build
+++ b/ui/thirdparty/meson.build
@@ -29,14 +29,9 @@ endif
 libfpng = static_library('fpng', sources: 'fpng/fpng.cpp', cpp_args: libfpng_cpp_args)
 fpng = declare_dependency(include_directories: 'fpng', link_with: libfpng)
 
-json = declare_dependency(include_directories: 'json')
+json = dependency('nlohmann_json')
 
-httplib_proj = subproject('cpp-httplib', default_options: ['cpp-httplib_openssl=enabled'])
-httplib_deps = [httplib_proj.get_variable('cpp_httplib_dep')]
-if host_os == 'windows'
-  httplib_deps += [crypt32]
-endif
-httplib = declare_dependency(dependencies: httplib_deps)
+httplib = dependency('httplib')
 
-libfatx = static_library('fatx', sources: 'fatx/fatx.c')
+libfatx = static_library('fatx', sources: files('fatx/fatx.c') + genh)
 fatx = declare_dependency(include_directories: 'fatx', link_with: libfatx)
diff --git a/ui/xui/reporting.cc b/ui/xui/reporting.cc
index c52bf00076..e692148b2a 100644
--- a/ui/xui/reporting.cc
+++ b/ui/xui/reporting.cc
@@ -23,7 +23,7 @@
 #include <stdio.h>
 #include "reporting.hh"
 #include <httplib.h>
-#include <json.hpp>
+#include <nlohmann/json.hpp>
 using json = nlohmann::json;
 
 #define DEBUG_COMPAT_SERVICE 0
