diff --git a/Lib/test/test_typing.py b/Lib/test/test_typing.py
index 088db9c012..00959a32d6 100644
--- a/Lib/test/test_typing.py
+++ b/Lib/test/test_typing.py
@@ -499,7 +499,7 @@ class LiteralTests(BaseTestCase):
         Literal[True]
         Literal[1, "2", False]
         Literal[Literal[1, 2], Literal[4, 5]]
-        Literal[b"foo", u"bar"]
+        Literal[b"foo", "bar"]
 
     def test_illegal_parameters_do_not_raise_runtime_errors(self):
         # Type checkers should reject these types, but we do not
