--- a/Lib/test/test_posix.py	2019-11-08 18:26:13.704894779 +0800
+++ b/Lib/test/test_posix.py	2019-11-08 18:26:34.575039504 +0800
@@ -1474,8 +1474,8 @@
     def test_pidfd_open(self):
         with self.assertRaises(OSError) as cm:
             os.pidfd_open(-1)
-        if cm.exception.errno == errno.ENOSYS:
-            self.skipTest("system does not support pidfd_open")
+        if cm.exception.errno in (errno.ENOSYS, errno.EPERM):
+            self.skipTest("system does not support or allow pidfd_open")
         self.assertEqual(cm.exception.errno, errno.EINVAL)
         os.close(os.pidfd_open(os.getpid(), 0))
 
