diff --git a/Lib/test/test_signal.py b/Lib/test/test_signal.py
index 119e9e079a..5bee8a8e91 100644
--- a/Lib/test/test_signal.py
+++ b/Lib/test/test_signal.py
@@ -1284,6 +1284,8 @@ class PidfdSignalTest(unittest.TestCase):
             signal.pidfd_send_signal(0, signal.SIGINT)
         if cm.exception.errno == errno.ENOSYS:
             self.skipTest("kernel does not support pidfds")
+        if isinstance(cm.exception, PermissionError):
+            self.skipTest(f"pidfd_send_signal syscall blocked: {cm.exception!r}")
         self.assertEqual(cm.exception.errno, errno.EBADF)
         my_pidfd = os.open(f'/proc/{os.getpid()}', os.O_DIRECTORY)
         self.addCleanup(os.close, my_pidfd)
