diff --git a/package.json b/package.json
index dc24620d251..880ed508d84 100644
--- a/package.json
+++ b/package.json
@@ -120,6 +120,7 @@
     "minimist": "^1.2.8",
     "native-is-elevated": "0.9.0",
     "native-keymap": "^3.3.5",
+    "node-ovsx-sign": "^1.2.0",
     "node-pty": "^1.2.0-beta.12",
     "open": "^10.1.2",
     "playwright-core": "1.59.0-alpha-2026-02-20",
diff --git a/src/vs/platform/extensionManagement/node/extensionSignatureVerificationService.ts b/src/vs/platform/extensionManagement/node/extensionSignatureVerificationService.ts
index 98535c5e648..1a07dbfb646 100644
--- a/src/vs/platform/extensionManagement/node/extensionSignatureVerificationService.ts
+++ b/src/vs/platform/extensionManagement/node/extensionSignatureVerificationService.ts
@@ -35,7 +35,7 @@ export interface IExtensionSignatureVerificationService {
 }
 
 declare namespace vsceSign {
-	export function verify(vsixFilePath: string, signatureArchiveFilePath: string, verbose: boolean): Promise<ExtensionSignatureVerificationResult>;
+	export function verify(vsixFilePath: string, signatureArchiveFilePath: string, verbose: boolean): Promise<boolean>;
 }
 
 /**
@@ -67,7 +67,7 @@ export class ExtensionSignatureVerificationService implements IExtensionSignatur
 	}
 
 	private async resolveVsceSign(): Promise<typeof vsceSign> {
-		const mod = '@vscode/vsce-sign';
+		const mod = 'node-ovsx-sign';
 		return import(mod);
 	}
 
@@ -87,7 +87,21 @@ export class ExtensionSignatureVerificationService implements IExtensionSignatur
 
 		try {
 			this.logService.trace(`Verifying extension signature for ${extensionId}...`);
-			result = await module.verify(vsixFilePath, signatureArchiveFilePath, this.logService.getLevel() === LogLevel.Trace);
+			let isSignatureValid =
+				await module.verify(vsixFilePath, signatureArchiveFilePath, this.logService.getLevel() === LogLevel.Trace);
+			if (isSignatureValid) {
+				result = {
+					code: ExtensionSignatureVerificationCode.Success,
+					didExecute: true,
+					output: "The extension signature is valid."
+				};
+			} else {
+				result = {
+					code: ExtensionSignatureVerificationCode.SignatureIsNotValid,
+					didExecute: true,
+					output: "The extension signature is invalid."
+				};
+			}
 		} catch (e) {
 			result = {
 				code: ExtensionSignatureVerificationCode.UnknownError,
