diff -rupN old/src/wl/sys/wl_linux.c new/src/wl/sys/wl_linux.c
--- old/src/wl/sys/wl_linux.c	2013-05-17 15:36:46.613375000 -0600
+++ new/src/wl/sys/wl_linux.c	2013-05-17 16:01:48.674658839 -0600
@@ -183,7 +183,7 @@ static int wl_set_radio_block(void *data
 static void wl_report_radio_state(wl_info_t *wl);
 #endif
 
-MODULE_LICENSE("MIXED/Proprietary");
+MODULE_LICENSE("GPL");
 
 static struct pci_device_id wl_id_table[] =
 {
