diff --git a/oneko.c b/oneko.c
index d2b81fe..207f7e2 100644
--- a/oneko.c
+++ b/oneko.c
@@ -267,7 +267,7 @@
     { &RightTogi2GC, &RightTogi2Msk } },	/* NekoState == NEKO_R_TOGI */
 };
 
-static void NullFunction();
+static void NullFunction(int signum);
 
 /*
  *	ビットマップデータ・GC 初期化
@@ -732,6 +732,11 @@
   exit(0);
 }
 
+void
+RestoreCursorHandler(int signum)
+{
+	RestoreCursor();
+}
 
 /*
  *	インターバル
@@ -1330,7 +1335,7 @@
  */
 
 static void
-NullFunction()
+NullFunction(int signum)
 {
   /* No Operation */
 #if defined(SYSV) || defined(SVR4)
@@ -1558,9 +1563,9 @@
   InitScreen(theDisplayName);
 
   signal(SIGALRM, NullFunction);
-  signal(SIGINT, RestoreCursor);
-  signal(SIGTERM, RestoreCursor);
-  signal(SIGQUIT, RestoreCursor);
+  signal(SIGINT, RestoreCursorHandler);
+  signal(SIGTERM, RestoreCursorHandler);
+  signal(SIGQUIT, RestoreCursorHandler);
 
   SinPiPer8Times3 = sin(PI_PER8 * (double)3);
   SinPiPer8 = sin(PI_PER8);
