diff --git a/Imakefile b/Imakefile
index 1ea4eca..552a85e 100644
--- a/Imakefile
+++ b/Imakefile
@@ -1,5 +1,5 @@
 /**/# Cannot compile in using Gcc
-#             CC = cc
+             CC = cc
 
   SYS_LIBRARIES = MathLibrary
 LOCAL_LIBRARIES = $(XLIB)
diff --git a/oneko.c b/oneko.c
index d2b81fe..207f7e2 100644
--- a/oneko.c
+++ b/oneko.c
@@ -6,6 +6,7 @@
 static char rcsid[] = "$Header: /home/sun/unix/kato/xsam/oneko/oneko.c,v 1.5 90/10/19 21:25:16 kato Exp $";
 #endif
 
+#include <unistd.h>
 #include "oneko.h"
 #include "patchlevel.h"
 /*
@@ -335,7 +336,7 @@ char	*resource;
  *	リソース・データベースからオプションを設定
  */
 
-GetResources()
+int GetResources()
 {
   char	*resource;
   int		num;
@@ -434,7 +435,7 @@ GetResources()
  *	ねずみ型カーソルを作る
  */
 
-MakeMouseCursor()
+int MakeMouseCursor()
 {
     Pixmap			theCursorSource;
     Pixmap			theCursorMask;
@@ -463,7 +464,7 @@ MakeMouseCursor()
  *	色を初期設定する
  */
 
-SetupColors()
+int SetupColors()
 {
     XColor	theExactColor;
     Colormap	theColormap;
