From 4122d72639f371da3dbcb08aae98faa1ff90d394 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Sun, 26 Feb 2023 14:11:02 -0500
Subject: [PATCH 2/3] Migrate old pids directory

---
 scripts/upgrade/minor-upgrade.sh | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/scripts/upgrade/minor-upgrade.sh b/scripts/upgrade/minor-upgrade.sh
index 55a1285a6..180006737 100755
--- a/scripts/upgrade/minor-upgrade.sh
+++ b/scripts/upgrade/minor-upgrade.sh
@@ -45,6 +45,33 @@ function migrate_avatars() {
     echo
 }
 
+function move_pids() {
+    echo
+    echo "------------------------------"
+    echo "Moving PID directories ..."
+    echo
+    # move "" directory outside
+    if [[ -d "${TOPDIR}/pids" ]]; then
+        echo "pid directory at correct location"
+    elif [[ -d "${TOPDIR}/seafile-data/pids" ]]; then
+        mv -iv "${TOPDIR}/seafile-data/pids" "${TOPDIR}/pids"
+    else
+        mkdir -v "${TOPDIR}/pids"
+    fi
+    if [[ -f "${default_conf_dir}/gunicorn.conf.py" ]]; then
+        if grep "'seafile-data', 'pids', " \
+                "${default_conf_dir}/gunicorn.conf.py" &> /dev/null; then
+            sed -i.bak "s/'seafile-data', 'pids', /'pids', /" "${default_conf_dir}/gunicorn.conf.py"
+            echo "PID file setting updated in conf/gunicorn.conf.py updated."
+            echo "Please verify and delete the backup file at conf/gunicorn.conf.py.bak"
+        fi
+    fi
+    echo
+    echo "DONE"
+    echo "------------------------------"
+    echo
+}
+
 function make_media_custom_symlink() {
     media_symlink=${INSTALLPATH}/seahub/media/custom
     if [[ -L "${media_symlink}" ]]; then
@@ -143,6 +170,7 @@ function rename_gunicorn_config() {
 validate_seafile_data_dir;
 rename_gunicorn_config;
 migrate_avatars;
+move_pids;
 
 move_old_customdir_outside;
 make_media_custom_symlink;
-- 
2.51.0

