From bf78847c690e5f0aa8b1aaa07c09b3d7339ee1b1 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Sun, 26 Feb 2023 14:11:52 -0500
Subject: [PATCH 3/3] Fix gunicorn binary path

---
 scripts/seahub.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/seahub.sh b/scripts/seahub.sh
index 2c6bb68cb..aeb9fa8d2 100755
--- a/scripts/seahub.sh
+++ b/scripts/seahub.sh
@@ -24,7 +24,7 @@ gunicorn_conf=${TOPDIR}/conf/gunicorn.conf.py
 pidfile=${TOPDIR}/pids/seahub.pid
 errorlog=${TOPDIR}/logs/gunicorn_error.log
 accesslog=${TOPDIR}/logs/gunicorn_access.log
-gunicorn_exe=${INSTALLPATH}/seahub/thirdpart/bin/gunicorn
+gunicorn_exe=/usr/bin/gunicorn
 pro_pylibs_dir=${INSTALLPATH}/pro/python
 seafesdir=$pro_pylibs_dir/seafes
 seahubdir=${INSTALLPATH}/seahub
-- 
2.51.0

