From f1f76f26f235b0ec4ab8eb63a37bcd75475435ef Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Wed, 13 Jul 2022 09:08:18 -0400
Subject: [PATCH] Do not force fusion style

---
 src/app/main.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/app/main.cpp b/src/app/main.cpp
index 32651d6430..b4fc503b50 100644
--- a/src/app/main.cpp
+++ b/src/app/main.cpp
@@ -87,7 +87,6 @@ int main(int argc, char** argv)
 
     app_init_qrc();
 
-    qputenv("QT_STYLE_OVERRIDE", "Fusion");
     qputenv("QML_DISABLE_DISK_CACHE", "true");
 
     // HACK: Workaround for crash #28840. This disables the incremental GC
-- 
2.53.0

