From de6d110d9cd67222c2d3fecc0b1d16a4554d3de9 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Sat, 28 Mar 2020 13:03:59 -0400
Subject: [PATCH] fix installpath detection

---
 controller/seafile-controller.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/controller/seafile-controller.c b/controller/seafile-controller.c
index 3dde027..bdebb1b 100644
--- a/controller/seafile-controller.c
+++ b/controller/seafile-controller.c
@@ -256,10 +256,8 @@ init_seafile_path ()
 
     bin_dir = g_path_get_dirname (binary);
 
-    tmp = g_path_get_dirname (bin_dir);
-    installpath = g_path_get_dirname (tmp);
-
-    topdir = g_path_get_dirname (installpath);
+    topdir = g_path_get_dirname (ctl->central_config_dir);
+    installpath = g_build_filename (topdir, "seafile-server", NULL);
 
     g_free (binary);
     g_free (tmp);
-- 
2.25.1

