From d09f77bdc33ede318a19afd114f769d6e758bd68 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Wed, 4 Sep 2024 11:59:23 -0400
Subject: [PATCH 3/3] Fix incorrect error usage and incorrect comparison
 function pointer

---
 server/repo-op.c | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/server/repo-op.c b/server/repo-op.c
index 6c89c35..744a319 100644
--- a/server/repo-op.c
+++ b/server/repo-op.c
@@ -1024,7 +1024,8 @@ check_files_with_same_name (SeafRepo *repo, const char *parent_dir, GList *filen
     char *canon_path = NULL;
     SeafCommit *commit = NULL;
     SeafDir *dir = NULL;
-    GError *error = NULL;
+    GError *_error = NULL;
+    GError **error = &_error;
     gboolean ret = FALSE;
 
     GET_COMMIT_OR_FAIL(commit, repo->id, repo->version, repo->head->commit_id);
@@ -1034,7 +1035,7 @@ check_files_with_same_name (SeafRepo *repo, const char *parent_dir, GList *filen
     dir = seaf_fs_manager_get_seafdir_by_path (seaf->fs_mgr,
                                                repo->store_id, repo->version,
                                                commit->root_id,
-                                               canon_path, &error);
+                                               canon_path, error);
     if (!dir) {
         goto out;
     }
@@ -1051,7 +1052,7 @@ check_files_with_same_name (SeafRepo *repo, const char *parent_dir, GList *filen
         g_free (unique_name);
     }
 out:
-    g_clear_error (&error);
+    g_clear_error (error);
     g_free (canon_path);
     seaf_dir_free (dir);
     seaf_commit_unref (commit);
@@ -6339,7 +6340,7 @@ hash_to_list (gpointer key, gpointer value, gpointer user_data)
 }
 
 static gint
-compare_commit_by_time (gconstpointer a, gconstpointer b, gpointer unused)
+compare_commit_by_time (gconstpointer a, gconstpointer b)
 {
     const SeafCommit *commit_a = a;
     const SeafCommit *commit_b = b;
@@ -6367,9 +6368,7 @@ insert_parent_commit (GList **list, GHashTable *hash,
         return -1;
     }
 
-    *list = g_list_insert_sorted_with_data (*list, p,
-                                           compare_commit_by_time,
-                                           NULL);
+    *list = g_list_insert_sorted (*list, p, compare_commit_by_time);
 
     key = g_strdup (parent_id);
     g_hash_table_replace (hash, key, key);
-- 
2.46.0

