diff --git a/scripts/setup-seafile-mysql.py b/scripts/setup-seafile-mysql.py
index 596687c..39ed520 100644
--- a/scripts/setup-seafile-mysql.py
+++ b/scripts/setup-seafile-mysql.py
@@ -301,9 +301,9 @@ class EnvManager(object):
     def __init__(self):
         self.install_path = os.path.dirname(os.path.abspath(__file__))
         self.top_dir = os.path.dirname(self.install_path)
-        self.bin_dir = os.path.join(self.install_path, 'seafile', 'bin')
+        self.bin_dir = '/usr/bin'
         self.central_config_dir = os.path.join(self.top_dir, 'conf')
         self.central_pids_dir = os.path.join(self.top_dir, 'pids')
         self.central_logs_dir = os.path.join(self.top_dir, 'logs')
         Utils.must_mkdir(self.central_config_dir)
 
@@ -313,7 +313,6 @@ class EnvManager(object):
                 Utils.error('"%s" not found' % path)
 
         paths = [
-            os.path.join(self.install_path, 'seafile'),
             os.path.join(self.install_path, 'seahub'),
             os.path.join(self.install_path, 'runtime'),
         ]
diff --git a/scripts/setup-seafile-mysql.py b/scripts/setup-seafile-mysql.py
index c5a7f1f..5cdb858 100644
--- a/scripts/setup-seafile-mysql.py
+++ b/scripts/setup-seafile-mysql.py
@@ -896,7 +896,7 @@ class CcnetConfigurator(AbstractConfigurator):
 
         cursor = conn.cursor()
 
-        sql_file = os.path.join(env_mgr.install_path, 'sql', 'mysql', 'ccnet.sql')
+        sql_file = '/usr/share/seafile-server/scripts/sql/mysql/ccnet.sql'
         with open(sql_file, 'r') as fp:
             content = fp.read()
 
@@ -1033,7 +1033,7 @@ class SeafileConfigurator(AbstractConfigurator):
 
         cursor = conn.cursor()
 
-        sql_file = os.path.join(env_mgr.install_path, 'sql', 'mysql', 'seafile.sql')
+        sql_file = '/usr/share/seafile-server/scripts/sql/mysql/seafile.sql'
         with open(sql_file, 'r') as fp:
             content = fp.read()
 
