diff -ruN a/tests/network/urlparse.c b/tests/network/urlparse.c
--- a/tests/network/urlparse.c	2020-02-19 11:27:00.000000000 +0100
+++ b/tests/network/urlparse.c	2020-02-20 17:29:46.667262285 +0100
@@ -61,18 +61,6 @@
 	cl_assert_equal_i(git_net_url_is_default_port(&conndata), 0);
 }
 
-void test_network_urlparse__implied_root_empty_port(void)
-{
-	cl_git_pass(git_net_url_parse(&conndata, "http://example.com:"));
-	cl_assert_equal_s(conndata.scheme, "http");
-	cl_assert_equal_s(conndata.host, "example.com");
-	cl_assert_equal_s(conndata.port, "80");
-	cl_assert_equal_s(conndata.path, "/");
-	cl_assert_equal_p(conndata.username, NULL);
-	cl_assert_equal_p(conndata.password, NULL);
-	cl_assert_equal_i(git_net_url_is_default_port(&conndata), 1);
-}
-
 void test_network_urlparse__encoded_password(void)
 {
 	cl_git_pass(git_net_url_parse(&conndata,
@@ -127,18 +115,6 @@
 	cl_assert_equal_i(git_net_url_is_default_port(&conndata), 0);
 }
 
-void test_network_urlparse__empty_port(void)
-{
-	cl_git_pass(git_net_url_parse(&conndata, "http://example.com:/resource"));
-	cl_assert_equal_s(conndata.scheme, "http");
-	cl_assert_equal_s(conndata.host, "example.com");
-	cl_assert_equal_s(conndata.port, "80");
-	cl_assert_equal_s(conndata.path, "/resource");
-	cl_assert_equal_p(conndata.username, NULL);
-	cl_assert_equal_p(conndata.password, NULL);
-	cl_assert_equal_i(git_net_url_is_default_port(&conndata), 1);
-}
-
 void test_network_urlparse__user_port(void)
 {
 	/* user@hostname.tld:port/resource */
