--- a/src/UPnPBase.cpp	2020-10-23 05:30:49.252207564 -0700
+++ b/src/UPnPBase.cpp	2020-10-23 05:26:15.927719211 -0700
@@ -824,13 +824,17 @@
 	// Null string at first
 	std::ostringstream msg;
 
+	// Declare those here to avoid 
+	// "jump to label ‘error’ [-fpermissive] crosses initialization
+	// of ‘char* ipAddress’"
+	unsigned short port;
+	char *ipAddress;
+
 	// Start UPnP
 	int ret;
-	char *ipAddress = NULL;
-	unsigned short port = 0;
-	ret = UpnpInit(ipAddress, udpPort);
+	ret = UpnpInit2(0, udpPort);
 	if (ret != UPNP_E_SUCCESS) {
-		msg << "error(UpnpInit): Error code ";
+		msg << "error(UpnpInit2): Error code ";
 		goto error;
 	}
 	port = UpnpGetServerPort();
