diff --git a/plugins/CMakeLists.txt b/plugins/CMakeLists.txt
index 1769a02a9..d4083666f 100644
--- a/plugins/CMakeLists.txt
+++ b/plugins/CMakeLists.txt
@@ -23,9 +23,7 @@ macro(check_obs_browser)
 endmacro()
 
 macro(check_obs_websocket)
-  if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/obs-websocket/CMakeLists.txt")
-    message(FATAL_ERROR "Required submodule 'obs-websocket' not available.")
-  else()
+  if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/obs-websocket/CMakeLists.txt")
     add_subdirectory(obs-websocket)
   endif()
 endmacro()
@@ -56,8 +54,6 @@ add_obs_plugin(mac-videotoolbox PLATFORMS MACOS)
 add_obs_plugin(mac-virtualcam PLATFORMS MACOS)
 add_obs_plugin(nv-filters PLATFORMS WINDOWS)
 
-check_obs_browser()
-
 add_obs_plugin(obs-ffmpeg)
 add_obs_plugin(obs-filters)
 add_obs_plugin(obs-libfdk)
