diff --git a/steam_helper/steam.cpp b/steam_helper/steam.cpp
index 190a8c05..afe7916b 100644
--- a/steam_helper/steam.cpp
+++ b/steam_helper/steam.cpp
@@ -34,7 +34,9 @@
 
 #include "ntstatus.h"
 #define WIN32_NO_STATUS
+extern "C" {
 #include <windows.h>
+}
 #include <winternl.h>
 #include <shlobj.h>
 #include <string.h>
diff --git a/vrclient_x64/vrclient_x64/json_converter.cpp b/vrclient_x64/vrclient_x64/json_converter.cpp
index 83fd2ec0..64d209a2 100644
--- a/vrclient_x64/vrclient_x64/json_converter.cpp
+++ b/vrclient_x64/vrclient_x64/json_converter.cpp
@@ -1,4 +1,6 @@
+extern "C" {
 #include <windows.h>
+}
 #include <wine/debug.h>
 
 WINE_DEFAULT_DEBUG_CHANNEL(vrclient);
