From 9e0ed865bbc68a14cf56354a576e824f013062bd Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Tue, 20 Feb 2024 11:37:38 -0500
Subject: [PATCH] Fix memory buffer support

---
 src/utils.jl | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/utils.jl b/src/utils.jl
index 6bfa36e..21c253f 100644
--- a/src/utils.jl
+++ b/src/utils.jl
@@ -260,6 +260,9 @@ consumeBOM(buf, pos) = (length(buf) >= 3 && buf[pos] == 0xef && buf[pos + 1] ==
         elseif x.data isa SubArray{UInt8}
             x = x.data
             return parent(x), first(x.indices), last(x.indices), tfile
+        else
+            x = take!(x)
+            return x, 1, length(x), tfile
         end
     elseif x isa Cmd || x isa IO
         if buffer_in_memory
-- 
2.42.1

