From 4394768e6a82ccaf1752c70f8f4c67d708d167ab Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Sat, 3 Oct 2020 08:45:51 -0400
Subject: [PATCH 1/4] Force enable GDB debug info registration

---
 src/init.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/init.c b/src/init.c
index 602583a922..927e0ab0bd 100644
--- a/src/init.c
+++ b/src/init.c
@@ -703,7 +703,7 @@ JL_DLLEXPORT void julia_init(JL_IMAGE_SEARCH rel)
     }
 #endif
 
-#if defined(JL_DEBUG_BUILD)
+#if defined(JL_DEBUG_BUILD) || 1
     jl_using_gdb_jitevents = 1;
 # else
     const char *jit_gdb = getenv("ENABLE_GDBLISTENER");
-- 
2.32.0

