From fdcbfb51afe1f7646afba9562ad5ac6b34e5857f Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Thu, 17 Sep 2020 02:08:31 -0400
Subject: [PATCH 1/7] Allow system libnghttp2

---
 deps/nghttp2.mk | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/deps/nghttp2.mk b/deps/nghttp2.mk
index a3c96789ce..843d037594 100644
--- a/deps/nghttp2.mk
+++ b/deps/nghttp2.mk
@@ -1,11 +1,9 @@
 ## nghttp2
 
 ifneq ($(USE_BINARYBUILDER_NGHTTP2), 1)
-$(error "Non-BinaryBuilder nghttp2 is not supported.")
+ifneq ($(USE_SYSTEM_NGHTTP2), 1)
+$(error "FIXME: Source build nghttp2 is not supported.")
 endif
-
-ifneq ($(USE_SYSTEM_NGHTTP2), 0)
-$(error "Non-BinaryBuilder nghttp2 is not supported.")
 endif
 
 NGHTTP2_BB_URL_BASE := https://github.com/JuliaBinaryWrappers/nghttp2_jll.jl/releases/download/nghttp2-v$(NGHTTP2_VER)+$(NGHTTP2_BB_REL)
-- 
2.28.0

