From d9cfe0880ad9922b715a7b1ffb52731c0152407d Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Tue, 11 Aug 2020 20:18:19 -0400
Subject: [PATCH 3/4] Mark load from DataType as constant

---
 src/codegen.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/codegen.cpp b/src/codegen.cpp
index f756de9271..f9f7c3f283 100644
--- a/src/codegen.cpp
+++ b/src/codegen.cpp
@@ -866,6 +866,9 @@ static void add_return_attr(T *f, Attribute::AttrKind Kind)
 
 static MDNode *best_tbaa(jl_value_t *jt) {
     jt = jl_unwrap_unionall(jt);
+    if (jt == (jl_value_t*)jl_datatype_type ||
+        (jl_is_type_type(jt) && jl_is_datatype(jl_tparam0(jt))))
+        return tbaa_const;
     if (!jl_is_datatype(jt))
         return tbaa_value;
     if (jl_is_abstracttype(jt))
-- 
2.28.0

