From 29df9129180c2f3a416e30938655b81caeb324b1 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Mon, 31 Aug 2020 12:44:58 -0400
Subject: [PATCH 3/4] Enable basic AA in O2

---
 src/aotcompile.cpp | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/aotcompile.cpp b/src/aotcompile.cpp
index 8388f238c2..e67bf7dec9 100644
--- a/src/aotcompile.cpp
+++ b/src/aotcompile.cpp
@@ -677,9 +677,7 @@ void addOptimizationPasses(legacy::PassManagerBase *PM, int opt_level,
     PM->add(createPropagateJuliaAddrspaces());
     PM->add(createScopedNoAliasAAWrapperPass());
     PM->add(createTypeBasedAAWrapperPass());
-    if (opt_level >= 3) {
-        PM->add(createBasicAAWrapperPass());
-    }
+    PM->add(createBasicAAWrapperPass());
 
     PM->add(createCFGSimplificationPass(simplifyCFGOptions));
     PM->add(createDeadCodeEliminationPass());
-- 
2.35.1

