From 4fed7a1ef59991b23b3db4f632c269d7f8c7bb56 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Fri, 30 Dec 2022 16:15:59 -0500
Subject: [PATCH 4/7] Use libc for libssp

---
 .../src/CompilerSupportLibraries_jll.jl                         | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/stdlib/CompilerSupportLibraries_jll/src/CompilerSupportLibraries_jll.jl b/stdlib/CompilerSupportLibraries_jll/src/CompilerSupportLibraries_jll.jl
index b4df77c516..a4cb82724a 100644
--- a/stdlib/CompilerSupportLibraries_jll/src/CompilerSupportLibraries_jll.jl
+++ b/stdlib/CompilerSupportLibraries_jll/src/CompilerSupportLibraries_jll.jl
@@ -49,7 +49,7 @@ else
     const libstdcxx = "libstdc++.so.6"
     const libgomp = "libgomp.so.1"
     if libc(HostPlatform()) != "musl"
-        const libssp = "libssp.so.0"
+        const libssp = "libc.so.6"
     end
 end
 
-- 
2.46.0

