From 9920804e7b17cc36c9c319df6297e3becd3e560f Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Sat, 19 Feb 2022 04:04:54 -0500
Subject: [PATCH 5/6] Update blastrampline version

---
 deps/blastrampoline.version | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/deps/blastrampoline.version b/deps/blastrampoline.version
index 86d77ab5bf..23074f7085 100644
--- a/deps/blastrampoline.version
+++ b/deps/blastrampoline.version
@@ -1,2 +1,2 @@
-BLASTRAMPOLINE_BRANCH=v3.0.4
-BLASTRAMPOLINE_SHA1=23de7a09bf354fe6f655c457bab5bf47fdd2486d
+BLASTRAMPOLINE_BRANCH=v5.0.1
+BLASTRAMPOLINE_SHA1=d32042273719672c6669f6442a0be5605d434b70
-- 
2.35.1

