From 230e28c4dd40deb1477ddcbf1fb0ef58f3b34891 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Tue, 3 Oct 2023 19:00:59 -0400
Subject: [PATCH 5/9] [julia-git] add a safeguard for bad unwinding

---
 src/stackwalk.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/stackwalk.c b/src/stackwalk.c
index 6f8f9c8b89..7fe23e26aa 100644
--- a/src/stackwalk.c
+++ b/src/stackwalk.c
@@ -288,6 +288,8 @@ JL_DLLEXPORT jl_value_t *jl_backtrace_from_here(int returnsp, int skip)
             size_t size_incr = 0;
             have_more_frames = jl_unw_stepn(&cursor, jl_array_data(ip, jl_bt_element_t) + offset,
                                             &size_incr, sp_ptr, maxincr, skip, &pgcstack, 0);
+            if (offset > JL_MAX_BT_SIZE * 10)
+                have_more_frames = 0;
             skip = 0;
             offset += size_incr;
         }
-- 
2.51.0

