From e8526039d65e91b3b0bbb64102b2b81d456bd0e0 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Sun, 13 Nov 2022 19:35:34 -0500
Subject: [PATCH 6/7] Remove library patching that should not be done for
 system libraries

---
 Makefile | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/Makefile b/Makefile
index d41f3e50d9..a09addfa72 100644
--- a/Makefile
+++ b/Makefile
@@ -420,16 +420,6 @@ else ifeq ($(JULIA_BUILD_MODE),debug)
 endif
 endif
 
-	# Fix rpaths for dependencies. This should be fixed in BinaryBuilder later.
-ifeq ($(OS), Linux)
-	-$(PATCHELF) --set-rpath '$$ORIGIN' $(DESTDIR)$(private_shlibdir)/libLLVM.$(SHLIB_EXT)
-endif
-
-	# Replace libstdc++ path, which is also moving from `lib` to `../lib/julia`.
-ifeq ($(OS),Linux)
-	$(call stringreplace,$(DESTDIR)$(shlibdir)/libjulia.$(JL_MAJOR_MINOR_SHLIB_EXT),\*libstdc++\.so\.6$$,*$(call dep_lib_path,$(shlibdir),$(private_shlibdir)/libstdc++.so.6))
-endif
-
 
 ifneq ($(LOADER_BUILD_DEP_LIBS),$(LOADER_INSTALL_DEP_LIBS))
 	# Next, overwrite relative path to libjulia-internal in our loader if $$(LOADER_BUILD_DEP_LIBS) != $$(LOADER_INSTALL_DEP_LIBS)
-- 
2.39.0

