From 334229142ad79164dc183c9b029c358ed1274df1 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Mon, 26 May 2025 23:20:16 -0400
Subject: [PATCH 06/14] Disable UCX signal handler

---
 src/init.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/init.c b/src/init.c
index e4efe1607c..1fff8a8057 100644
--- a/src/init.c
+++ b/src/init.c
@@ -697,6 +697,8 @@ static void init_global_mutexes(void) {
 
 JL_DLLEXPORT void jl_init_(jl_image_buf_t sysimage)
 {
+    putenv("UCX_HANDLE_ERRORS=none"); // disable openucx signal handler
+
     // initialize many things, in no particular order
     // but generally running from simple platform things to optional
     // configuration features
-- 
2.53.0

