From 45a5aeead6955fe71ea12d0bb6a7f9e2c6c4c28c Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Fri, 30 Dec 2022 16:15:59 -0500
Subject: [PATCH 6/7] Use libc for libssp

---
 .../src/CompilerSupportLibraries_jll.jl                         | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/stdlib/CompilerSupportLibraries_jll/src/CompilerSupportLibraries_jll.jl b/stdlib/CompilerSupportLibraries_jll/src/CompilerSupportLibraries_jll.jl
index 097659e01b..ada016aed1 100644
--- a/stdlib/CompilerSupportLibraries_jll/src/CompilerSupportLibraries_jll.jl
+++ b/stdlib/CompilerSupportLibraries_jll/src/CompilerSupportLibraries_jll.jl
@@ -48,7 +48,7 @@ else
     const libstdcxx = "libstdc++.so.6"
     const libgomp = "libgomp.so.1"
     if libc(HostPlatform()) != "musl"
-        const libssp = "libssp.so.0"
+        const libssp = "libc.so.6"
     end
 end
 
-- 
2.40.0

