From 4e5979ff72682cbbdbc1fd6ead5a348eaaf088d3 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Fri, 15 Aug 2025 12:24:38 -0400
Subject: [PATCH 07/18] [julia-git] Better support for loading packages from
 stdlib

---
 base/loading.jl | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/base/loading.jl b/base/loading.jl
index bb991ffd7a..18850dfbc4 100644
--- a/base/loading.jl
+++ b/base/loading.jl
@@ -1155,6 +1155,10 @@ function explicit_manifest_entry_path(manifest_file::String, pkg::PkgId, entry::
             ispath(path) && return entry_path(abspath(path), pkg.name, entryfile)
         end
     end
+    mbypath = manifest_uuid_path(Sys.STDLIB, pkg)
+    if mbypath isa String && isfile(mbypath)
+        return mbypath
+    end
     # no depot contains the package, return missing to stop looking
     return missing
 end
-- 
2.51.2

