From 5f314b75a300399dcc6f8e6d08ba40bbe75b84b2 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Fri, 3 Feb 2023 10:18:30 -0500
Subject: [PATCH 7/7] LLVM 15 fix

---
 src/codegen.cpp | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/src/codegen.cpp b/src/codegen.cpp
index ce5a9aa539..67fa470b33 100644
--- a/src/codegen.cpp
+++ b/src/codegen.cpp
@@ -525,6 +525,13 @@ AttributeSet Attributes(LLVMContext &C, std::initializer_list<Attribute::AttrKin
     return AttributeSet::get(C, makeArrayRef(attrs));
 }
 
+AttributeSet Attributes(LLVMContext &C, std::initializer_list<Attribute::AttrKind> attrkinds, SmallVector<Attribute, 8> attrs)
+{
+    for (size_t i = 0; i < attrkinds.size(); i++)
+        attrs.push_back(Attribute::get(C, attrkinds.begin()[i]));
+    return AttributeSet::get(C, makeArrayRef(attrs));
+}
+
 static Type *get_pjlvalue(LLVMContext &C) { return JuliaType::get_pjlvalue_ty(C); }
 
 static FunctionType *get_func_sig(LLVMContext &C) { return JuliaType::get_jlfunc_ty(C); }
@@ -898,11 +905,11 @@ static const auto jl_alloc_obj_func = new JuliaFunction{
     [](LLVMContext &C) { return AttributeList::get(C,
             AttributeSet::get(C, makeArrayRef({Attribute::getWithAllocSizeArgs(C, 1, None)})), // returns %1 bytes
 
-            Attributes(C, {Attribute::NoAlias, Attribute::NonNull,
+            Attributes(C, {Attribute::NoAlias, Attribute::NonNull}
 #if JL_LLVM_VERSION >= 150000
-            Attribute::get(C, Attribute::AllocKind, AllocFnKind::Alloc | AllocFnKind::Uninitialized | AllocFnKind::Aligned),
+                       , { Attribute::get(C, Attribute::AllocKind, static_cast<uint64_t>(AllocFnKind::Alloc | AllocFnKind::Uninitialized | AllocFnKind::Aligned)) }
 #endif
-            }),
+            ),
             None); },
 };
 static const auto jl_newbits_func = new JuliaFunction{
-- 
2.39.0

