From ef43dcd481760d15a4166c3eebe8eb500869fe2b Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Sun, 2 Jun 2024 15:24:06 -0500
Subject: [PATCH 7/8] Fix gcc12 error on aarch64

---
 src/stackwalk.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/stackwalk.c b/src/stackwalk.c
index 33fbd07a5a..2f9ad76c4d 100644
--- a/src/stackwalk.c
+++ b/src/stackwalk.c
@@ -988,7 +988,7 @@ JL_DLLEXPORT size_t jl_record_backtrace(jl_task_t *t, jl_bt_element_t *bt_data,
         memset(&c, 0, sizeof(c));
      #if defined(_OS_LINUX_) && defined(__GLIBC__)
         __jmp_buf *mctx = &t->ctx.ctx->uc_mcontext->__jmpbuf;
-        mcontext_t *mc = &c.uc_mcontext;
+        mcontext_t *mc = (mcontext_t*)&c.uc_mcontext;
       #if defined(_CPU_X86_)
         // https://github.com/bminor/glibc/blame/master/sysdeps/i386/__longjmp.S
         // https://github.com/bminor/glibc/blame/master/sysdeps/i386/jmpbuf-offsets.h
-- 
2.45.2

