From 0e26349fcd0533f3e9a73de3f24ad9c9a82da74d Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Fri, 15 Aug 2025 12:24:38 -0400
Subject: [PATCH 7/8] [julia-git] Better support for loading packages from
 stdlib

---
 base/loading.jl | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/base/loading.jl b/base/loading.jl
index 6489e0a78c..ad034c27a7 100644
--- a/base/loading.jl
+++ b/base/loading.jl
@@ -1313,6 +1313,10 @@ function explicit_manifest_entry_load_spec(manifest_file::String, pkg::PkgId, en
             ispath(path) && return PkgLoadSpec(entry_path(abspath(path), pkg.name, entryfile), syntax_version)
         end
     end
+    mbypath = manifest_uuid_path(Sys.STDLIB, pkg)
+    if mbypath isa String && isfile(mbypath)
+        return mbypath
+    end
     # no depot contains the package, return missing to stop looking
     return missing
 end
-- 
2.53.0

