diff --git a/src/debuginfo.cpp b/src/debuginfo.cpp
index a03b7447fa..46edf285d3 100644
--- a/src/debuginfo.cpp
+++ b/src/debuginfo.cpp
@@ -243,14 +243,17 @@ public:
                 auto sName = section.getName();
                 if (!sName)
                     continue;
+                if (sName.get() != ".ARM.exidx") {
+                    continue;
+                }
 #else
                 StringRef sName;
                 if (section.getName(sName))
                     continue;
-#endif
                 if (sName != ".ARM.exidx") {
                     continue;
                 }
+#endif
             }
             uint64_t loadaddr = L.getSectionLoadAddress(section);
             size_t seclen = section.getSize();
