diff --git a/src/processor_x86.cpp b/src/processor_x86.cpp
index 06b3e42f07..33dd2e09a7 100644
--- a/src/processor_x86.cpp
+++ b/src/processor_x86.cpp
@@ -833,6 +833,9 @@ get_llvm_target_noext(const TargetData<feature_sz> &data)
     features.push_back("+sse2");
     features.push_back("+mmx");
     features.push_back("+fxsr");
+#ifdef _CPU_X86_64_
+    features.push_back("+64bit");
+#endif
     return std::make_pair(std::move(name), std::move(features));
 }
 
