--- src/PlatformEngines.jl      2021-06-17 23:25:32.000000000 -0400
+++ src/PlatformEngines.jl      2021-06-22 08:18:56.251328138 -0400
@@ -12,9 +12,10 @@
 export probe_platform_engines!, verify, unpack, package, download_verify_unpack

 const EXE7Z_LOCK = ReentrantLock()
-const EXE7Z = Ref{String}()
+const EXE7Z = Ref{String}("/usr/bin/7z")

 function exe7z()
+    return `/usr/bin/7z`
     # If the JLL is available, use the wrapper function defined in there
     if p7zip_jll.is_available()
         return p7zip_jll.p7zip()
