--- src/openvpn/console.c.orig	2013-01-30 09:42:57.480257300 +0100
+++ src/openvpn/console.c	2013-01-30 09:43:07.446179811 +0100
@@ -162,7 +162,7 @@ get_console_input_systemd (const char *p
   struct argv argv;
 
   argv_init (&argv);
-  argv_printf (&argv, "/bin/systemd-ask-password");
+  argv_printf (&argv, "/usr/bin/systemd-ask-password");
   argv_printf_cat (&argv, "%s", prompt);
 
   if ((std_out = openvpn_popen (&argv, NULL)) < 0) {
