diff --git a/src/utils/common.py b/src/utils/common.py
index 5dbe664..b975708 100644
--- a/src/utils/common.py
+++ b/src/utils/common.py
@@ -29,11 +29,10 @@ makedirs(str(BASE_PATH / 'endpoints'), exist_ok=True)
 # Constructing paths - executables
 
 external = Path(dirname(realpath(__file__))) / 'external'
-arch = '64' * (architecture()[0] == '64bit')
 
-protoc = str(external / 'protoc' / ('protoc' + {'win32': '.exe', 'darwin': '_osx'}.get(platform, arch)))
+protoc = '/usr/bin/protoc'
 dex2jar = str(external / 'dex2jar' / ('d2j-dex2jar.' + {'win32': 'bat'}.get(platform, 'sh')))
-jad = str(external / 'jad' / ('jad' + {'win32': '.exe', 'darwin': '_osx'}.get(platform, '')))
+jad = '/usr/bin/jad'
 
 # Disable the C++ extension for Python-Protobuf (for consistent behaviour) [1]
 # [1] https://github.com/google/protobuf/blob/cf1418/python/google/protobuf/internal/api_implementation.py#L72
