diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/3columnA4.py trunk/Scribus/scribus/plugins/scriptplugin/samples/3columnA4.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/3columnA4.py	2017-03-29 21:17:58.546722296 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/samples/3columnA4.py	2017-03-29 21:44:22.445693514 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """ Creates 3 column layout on A4 paper and save it under 3columnA4.sla filename. This is a simple way to demonstrate creating a doc on the fly. """
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/3columnUSLTR.py trunk/Scribus/scribus/plugins/scriptplugin/samples/3columnUSLTR.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/3columnUSLTR.py	2017-03-29 21:17:58.546722296 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/samples/3columnUSLTR.py	2017-03-29 21:44:30.395656189 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """ Creates 3 column layout on US Letter  paper and save it under 3columnUSLTR.sla filename. This is a simple way to demonstrate creating a doc on the fly. """
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/Calender.py trunk/Scribus/scribus/plugins/scriptplugin/samples/Calender.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/Calender.py	2017-03-29 21:17:58.546722296 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/samples/Calender.py	2017-03-29 21:44:43.712260230 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """ This Script creates a Calendar Sheet for the Current Month """
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/ExtractText.py trunk/Scribus/scribus/plugins/scriptplugin/samples/ExtractText.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/ExtractText.py	2017-03-29 21:17:58.543388972 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/samples/ExtractText.py	2017-03-29 21:44:48.508904307 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/Sample1.py trunk/Scribus/scribus/plugins/scriptplugin/samples/Sample1.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/Sample1.py	2017-03-29 21:17:58.546722296 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/samples/Sample1.py	2017-03-29 21:44:57.488861973 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """ A sample script """
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/boilerplate.py trunk/Scribus/scribus/plugins/scriptplugin/samples/boilerplate.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/boilerplate.py	2017-03-29 21:17:58.546722296 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/samples/boilerplate.py	2017-03-29 21:45:05.005493161 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 import sys
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/golden-mean.py trunk/Scribus/scribus/plugins/scriptplugin/samples/golden-mean.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/golden-mean.py	2017-03-29 21:17:58.546722296 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/samples/golden-mean.py	2017-03-29 21:45:09.598804794 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """Golden Mean for Scribus.
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/legende.py trunk/Scribus/scribus/plugins/scriptplugin/samples/legende.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/legende.py	2017-03-29 21:17:58.550055620 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/samples/legende.py	2017-03-29 21:45:15.442110500 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """ When you have an image selected this script creates small text legende
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/moins_10_pourcent_group.py trunk/Scribus/scribus/plugins/scriptplugin/samples/moins_10_pourcent_group.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/moins_10_pourcent_group.py	2017-03-29 21:17:58.550055620 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/samples/moins_10_pourcent_group.py	2017-03-29 21:45:22.102078985 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """ Make selected group smaller by 10% """
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/plus_10_pourcent_group.py trunk/Scribus/scribus/plugins/scriptplugin/samples/plus_10_pourcent_group.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/plus_10_pourcent_group.py	2017-03-29 21:17:58.546722296 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/samples/plus_10_pourcent_group.py	2017-03-29 21:45:28.348716065 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """ Make selected group larger by 10% """
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/pochette_cd.py trunk/Scribus/scribus/plugins/scriptplugin/samples/pochette_cd.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/pochette_cd.py	2017-03-29 21:17:58.546722296 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/samples/pochette_cd.py	2017-03-29 21:45:34.875351791 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """ This script creates a CD Pochette - a paper pocket for CD/DVD disc """
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/quote.py trunk/Scribus/scribus/plugins/scriptplugin/samples/quote.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/quote.py	2017-03-29 21:17:58.546722296 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/samples/quote.py	2017-03-29 21:47:58.714663426 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: iso-8859-1 -*-
 
 """ This script changes quotation marks from " " to french style """
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/sample_db_usage.py trunk/Scribus/scribus/plugins/scriptplugin/samples/sample_db_usage.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/sample_db_usage.py	2017-03-29 21:17:58.546722296 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/samples/sample_db_usage.py	2017-03-29 21:45:48.628619833 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """OK. This is a sample Scribus database connection with Python HOWTO.
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/trait_de_coupe.py trunk/Scribus/scribus/plugins/scriptplugin/samples/trait_de_coupe.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/trait_de_coupe.py	2017-03-29 21:17:58.550055620 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/samples/trait_de_coupe.py	2017-03-29 21:45:53.861928292 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """ Draws a "crop marks" around selected object """
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/wordcount.py trunk/Scribus/scribus/plugins/scriptplugin/samples/wordcount.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/samples/wordcount.py	2017-03-29 21:17:58.546722296 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/samples/wordcount.py	2017-03-29 21:46:00.898561484 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """ Counts the words in the whole document or in a textframe """
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/scripts/Autoquote.py trunk/Scribus/scribus/plugins/scriptplugin/scripts/Autoquote.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/scripts/Autoquote.py	2017-03-29 21:17:58.813388241 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/scripts/Autoquote.py	2017-03-29 21:42:53.636107065 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 # File: quotes.py - changes typewriter quotes to typographic quotes
 # © 2010.08.28 Gregory Pittman
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/scripts/Autoquote2.py trunk/Scribus/scribus/plugins/scriptplugin/scripts/Autoquote2.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/scripts/Autoquote2.py	2017-03-29 21:17:58.813388241 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/scripts/Autoquote2.py	2017-03-29 21:42:58.732750178 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 # File: spacenquotes.py - changes typewriter quotes to typographic quotes with option to add spaces
 # © 2010.08.28 Gregory Pittman
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/scripts/Caption.py trunk/Scribus/scribus/plugins/scriptplugin/scripts/Caption.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/scripts/Caption.py	2017-03-29 21:17:58.813388241 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/scripts/Caption.py	2017-03-29 21:42:05.079663595 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8  -*-
 
 # ****************************************************************************
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/scripts/ColorChart.py trunk/Scribus/scribus/plugins/scriptplugin/scripts/ColorChart.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/scripts/ColorChart.py	2017-03-29 21:17:58.816721565 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/scripts/ColorChart.py	2017-03-29 21:43:22.059308755 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 
 """
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/scripts/DirectImageImport.py trunk/Scribus/scribus/plugins/scriptplugin/scripts/DirectImageImport.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/scripts/DirectImageImport.py	2017-03-29 21:17:58.816721565 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/scripts/DirectImageImport.py	2017-03-29 21:43:30.555935934 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """ 
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/scripts/FontSample.py trunk/Scribus/scribus/plugins/scriptplugin/scripts/FontSample.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/scripts/FontSample.py	2017-03-29 21:17:58.816721565 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/scripts/FontSample.py	2017-03-29 21:43:36.055910341 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 # ****************************************************************************
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/scripts/InfoBox.py trunk/Scribus/scribus/plugins/scriptplugin/scripts/InfoBox.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/scripts/InfoBox.py	2017-03-29 21:17:58.816721565 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/scripts/InfoBox.py	2017-03-29 21:42:20.419593323 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8  -*-
 
 # ****************************************************************************
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/scripts/Ligatursatz.py trunk/Scribus/scribus/plugins/scriptplugin/scripts/Ligatursatz.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/scripts/Ligatursatz.py	2017-03-29 21:17:58.816721565 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/scripts/Ligatursatz.py	2017-03-29 21:44:35.838963939 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf8 -*-
 
 """
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/scripts/color2csv.py trunk/Scribus/scribus/plugins/scriptplugin/scripts/color2csv.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/scripts/color2csv.py	2017-03-29 21:17:58.816721565 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/scripts/color2csv.py	2017-03-29 21:43:50.619175789 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 """
 ABOUT THIS SCRIPT:
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/scripts/csv2color.py trunk/Scribus/scribus/plugins/scriptplugin/scripts/csv2color.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/scripts/csv2color.py	2017-03-29 21:17:58.816721565 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/scripts/csv2color.py	2017-03-29 21:43:55.282487348 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 """
 ABOUT THIS SCRIPT:
diff -Naur trunk.orig/Scribus/scribus/plugins/scriptplugin/scripts/importcsv2table.py trunk/Scribus/scribus/plugins/scriptplugin/scripts/importcsv2table.py
--- trunk.orig/Scribus/scribus/plugins/scriptplugin/scripts/importcsv2table.py	2017-03-29 21:17:58.816721565 +0200
+++ trunk/Scribus/scribus/plugins/scriptplugin/scripts/importcsv2table.py	2017-03-29 21:44:02.932451592 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 """
 ABOUT THIS SCRIPT:
