diff --git a/extern/audaspace/CMakeLists.txt b/extern/audaspace/CMakeLists.txt
index 9e2921ef35a..67f8ada7802 100644
--- a/extern/audaspace/CMakeLists.txt
+++ b/extern/audaspace/CMakeLists.txt
@@ -412,6 +412,10 @@ if(WITH_FFMPEG)
 			plugins/ffmpeg/FFMPEGWriter.h
 		)
 
+		list(APPEND INC
+			../../intern/ffmpeg
+		)
+
 		if(NOT PLUGIN_FFMPEG)
 			list(APPEND INCLUDE ${FFMPEG_INCLUDE_DIRS})
 			list(APPEND LIBRARIES ${FFMPEG_LIBRARIES})
diff --git a/extern/audaspace/plugins/ffmpeg/FFMPEGReader.cpp b/extern/audaspace/plugins/ffmpeg/FFMPEGReader.cpp
index 6b79cc5abfd..5e9ec29f302 100644
--- a/extern/audaspace/plugins/ffmpeg/FFMPEGReader.cpp
+++ b/extern/audaspace/plugins/ffmpeg/FFMPEGReader.cpp
@@ -16,6 +16,7 @@
 
 #include "FFMPEGReader.h"
 #include "Exception.h"
+#include "../../../../intern/ffmpeg/ffmpeg_compat.h"
 
 #include <algorithm>
 
diff --git a/extern/audaspace/plugins/ffmpeg/FFMPEGWriter.cpp b/extern/audaspace/plugins/ffmpeg/FFMPEGWriter.cpp
index f79f0f7fc6b..6b89a83613c 100644
--- a/extern/audaspace/plugins/ffmpeg/FFMPEGWriter.cpp
+++ b/extern/audaspace/plugins/ffmpeg/FFMPEGWriter.cpp
@@ -16,6 +16,7 @@
 
 #include "FFMPEGWriter.h"
 #include "Exception.h"
+#include "../../../../intern/ffmpeg/ffmpeg_compat.h"
 
 #include <algorithm>
 #include <cstring>
