From d01377c9414837a8a1c7771bc7ecec0c8bffd9f8 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Wed, 28 Dec 2022 11:07:17 -0500
Subject: [PATCH] Get rid of problematic path dance

---
 GR/src/preferences.jl | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/GR/src/preferences.jl b/GR/src/preferences.jl
index 42dbe3c..3fb569d 100644
--- a/GR/src/preferences.jl
+++ b/GR/src/preferences.jl
@@ -77,14 +77,6 @@ module GRPreferences
             libGRM[]  = GR_jll.libGRM
             libGKS[]  = GR_jll.libGKS
 
-            # Because GR_jll does not dlopen as of 0.69.1+1, we need to append
-            # the LIBPATH_list similar to JLLWrappers.@init_library_product
-            push!(GR_jll.LIBPATH_list, dirname(GR_jll.libGR))
-            # Recompute LIBPATH similar to JLLWrappers.@generate_init_footer
-            unique!(GR_jll.LIBPATH_list)
-            pathsep = GR_jll.JLLWrappers.pathsep
-            GR_jll.LIBPATH[] = join(vcat(GR_jll.LIBPATH_list, Base.invokelatest(GR_jll.JLLWrappers.get_julia_libpaths))::Vector{String}, pathsep)
-
             libpath[] = GR_jll.LIBPATH[]
             ENV["GRDIR"] = grdir[]
         elseif binary == "system"
-- 
2.52.0

