--- configure.ac	2016-05-27 18:15:33.000000000 +0200
+++ configure.ac.new	2016-05-28 11:17:11.622530477 +0200
@@ -1,8 +1,7 @@
 AC_INIT([Paper ], 1.1, [https://github.com/snwh/paper-icon-theme/issues], [paper-icon-theme], [http://snwh.org/paper])
 AM_INIT_AUTOMAKE([-Wno-portability no-dist-gzip dist-xz foreign subdir-objects 1.9 tar-ustar])
 AC_PREFIX_DEFAULT(/usr/local)
-AM_SILENT_RULES([yes])
-
+m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 
 AC_CONFIG_FILES([Makefile])
 AC_OUTPUT
@@ -15,4 +14,4 @@
         exec_prefix:            ${exec_prefix}
         datarootdir:            ${datarootdir}
 
-])
\ No newline at end of file
+])
