diff -Naur glib-1.2.10-orig/configure.in glib-1.2.10/configure.in
--- glib-1.2.10-orig/configure.in	2013-07-22 01:33:26.930091236 -0400
+++ glib-1.2.10/configure.in	2013-07-22 01:35:19.988776415 -0400
@@ -17,7 +17,7 @@
 
 dnl we need to AC_DIVERT_PUSH/AC_DIVERT_POP these variable definitions so they
 dnl are available for $ac_help expansion (don't we all *love* autoconf?)
-AC_DIVERT_PUSH(AC_DIVERSION_NOTICE)dnl
+AC_DIVERT_PUSH()dnl
 #
 # The following version number definitions apply to GLib, GModule and GThread
 # as a whole, so if changes occoured in any of them, they are all
@@ -81,7 +81,7 @@
 
 dnl figure debugging default, prior to $ac_help setup
 dnl
-AC_DIVERT_PUSH(AC_DIVERSION_NOTICE)dnl
+AC_DIVERT_PUSH()dnl
 if test `expr $GLIB_MINOR_VERSION \% 2` = 1 ; then
 	debug_default=yes
 else
diff -Naur glib-1.2.10-orig/gmodule/Makefile.am glib-1.2.10/gmodule/Makefile.am
--- glib-1.2.10-orig/gmodule/Makefile.am	2013-07-22 01:33:26.933424531 -0400
+++ glib-1.2.10/gmodule/Makefile.am	2013-07-22 01:38:46.553041607 -0400
@@ -42,7 +42,7 @@
 libgplugin_b_la_LIBADD = @G_MODULE_LIBS@ # $(libglib)
 
 noinst_PROGRAMS = testgmodule
-testgmodule_LDFLAGS += @G_MODULE_LDFLAGS@
+testgmodule_LDFLAGS = @G_MODULE_LDFLAGS@
 testgmodule_LDADD = libgmodule.la $(libglib) @G_MODULE_LIBS@
 
 .PHONY: files release
