From e314deae474cd41781e15a962ff23baabbcf2d84 Mon Sep 17 00:00:00 2001
From: Coelacanthus <CoelacanthusHex@gmail.com>
Date: Fri, 14 Jun 2024 15:10:53 +0800
Subject: [PATCH] chore: allow None as BUILD_TYPE

Signed-off-by: Coelacanthus <CoelacanthusHex@gmail.com>
---
 cmake/BuildTypes.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/BuildTypes.cmake b/cmake/BuildTypes.cmake
index e90d8f8..167579b 100644
--- a/cmake/BuildTypes.cmake
+++ b/cmake/BuildTypes.cmake
@@ -49,7 +49,7 @@ foreach(build IN ITEMS ASan TSan MSan)
     )
     list(APPEND KNOWN_BUILD_TYPES ${build})
 endforeach()
-list(APPEND KNOWN_BUILD_TYPES Debug Release RelWithDebInfo MinSizeRel)
+list(APPEND KNOWN_BUILD_TYPES Debug Release RelWithDebInfo MinSizeRel None)
 
 if (NOT CMAKE_BUILD_TYPE IN_LIST KNOWN_BUILD_TYPES)
     message(FATAL_ERROR "Unknown build type: ${CMAKE_BUILD_TYPE}. Choices are ${KNOWN_BUILD_TYPES}")
-- 
2.45.2

