diff --git a/contents/ui/main.qml b/contents/ui/main.qml
index 3107699..fe66e03 100644
--- a/contents/ui/main.qml
+++ b/contents/ui/main.qml
@@ -9,7 +9,6 @@ import org.kde.taskmanager       as TaskManager
 import org.kde.kirigami          as Kirigami
 import org.kde.activities        as Activities
 import "../tools/Tools.js"       as Tools
-import org.kde.plasma.private.appmenu 1.0 as AppMenuPrivate
 
 PlasmoidItem {
     id: root
