From e4f3220e535a7bc93730e50d1f10c89ef3996075 Mon Sep 17 00:00:00 2001
From: Chocobo1 <Chocobo1@users.noreply.github.com>
Date: Fri, 13 May 2022 18:15:52 +0800
Subject: [PATCH] Provide project information to autotools

---
 configure.ac | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index f0b1bdb..03bf9a3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -30,7 +30,8 @@
 # POSSIBILITY OF SUCH DAMAGE.
 
 AC_PREREQ(2.69)
-AC_INIT(package-unused, version-unused,, libbacktrace)
+AC_INIT(libbacktrace, 1.0, https://github.com/ianlancetaylor/libbacktrace/issues,
+        libbacktrace, https://github.com/ianlancetaylor/libbacktrace)
 AC_CONFIG_SRCDIR(backtrace.h)
 AC_CONFIG_HEADER(config.h)
 AC_CONFIG_MACRO_DIR(config)
