diff --git i/CMakeLists.txt w/CMakeLists.txt
index e913122..2cc3c3d 100644
--- i/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -634,7 +634,6 @@ macro(srt_make_application name)
 		# be placed into the binrary directory anyway.
 		# XXX not sure about Mac.
 		# See this name used already in install(${TARGET_srt} LIBRARY DESTINATION...).
-		set(FORCE_RPATH LINK_FLAGS -Wl,-rpath,.,-rpath,../${CMAKE_INSTALL_LIBDIR} BUILD_WITH_INSTALL_RPATH TRUE INSTALL_RPATH_USE_LINK_PATH TRUE)
 	endif()
 
 	# We state that Darwin always uses CLANG compiler, which honors this flag the same way.
