diff -ur usr/share/applications/google-chrome.desktop usr-patched/share/applications/google-chrome.desktop
--- usr/share/applications/google-chrome.desktop	2016-09-25 06:31:36.000000000 +0300
+++ usr-patched/share/applications/google-chrome.desktop	2016-10-09 16:58:01.694853900 +0300
@@ -105,6 +105,7 @@
 Comment[zh_CN]=访问互联网
 Comment[zh_HK]=連線到網際網路
 Comment[zh_TW]=連線到網際網路
+StartupWMClass=Google-chrome
 Exec=/usr/bin/google-chrome-stable %U
 Terminal=false
 Icon=google-chrome
@@ -112,6 +113,17 @@
 Categories=Network;WebBrowser;
 MimeType=text/html;text/xml;application/xhtml_xml;image/webp;x-scheme-handler/http;x-scheme-handler/https;x-scheme-handler/ftp;
 X-Ayatana-Desktop-Shortcuts=NewWindow;NewIncognito
+Actions=new-window;new-private-window;
+
+[Desktop Action new-window]
+Name=New Window
+StartupWMClass=Google-chrome
+Exec=/usr/bin/google-chrome-stable
+
+[Desktop Action new-private-window]
+Name=New Private Window
+StartupWMClass=Google-chrome
+Exec=/usr/bin/google-chrome-stable --incognito
 
 [NewWindow Shortcut Group]
 Name=New Window
