diff --git a/Lib/ufo2ft/featureWriters/markFeatureWriter.py b/Lib/ufo2ft/featureWriters/markFeatureWriter.py
index a71a88b..9542bab 100644
--- a/Lib/ufo2ft/featureWriters/markFeatureWriter.py
+++ b/Lib/ufo2ft/featureWriters/markFeatureWriter.py
@@ -2,6 +2,7 @@ import itertools
 import re
 from collections import OrderedDict, defaultdict
 from functools import partial
+from typing import Dict, Optional, Set, Tuple
 
 from ufo2ft.constants import INDIC_SCRIPTS, OBJECT_LIBS_KEY, USE_SCRIPTS
 from ufo2ft.featureWriters import BaseFeatureWriter, ast
@@ -725,25 +726,40 @@ class MarkFeatureWriter(BaseFeatureWriter):
             result.append(MarkToLigaPos(glyphName, ligatureMarks))
         return result
 
-    def _makeContextualAttachments(self, glyphClass, liga=False):
-        ctx = self.context
-        result = defaultdict(list)
-        markGlyphNames = ctx.markGlyphNames
-        for glyphName, anchors in sorted(ctx.anchorLists.items()):
-            if glyphName in markGlyphNames:
-                continue
-            if glyphClass and glyphName not in glyphClass:
+    def _makeContextualAttachments(
+        self, baseClass: Optional[Set[str]], ligatureClass: Optional[Set[str]]
+    ) -> Tuple[Dict[str, Tuple[str, NamedAnchor]], Dict[str, Tuple[str, NamedAnchor]]]:
+        def includedOrNoClass(gdefClass: Optional[Set[str]], glyphName: str) -> bool:
+            return glyphName in gdefClass if gdefClass is not None else True
+
+        def includedInClass(gdefClass: Optional[Set[str]], glyphName: str) -> bool:
+            return glyphName in gdefClass if gdefClass is not None else False
+
+        baseResult = defaultdict(list)
+        ligatureResult = defaultdict(list)
+
+        for glyphName, anchors in sorted(self.context.anchorLists.items()):
+            if glyphName in self.context.markGlyphNames:
                 continue
             for anchor in anchors:
                 # Skip non-contextual anchors
                 if not anchor.isContextual:
                     continue
-                # If we are building the mark2liga lookup, skip anchors without a number
-                if liga and anchor.number is None:
-                    continue
-                # If we are building the mark2base lookup, skip anchors with a number
-                if not liga and anchor.number is not None:
+
+                # See "after" truth table for what this logic hopes to achieve:
+                # https://github.com/googlefonts/ufo2ft/pull/890#issuecomment-2498032081
+                if anchor.number is not None and includedOrNoClass(
+                    ligatureClass, glyphName
+                ):
+                    dest = ligatureResult
+                elif anchor.number is None and (
+                    includedOrNoClass(baseClass, glyphName)
+                    or includedInClass(ligatureClass, glyphName)
+                ):
+                    dest = baseResult
+                else:
                     continue
+
                 anchor_context = anchor.libData.get("GPOS_Context", "").strip()
                 if not anchor_context:
                     self.log.warning(
@@ -752,8 +768,8 @@ class MarkFeatureWriter(BaseFeatureWriter):
                         glyphName,
                     )
                     continue
-                result[anchor_context].append((glyphName, anchor))
-        return result
+                dest[anchor_context].append((glyphName, anchor))
+        return baseResult, ligatureResult
 
     @staticmethod
     def _iterAttachments(attachments, include=None, marksFilter=None):
@@ -1031,12 +1047,9 @@ class MarkFeatureWriter(BaseFeatureWriter):
         ctx.markToMarkAttachments = self._makeMarkToMarkAttachments()
 
         baseClass = self.context.gdefClasses.base
-        ctx.contextualMarkToBaseAnchors = self._makeContextualAttachments(baseClass)
-
         ligatureClass = self.context.gdefClasses.ligature
-        ctx.contextualMarkToLigaAnchors = self._makeContextualAttachments(
-            ligatureClass,
-            True,
+        ctx.contextualMarkToBaseAnchors, ctx.contextualMarkToLigaAnchors = (
+            self._makeContextualAttachments(baseClass, ligatureClass)
         )
 
         abvmGlyphs, notAbvmGlyphs = self._getAbvmGlyphs()
diff --git a/Lib/ufo2ft/instantiator.py b/Lib/ufo2ft/instantiator.py
index 067eb1f..fcc7f97 100644
--- a/Lib/ufo2ft/instantiator.py
+++ b/Lib/ufo2ft/instantiator.py
@@ -251,7 +251,7 @@ class Instantiator:
         do_glyphs=True,
     ):
         """Instantiates a new data class from a Designspace object."""
-        if designspace.default is None:
+        if designspace.findDefault() is None:
             raise InstantiatorError(_error_msg_no_default(designspace))
 
         if any(hasattr(axis, "values") for axis in designspace.axes):
diff --git a/Lib/ufo2ft/outlineCompiler.py b/Lib/ufo2ft/outlineCompiler.py
index d8816c6..067d3e7 100644
--- a/Lib/ufo2ft/outlineCompiler.py
+++ b/Lib/ufo2ft/outlineCompiler.py
@@ -988,13 +988,31 @@ class BaseOutlineCompiler:
                 for glyphs, box in self.ufo.lib.get(COLR_CLIP_BOXES_KEY, ())
                 for glyphName in glyphs
             }
-            self.otf["COLR"] = buildCOLR(
+            colr = buildCOLR(
                 layerInfo,
                 glyphMap=glyphMap,
                 clipBoxes=clipBoxes,
                 allowLayerReuse=self.colrLayerReuse,
             )
 
+            # Warn if there are any COLRv0 base glyphs and the gid1 isn't blank
+            # https://github.com/MicrosoftDocs/typography-issues/issues/346
+            if (colr.version == 0 or colr.table.BaseGlyphRecordCount > 0) and len(
+                self.glyphOrder
+            ) > 1:
+                g1 = self.allGlyphs[self.glyphOrder[1]]
+                if len(g1) > 0 or len(g1.components) > 0:
+                    logger.warning(
+                        "COLRv0 might not render correctly on Windows because "
+                        "the glyph at index 1 is not empty ('%s'). DirectWrite's "
+                        "COLRv0 implementation in Windows 10 used to require glyph ID "
+                        "1 to be blank, see: "
+                        "https://github.com/MicrosoftDocs/typography-issues/issues/346",
+                        g1.name,
+                    )
+
+            self.otf["COLR"] = colr
+
     def _computeCOLRClipBoxes(self):
         if (
             "COLR" not in self.otf
diff --git a/Lib/ufo2ft/postProcessor.py b/Lib/ufo2ft/postProcessor.py
index c5378c6..4e2bcbd 100644
--- a/Lib/ufo2ft/postProcessor.py
+++ b/Lib/ufo2ft/postProcessor.py
@@ -3,6 +3,7 @@ import logging
 import re
 from io import BytesIO
 
+from fontTools.cffLib.CFFToCFF2 import convertCFFToCFF2
 from fontTools.ttLib import TTFont
 from fontTools.ttLib.standardGlyphOrder import standardGlyphOrder
 
@@ -134,7 +135,8 @@ class PostProcessor:
                 cffInputVersion == CFFVersion.CFF
                 and cffOutputVersion == CFFVersion.CFF2
             ):
-                self._convert_cff_to_cff2(self.otf)
+                logger.info("Converting CFF table to CFF2")
+                convertCFFToCFF2(self.otf)
             else:
                 raise NotImplementedError(
                     "Unsupported CFF conversion {cffInputVersion} => {cffOutputVersion}"
@@ -316,22 +318,6 @@ class PostProcessor:
         else:
             return None
 
-    @staticmethod
-    def _convert_cff_to_cff2(otf):
-        from fontTools.varLib.cff import convertCFFtoCFF2
-
-        logger.info("Converting CFF table to CFF2")
-
-        # convertCFFtoCFF2 doesn't strip T2CharStrings' widths, so we do it ourselves
-        # https://github.com/fonttools/fonttools/issues/1835
-        charstrings = otf["CFF "].cff[0].CharStrings
-        for glyph_name in otf.getGlyphOrder():
-            cs = charstrings[glyph_name]
-            cs.decompile()
-            cs.program = _stripCharStringWidth(cs.program)
-
-        convertCFFtoCFF2(otf)
-
     @classmethod
     def _subroutinize(cls, backend, otf, cffVersion):
         subroutinize = getattr(cls, f"_subroutinize_with_{backend.value}")
@@ -375,47 +361,6 @@ class PostProcessor:
         compiler.compile()
 
 
-# Adapted from fontTools.cff.specializer.programToCommands
-# https://github.com/fonttools/fonttools/blob/babca16
-# /Lib/fontTools/cffLib/specializer.py#L40-L122
-# When converting from CFF to CFF2 we need to drop the charstrings' widths.
-# This function returns a new charstring program without the initial width value.
-# TODO: Move to fontTools?
-def _stripCharStringWidth(program):
-    seenWidthOp = False
-    result = []
-    stack = []
-    for token in program:
-        if not isinstance(token, str):
-            stack.append(token)
-            continue
-
-        if (not seenWidthOp) and token in {
-            "hstem",
-            "hstemhm",
-            "vstem",
-            "vstemhm",
-            "cntrmask",
-            "hintmask",
-            "hmoveto",
-            "vmoveto",
-            "rmoveto",
-            "endchar",
-        }:
-            seenWidthOp = True
-            parity = token in {"hmoveto", "vmoveto"}
-            numArgs = len(stack)
-            if numArgs and (numArgs % 2) ^ parity:
-                stack.pop(0)  # pop width
-
-        result.extend(stack)
-        result.append(token)
-        stack = []
-    if stack:
-        result.extend(stack)
-    return result
-
-
 def _reloadFont(font: TTFont) -> TTFont:
     """Recompile a font to arrive at the final internal layout."""
     stream = BytesIO()
diff --git a/tests/data/ColorTest.ufo/glyphs/contents.plist b/tests/data/ColorTest.ufo/glyphs/contents.plist
index 3a55d6a..90e348b 100644
--- a/tests/data/ColorTest.ufo/glyphs/contents.plist
+++ b/tests/data/ColorTest.ufo/glyphs/contents.plist
@@ -8,5 +8,7 @@
     <string>b.glif</string>
     <key>c</key>
     <string>c.glif</string>
+    <key>space</key>
+    <string>space.glif</string>
   </dict>
 </plist>
diff --git a/tests/data/ColorTest.ufo/glyphs/space.glif b/tests/data/ColorTest.ufo/glyphs/space.glif
new file mode 100644
index 0000000..ea10aa5
--- /dev/null
+++ b/tests/data/ColorTest.ufo/glyphs/space.glif
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<glyph name="space" format="2">
+  <unicode hex="0020"/>
+  <advance width="250"/>
+</glyph>
diff --git a/tests/data/ColorTestRaw.ufo/glyphs/contents.plist b/tests/data/ColorTestRaw.ufo/glyphs/contents.plist
index 0d46f08..e22d7b9 100644
--- a/tests/data/ColorTestRaw.ufo/glyphs/contents.plist
+++ b/tests/data/ColorTestRaw.ufo/glyphs/contents.plist
@@ -8,5 +8,7 @@
     <string>a.color1.glif</string>
     <key>a.color2</key>
     <string>a.color2.glif</string>
+    <key>space</key>
+    <string>space.glif</string>
   </dict>
 </plist>
diff --git a/tests/data/ColorTestRaw.ufo/glyphs/space.glif b/tests/data/ColorTestRaw.ufo/glyphs/space.glif
new file mode 100644
index 0000000..ea10aa5
--- /dev/null
+++ b/tests/data/ColorTestRaw.ufo/glyphs/space.glif
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<glyph name="space" format="2">
+  <unicode hex="0020"/>
+  <advance width="250"/>
+</glyph>
diff --git a/tests/data/DSv5/MutatorSansVariable_Weight-CFF2.ttx b/tests/data/DSv5/MutatorSansVariable_Weight-CFF2.ttx
index ff1d945..915e567 100644
--- a/tests/data/DSv5/MutatorSansVariable_Weight-CFF2.ttx
+++ b/tests/data/DSv5/MutatorSansVariable_Weight-CFF2.ttx
@@ -128,51 +128,6 @@
   </cmap>
 
   <name>
-    <namerecord nameID="256" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Weight
-    </namerecord>
-    <namerecord nameID="257" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans Light Condensed
-    </namerecord>
-    <namerecord nameID="258" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansLightCondensed
-    </namerecord>
-    <namerecord nameID="259" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans Bold Condensed
-    </namerecord>
-    <namerecord nameID="260" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansBoldCondensed
-    </namerecord>
-    <namerecord nameID="261" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansLightCondensed
-    </namerecord>
-    <namerecord nameID="262" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Regular
-    </namerecord>
-    <namerecord nameID="263" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      serif
-    </namerecord>
-    <namerecord nameID="264" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans
-    </namerecord>
-    <namerecord nameID="265" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      weight
-    </namerecord>
-    <namerecord nameID="266" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Light
-    </namerecord>
-    <namerecord nameID="267" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Medium
-    </namerecord>
-    <namerecord nameID="268" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Bold
-    </namerecord>
-    <namerecord nameID="269" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      width
-    </namerecord>
-    <namerecord nameID="270" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Condensed
-    </namerecord>
     <namerecord nameID="1" platformID="3" platEncID="1" langID="0x409">
       New Font
     </namerecord>
@@ -210,30 +165,27 @@
       MutatorMathTest-SansLightCondensed
     </namerecord>
     <namerecord nameID="262" platformID="3" platEncID="1" langID="0x409">
-      Regular
-    </namerecord>
-    <namerecord nameID="263" platformID="3" platEncID="1" langID="0x409">
       serif
     </namerecord>
-    <namerecord nameID="264" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="263" platformID="3" platEncID="1" langID="0x409">
       Sans
     </namerecord>
-    <namerecord nameID="265" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="264" platformID="3" platEncID="1" langID="0x409">
       weight
     </namerecord>
-    <namerecord nameID="266" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="265" platformID="3" platEncID="1" langID="0x409">
       Light
     </namerecord>
-    <namerecord nameID="267" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="266" platformID="3" platEncID="1" langID="0x409">
       Medium
     </namerecord>
-    <namerecord nameID="268" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="267" platformID="3" platEncID="1" langID="0x409">
       Bold
     </namerecord>
-    <namerecord nameID="269" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="268" platformID="3" platEncID="1" langID="0x409">
       width
     </namerecord>
-    <namerecord nameID="270" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="269" platformID="3" platEncID="1" langID="0x409">
       Condensed
     </namerecord>
   </name>
@@ -484,17 +436,17 @@
     <DesignAxisRecord>
       <Axis index="0">
         <AxisTag value="SRIF"/>
-        <AxisNameID value="263"/>  <!-- serif -->
+        <AxisNameID value="262"/>  <!-- serif -->
         <AxisOrdering value="0"/>
       </Axis>
       <Axis index="1">
         <AxisTag value="wght"/>
-        <AxisNameID value="265"/>  <!-- weight -->
+        <AxisNameID value="264"/>  <!-- weight -->
         <AxisOrdering value="1"/>
       </Axis>
       <Axis index="2">
         <AxisTag value="wdth"/>
-        <AxisNameID value="269"/>  <!-- width -->
+        <AxisNameID value="268"/>  <!-- width -->
         <AxisOrdering value="2"/>
       </Axis>
     </DesignAxisRecord>
@@ -503,13 +455,13 @@
       <AxisValue index="0" Format="1">
         <AxisIndex value="0"/>
         <Flags value="0"/>
-        <ValueNameID value="264"/>  <!-- Sans -->
+        <ValueNameID value="263"/>  <!-- Sans -->
         <Value value="0.0"/>
       </AxisValue>
       <AxisValue index="1" Format="2">
         <AxisIndex value="1"/>
         <Flags value="0"/>
-        <ValueNameID value="266"/>  <!-- Light -->
+        <ValueNameID value="265"/>  <!-- Light -->
         <NominalValue value="300.0"/>
         <RangeMinValue value="300.0"/>
         <RangeMaxValue value="400.0"/>
@@ -517,7 +469,7 @@
       <AxisValue index="2" Format="2">
         <AxisIndex value="1"/>
         <Flags value="0"/>
-        <ValueNameID value="267"/>  <!-- Medium -->
+        <ValueNameID value="266"/>  <!-- Medium -->
         <NominalValue value="500.0"/>
         <RangeMinValue value="400.0"/>
         <RangeMaxValue value="600.0"/>
@@ -525,7 +477,7 @@
       <AxisValue index="3" Format="2">
         <AxisIndex value="1"/>
         <Flags value="0"/>
-        <ValueNameID value="268"/>  <!-- Bold -->
+        <ValueNameID value="267"/>  <!-- Bold -->
         <NominalValue value="700.0"/>
         <RangeMinValue value="600.0"/>
         <RangeMaxValue value="700.0"/>
@@ -533,13 +485,13 @@
       <AxisValue index="4" Format="2">
         <AxisIndex value="2"/>
         <Flags value="0"/>
-        <ValueNameID value="270"/>  <!-- Condensed -->
+        <ValueNameID value="269"/>  <!-- Condensed -->
         <NominalValue value="50.0"/>
         <RangeMinValue value="50.0"/>
         <RangeMaxValue value="75.0"/>
       </AxisValue>
     </AxisValueArray>
-    <ElidedFallbackNameID value="262"/>  <!-- Regular -->
+    <ElidedFallbackNameID value="2"/>  <!-- Regular -->
   </STAT>
 
   <fvar>
diff --git a/tests/data/DSv5/MutatorSansVariable_Weight-TTF.ttx b/tests/data/DSv5/MutatorSansVariable_Weight-TTF.ttx
index e87c86f..f9ced7d 100644
--- a/tests/data/DSv5/MutatorSansVariable_Weight-TTF.ttx
+++ b/tests/data/DSv5/MutatorSansVariable_Weight-TTF.ttx
@@ -251,51 +251,6 @@
   </glyf>
 
   <name>
-    <namerecord nameID="256" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Weight
-    </namerecord>
-    <namerecord nameID="257" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans Light Condensed
-    </namerecord>
-    <namerecord nameID="258" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansLightCondensed
-    </namerecord>
-    <namerecord nameID="259" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans Bold Condensed
-    </namerecord>
-    <namerecord nameID="260" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansBoldCondensed
-    </namerecord>
-    <namerecord nameID="261" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansLightCondensed
-    </namerecord>
-    <namerecord nameID="262" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Regular
-    </namerecord>
-    <namerecord nameID="263" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      serif
-    </namerecord>
-    <namerecord nameID="264" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans
-    </namerecord>
-    <namerecord nameID="265" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      weight
-    </namerecord>
-    <namerecord nameID="266" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Light
-    </namerecord>
-    <namerecord nameID="267" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Medium
-    </namerecord>
-    <namerecord nameID="268" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Bold
-    </namerecord>
-    <namerecord nameID="269" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      width
-    </namerecord>
-    <namerecord nameID="270" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Condensed
-    </namerecord>
     <namerecord nameID="1" platformID="3" platEncID="1" langID="0x409">
       New Font
     </namerecord>
@@ -333,30 +288,27 @@
       MutatorMathTest-SansLightCondensed
     </namerecord>
     <namerecord nameID="262" platformID="3" platEncID="1" langID="0x409">
-      Regular
-    </namerecord>
-    <namerecord nameID="263" platformID="3" platEncID="1" langID="0x409">
       serif
     </namerecord>
-    <namerecord nameID="264" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="263" platformID="3" platEncID="1" langID="0x409">
       Sans
     </namerecord>
-    <namerecord nameID="265" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="264" platformID="3" platEncID="1" langID="0x409">
       weight
     </namerecord>
-    <namerecord nameID="266" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="265" platformID="3" platEncID="1" langID="0x409">
       Light
     </namerecord>
-    <namerecord nameID="267" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="266" platformID="3" platEncID="1" langID="0x409">
       Medium
     </namerecord>
-    <namerecord nameID="268" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="267" platformID="3" platEncID="1" langID="0x409">
       Bold
     </namerecord>
-    <namerecord nameID="269" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="268" platformID="3" platEncID="1" langID="0x409">
       width
     </namerecord>
-    <namerecord nameID="270" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="269" platformID="3" platEncID="1" langID="0x409">
       Condensed
     </namerecord>
   </name>
@@ -508,17 +460,17 @@
     <DesignAxisRecord>
       <Axis index="0">
         <AxisTag value="SRIF"/>
-        <AxisNameID value="263"/>  <!-- serif -->
+        <AxisNameID value="262"/>  <!-- serif -->
         <AxisOrdering value="0"/>
       </Axis>
       <Axis index="1">
         <AxisTag value="wght"/>
-        <AxisNameID value="265"/>  <!-- weight -->
+        <AxisNameID value="264"/>  <!-- weight -->
         <AxisOrdering value="1"/>
       </Axis>
       <Axis index="2">
         <AxisTag value="wdth"/>
-        <AxisNameID value="269"/>  <!-- width -->
+        <AxisNameID value="268"/>  <!-- width -->
         <AxisOrdering value="2"/>
       </Axis>
     </DesignAxisRecord>
@@ -527,13 +479,13 @@
       <AxisValue index="0" Format="1">
         <AxisIndex value="0"/>
         <Flags value="0"/>
-        <ValueNameID value="264"/>  <!-- Sans -->
+        <ValueNameID value="263"/>  <!-- Sans -->
         <Value value="0.0"/>
       </AxisValue>
       <AxisValue index="1" Format="2">
         <AxisIndex value="1"/>
         <Flags value="0"/>
-        <ValueNameID value="266"/>  <!-- Light -->
+        <ValueNameID value="265"/>  <!-- Light -->
         <NominalValue value="300.0"/>
         <RangeMinValue value="300.0"/>
         <RangeMaxValue value="400.0"/>
@@ -541,7 +493,7 @@
       <AxisValue index="2" Format="2">
         <AxisIndex value="1"/>
         <Flags value="0"/>
-        <ValueNameID value="267"/>  <!-- Medium -->
+        <ValueNameID value="266"/>  <!-- Medium -->
         <NominalValue value="500.0"/>
         <RangeMinValue value="400.0"/>
         <RangeMaxValue value="600.0"/>
@@ -549,7 +501,7 @@
       <AxisValue index="3" Format="2">
         <AxisIndex value="1"/>
         <Flags value="0"/>
-        <ValueNameID value="268"/>  <!-- Bold -->
+        <ValueNameID value="267"/>  <!-- Bold -->
         <NominalValue value="700.0"/>
         <RangeMinValue value="600.0"/>
         <RangeMaxValue value="700.0"/>
@@ -557,13 +509,13 @@
       <AxisValue index="4" Format="2">
         <AxisIndex value="2"/>
         <Flags value="0"/>
-        <ValueNameID value="270"/>  <!-- Condensed -->
+        <ValueNameID value="269"/>  <!-- Condensed -->
         <NominalValue value="50.0"/>
         <RangeMinValue value="50.0"/>
         <RangeMaxValue value="75.0"/>
       </AxisValue>
     </AxisValueArray>
-    <ElidedFallbackNameID value="262"/>  <!-- Regular -->
+    <ElidedFallbackNameID value="2"/>  <!-- Regular -->
   </STAT>
 
   <cvar>
diff --git a/tests/data/DSv5/MutatorSansVariable_Weight_Width-CFF2.ttx b/tests/data/DSv5/MutatorSansVariable_Weight_Width-CFF2.ttx
index bcde82d..4d4c119 100644
--- a/tests/data/DSv5/MutatorSansVariable_Weight_Width-CFF2.ttx
+++ b/tests/data/DSv5/MutatorSansVariable_Weight_Width-CFF2.ttx
@@ -128,99 +128,6 @@
   </cmap>
 
   <name>
-    <namerecord nameID="256" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Weight
-    </namerecord>
-    <namerecord nameID="257" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Width
-    </namerecord>
-    <namerecord nameID="258" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans Light Condensed
-    </namerecord>
-    <namerecord nameID="259" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansLightCondensed
-    </namerecord>
-    <namerecord nameID="260" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans Bold Condensed
-    </namerecord>
-    <namerecord nameID="261" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansBoldCondensed
-    </namerecord>
-    <namerecord nameID="262" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans Light Extended
-    </namerecord>
-    <namerecord nameID="263" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansLightExtended
-    </namerecord>
-    <namerecord nameID="264" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans Bold Extended
-    </namerecord>
-    <namerecord nameID="265" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansBoldExtended
-    </namerecord>
-    <namerecord nameID="266" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans Medium
-    </namerecord>
-    <namerecord nameID="267" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansMedium
-    </namerecord>
-    <namerecord nameID="268" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansMedium
-    </namerecord>
-    <namerecord nameID="269" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans Bold
-    </namerecord>
-    <namerecord nameID="270" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansBold
-    </namerecord>
-    <namerecord nameID="271" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans Medium Extended
-    </namerecord>
-    <namerecord nameID="272" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansMediumExtended
-    </namerecord>
-    <namerecord nameID="273" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansLightCondensed
-    </namerecord>
-    <namerecord nameID="274" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Regular
-    </namerecord>
-    <namerecord nameID="275" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      serif
-    </namerecord>
-    <namerecord nameID="276" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans
-    </namerecord>
-    <namerecord nameID="277" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      weight
-    </namerecord>
-    <namerecord nameID="278" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Light
-    </namerecord>
-    <namerecord nameID="279" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Medium
-    </namerecord>
-    <namerecord nameID="280" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Bold
-    </namerecord>
-    <namerecord nameID="281" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      width
-    </namerecord>
-    <namerecord nameID="282" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Condensed
-    </namerecord>
-    <namerecord nameID="283" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Normal
-    </namerecord>
-    <namerecord nameID="284" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Extended
-    </namerecord>
-    <namerecord nameID="285" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      S1
-    </namerecord>
-    <namerecord nameID="286" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      S2
-    </namerecord>
     <namerecord nameID="1" platformID="3" platEncID="1" langID="0x409">
       New Font
     </namerecord>
@@ -294,42 +201,39 @@
       MutatorMathTest-SansLightCondensed
     </namerecord>
     <namerecord nameID="274" platformID="3" platEncID="1" langID="0x409">
-      Regular
-    </namerecord>
-    <namerecord nameID="275" platformID="3" platEncID="1" langID="0x409">
       serif
     </namerecord>
-    <namerecord nameID="276" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="275" platformID="3" platEncID="1" langID="0x409">
       Sans
     </namerecord>
-    <namerecord nameID="277" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="276" platformID="3" platEncID="1" langID="0x409">
       weight
     </namerecord>
-    <namerecord nameID="278" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="277" platformID="3" platEncID="1" langID="0x409">
       Light
     </namerecord>
-    <namerecord nameID="279" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="278" platformID="3" platEncID="1" langID="0x409">
       Medium
     </namerecord>
-    <namerecord nameID="280" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="279" platformID="3" platEncID="1" langID="0x409">
       Bold
     </namerecord>
-    <namerecord nameID="281" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="280" platformID="3" platEncID="1" langID="0x409">
       width
     </namerecord>
-    <namerecord nameID="282" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="281" platformID="3" platEncID="1" langID="0x409">
       Condensed
     </namerecord>
-    <namerecord nameID="283" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="282" platformID="3" platEncID="1" langID="0x409">
       Normal
     </namerecord>
-    <namerecord nameID="284" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="283" platformID="3" platEncID="1" langID="0x409">
       Extended
     </namerecord>
-    <namerecord nameID="285" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="284" platformID="3" platEncID="1" langID="0x409">
       S1
     </namerecord>
-    <namerecord nameID="286" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="285" platformID="3" platEncID="1" langID="0x409">
       S2
     </namerecord>
   </name>
@@ -692,17 +596,17 @@
     <DesignAxisRecord>
       <Axis index="0">
         <AxisTag value="SRIF"/>
-        <AxisNameID value="275"/>  <!-- serif -->
+        <AxisNameID value="274"/>  <!-- serif -->
         <AxisOrdering value="0"/>
       </Axis>
       <Axis index="1">
         <AxisTag value="wght"/>
-        <AxisNameID value="277"/>  <!-- weight -->
+        <AxisNameID value="276"/>  <!-- weight -->
         <AxisOrdering value="1"/>
       </Axis>
       <Axis index="2">
         <AxisTag value="wdth"/>
-        <AxisNameID value="281"/>  <!-- width -->
+        <AxisNameID value="280"/>  <!-- width -->
         <AxisOrdering value="2"/>
       </Axis>
     </DesignAxisRecord>
@@ -711,7 +615,7 @@
       <AxisValue index="0" Format="4">
         <!-- AxisCount=3 -->
         <Flags value="0"/>
-        <ValueNameID value="285"/>  <!-- S1 -->
+        <ValueNameID value="284"/>  <!-- S1 -->
         <AxisValueRecord index="0">
           <AxisIndex value="0"/>
           <Value value="0.0"/>
@@ -728,7 +632,7 @@
       <AxisValue index="1" Format="4">
         <!-- AxisCount=3 -->
         <Flags value="0"/>
-        <ValueNameID value="286"/>  <!-- S2 -->
+        <ValueNameID value="285"/>  <!-- S2 -->
         <AxisValueRecord index="0">
           <AxisIndex value="0"/>
           <Value value="0.0"/>
@@ -745,13 +649,13 @@
       <AxisValue index="2" Format="1">
         <AxisIndex value="0"/>
         <Flags value="0"/>
-        <ValueNameID value="276"/>  <!-- Sans -->
+        <ValueNameID value="275"/>  <!-- Sans -->
         <Value value="0.0"/>
       </AxisValue>
       <AxisValue index="3" Format="2">
         <AxisIndex value="1"/>
         <Flags value="0"/>
-        <ValueNameID value="278"/>  <!-- Light -->
+        <ValueNameID value="277"/>  <!-- Light -->
         <NominalValue value="300.0"/>
         <RangeMinValue value="300.0"/>
         <RangeMaxValue value="400.0"/>
@@ -759,7 +663,7 @@
       <AxisValue index="4" Format="2">
         <AxisIndex value="1"/>
         <Flags value="0"/>
-        <ValueNameID value="279"/>  <!-- Medium -->
+        <ValueNameID value="278"/>  <!-- Medium -->
         <NominalValue value="500.0"/>
         <RangeMinValue value="400.0"/>
         <RangeMaxValue value="600.0"/>
@@ -767,7 +671,7 @@
       <AxisValue index="5" Format="2">
         <AxisIndex value="1"/>
         <Flags value="0"/>
-        <ValueNameID value="280"/>  <!-- Bold -->
+        <ValueNameID value="279"/>  <!-- Bold -->
         <NominalValue value="700.0"/>
         <RangeMinValue value="600.0"/>
         <RangeMaxValue value="700.0"/>
@@ -775,7 +679,7 @@
       <AxisValue index="6" Format="2">
         <AxisIndex value="2"/>
         <Flags value="0"/>
-        <ValueNameID value="282"/>  <!-- Condensed -->
+        <ValueNameID value="281"/>  <!-- Condensed -->
         <NominalValue value="50.0"/>
         <RangeMinValue value="50.0"/>
         <RangeMaxValue value="75.0"/>
@@ -783,7 +687,7 @@
       <AxisValue index="7" Format="2">
         <AxisIndex value="2"/>
         <Flags value="2"/>  <!-- ElidableAxisValueName -->
-        <ValueNameID value="283"/>  <!-- Normal -->
+        <ValueNameID value="282"/>  <!-- Normal -->
         <NominalValue value="100.0"/>
         <RangeMinValue value="75.0"/>
         <RangeMaxValue value="125.0"/>
@@ -791,13 +695,13 @@
       <AxisValue index="8" Format="2">
         <AxisIndex value="2"/>
         <Flags value="0"/>
-        <ValueNameID value="284"/>  <!-- Extended -->
+        <ValueNameID value="283"/>  <!-- Extended -->
         <NominalValue value="200.0"/>
         <RangeMinValue value="125.0"/>
         <RangeMaxValue value="200.0"/>
       </AxisValue>
     </AxisValueArray>
-    <ElidedFallbackNameID value="274"/>  <!-- Regular -->
+    <ElidedFallbackNameID value="2"/>  <!-- Regular -->
   </STAT>
 
   <avar>
diff --git a/tests/data/DSv5/MutatorSansVariable_Weight_Width-TTF.ttx b/tests/data/DSv5/MutatorSansVariable_Weight_Width-TTF.ttx
index 1e36971..75ca769 100644
--- a/tests/data/DSv5/MutatorSansVariable_Weight_Width-TTF.ttx
+++ b/tests/data/DSv5/MutatorSansVariable_Weight_Width-TTF.ttx
@@ -251,99 +251,6 @@
   </glyf>
 
   <name>
-    <namerecord nameID="256" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Weight
-    </namerecord>
-    <namerecord nameID="257" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Width
-    </namerecord>
-    <namerecord nameID="258" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans Light Condensed
-    </namerecord>
-    <namerecord nameID="259" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansLightCondensed
-    </namerecord>
-    <namerecord nameID="260" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans Bold Condensed
-    </namerecord>
-    <namerecord nameID="261" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansBoldCondensed
-    </namerecord>
-    <namerecord nameID="262" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans Light Extended
-    </namerecord>
-    <namerecord nameID="263" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansLightExtended
-    </namerecord>
-    <namerecord nameID="264" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans Bold Extended
-    </namerecord>
-    <namerecord nameID="265" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansBoldExtended
-    </namerecord>
-    <namerecord nameID="266" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans Medium
-    </namerecord>
-    <namerecord nameID="267" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansMedium
-    </namerecord>
-    <namerecord nameID="268" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansMedium
-    </namerecord>
-    <namerecord nameID="269" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans Bold
-    </namerecord>
-    <namerecord nameID="270" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansBold
-    </namerecord>
-    <namerecord nameID="271" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans Medium Extended
-    </namerecord>
-    <namerecord nameID="272" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansMediumExtended
-    </namerecord>
-    <namerecord nameID="273" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansLightCondensed
-    </namerecord>
-    <namerecord nameID="274" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Regular
-    </namerecord>
-    <namerecord nameID="275" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      serif
-    </namerecord>
-    <namerecord nameID="276" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans
-    </namerecord>
-    <namerecord nameID="277" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      weight
-    </namerecord>
-    <namerecord nameID="278" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Light
-    </namerecord>
-    <namerecord nameID="279" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Medium
-    </namerecord>
-    <namerecord nameID="280" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Bold
-    </namerecord>
-    <namerecord nameID="281" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      width
-    </namerecord>
-    <namerecord nameID="282" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Condensed
-    </namerecord>
-    <namerecord nameID="283" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Normal
-    </namerecord>
-    <namerecord nameID="284" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Extended
-    </namerecord>
-    <namerecord nameID="285" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      S1
-    </namerecord>
-    <namerecord nameID="286" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      S2
-    </namerecord>
     <namerecord nameID="1" platformID="3" platEncID="1" langID="0x409">
       New Font
     </namerecord>
@@ -417,42 +324,39 @@
       MutatorMathTest-SansLightCondensed
     </namerecord>
     <namerecord nameID="274" platformID="3" platEncID="1" langID="0x409">
-      Regular
-    </namerecord>
-    <namerecord nameID="275" platformID="3" platEncID="1" langID="0x409">
       serif
     </namerecord>
-    <namerecord nameID="276" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="275" platformID="3" platEncID="1" langID="0x409">
       Sans
     </namerecord>
-    <namerecord nameID="277" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="276" platformID="3" platEncID="1" langID="0x409">
       weight
     </namerecord>
-    <namerecord nameID="278" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="277" platformID="3" platEncID="1" langID="0x409">
       Light
     </namerecord>
-    <namerecord nameID="279" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="278" platformID="3" platEncID="1" langID="0x409">
       Medium
     </namerecord>
-    <namerecord nameID="280" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="279" platformID="3" platEncID="1" langID="0x409">
       Bold
     </namerecord>
-    <namerecord nameID="281" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="280" platformID="3" platEncID="1" langID="0x409">
       width
     </namerecord>
-    <namerecord nameID="282" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="281" platformID="3" platEncID="1" langID="0x409">
       Condensed
     </namerecord>
-    <namerecord nameID="283" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="282" platformID="3" platEncID="1" langID="0x409">
       Normal
     </namerecord>
-    <namerecord nameID="284" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="283" platformID="3" platEncID="1" langID="0x409">
       Extended
     </namerecord>
-    <namerecord nameID="285" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="284" platformID="3" platEncID="1" langID="0x409">
       S1
     </namerecord>
-    <namerecord nameID="286" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="285" platformID="3" platEncID="1" langID="0x409">
       S2
     </namerecord>
   </name>
@@ -635,17 +539,17 @@
     <DesignAxisRecord>
       <Axis index="0">
         <AxisTag value="SRIF"/>
-        <AxisNameID value="275"/>  <!-- serif -->
+        <AxisNameID value="274"/>  <!-- serif -->
         <AxisOrdering value="0"/>
       </Axis>
       <Axis index="1">
         <AxisTag value="wght"/>
-        <AxisNameID value="277"/>  <!-- weight -->
+        <AxisNameID value="276"/>  <!-- weight -->
         <AxisOrdering value="1"/>
       </Axis>
       <Axis index="2">
         <AxisTag value="wdth"/>
-        <AxisNameID value="281"/>  <!-- width -->
+        <AxisNameID value="280"/>  <!-- width -->
         <AxisOrdering value="2"/>
       </Axis>
     </DesignAxisRecord>
@@ -654,7 +558,7 @@
       <AxisValue index="0" Format="4">
         <!-- AxisCount=3 -->
         <Flags value="0"/>
-        <ValueNameID value="285"/>  <!-- S1 -->
+        <ValueNameID value="284"/>  <!-- S1 -->
         <AxisValueRecord index="0">
           <AxisIndex value="0"/>
           <Value value="0.0"/>
@@ -671,7 +575,7 @@
       <AxisValue index="1" Format="4">
         <!-- AxisCount=3 -->
         <Flags value="0"/>
-        <ValueNameID value="286"/>  <!-- S2 -->
+        <ValueNameID value="285"/>  <!-- S2 -->
         <AxisValueRecord index="0">
           <AxisIndex value="0"/>
           <Value value="0.0"/>
@@ -688,13 +592,13 @@
       <AxisValue index="2" Format="1">
         <AxisIndex value="0"/>
         <Flags value="0"/>
-        <ValueNameID value="276"/>  <!-- Sans -->
+        <ValueNameID value="275"/>  <!-- Sans -->
         <Value value="0.0"/>
       </AxisValue>
       <AxisValue index="3" Format="2">
         <AxisIndex value="1"/>
         <Flags value="0"/>
-        <ValueNameID value="278"/>  <!-- Light -->
+        <ValueNameID value="277"/>  <!-- Light -->
         <NominalValue value="300.0"/>
         <RangeMinValue value="300.0"/>
         <RangeMaxValue value="400.0"/>
@@ -702,7 +606,7 @@
       <AxisValue index="4" Format="2">
         <AxisIndex value="1"/>
         <Flags value="0"/>
-        <ValueNameID value="279"/>  <!-- Medium -->
+        <ValueNameID value="278"/>  <!-- Medium -->
         <NominalValue value="500.0"/>
         <RangeMinValue value="400.0"/>
         <RangeMaxValue value="600.0"/>
@@ -710,7 +614,7 @@
       <AxisValue index="5" Format="2">
         <AxisIndex value="1"/>
         <Flags value="0"/>
-        <ValueNameID value="280"/>  <!-- Bold -->
+        <ValueNameID value="279"/>  <!-- Bold -->
         <NominalValue value="700.0"/>
         <RangeMinValue value="600.0"/>
         <RangeMaxValue value="700.0"/>
@@ -718,7 +622,7 @@
       <AxisValue index="6" Format="2">
         <AxisIndex value="2"/>
         <Flags value="0"/>
-        <ValueNameID value="282"/>  <!-- Condensed -->
+        <ValueNameID value="281"/>  <!-- Condensed -->
         <NominalValue value="50.0"/>
         <RangeMinValue value="50.0"/>
         <RangeMaxValue value="75.0"/>
@@ -726,7 +630,7 @@
       <AxisValue index="7" Format="2">
         <AxisIndex value="2"/>
         <Flags value="2"/>  <!-- ElidableAxisValueName -->
-        <ValueNameID value="283"/>  <!-- Normal -->
+        <ValueNameID value="282"/>  <!-- Normal -->
         <NominalValue value="100.0"/>
         <RangeMinValue value="75.0"/>
         <RangeMaxValue value="125.0"/>
@@ -734,13 +638,13 @@
       <AxisValue index="8" Format="2">
         <AxisIndex value="2"/>
         <Flags value="0"/>
-        <ValueNameID value="284"/>  <!-- Extended -->
+        <ValueNameID value="283"/>  <!-- Extended -->
         <NominalValue value="200.0"/>
         <RangeMinValue value="125.0"/>
         <RangeMaxValue value="200.0"/>
       </AxisValue>
     </AxisValueArray>
-    <ElidedFallbackNameID value="274"/>  <!-- Regular -->
+    <ElidedFallbackNameID value="2"/>  <!-- Regular -->
   </STAT>
 
   <avar>
diff --git a/tests/data/DSv5/MutatorSansVariable_Width-CFF2.ttx b/tests/data/DSv5/MutatorSansVariable_Width-CFF2.ttx
index 201eccb..3bbab81 100644
--- a/tests/data/DSv5/MutatorSansVariable_Width-CFF2.ttx
+++ b/tests/data/DSv5/MutatorSansVariable_Width-CFF2.ttx
@@ -128,51 +128,6 @@
   </cmap>
 
   <name>
-    <namerecord nameID="256" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Width
-    </namerecord>
-    <namerecord nameID="257" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans Light Condensed
-    </namerecord>
-    <namerecord nameID="258" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansLightCondensed
-    </namerecord>
-    <namerecord nameID="259" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans Light Extended
-    </namerecord>
-    <namerecord nameID="260" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansLightExtended
-    </namerecord>
-    <namerecord nameID="261" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansLightCondensed
-    </namerecord>
-    <namerecord nameID="262" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Regular
-    </namerecord>
-    <namerecord nameID="263" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      serif
-    </namerecord>
-    <namerecord nameID="264" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans
-    </namerecord>
-    <namerecord nameID="265" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      weight
-    </namerecord>
-    <namerecord nameID="266" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Light
-    </namerecord>
-    <namerecord nameID="267" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      width
-    </namerecord>
-    <namerecord nameID="268" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Condensed
-    </namerecord>
-    <namerecord nameID="269" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Normal
-    </namerecord>
-    <namerecord nameID="270" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Extended
-    </namerecord>
     <namerecord nameID="1" platformID="3" platEncID="1" langID="0x409">
       New Font
     </namerecord>
@@ -210,30 +165,27 @@
       MutatorMathTest-SansLightCondensed
     </namerecord>
     <namerecord nameID="262" platformID="3" platEncID="1" langID="0x409">
-      Regular
-    </namerecord>
-    <namerecord nameID="263" platformID="3" platEncID="1" langID="0x409">
       serif
     </namerecord>
-    <namerecord nameID="264" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="263" platformID="3" platEncID="1" langID="0x409">
       Sans
     </namerecord>
-    <namerecord nameID="265" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="264" platformID="3" platEncID="1" langID="0x409">
       weight
     </namerecord>
-    <namerecord nameID="266" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="265" platformID="3" platEncID="1" langID="0x409">
       Light
     </namerecord>
-    <namerecord nameID="267" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="266" platformID="3" platEncID="1" langID="0x409">
       width
     </namerecord>
-    <namerecord nameID="268" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="267" platformID="3" platEncID="1" langID="0x409">
       Condensed
     </namerecord>
-    <namerecord nameID="269" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="268" platformID="3" platEncID="1" langID="0x409">
       Normal
     </namerecord>
-    <namerecord nameID="270" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="269" platformID="3" platEncID="1" langID="0x409">
       Extended
     </namerecord>
   </name>
@@ -462,17 +414,17 @@
     <DesignAxisRecord>
       <Axis index="0">
         <AxisTag value="SRIF"/>
-        <AxisNameID value="263"/>  <!-- serif -->
+        <AxisNameID value="262"/>  <!-- serif -->
         <AxisOrdering value="0"/>
       </Axis>
       <Axis index="1">
         <AxisTag value="wght"/>
-        <AxisNameID value="265"/>  <!-- weight -->
+        <AxisNameID value="264"/>  <!-- weight -->
         <AxisOrdering value="1"/>
       </Axis>
       <Axis index="2">
         <AxisTag value="wdth"/>
-        <AxisNameID value="267"/>  <!-- width -->
+        <AxisNameID value="266"/>  <!-- width -->
         <AxisOrdering value="2"/>
       </Axis>
     </DesignAxisRecord>
@@ -481,13 +433,13 @@
       <AxisValue index="0" Format="1">
         <AxisIndex value="0"/>
         <Flags value="0"/>
-        <ValueNameID value="264"/>  <!-- Sans -->
+        <ValueNameID value="263"/>  <!-- Sans -->
         <Value value="0.0"/>
       </AxisValue>
       <AxisValue index="1" Format="2">
         <AxisIndex value="1"/>
         <Flags value="0"/>
-        <ValueNameID value="266"/>  <!-- Light -->
+        <ValueNameID value="265"/>  <!-- Light -->
         <NominalValue value="300.0"/>
         <RangeMinValue value="300.0"/>
         <RangeMaxValue value="400.0"/>
@@ -495,7 +447,7 @@
       <AxisValue index="2" Format="2">
         <AxisIndex value="2"/>
         <Flags value="0"/>
-        <ValueNameID value="268"/>  <!-- Condensed -->
+        <ValueNameID value="267"/>  <!-- Condensed -->
         <NominalValue value="50.0"/>
         <RangeMinValue value="50.0"/>
         <RangeMaxValue value="75.0"/>
@@ -503,7 +455,7 @@
       <AxisValue index="3" Format="2">
         <AxisIndex value="2"/>
         <Flags value="2"/>  <!-- ElidableAxisValueName -->
-        <ValueNameID value="269"/>  <!-- Normal -->
+        <ValueNameID value="268"/>  <!-- Normal -->
         <NominalValue value="100.0"/>
         <RangeMinValue value="75.0"/>
         <RangeMaxValue value="125.0"/>
@@ -511,13 +463,13 @@
       <AxisValue index="4" Format="2">
         <AxisIndex value="2"/>
         <Flags value="0"/>
-        <ValueNameID value="270"/>  <!-- Extended -->
+        <ValueNameID value="269"/>  <!-- Extended -->
         <NominalValue value="200.0"/>
         <RangeMinValue value="125.0"/>
         <RangeMaxValue value="200.0"/>
       </AxisValue>
     </AxisValueArray>
-    <ElidedFallbackNameID value="262"/>  <!-- Regular -->
+    <ElidedFallbackNameID value="2"/>  <!-- Regular -->
   </STAT>
 
   <avar>
diff --git a/tests/data/DSv5/MutatorSansVariable_Width-TTF.ttx b/tests/data/DSv5/MutatorSansVariable_Width-TTF.ttx
index 4b2eb6e..73d7d90 100644
--- a/tests/data/DSv5/MutatorSansVariable_Width-TTF.ttx
+++ b/tests/data/DSv5/MutatorSansVariable_Width-TTF.ttx
@@ -251,51 +251,6 @@
   </glyf>
 
   <name>
-    <namerecord nameID="256" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Width
-    </namerecord>
-    <namerecord nameID="257" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans Light Condensed
-    </namerecord>
-    <namerecord nameID="258" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansLightCondensed
-    </namerecord>
-    <namerecord nameID="259" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans Light Extended
-    </namerecord>
-    <namerecord nameID="260" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansLightExtended
-    </namerecord>
-    <namerecord nameID="261" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SansLightCondensed
-    </namerecord>
-    <namerecord nameID="262" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Regular
-    </namerecord>
-    <namerecord nameID="263" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      serif
-    </namerecord>
-    <namerecord nameID="264" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Sans
-    </namerecord>
-    <namerecord nameID="265" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      weight
-    </namerecord>
-    <namerecord nameID="266" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Light
-    </namerecord>
-    <namerecord nameID="267" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      width
-    </namerecord>
-    <namerecord nameID="268" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Condensed
-    </namerecord>
-    <namerecord nameID="269" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Normal
-    </namerecord>
-    <namerecord nameID="270" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Extended
-    </namerecord>
     <namerecord nameID="1" platformID="3" platEncID="1" langID="0x409">
       New Font
     </namerecord>
@@ -333,30 +288,27 @@
       MutatorMathTest-SansLightCondensed
     </namerecord>
     <namerecord nameID="262" platformID="3" platEncID="1" langID="0x409">
-      Regular
-    </namerecord>
-    <namerecord nameID="263" platformID="3" platEncID="1" langID="0x409">
       serif
     </namerecord>
-    <namerecord nameID="264" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="263" platformID="3" platEncID="1" langID="0x409">
       Sans
     </namerecord>
-    <namerecord nameID="265" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="264" platformID="3" platEncID="1" langID="0x409">
       weight
     </namerecord>
-    <namerecord nameID="266" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="265" platformID="3" platEncID="1" langID="0x409">
       Light
     </namerecord>
-    <namerecord nameID="267" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="266" platformID="3" platEncID="1" langID="0x409">
       width
     </namerecord>
-    <namerecord nameID="268" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="267" platformID="3" platEncID="1" langID="0x409">
       Condensed
     </namerecord>
-    <namerecord nameID="269" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="268" platformID="3" platEncID="1" langID="0x409">
       Normal
     </namerecord>
-    <namerecord nameID="270" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="269" platformID="3" platEncID="1" langID="0x409">
       Extended
     </namerecord>
   </name>
@@ -508,17 +460,17 @@
     <DesignAxisRecord>
       <Axis index="0">
         <AxisTag value="SRIF"/>
-        <AxisNameID value="263"/>  <!-- serif -->
+        <AxisNameID value="262"/>  <!-- serif -->
         <AxisOrdering value="0"/>
       </Axis>
       <Axis index="1">
         <AxisTag value="wght"/>
-        <AxisNameID value="265"/>  <!-- weight -->
+        <AxisNameID value="264"/>  <!-- weight -->
         <AxisOrdering value="1"/>
       </Axis>
       <Axis index="2">
         <AxisTag value="wdth"/>
-        <AxisNameID value="267"/>  <!-- width -->
+        <AxisNameID value="266"/>  <!-- width -->
         <AxisOrdering value="2"/>
       </Axis>
     </DesignAxisRecord>
@@ -527,13 +479,13 @@
       <AxisValue index="0" Format="1">
         <AxisIndex value="0"/>
         <Flags value="0"/>
-        <ValueNameID value="264"/>  <!-- Sans -->
+        <ValueNameID value="263"/>  <!-- Sans -->
         <Value value="0.0"/>
       </AxisValue>
       <AxisValue index="1" Format="2">
         <AxisIndex value="1"/>
         <Flags value="0"/>
-        <ValueNameID value="266"/>  <!-- Light -->
+        <ValueNameID value="265"/>  <!-- Light -->
         <NominalValue value="300.0"/>
         <RangeMinValue value="300.0"/>
         <RangeMaxValue value="400.0"/>
@@ -541,7 +493,7 @@
       <AxisValue index="2" Format="2">
         <AxisIndex value="2"/>
         <Flags value="0"/>
-        <ValueNameID value="268"/>  <!-- Condensed -->
+        <ValueNameID value="267"/>  <!-- Condensed -->
         <NominalValue value="50.0"/>
         <RangeMinValue value="50.0"/>
         <RangeMaxValue value="75.0"/>
@@ -549,7 +501,7 @@
       <AxisValue index="3" Format="2">
         <AxisIndex value="2"/>
         <Flags value="2"/>  <!-- ElidableAxisValueName -->
-        <ValueNameID value="269"/>  <!-- Normal -->
+        <ValueNameID value="268"/>  <!-- Normal -->
         <NominalValue value="100.0"/>
         <RangeMinValue value="75.0"/>
         <RangeMaxValue value="125.0"/>
@@ -557,13 +509,13 @@
       <AxisValue index="4" Format="2">
         <AxisIndex value="2"/>
         <Flags value="0"/>
-        <ValueNameID value="270"/>  <!-- Extended -->
+        <ValueNameID value="269"/>  <!-- Extended -->
         <NominalValue value="200.0"/>
         <RangeMinValue value="125.0"/>
         <RangeMaxValue value="200.0"/>
       </AxisValue>
     </AxisValueArray>
-    <ElidedFallbackNameID value="262"/>  <!-- Regular -->
+    <ElidedFallbackNameID value="2"/>  <!-- Regular -->
   </STAT>
 
   <avar>
diff --git a/tests/data/DSv5/MutatorSerifVariable_Width-CFF2.ttx b/tests/data/DSv5/MutatorSerifVariable_Width-CFF2.ttx
index c59cf86..f788eb4 100644
--- a/tests/data/DSv5/MutatorSerifVariable_Width-CFF2.ttx
+++ b/tests/data/DSv5/MutatorSerifVariable_Width-CFF2.ttx
@@ -128,42 +128,6 @@
   </cmap>
 
   <name>
-    <namerecord nameID="256" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Width
-    </namerecord>
-    <namerecord nameID="257" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Serif Light Condensed
-    </namerecord>
-    <namerecord nameID="258" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SerifLightCondensed
-    </namerecord>
-    <namerecord nameID="259" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Regular
-    </namerecord>
-    <namerecord nameID="260" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      serif
-    </namerecord>
-    <namerecord nameID="261" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Serif
-    </namerecord>
-    <namerecord nameID="262" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      weight
-    </namerecord>
-    <namerecord nameID="263" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Light
-    </namerecord>
-    <namerecord nameID="264" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      width
-    </namerecord>
-    <namerecord nameID="265" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Condensed
-    </namerecord>
-    <namerecord nameID="266" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Normal
-    </namerecord>
-    <namerecord nameID="267" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Extended
-    </namerecord>
     <namerecord nameID="1" platformID="3" platEncID="1" langID="0x409">
       New Font
     </namerecord>
@@ -192,30 +156,27 @@
       MutatorMathTest-SerifLightCondensed
     </namerecord>
     <namerecord nameID="259" platformID="3" platEncID="1" langID="0x409">
-      Regular
-    </namerecord>
-    <namerecord nameID="260" platformID="3" platEncID="1" langID="0x409">
       serif
     </namerecord>
-    <namerecord nameID="261" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="260" platformID="3" platEncID="1" langID="0x409">
       Serif
     </namerecord>
-    <namerecord nameID="262" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="261" platformID="3" platEncID="1" langID="0x409">
       weight
     </namerecord>
-    <namerecord nameID="263" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="262" platformID="3" platEncID="1" langID="0x409">
       Light
     </namerecord>
-    <namerecord nameID="264" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="263" platformID="3" platEncID="1" langID="0x409">
       width
     </namerecord>
-    <namerecord nameID="265" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="264" platformID="3" platEncID="1" langID="0x409">
       Condensed
     </namerecord>
-    <namerecord nameID="266" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="265" platformID="3" platEncID="1" langID="0x409">
       Normal
     </namerecord>
-    <namerecord nameID="267" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="266" platformID="3" platEncID="1" langID="0x409">
       Extended
     </namerecord>
   </name>
@@ -371,17 +332,17 @@
     <DesignAxisRecord>
       <Axis index="0">
         <AxisTag value="SRIF"/>
-        <AxisNameID value="260"/>  <!-- serif -->
+        <AxisNameID value="259"/>  <!-- serif -->
         <AxisOrdering value="0"/>
       </Axis>
       <Axis index="1">
         <AxisTag value="wght"/>
-        <AxisNameID value="262"/>  <!-- weight -->
+        <AxisNameID value="261"/>  <!-- weight -->
         <AxisOrdering value="1"/>
       </Axis>
       <Axis index="2">
         <AxisTag value="wdth"/>
-        <AxisNameID value="264"/>  <!-- width -->
+        <AxisNameID value="263"/>  <!-- width -->
         <AxisOrdering value="2"/>
       </Axis>
     </DesignAxisRecord>
@@ -390,13 +351,13 @@
       <AxisValue index="0" Format="1">
         <AxisIndex value="0"/>
         <Flags value="0"/>
-        <ValueNameID value="261"/>  <!-- Serif -->
+        <ValueNameID value="260"/>  <!-- Serif -->
         <Value value="1.0"/>
       </AxisValue>
       <AxisValue index="1" Format="2">
         <AxisIndex value="1"/>
         <Flags value="0"/>
-        <ValueNameID value="263"/>  <!-- Light -->
+        <ValueNameID value="262"/>  <!-- Light -->
         <NominalValue value="300.0"/>
         <RangeMinValue value="300.0"/>
         <RangeMaxValue value="400.0"/>
@@ -404,7 +365,7 @@
       <AxisValue index="2" Format="2">
         <AxisIndex value="2"/>
         <Flags value="0"/>
-        <ValueNameID value="265"/>  <!-- Condensed -->
+        <ValueNameID value="264"/>  <!-- Condensed -->
         <NominalValue value="50.0"/>
         <RangeMinValue value="50.0"/>
         <RangeMaxValue value="75.0"/>
@@ -412,7 +373,7 @@
       <AxisValue index="3" Format="2">
         <AxisIndex value="2"/>
         <Flags value="2"/>  <!-- ElidableAxisValueName -->
-        <ValueNameID value="266"/>  <!-- Normal -->
+        <ValueNameID value="265"/>  <!-- Normal -->
         <NominalValue value="100.0"/>
         <RangeMinValue value="75.0"/>
         <RangeMaxValue value="125.0"/>
@@ -420,13 +381,13 @@
       <AxisValue index="4" Format="2">
         <AxisIndex value="2"/>
         <Flags value="0"/>
-        <ValueNameID value="267"/>  <!-- Extended -->
+        <ValueNameID value="266"/>  <!-- Extended -->
         <NominalValue value="200.0"/>
         <RangeMinValue value="125.0"/>
         <RangeMaxValue value="200.0"/>
       </AxisValue>
     </AxisValueArray>
-    <ElidedFallbackNameID value="259"/>  <!-- Regular -->
+    <ElidedFallbackNameID value="2"/>  <!-- Regular -->
   </STAT>
 
   <avar>
diff --git a/tests/data/DSv5/MutatorSerifVariable_Width-TTF.ttx b/tests/data/DSv5/MutatorSerifVariable_Width-TTF.ttx
index d442017..354fc6c 100644
--- a/tests/data/DSv5/MutatorSerifVariable_Width-TTF.ttx
+++ b/tests/data/DSv5/MutatorSerifVariable_Width-TTF.ttx
@@ -256,42 +256,6 @@
   </glyf>
 
   <name>
-    <namerecord nameID="256" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Width
-    </namerecord>
-    <namerecord nameID="257" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Serif Light Condensed
-    </namerecord>
-    <namerecord nameID="258" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      MutatorMathTest-SerifLightCondensed
-    </namerecord>
-    <namerecord nameID="259" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Regular
-    </namerecord>
-    <namerecord nameID="260" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      serif
-    </namerecord>
-    <namerecord nameID="261" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Serif
-    </namerecord>
-    <namerecord nameID="262" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      weight
-    </namerecord>
-    <namerecord nameID="263" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Light
-    </namerecord>
-    <namerecord nameID="264" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      width
-    </namerecord>
-    <namerecord nameID="265" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Condensed
-    </namerecord>
-    <namerecord nameID="266" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Normal
-    </namerecord>
-    <namerecord nameID="267" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Extended
-    </namerecord>
     <namerecord nameID="1" platformID="3" platEncID="1" langID="0x409">
       New Font
     </namerecord>
@@ -320,30 +284,27 @@
       MutatorMathTest-SerifLightCondensed
     </namerecord>
     <namerecord nameID="259" platformID="3" platEncID="1" langID="0x409">
-      Regular
-    </namerecord>
-    <namerecord nameID="260" platformID="3" platEncID="1" langID="0x409">
       serif
     </namerecord>
-    <namerecord nameID="261" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="260" platformID="3" platEncID="1" langID="0x409">
       Serif
     </namerecord>
-    <namerecord nameID="262" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="261" platformID="3" platEncID="1" langID="0x409">
       weight
     </namerecord>
-    <namerecord nameID="263" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="262" platformID="3" platEncID="1" langID="0x409">
       Light
     </namerecord>
-    <namerecord nameID="264" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="263" platformID="3" platEncID="1" langID="0x409">
       width
     </namerecord>
-    <namerecord nameID="265" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="264" platformID="3" platEncID="1" langID="0x409">
       Condensed
     </namerecord>
-    <namerecord nameID="266" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="265" platformID="3" platEncID="1" langID="0x409">
       Normal
     </namerecord>
-    <namerecord nameID="267" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="266" platformID="3" platEncID="1" langID="0x409">
       Extended
     </namerecord>
   </name>
@@ -407,17 +368,17 @@
     <DesignAxisRecord>
       <Axis index="0">
         <AxisTag value="SRIF"/>
-        <AxisNameID value="260"/>  <!-- serif -->
+        <AxisNameID value="259"/>  <!-- serif -->
         <AxisOrdering value="0"/>
       </Axis>
       <Axis index="1">
         <AxisTag value="wght"/>
-        <AxisNameID value="262"/>  <!-- weight -->
+        <AxisNameID value="261"/>  <!-- weight -->
         <AxisOrdering value="1"/>
       </Axis>
       <Axis index="2">
         <AxisTag value="wdth"/>
-        <AxisNameID value="264"/>  <!-- width -->
+        <AxisNameID value="263"/>  <!-- width -->
         <AxisOrdering value="2"/>
       </Axis>
     </DesignAxisRecord>
@@ -426,13 +387,13 @@
       <AxisValue index="0" Format="1">
         <AxisIndex value="0"/>
         <Flags value="0"/>
-        <ValueNameID value="261"/>  <!-- Serif -->
+        <ValueNameID value="260"/>  <!-- Serif -->
         <Value value="1.0"/>
       </AxisValue>
       <AxisValue index="1" Format="2">
         <AxisIndex value="1"/>
         <Flags value="0"/>
-        <ValueNameID value="263"/>  <!-- Light -->
+        <ValueNameID value="262"/>  <!-- Light -->
         <NominalValue value="300.0"/>
         <RangeMinValue value="300.0"/>
         <RangeMaxValue value="400.0"/>
@@ -440,7 +401,7 @@
       <AxisValue index="2" Format="2">
         <AxisIndex value="2"/>
         <Flags value="0"/>
-        <ValueNameID value="265"/>  <!-- Condensed -->
+        <ValueNameID value="264"/>  <!-- Condensed -->
         <NominalValue value="50.0"/>
         <RangeMinValue value="50.0"/>
         <RangeMaxValue value="75.0"/>
@@ -448,7 +409,7 @@
       <AxisValue index="3" Format="2">
         <AxisIndex value="2"/>
         <Flags value="2"/>  <!-- ElidableAxisValueName -->
-        <ValueNameID value="266"/>  <!-- Normal -->
+        <ValueNameID value="265"/>  <!-- Normal -->
         <NominalValue value="100.0"/>
         <RangeMinValue value="75.0"/>
         <RangeMaxValue value="125.0"/>
@@ -456,13 +417,13 @@
       <AxisValue index="4" Format="2">
         <AxisIndex value="2"/>
         <Flags value="0"/>
-        <ValueNameID value="267"/>  <!-- Extended -->
+        <ValueNameID value="266"/>  <!-- Extended -->
         <NominalValue value="200.0"/>
         <RangeMinValue value="125.0"/>
         <RangeMaxValue value="200.0"/>
       </AxisValue>
     </AxisValueArray>
-    <ElidedFallbackNameID value="259"/>  <!-- Regular -->
+    <ElidedFallbackNameID value="2"/>  <!-- Regular -->
   </STAT>
 
   <avar>
diff --git a/tests/data/TestVarFont-MyFontVF1.ttx b/tests/data/TestVarFont-MyFontVF1.ttx
index ba4fd42..539c2c4 100644
--- a/tests/data/TestVarFont-MyFontVF1.ttx
+++ b/tests/data/TestVarFont-MyFontVF1.ttx
@@ -23,9 +23,6 @@
   </head>
 
   <name>
-    <namerecord nameID="256" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Weight
-    </namerecord>
     <namerecord nameID="1" platformID="3" platEncID="1" langID="0x409">
       TestVarFont
     </namerecord>
diff --git a/tests/data/TestVarFont-MyFontVF2.ttx b/tests/data/TestVarFont-MyFontVF2.ttx
index 3893a29..7b4be8b 100644
--- a/tests/data/TestVarFont-MyFontVF2.ttx
+++ b/tests/data/TestVarFont-MyFontVF2.ttx
@@ -23,9 +23,6 @@
   </head>
 
   <name>
-    <namerecord nameID="256" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Weight
-    </namerecord>
     <namerecord nameID="1" platformID="3" platEncID="1" langID="0x409">
       My Font Narrow VF
     </namerecord>
diff --git a/tests/data/TestVariableFont-CFF2-cffsubr.ttx b/tests/data/TestVariableFont-CFF2-cffsubr.ttx
index 48af7a3..2d0d77e 100644
--- a/tests/data/TestVariableFont-CFF2-cffsubr.ttx
+++ b/tests/data/TestVariableFont-CFF2-cffsubr.ttx
@@ -138,9 +138,6 @@
   </cmap>
 
   <name>
-    <namerecord nameID="256" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Weight
-    </namerecord>
     <namerecord nameID="1" platformID="3" platEncID="1" langID="0x409">
       Layer Font
     </namerecord>
diff --git a/tests/data/TestVariableFont-CFF2-post3.ttx b/tests/data/TestVariableFont-CFF2-post3.ttx
index f0753e8..209bfb5 100644
--- a/tests/data/TestVariableFont-CFF2-post3.ttx
+++ b/tests/data/TestVariableFont-CFF2-post3.ttx
@@ -138,9 +138,6 @@
   </cmap>
 
   <name>
-    <namerecord nameID="256" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Weight
-    </namerecord>
     <namerecord nameID="1" platformID="3" platEncID="1" langID="0x409">
       Layer Font
     </namerecord>
diff --git a/tests/data/TestVariableFont-CFF2-useProductionNames.ttx b/tests/data/TestVariableFont-CFF2-useProductionNames.ttx
index e48835f..d6a9d5e 100644
--- a/tests/data/TestVariableFont-CFF2-useProductionNames.ttx
+++ b/tests/data/TestVariableFont-CFF2-useProductionNames.ttx
@@ -138,9 +138,6 @@
   </cmap>
 
   <name>
-    <namerecord nameID="256" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Weight
-    </namerecord>
     <namerecord nameID="1" platformID="3" platEncID="1" langID="0x409">
       Layer Font
     </namerecord>
diff --git a/tests/data/TestVariableFont-CFF2.ttx b/tests/data/TestVariableFont-CFF2.ttx
index 4922528..fdb9295 100644
--- a/tests/data/TestVariableFont-CFF2.ttx
+++ b/tests/data/TestVariableFont-CFF2.ttx
@@ -138,9 +138,6 @@
   </cmap>
 
   <name>
-    <namerecord nameID="256" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Weight
-    </namerecord>
     <namerecord nameID="1" platformID="3" platEncID="1" langID="0x409">
       Layer Font
     </namerecord>
diff --git a/tests/data/TestVariableFont-TTF-post3.ttx b/tests/data/TestVariableFont-TTF-post3.ttx
index 2bf1d96..c730d63 100644
--- a/tests/data/TestVariableFont-TTF-post3.ttx
+++ b/tests/data/TestVariableFont-TTF-post3.ttx
@@ -257,9 +257,6 @@
   </glyf>
 
   <name>
-    <namerecord nameID="256" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Weight
-    </namerecord>
     <namerecord nameID="1" platformID="3" platEncID="1" langID="0x409">
       Layer Font
     </namerecord>
diff --git a/tests/data/TestVariableFont-TTF-useProductionNames.ttx b/tests/data/TestVariableFont-TTF-useProductionNames.ttx
index 3382109..4f365f3 100644
--- a/tests/data/TestVariableFont-TTF-useProductionNames.ttx
+++ b/tests/data/TestVariableFont-TTF-useProductionNames.ttx
@@ -257,9 +257,6 @@
   </glyf>
 
   <name>
-    <namerecord nameID="256" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Weight
-    </namerecord>
     <namerecord nameID="1" platformID="3" platEncID="1" langID="0x409">
       Layer Font
     </namerecord>
diff --git a/tests/data/TestVariableFont-TTF.ttx b/tests/data/TestVariableFont-TTF.ttx
index 6488325..b36a9c5 100644
--- a/tests/data/TestVariableFont-TTF.ttx
+++ b/tests/data/TestVariableFont-TTF.ttx
@@ -257,9 +257,6 @@
   </glyf>
 
   <name>
-    <namerecord nameID="256" platformID="1" platEncID="0" langID="0x0" unicode="True">
-      Weight
-    </namerecord>
     <namerecord nameID="1" platformID="3" platEncID="1" langID="0x409">
       Layer Font
     </namerecord>
diff --git a/tests/featureWriters/markFeatureWriter_test.py b/tests/featureWriters/markFeatureWriter_test.py
index a504f60..5edcaef 100644
--- a/tests/featureWriters/markFeatureWriter_test.py
+++ b/tests/featureWriters/markFeatureWriter_test.py
@@ -2011,6 +2011,81 @@ class MarkFeatureWriterTest(FeatureWriterTest):
             """
         )
 
+    def test_contextual_liga_anchor_no_number(self, testufo):
+        fi = testufo["f_i"]
+        fi.appendAnchor(
+            {"name": "*top.tilde", "x": 300, "y": 500, "identifier": "*top.tilde"}
+        )
+        fi.appendAnchor(
+            {"name": "*top_1.acute", "x": 200, "y": 300, "identifier": "*top_1.acute"}
+        )
+        fi.lib[OBJECT_LIBS_KEY] = {
+            "*top.tilde": {
+                "GPOS_Context": "* tildecomb",
+            },
+            "*top_1.acute": {
+                "GPOS_Context": "* acutecomb",
+            },
+        }
+
+        writer = MarkFeatureWriter()
+        feaFile = ast.FeatureFile()
+        assert writer.write(testufo, feaFile)
+
+        assert str(feaFile) == dedent(
+            """\
+            markClass acutecomb <anchor 100 200> @MC_top;
+            markClass tildecomb <anchor 100 200> @MC_top;
+
+            lookup mark2base {
+                pos base a
+                    <anchor 100 200> mark @MC_top;
+            } mark2base;
+
+            lookup mark2liga {
+                pos ligature f_i
+                        <anchor 100 500> mark @MC_top
+                    ligComponent
+                        <anchor 600 500> mark @MC_top;
+            } mark2liga;
+
+            lookup ContextualMark_0 {
+                pos base f_i
+                    <anchor 300 500> mark @MC_top;
+            } ContextualMark_0;
+
+            lookup ContextualMark_1 {
+                pos ligature f_i
+                        <anchor 200 300> mark @MC_top
+                    ligComponent
+                        <anchor NULL>;
+            } ContextualMark_1;
+
+            lookup ContextualMarkDispatch_0 {
+                # * tildecomb
+                pos [f_i] @MC_top' lookup ContextualMark_0 tildecomb;
+                # * acutecomb
+                pos [f_i] @MC_top' lookup ContextualMark_1 acutecomb;
+            } ContextualMarkDispatch_0;
+
+            feature mark {
+                lookup mark2base;
+                lookup mark2liga;
+                lookup ContextualMarkDispatch_0;
+            } mark;
+
+            feature mkmk {
+                lookup mark2mark_top {
+                    @MFS_mark2mark_top = [acutecomb tildecomb];
+                    lookupflag UseMarkFilteringSet @MFS_mark2mark_top;
+                    pos mark tildecomb
+                        <anchor 100 300> mark @MC_top;
+                } mark2mark_top;
+
+            } mkmk;
+            """
+        )
+
     def test_contextual_anchor_no_context(self, testufo, caplog):
         a = testufo["a"]
         a.appendAnchor({"name": "*top", "x": 200, "y": 200, "identifier": "*top"})
diff --git a/tests/instantiator_test.py b/tests/instantiator_test.py
index 6112db2..fa67331 100644
--- a/tests/instantiator_test.py
+++ b/tests/instantiator_test.py
@@ -71,7 +71,6 @@ def test_default_groups_only(ufo_module, data_dir, caplog):
     d.addSourceDescriptor(location={"Weight": 300}, font=ufo_module.Font())
     d.addSourceDescriptor(location={"Weight": 900}, font=ufo_module.Font())
     d.addInstanceDescriptor(styleName="2", location={"Weight": 400})
-    d.findDefault()
 
     d.sources[0].font.groups["public.kern1.GRK_alpha_alt_LC_1ST"] = [
         "alpha.alt",
@@ -102,7 +101,6 @@ def test_default_groups_only2(ufo_module, data_dir, caplog):
     d.addSourceDescriptor(location={"Weight": 300}, font=ufo_module.Font())
     d.addSourceDescriptor(location={"Weight": 900}, font=ufo_module.Font())
     d.addInstanceDescriptor(styleName="2", location={"Weight": 400})
-    d.findDefault()
 
     d.sources[0].font.groups["public.kern1.GRK_alpha_alt_LC_1ST"] = [
         "alpha.alt",
diff --git a/tests/outlineCompiler_test.py b/tests/outlineCompiler_test.py
index 5755ce5..7219794 100644
--- a/tests/outlineCompiler_test.py
+++ b/tests/outlineCompiler_test.py
@@ -1009,6 +1009,21 @@ class ColrCpalTest:
             "c": [("c.color2", 1), ("c.color1", 0)],
         }
 
+    def test_colr_cpal_gid1_not_blank(self, FontClass, caplog):
+        # https://github.com/MicrosoftDocs/typography-issues/issues/346
+        testufo = FontClass(getpath("ColorTest.ufo"))
+        del testufo["space"]
+
+        with caplog.at_level(logging.WARNING, logger="ufo2ft.outlineCompiler"):
+            ttf = compileTTF(testufo)
+
+        assert ttf["COLR"].version == 0
+        assert ttf.getGlyphOrder()[1] == "a"
+        assert (
+            "COLRv0 might not render correctly on Windows because "
+            "the glyph at index 1 is not empty ('a')."
+        ) in caplog.text
+
     @pytest.mark.parametrize("compileFunc", [compileTTF, compileOTF])
     @pytest.mark.parametrize("manualClipBoxes", [True, False])
     @pytest.mark.parametrize(
