diff -Naur The Unarchiver/UniversalDetector/Makefile.linux The Unarchiver_patched/UniversalDetector/Makefile.linux
--- The Unarchiver/UniversalDetector/Makefile.linux	2011-04-27 03:32:21.000000000 +0200
+++ The Unarchiver_patched/UniversalDetector/Makefile.linux	2011-12-22 10:48:32.138621163 +0100
@@ -16,7 +16,6 @@
 
 GNUSTEP_OPTS =	-DGNUSTEP \
 		-DGNU_RUNTIME=1 \
-		-D_NATIVE_OBJC_EXCEPTIONS \
 		-fgnu-runtime \
 		-fexceptions \
 		-fobjc-exceptions \
diff -Naur The Unarchiver/XADMaster/Makefile.linux The Unarchiver_patched/XADMaster/Makefile.linux
--- The Unarchiver/XADMaster/Makefile.linux	2011-04-27 03:32:21.000000000 +0200
+++ The Unarchiver_patched/XADMaster/Makefile.linux	2011-12-22 10:48:15.488721721 +0100
@@ -16,7 +16,6 @@
 
 GNUSTEP_OPTS =	-DGNUSTEP \
 		-DGNU_RUNTIME=1 \
-		-D_NATIVE_OBJC_EXCEPTIONS \
 		-fgnu-runtime \
 		-fexceptions \
 		-fobjc-exceptions \
