From 8f8ef67d50737f83515a66146e8cd2cf811480aa Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Tue, 21 Jan 2025 08:54:11 -0500
Subject: [PATCH 2/2] Handle new rwm types in llvm 19

---
 deps/LLVMExtra/lib/Core.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/deps/LLVMExtra/lib/Core.cpp b/deps/LLVMExtra/lib/Core.cpp
index 8e6a24dd..519e2ff4 100644
--- a/deps/LLVMExtra/lib/Core.cpp
+++ b/deps/LLVMExtra/lib/Core.cpp
@@ -714,6 +714,10 @@ static AtomicRMWInst::BinOp mapFromLLVMRMWBinOp(LLVMAtomicRMWBinOp BinOp) {
     case LLVMAtomicRMWBinOpFSub: return AtomicRMWInst::FSub;
     case LLVMAtomicRMWBinOpFMax: return AtomicRMWInst::FMax;
     case LLVMAtomicRMWBinOpFMin: return AtomicRMWInst::FMin;
+#if LLVM_VERSION_MAJOR >= 19
+    case LLVMAtomicRMWBinOpUIncWrap: return AtomicRMWInst::UIncWrap;
+    case LLVMAtomicRMWBinOpUDecWrap: return AtomicRMWInst::UDecWrap;
+#endif
   }
 
   llvm_unreachable("Invalid LLVMAtomicRMWBinOp value!");
-- 
2.47.1

