diff --git a/third_party/libwebrtc/modules/desktop_capture/desktop_capture_gn/moz.build b/third_party/libwebrtc/modules/desktop_capture/desktop_capture_gn/moz.build
index 366ea67aa9ab..c57962ae3dc4 100644
--- a/third_party/libwebrtc/modules/desktop_capture/desktop_capture_gn/moz.build
+++ b/third_party/libwebrtc/modules/desktop_capture/desktop_capture_gn/moz.build
@@ -313,6 +313,7 @@ if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
       "/third_party/libwebrtc/modules/desktop_capture/linux/wayland/egl_dmabuf.cc",
       "/third_party/libwebrtc/modules/desktop_capture/linux/wayland/mouse_cursor_monitor_pipewire.cc",
       "/third_party/libwebrtc/modules/desktop_capture/linux/wayland/restore_token_manager.cc",
+      "/third_party/libwebrtc/modules/desktop_capture/linux/wayland/scoped_glib.cc",
       "/third_party/libwebrtc/modules/desktop_capture/linux/wayland/screen_capture_portal_interface.cc",
       "/third_party/libwebrtc/modules/desktop_capture/linux/wayland/screencast_portal.cc",
       "/third_party/libwebrtc/modules/desktop_capture/linux/wayland/screencast_stream_utils.cc",
