diff --git a/build/moz.configure/init.configure b/build/moz.configure/init.configure
--- a/build/moz.configure/init.configure
+++ b/build/moz.configure/init.configure
@@ -136,11 +136,12 @@
 @checking("for Python 3", callback=lambda x: "%s (%s)" % (x.path, x.str_version))
 @imports("sys")
 @imports(_from="mach.site", _import="PythonVirtualenv")
+@imports(_from="os.path", _import="realpath")
 def virtualenv_python3():
     return namespace(
         # sys.executable is currently not updated for in-process activations. However,
         # sys.prefix is, so we can calculate the python executable's path from there.
-        path=normsep(PythonVirtualenv(sys.prefix).python_path),
+        path=normsep(PythonVirtualenv(realpath(sys.prefix)).python_path),
         str_version=".".join(str(i) for i in sys.version_info[0:3]),
     )
 

