diff --unified --recursive --text fuzzylite-6.0.orig/fuzzylite/CMakeLists.txt fuzzylite-6.0/fuzzylite/CMakeLists.txt
--- fuzzylite-6.0.orig/fuzzylite/CMakeLists.txt	2023-06-19 13:11:52.635098242 +0200
+++ fuzzylite-6.0/fuzzylite/CMakeLists.txt	2023-06-19 13:12:27.115350609 +0200
@@ -74,7 +74,7 @@
 if(NOT MSVC)
 #TODO: Remove -Werror before release.
 #Add Unix compilation flags
-    set(CMAKE_CXX_FLAGS "-pedantic -Wall -Wextra -Werror ${CMAKE_CXX_FLAGS}")
+    set(CMAKE_CXX_FLAGS "-pedantic -Wall -Wextra ${CMAKE_CXX_FLAGS}")
 
     set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG}")
     set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}")
