--- ./src/bridge.c.orig	2013-04-24 15:36:13.486078355 -0400
+++ ./src/bridge.c	2013-04-24 15:44:24.063045320 -0400
@@ -970,7 +970,6 @@
       || (g_strcmp0 (prg, "qtoctave") == 0)
       || (g_strcmp0 (prg, "firefox-bin") == 0)
       || (g_strcmp0 (prg, "thunderbird-bin") == 0)
-      || (g_strcmp0 (prg, "Eclipse") == 0)
       || g_str_has_prefix (prg, "emacs")
       || (g_strcmp0 (prg, "glade") == 0)
       || (g_strcmp0 (prg, "gnome-panel") == 0)
