--- a/src/bridge.c	2012-10-03 13:21:06.000000000 -0400
+++ b/src/bridge.c	2013-02-09 12:40:43.441589211 -0500
@@ -970,7 +970,6 @@
       || (g_strcmp0 (prg, "qtoctave") == 0)
       || (g_strcmp0 (prg, "firefox-bin") == 0)
       || (g_strcmp0 (prg, "thunderbird-bin") == 0)
-      || (g_strcmp0 (prg, "Eclipse") == 0)
       || (g_strcmp0 (prg, "emacs") == 0)
       || (g_strcmp0 (prg, "emacs23") == 0)
       || (g_strcmp0 (prg, "emacs23-lucid") == 0)
