diff --git a/pyoxidizer.template.bzl b/pyoxidizer.template.bzl
index 612ecde..5efea3c 100644
--- a/pyoxidizer.template.bzl
+++ b/pyoxidizer.template.bzl
@@ -27,7 +27,7 @@ def make_exe(dist):
         name="iredis",
         config=python_config,
         # Embed all extension modules, making this a fully-featured Python.
-        extension_module_filter='all',
+        #extension_module_filter='all',
 
         # Only package the minimal set of extension modules needed to initialize
         # a Python interpreter. Many common packages in Python's standard
@@ -37,7 +37,7 @@ def make_exe(dist):
         # Only package extension modules that don't require linking against
         # non-Python libraries. e.g. will exclude support for OpenSSL, SQLite3,
         # other features that require external libraries.
-        #extension_module_filter='no-libraries',
+        extension_module_filter='no-libraries',
 
         # Only package extension modules that don't link against GPL licensed
         # libraries.
@@ -46,7 +46,7 @@ def make_exe(dist):
         # Include Python module sources. This isn't strictly required and it does
         # make binary sizes larger. But having the sources can be useful for
         # activities such as debugging.
-        include_sources=True,
+        include_sources=False,
 
         # Whether to include non-module resource data/files.
         include_resources=False,
