From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Sat, 20 Jun 2015 17:22:38 +0200
Subject: [PATCH] Xsession: Don't start ssh-agent by default

---
 data/Xsession.in | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/data/Xsession.in b/data/Xsession.in
index 2e4de4fe384f..29ebc30ea0c5 100755
--- a/data/Xsession.in
+++ b/data/Xsession.in
@@ -207,14 +207,6 @@ if [ "x$command" = "xdefault" ] ; then
   fi
 fi
 
-# add ssh-agent if found
-sshagent="`gdmwhich ssh-agent`"
-if [ -n "$sshagent" ] && [ -x "$sshagent" ] && [ -z "$SSH_AUTH_SOCK" ]; then
-    command="$sshagent -- $command"
-elif [ -z "$sshagent" ] ; then
-    echo "$0: ssh-agent not found!"
-fi
-
 echo "$0: Setup done, will execute: $command"
 
 eval exec $command
