From 328a315c21ec71e563d00699f0a79186b229270a Mon Sep 17 00:00:00 2001
Message-Id: <328a315c21ec71e563d00699f0a79186b229270a.1541542184.git.jan.steffens@gmail.com>
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Sat, 20 Jun 2015 17:22:38 +0200
Subject: [PATCH] Xsession: Don't start ssh-agent by default

---
 data/Xsession.in | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/data/Xsession.in b/data/Xsession.in
index 9d79558c..ff6d9de0 100755
--- a/data/Xsession.in
+++ b/data/Xsession.in
@@ -175,14 +175,6 @@ if [ "x$command" = "xdefault" ] ; then
   fi
 fi
 
-# add ssh-agent if found
-sshagent="`gdmwhich ssh-agent`"
-if [ -n "$sshagent" ] && [ -x "$sshagent" ] && [ -z "$SSH_AUTH_SOCK" ]; then
-    command="$sshagent -- $command"
-elif [ -z "$sshagent" ] ; then
-    echo "$0: ssh-agent not found!"
-fi
-
 echo "$0: Setup done, will execute: $command"
 
 eval exec $command
-- 
2.19.1

