--- wxPython/wx/tools/XRCed/plugins/xh_wxlib.py (revision 50185)
+++ wxPython/wx/tools/XRCed/plugins/xh_wxlib.py (revision 69431)
@@ -18,5 +18,4 @@
         self.AddWindowStyles()
         # Custom styles
-        self.AddStyle('FPB_DEFAULT_STYLE', fpb.FPB_DEFAULT_STYLE)
         self.AddStyle('FPB_SINGLE_FOLD', fpb.FPB_SINGLE_FOLD)
         self.AddStyle('FPB_COLLAPSE_TO_BOTTOM', fpb.FPB_COLLAPSE_TO_BOTTOM)
--- wxPython/wx/tools/XRCed/plugins/wxlib.py (revision 57250)
+++ wxPython/wx/tools/XRCed/plugins/wxlib.py (revision 69431)
@@ -20,5 +20,5 @@
                    implicit_attributes=['label', 'collapsed'],
                    implicit_params={'collapsed': params.ParamBool})
-c.addStyles('FPB_DEFAULT_STYLE', 'FPB_SINGLE_FOLD', 'FPB_COLLAPSE_TO_BOTTOM',
+c.addStyles('FPB_SINGLE_FOLD', 'FPB_COLLAPSE_TO_BOTTOM',
             'FPB_EXCLUSIVE_FOLD', 'FPB_HORIZONTAL', 'FPB_VERTICAL')
 component.Manager.register(c)

