diff --git source/tools/toolutil/toolutil.cpp source/tools/toolutil/toolutil.cpp
index 038f21b..0dbdaae 100644
--- source/tools/toolutil/toolutil.cpp
+++ source/tools/toolutil/toolutil.cpp
@@ -1,7 +1,7 @@
 /*
 *******************************************************************************
 *
-*   Copyright (C) 1999-2013, International Business Machines
+*   Copyright (C) 1999-2014, International Business Machines
 *   Corporation and others.  All Rights Reserved.
 *
 *******************************************************************************
@@ -18,6 +18,7 @@
 *   This file contains utility functions for ICU tools like genccode.
 */
 
+#include "unicode/platform.h"
 #if U_PLATFORM == U_PF_MINGW
 // *cough* - for struct stat
 #ifdef __STRICT_ANSI__
