diff -ura package.orig/PyMuPDF-1.14.8/setup.py package.new/PyMuPDF-1.14.8/setup.py
--- package.orig/PyMuPDF-1.14.8/setup.py	2019-01-31 01:02:59.000000000 +0800
+++ package.new/PyMuPDF-1.14.8/setup.py	2019-02-13 11:24:16.425205772 +0800
@@ -14,8 +14,8 @@
                        libraries=[
                            'mupdf',
                            'mupdf-third',
-                           # 'jbig2dec', 'openjp2', 'jpeg', 'freetype',
-                           # 'crypto', #openssl is required by mupdf on archlinux
+                           'jbig2dec', 'openjp2', 'jpeg', 'freetype',
+                           'crypto', #openssl is required by mupdf on archlinux
                            ], # the libraries to link with
                       )
 elif sys.platform.startswith(('darwin', 'freebsd')):
