--- orig/src/Makefile	2016-08-26 11:52:26.530936797 -0400
+++ src/Makefile	2016-08-26 11:37:11.178797636 -0400
@@ -35,7 +35,7 @@
 ifdef STATIC
 STATIC_LIBS += -lboost_filesystem -lboost_system -lbsl430 -lboost_date_time -lboost_chrono -lboost_thread
 else
-LIBS += -lboost_filesystem -lboost_system -lbsl430 -lboost_date_time -lboost_chrono -lboost_thread
+LIBS += -lboost_filesystem -lboost_system -lbsl430 -lboost_date_time -lboost_chrono -lboost_thread -lhidapi-libusb
 endif
 
 LIBTHIRD := ./ThirdParty/lib64
@@ -128,7 +128,7 @@
 OBJS := $(patsubst %.cpp, %.o, $(SRC))
 
 all: $(BSLLIB) $(OBJS)
-	$(CXX) $(CXXFLAGS) -shared $(OUTNAME)$(OUTPUT) -o $(OUTPUT) $(OBJS) $(HIDOBJ) $(LIBDIRS) $(BSTATIC) $(STATIC_LIBS) $(BDYNAMIC) $(LIBS)
+	$(CXX) $(CXXFLAGS) -shared $(OUTNAME)$(OUTPUT) -o $(OUTPUT) $(OBJS) $(LIBDIRS) $(BSTATIC) $(STATIC_LIBS) $(BDYNAMIC) $(LIBS)
 	rm -f $(STATICOUTPUT).a
 	ar -rs $(STATICOUTPUT).a $(OBJS)
 
--- orig/src/ThirdParty/BSL430_DLL/BSL430_DLL/Physical_Interfaces/MSPBSL_PhysicalInterfaceUSB.h	2016-08-26 12:12:09.657835783 -0400
+++ src/ThirdParty/BSL430_DLL/BSL430_DLL/Physical_Interfaces/MSPBSL_PhysicalInterfaceUSB.h	2016-08-26 11:35:52.380386062 -0400
@@ -38,7 +38,7 @@
 
 #pragma once
 
-#include <hidapi.h>
+#include <hidapi/hidapi.h>
 
 #include "MSPBSL_PhysicalInterface.h"
 
--- orig/src/DLL430_v3/src/TI/DLL430/HidUpdateManager.cpp	2016-08-26 12:12:09.707835250 -0400
+++ src/DLL430_v3/src/TI/DLL430/HidUpdateManager.cpp	2016-08-26 11:35:52.380386062 -0400
@@ -38,7 +38,7 @@
 #include <pch.h>
 #include "HidUpdateManager.h"
 
-#include <hidapi.h>
+#include <hidapi/hidapi.h>
 
 #include <BSL430_DLL/Connections/MSPBSL_Connection5xxUSB.h>
 #include <BSL430_DLL/MSPBSL_Factory.h>
