--- dislocker-0.7.3/src/CMakeLists.txt.orig
+++ dislocker-0.7.3/src/CMakeLists.txt
@@ -129,12 +129,15 @@
 	return ()
 endif()
 
-find_package (Ruby)
-if(RUBY_FOUND  AND  RUBY_INCLUDE_DIRS  AND  RUBY_LIBRARY)
-	include_directories (${RUBY_INCLUDE_DIRS})
-	set (LIB ${LIB} ${RUBY_LIBRARY})
-	add_definitions (-D_HAVE_RUBY=${RUBY_VERSION_STRING})
-	set (SOURCES ${SOURCES} ruby.c)
+option (WITH_RUBY "Enable Ruby bindings" ON)
+if (WITH_RUBY)
+	find_package (Ruby)
+	if(RUBY_FOUND  AND  RUBY_INCLUDE_DIRS  AND  RUBY_LIBRARY)
+		include_directories (${RUBY_INCLUDE_DIRS})
+		set (LIB ${LIB} ${RUBY_LIBRARY})
+		add_definitions (-D_HAVE_RUBY=${RUBY_VERSION_STRING})
+		set (SOURCES ${SOURCES} ruby.c)
+	endif()
 endif()
 
 find_package (FUSE)
