--- webkitgtk-2.4.9/Source/WebCore/platform/text/enchant/TextCheckerEnchant.cpp.orig	2017-12-06 14:59:40.768262788 -0500
+++ webkitgtk-2.4.9/Source/WebCore/platform/text/enchant/TextCheckerEnchant.cpp	2017-12-06 15:03:10.000000000 -0500
@@ -128,7 +128,7 @@
         for (i = 0; i < numberOfSuggestions; i++)
             guesses.append(String::fromUTF8(suggestions[i]));
 
-        enchant_dict_free_suggestions(*iter, suggestions);
+        enchant_dict_free_string_list(*iter, suggestions);
     }
 
     return guesses;
