diff -ura amarok-eb9706eaee14e07d894cf2ad442268a6f62d915a/cmake/modules/FindMySQL.cmake amarok-diff/cmake/modules/FindMySQL.cmake
--- amarok-eb9706eaee14e07d894cf2ad442268a6f62d915a/cmake/modules/FindMySQL.cmake	2019-01-12 04:54:33.000000000 +0800
+++ amarok-diff/cmake/modules/FindMySQL.cmake	2019-01-27 21:52:42.770416475 +0800
@@ -42,7 +42,16 @@
         # At least on OpenSUSE --libmysql-libs doesn't exist, so we just use
         # MYSQL_LIBRARIES for that. We'll see if that's enough when testing
         # below.
-        set(MYSQL_EMBEDDED_LIBRARIES ${MYSQL_LIBRARIES})
+        # mysql-config removed --libmysql-libs, but amarok need libmysqld other
+        # than libmysqlclient to run mysql embedded server.
+        find_library(MYSQL_EMBEDDED_LIBRARIES NAMES mysqld libmysqld
+            PATHS
+                $ENV{MYSQL_DIR}/libmysql_r/.libs
+                $ENV{MYSQL_DIR}/lib
+                $ENV{MYSQL_DIR}/lib/mysql
+            PATH_SUFFIXES
+                mysql
+        )
     else()
         set(MYSQL_EMBEDDED_LIBRARIES ${MC_MYSQL_EMBEDDED_LIBRARIES})
     endif()
