diff --git a/src/org.kde.amarok.desktop b/src/org.kde.amarok.desktop
index 198a0c1..3d82037 100644
--- a/src/org.kde.amarok.desktop
+++ b/src/org.kde.amarok.desktop
@@ -234,6 +234,5 @@ X-KDE-Keywords[x-test]=xxmusicxx,xxpodcastxx
 X-KDE-Keywords[zh_CN]=music,podcast,音乐,播客
 X-KDE-Keywords[zh_TW]=music,podcast
 X-KDE-Protocols=http
-DBusActivatable=true
 Terminal=false
 Categories=Qt;KDE;AudioVideo;Audio;Player;
