diff --git a/tests/ut_dconfig.cpp b/tests/ut_dconfig.cpp
index 4cf17b9..b55f566 100644
--- a/tests/ut_dconfig.cpp
+++ b/tests/ut_dconfig.cpp
@@ -29,6 +29,9 @@
 
 DCORE_USE_NAMESPACE
 
+static constexpr char const *APP_ID = "tests";
+static constexpr char const *FILE_NAME = "example";
+
 class ut_DConfig : public testing::Test
 {
 protected:
@@ -50,8 +53,7 @@ protected:
     static EnvGuard backendType;
     static EnvGuard fileBackendLocalPerfix;
     static FileCopyGuard *metaGuard;
-    static constexpr char const *APP_ID = "tests";
-    static constexpr char const *FILE_NAME = "example";
+
 };
 EnvGuard ut_DConfig::fileBackendLocalPerfix;
 EnvGuard ut_DConfig::backendType;
diff --git a/tests/ut_dconfigfile.cpp b/tests/ut_dconfigfile.cpp
index da185b2..728d4e0 100644
--- a/tests/ut_dconfigfile.cpp
+++ b/tests/ut_dconfigfile.cpp
@@ -29,6 +29,8 @@
 
 DCORE_USE_NAMESPACE
 
+static constexpr char const *LocalPrefix = "/tmp/example";
+
 class ut_DConfigFile : public testing::Test
 {
 protected:
@@ -42,7 +44,7 @@ protected:
     }
     virtual void TearDown() override;
 
-    static constexpr char const *LocalPrefix = "/tmp/example";
+
 
     const char *APP_ID = "org.foo.appid";
     const char *FILE_NAME = "org.foo.name";
