--- hid-nintendo.c	2021-10-31 00:11:46.248903347 -0400
+++ hid-nintendo.c	2021-10-31 00:13:06.356102213 -0400
@@ -35,6 +35,14 @@
 #include <linux/power_supply.h>
 #include <linux/spinlock.h>

+#ifndef LED_FUNCTION_PLAYER1
+#define LED_FUNCTION_PLAYER1 "player-1"
+#define LED_FUNCTION_PLAYER2 "player-2"
+#define LED_FUNCTION_PLAYER3 "player-3"
+#define LED_FUNCTION_PLAYER4 "player-4"
+#define LED_FUNCTION_PLAYER5 "player-5"
+#endif
+
 /*
  * Reference the url below for the following HID report defines:
  * https://github.com/dekuNukem/Nintendo_Switch_Reverse_Engineering
