--- src/gn/build/gen.py.orig	2018-07-21 16:28:08.909054488 +0200
+++ src/gn/build/gen.py	2018-07-21 19:15:30.284412730 +0200
@@ -308,8 +312,11 @@
       ldflags.append(min_mac_version_flag)
 
     if options.use_lto:
-      cflags.extend(['-flto', '-fwhole-program-vtables'])
-      ldflags.extend(['-flto', '-fwhole-program-vtables'])
+      cflags.append('-flto')
+      ldflags.append('-flto')
+      if "clang" in cc:
+        cflags.append('-fwhole-program-vtables')
+        ldflags.append('-fwhole-program-vtables')
 
   elif is_win:
     if not options.debug:
