--- Cutegram.pro 2015-03-12 02:06:01.000000000 +0800
+++ Cutegram.pro 2015-05-03 12:30:35.000000000 +0800
@@ -104,6 +104,7 @@
     translations/lang-it.qm \
     translations/lang-nl_NL.qm \
     translations/lang-pt_BR.qm \
+    translations/lang-zh_CN.qm \
     translations/lang-sr.qm

 isEmpty(PREFIX) {
@@ -157,4 +158,5 @@
     translations_sources/lang-nl_NL.ts \
     translations_sources/lang-pt_BR.ts \
     translations_sources/lang-sr.ts \
+    translations_sources/lang-zh_CN.ts \
     objects/themeitem.sco

