From d24c6ff4fceb06dd432d6b2d4b307dff78e3656b Mon Sep 17 00:00:00 2001
From: Thomas Deutschmann <whissi@gentoo.org>
Date: Mon, 6 Apr 2020 20:13:34 +0200
Subject: [PATCH 20/35] Make PGO use toolchain

Signed-off-by: Thomas Deutschmann <whissi@gentoo.org>
---
 build/unix/mozconfig.unix | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/build/unix/mozconfig.unix b/build/unix/mozconfig.unix
index 539dd4d44b..d10ae3df4a 100644
--- a/build/unix/mozconfig.unix
+++ b/build/unix/mozconfig.unix
@@ -6,6 +6,15 @@ if [ -n "$FORCE_GCC" ]; then
     CC="$MOZ_FETCHES_DIR/gcc/bin/gcc"
     CXX="$MOZ_FETCHES_DIR/gcc/bin/g++"
 
+    if [ -n "$MOZ_PGO" ]; then
+        if [ -z "$USE_ARTIFACT" ]; then
+            ac_add_options --enable-lto
+        fi
+        export AR="$topsrcdir/gcc/bin/gcc-ar"
+        export NM="$topsrcdir/gcc/bin/gcc-nm"
+        export RANLIB="$topsrcdir/gcc/bin/gcc-ranlib"
+    fi
+
     # We want to make sure we use binutils and other binaries in the tooltool
     # package.
     mk_add_options "export PATH=$MOZ_FETCHES_DIR/gcc/bin:$PATH"
-- 
2.29.2

