diff -up firefox-65.0/toolkit/moz.configure.disable-elfhack firefox-65.0/toolkit/moz.configure
--- firefox-65.0/toolkit/moz.configure.disable-elfhack	2019-01-28 14:16:48.530345132 +0100
+++ firefox-65.0/toolkit/moz.configure	2019-01-28 14:18:03.231029682 +0100
@@ -1036,7 +1036,7 @@ with only_when('--enable-compile-environ
                help='{Enable|Disable} elf hacks')
 
         set_config('USE_ELF_HACK',
-                   depends_if('--enable-elf-hack')(lambda _: True))
+                   depends_if('--enable-elf-hack')(lambda _: False))
 
 
 @depends(check_build_environment)
