From 04adb3b0cb7c2e241112e8affbbb95e11d288ac3 Mon Sep 17 00:00:00 2001
From: Carl Lei <me@xecycle.info>
Date: Sun, 3 Nov 2024 13:12:20 +0800
Subject: [PATCH] inhibit hex check

---
 gtk/gtkimcontextsimple.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/gtk/gtkimcontextsimple.c b/gtk/gtkimcontextsimple.c
index adcca19722..705836ee25 100644
--- a/gtk/gtkimcontextsimple.c
+++ b/gtk/gtkimcontextsimple.c
@@ -895,12 +895,8 @@ gtk_im_context_simple_filter_keypress (GtkIMContext *context,
     have_hex_mods = TRUE;
   else
     have_hex_mods = (state & (hex_mod_mask)) == hex_mod_mask;
-  is_hex_start = keyval == GDK_KEY_U;
-  is_hex_end = (keyval == GDK_KEY_space ||
-                keyval == GDK_KEY_KP_Space ||
-                keyval == GDK_KEY_Return ||
-                keyval == GDK_KEY_ISO_Enter ||
-                keyval == GDK_KEY_KP_Enter);
+  is_hex_start = FALSE;
+  is_hex_end = FALSE;
   is_backspace = keyval == GDK_KEY_BackSpace;
   is_escape = keyval == GDK_KEY_Escape;
   hex_keyval = canonical_hex_keyval (event);
-- 
2.54.0

