From cf72fe1f658dd5451f3397f132e83f275f9ea982 Mon Sep 17 00:00:00 2001
From: axionl <i@axionl.me>
Date: Mon, 20 Dec 2021 09:08:53 +0800
Subject: [PATCH] fix-normal-color-takes-no-prefix

---
 nanorc.nanorc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nanorc.nanorc b/nanorc.nanorc
index dd889fd..eaf085d 100644
--- a/nanorc.nanorc
+++ b/nanorc.nanorc
@@ -23,7 +23,7 @@ icolor brightblue " brightblue"
 icolor brightmagenta " brightmagenta"
 icolor brightcyan " brightcyan"
 icolor brightwhite " brightwhite"
-icolor brightnormal " brightnormal"
+# icolor brightnormal " brightnormal"
 icolor ,black ",black "
 icolor ,red ",red "
 icolor ,green ",green "
-- 
2.34.1

