From 8930000d55a926e347502ca5f0d94b4a09ae06bf Mon Sep 17 00:00:00 2001
From: Jakub Janeczko <kuba@ev1.pl>
Date: Mon, 20 Apr 2026 17:59:23 +0200
Subject: [PATCH 1/5] fix ????

---
 transactron/core/manager.py | 20 +++++++++++---------
 1 file changed, 11 insertions(+), 9 deletions(-)

diff --git a/transactron/core/manager.py b/transactron/core/manager.py
index 66be51e..cec2050 100644
--- a/transactron/core/manager.py
+++ b/transactron/core/manager.py
@@ -245,8 +245,8 @@ def rec(transaction: TBody, source: Body):
         return method_enables
 
     @staticmethod
-    def _ready_dependencies(method_map: MethodMap) -> TransactionGraph:
-        ready_dependencies = defaultdict[TBody, set[TBody]](set)
+    def _ready_dependencies(method_map: MethodMap) -> Graph[Body]:
+        ready_dependencies = defaultdict[Body, set[Body]](set)
 
         relations = TransactionManager._relations(method_map)
 
@@ -254,12 +254,14 @@ def _ready_dependencies(method_map: MethodMap) -> TransactionGraph:
             if not relation.ready_dependent:
                 continue
 
-            for trans_start in method_map.transactions_for(relation.start):
-                for trans_end in method_map.transactions_for(relation.end):
-                    if trans_start is not trans_end and not TransactionManager._transactions_exclusive(
-                        method_map, trans_start, trans_end
-                    ):
-                        ready_dependencies[trans_end].add(trans_start)
+            # Transitively trickle down ready dependencies
+            todo = {relation.end}
+            while todo:
+                current = todo.pop()
+                if relation.start not in ready_dependencies[current]:
+                    ready_dependencies[current].add(relation.start)
+                    for parent in method_map.method_parents[current]:
+                        todo.add(parent)
 
         return ready_dependencies
 
@@ -452,7 +454,7 @@ def elaborate(self, platform):
                 method._validate_arguments(method_map.argument_by_call[transaction, method])
                 for method in method_map.methods_by_transaction[transaction]
             ]
-            runnable_terms.extend(dep.ready for dep in ready_dependencies[transaction])
+            runnable_terms.extend(dep.run for dep in ready_dependencies[transaction])
             m.d.comb += transaction.runnable.eq(Cat(runnable_terms).all())
 
         ccs = _graph_ccs(cgr)

From a7cb7d75f8aef31bd73b4c9d854b4c6b569f7d0a Mon Sep 17 00:00:00 2001
From: Jakub Janeczko <kuba@ev1.pl>
Date: Mon, 20 Apr 2026 18:00:54 +0200
Subject: [PATCH 2/5] fix 2

---
 transactron/core/method.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/transactron/core/method.py b/transactron/core/method.py
index c3c8ea2..af46783 100644
--- a/transactron/core/method.py
+++ b/transactron/core/method.py
@@ -314,7 +314,7 @@ def __call__(
             arg = kwargs
 
         enable_sig = Signal(name=self.owned_name + "_enable")
-        m.d.comb += enable_sig.eq(enable_call)
+        m.d.av_comb += enable_sig.eq(enable_call)
         m.d.top_comb += assign(arg_rec, arg, fields=AssignType.ALL)
 
         caller = Body.get()

From 2b0a07630dd985081e6876b1e76cb7eb51854401 Mon Sep 17 00:00:00 2001
From: Jakub Janeczko <kuba@ev1.pl>
Date: Mon, 20 Apr 2026 20:07:53 +0200
Subject: [PATCH 3/5] add test

---
 test/core/test_transactions.py | 54 ++++++++++++++++++++++++++++++++++
 1 file changed, 54 insertions(+)

diff --git a/test/core/test_transactions.py b/test/core/test_transactions.py
index f3fc383..354579c 100644
--- a/test/core/test_transactions.py
+++ b/test/core/test_transactions.py
@@ -399,6 +399,38 @@ def _():
         return m
 
 
+class NestedTransactionInMethodMultipleCallersCircuit(Elaboratable):
+    def __init__(self):
+        self.r1 = Signal()
+        self.r2 = Signal()
+
+        self.c1 = Signal()
+        self.c2 = Signal()
+        self.method_run = Signal()
+        self.inner_run = Signal()
+
+    def elaborate(self, platform):
+        m = TModule()
+
+        shared_method = Method()
+
+        @def_method(m, shared_method)
+        def _():
+            m.d.comb += self.method_run.eq(1)
+            with Transaction().body(m):
+                m.d.comb += self.inner_run.eq(1)
+
+        with Transaction().body(m, ready=self.r1):
+            m.d.comb += self.c1.eq(1)
+            shared_method(m)
+
+        with Transaction().body(m, ready=self.r2):
+            m.d.comb += self.c2.eq(1)
+            shared_method(m)
+
+        return m
+
+
 @pytest.mark.parametrize(
     "circuit",
     [
@@ -430,6 +462,28 @@ async def process(sim):
             sim.add_testbench(process)
 
 
+class TestNestedMethodMultipleCallers(TestCaseWithSimulator):
+    def test_nested_transaction_runs_if_any_caller_runs(self):
+        m = NestedTransactionInMethodMultipleCallersCircuit()
+
+        async def process(sim):
+            to_test = 5 * [(0, 0), (0, 1), (1, 0), (1, 1)]
+            random.shuffle(to_test)
+
+            for r1, r2 in to_test:
+                sim.set(m.r1, r1)
+                sim.set(m.r2, r2)
+                *_, c1, c2, method_run, inner_run = await sim.tick().sample(m.c1, m.c2, m.method_run, m.inner_run)
+
+                caller_run = bool(c1) or bool(c2)
+                assert caller_run == bool(r1 or r2)
+                assert bool(method_run) == caller_run
+                assert bool(inner_run) == caller_run
+
+        with self.run_simulation(m) as sim:
+            sim.add_testbench(process)
+
+
 class NestedTransactionCycleStealTestCircuit(SchedulingTestCircuit):
     def elaborate(self, platform):
         m = TModule()

From 856b025c0331419526e9179f1fa33fdc7b30ebf1 Mon Sep 17 00:00:00 2001
From: Jakub Janeczko <kuba@ev1.pl>
Date: Mon, 20 Apr 2026 20:36:31 +0200
Subject: [PATCH 4/5] simplify ready_dependencies collection

---
 transactron/core/manager.py | 14 ++++----------
 1 file changed, 4 insertions(+), 10 deletions(-)

diff --git a/transactron/core/manager.py b/transactron/core/manager.py
index cec2050..fe88caa 100644
--- a/transactron/core/manager.py
+++ b/transactron/core/manager.py
@@ -245,8 +245,8 @@ def rec(transaction: TBody, source: Body):
         return method_enables
 
     @staticmethod
-    def _ready_dependencies(method_map: MethodMap) -> Graph[Body]:
-        ready_dependencies = defaultdict[Body, set[Body]](set)
+    def _ready_dependencies(method_map: MethodMap) -> Mapping[TBody, set[Body]]:
+        ready_dependencies = defaultdict[TBody, set[Body]](set)
 
         relations = TransactionManager._relations(method_map)
 
@@ -254,14 +254,8 @@ def _ready_dependencies(method_map: MethodMap) -> Graph[Body]:
             if not relation.ready_dependent:
                 continue
 
-            # Transitively trickle down ready dependencies
-            todo = {relation.end}
-            while todo:
-                current = todo.pop()
-                if relation.start not in ready_dependencies[current]:
-                    ready_dependencies[current].add(relation.start)
-                    for parent in method_map.method_parents[current]:
-                        todo.add(parent)
+            for trans in method_map.transactions_for(relation.end):
+                ready_dependencies[trans].add(relation.start)
 
         return ready_dependencies
 

From 705816b8c7ec4b8835f8924a8bf8de1dc489bab6 Mon Sep 17 00:00:00 2001
From: Jakub Janeczko <kuba@ev1.pl>
Date: Mon, 20 Apr 2026 21:05:15 +0200
Subject: [PATCH 5/5] simplify

---
 transactron/core/manager.py | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/transactron/core/manager.py b/transactron/core/manager.py
index fe88caa..0eda91c 100644
--- a/transactron/core/manager.py
+++ b/transactron/core/manager.py
@@ -245,8 +245,8 @@ def rec(transaction: TBody, source: Body):
         return method_enables
 
     @staticmethod
-    def _ready_dependencies(method_map: MethodMap) -> Mapping[TBody, set[Body]]:
-        ready_dependencies = defaultdict[TBody, set[Body]](set)
+    def _ready_dependencies(method_map: MethodMap) -> Graph[Body]:
+        ready_dependencies = defaultdict[Body, set[Body]](set)
 
         relations = TransactionManager._relations(method_map)
 
@@ -254,8 +254,7 @@ def _ready_dependencies(method_map: MethodMap) -> Mapping[TBody, set[Body]]:
             if not relation.ready_dependent:
                 continue
 
-            for trans in method_map.transactions_for(relation.end):
-                ready_dependencies[trans].add(relation.start)
+            ready_dependencies[relation.end].add(relation.start)
 
         return ready_dependencies
 
@@ -445,7 +444,10 @@ def elaborate(self, platform):
 
         for transaction in method_map.transactions:
             runnable_terms = [
-                method._validate_arguments(method_map.argument_by_call[transaction, method])
+                (
+                    method._validate_arguments(method_map.argument_by_call[transaction, method])
+                    & Cat(dep.run for dep in ready_dependencies[method]).all()
+                )
                 for method in method_map.methods_by_transaction[transaction]
             ]
             runnable_terms.extend(dep.run for dep in ready_dependencies[transaction])
