--- a/services/device/serial/serial_io_handler_posix.cc	2025-11-09 13:57:16.132379788 +0100
+++ b/services/device/serial/serial_io_handler_posix.cc	2025-11-09 14:07:13.488376598 +0100
@@ -51,7 +51,7 @@
 bool BitrateToSpeedConstant(int bitrate, speed_t* speed) {
 #define BITRATE_TO_SPEED_CASE(x) \
   case x:                        \
-    *speed = B##x;               \
+    *speed = __B##x;             \
     return true;
   switch (bitrate) {
     BITRATE_TO_SPEED_CASE(0)
