--- /usr/share/devtools/pacman.conf.d/extra.conf	2023-05-21 18:50:42.000000000 +0800
+++ extra-riscv64.conf	2023-05-22 10:32:48.462290337 +0800
@@ -19,7 +19,7 @@
 #XferCommand = /usr/bin/curl -L -C - -f -o %o %u
 #XferCommand = /usr/bin/wget --passive-ftp -c -O %o %u
 #CleanMethod = KeepInstalled
-Architecture = auto
+Architecture = riscv64
 
 # Pacman won't upgrade packages listed in IgnorePkg and members of IgnoreGroup
 #IgnorePkg   =
@@ -71,16 +71,16 @@
 # after the header, and they will be used before the default mirrors.
 
 #[core-testing]
-#Include = /etc/pacman.d/mirrorlist
+#Server = https://archriscv.felixc.at/repo/$repo
 
 [core]
-Include = /etc/pacman.d/mirrorlist
+Server = https://archriscv.felixc.at/repo/$repo
 
 #[extra-testing]
-#Include = /etc/pacman.d/mirrorlist
+#Server = https://archriscv.felixc.at/repo/$repo
 
 [extra]
-Include = /etc/pacman.d/mirrorlist
+Server = https://archriscv.felixc.at/repo/$repo
 
 # An example of a custom package repository.  See the pacman manpage for
 # tips on creating your own repositories.
