--- /usr/share/devtools/pacman-extra.conf	2023-03-08 01:56:48.000000000 +0200
+++ pacman-extra-riscv64.conf	2023-03-15 07:59:42.601735222 +0200
@@ -19,7 +19,7 @@
 #XferCommand = /usr/bin/curl -L -C - -f -o %o %u
 #XferCommand = /usr/bin/wget --passive-ftp -c -O %o %u
 #CleanMethod = KeepInstalled
-Architecture = auto
+Architecture = riscv64
 
 # Pacman won't upgrade packages listed in IgnorePkg and members of IgnoreGroup
 #IgnorePkg   =
@@ -71,19 +71,19 @@
 # after the header, and they will be used before the default mirrors.
 
 #[testing]
-#Include = /etc/pacman.d/mirrorlist
+#Server = https://archriscv.felixc.at/repo/$repo
 
 [core]
-Include = /etc/pacman.d/mirrorlist
+Server = https://archriscv.felixc.at/repo/$repo
 
 [extra]
-Include = /etc/pacman.d/mirrorlist
+Server = https://archriscv.felixc.at/repo/$repo
 
 #[community-testing]
-#Include = /etc/pacman.d/mirrorlist
+#Server = https://archriscv.felixc.at/repo/$repo
 
 [community]
-Include = /etc/pacman.d/mirrorlist
+Server = https://archriscv.felixc.at/repo/$repo
 
 # An example of a custom package repository.  See the pacman manpage for
 # tips on creating your own repositories.
