Index: b/gtk/ui/gtkstatusbar.ui
===================================================================
--- a/gtk/ui/gtkstatusbar.ui
+++ b/gtk/ui/gtkstatusbar.ui
@@ -5,8 +5,8 @@
     <property name="spacing">2</property>
     <property name="margin-start">10</property>
     <property name="margin-end">10</property>
-    <property name="margin-top">6</property>
-    <property name="margin-bottom">6</property>
+    <property name="margin-top">1</property>
+    <property name="margin-bottom">1</property>
     <child>
       <object class="GtkFrame" id="frame">
         <property name="visible">1</property>
