Index: b/gtk/gtkentry.c
===================================================================
--- a/gtk/gtkentry.c
+++ b/gtk/gtkentry.c
@@ -9627,7 +9627,8 @@ popup_targets_received (GtkClipboard
           g_signal_connect_swapped (menuitem, "activate",
                                     G_CALLBACK (gtk_entry_insert_emoji), entry);
           gtk_widget_show (menuitem);
-          gtk_menu_shell_append (GTK_MENU_SHELL (menu), menuitem);
+          if (g_getenv ("GTKM_INSERT_EMOJI") != FALSE)
+            gtk_menu_shell_append (GTK_MENU_SHELL (menu), menuitem);
         }
 
       g_signal_emit (entry, signals[POPULATE_POPUP], 0, menu);
Index: b/gtk/gtktextview.c
===================================================================
--- a/gtk/gtktextview.c
+++ b/gtk/gtktextview.c
@@ -9555,7 +9555,8 @@ popup_targets_received (GtkClipboard
           g_signal_connect_swapped (menuitem, "activate",
                                     G_CALLBACK (gtk_text_view_insert_emoji), text_view);
           gtk_widget_show (menuitem);
-          gtk_menu_shell_append (GTK_MENU_SHELL (priv->popup_menu), menuitem);
+          if (g_getenv ("GTKM_INSERT_EMOJI") != FALSE)
+            gtk_menu_shell_append (GTK_MENU_SHELL (priv->popup_menu), menuitem);
         }
 
       g_signal_emit (text_view, signals[POPULATE_POPUP],
