Index: b/gtk/gtkcolorswatch.c
===================================================================
--- a/gtk/gtkcolorswatch.c
+++ b/gtk/gtkcolorswatch.c
@@ -349,6 +349,17 @@ do_popup (GtkColorSwatch *swatch)
 {
   if (swatch->priv->popover == NULL)
     {
+      GtkWidget *item;
+      swatch->priv->popover = gtk_menu_new ();
+      item = gtk_menu_item_new_with_mnemonic (_ ("C_ustomize"));
+      g_signal_connect_swapped (item, "activate", G_CALLBACK (emit_customize), swatch);
+      gtk_widget_set_visible (GTK_WIDGET (item), TRUE);
+      gtk_menu_shell_append (GTK_MENU_SHELL (swatch->priv->popover), item);
+    }
+  gtk_menu_popup_at_pointer (GTK_MENU (swatch->priv->popover), NULL);
+  return;
+  if (swatch->priv->popover == NULL)
+    {
       GtkWidget *box;
       GtkWidget *item;
 
