diff --git a/Makefile b/Makefile
index fa5aa64..2623ac9 100644
--- a/Makefile
+++ b/Makefile
@@ -1,13 +1,13 @@
 BUILD := DEBUG
 VERSION := $(shell git describe --tags)
 
-FLAGS := -Wall -Wpedantic -std=c++11 -MMD -MP
+FLAGS := -Wall -Wpedantic -std=c++11 -MMD -MP -I../libusb/target/include
 DEBUG_FLAGS := -Og -g -DDEBUG
 RELEASE_FLAGS := -O3
 DEFINES := -DVERSION=\"$(VERSION)\"
 
 CXXFLAGS += $(FLAGS) $($(BUILD)_FLAGS) $(DEFINES)
-LDLIBS += -lpthread -lusb-1.0
+LDLIBS += ../libusb/target/lib/libusb-1.0.a -lpthread -ludev
 SOURCES := $(wildcard *.cpp) $(wildcard */*.cpp)
 OBJECTS := $(SOURCES:.cpp=.o) firmware.o
 DEPENDENCIES := $(SOURCES:.cpp=.d)
