--- PluginEngine.cpp	2012-02-28 00:25:30.000000000 +0100
+++ PluginEngine-patched.cpp	2020-01-17 12:25:13.266618206 +0100
@@ -36,7 +36,6 @@
   globalObject().setProperty("importExtension", newFunction(importExtension));
   globalObject().setProperty("vdebug", newFunction(vdebug));
   globalObject().setProperty("findWidget", newFunction(findWidget));
-  globalObject().setProperty("sleep", newFunction(sleep));
 
   VidaliaSettings settings;
   globalObject().setProperty("pluginPath", QScriptValue(settings.pluginPath()));
@@ -202,20 +201,3 @@
 
   return ret;
 }
-
-QScriptValue
-PluginEngine::sleep(QScriptContext *context, QScriptEngine *engine)
-{
-  if(context->argumentCount() != 1)
-    return context->throwError(QString("sleep must be called with 1 parameter."));
-
-  int s = context->argument(1).toInt32();
-
-#if defined(Q_WS_WIN)
-  Sleep(s*1000);
-#else
-  !::sleep(s);
-#endif
-
-  return engine->nullValue();
-}
