--- PluginEngine.h	2012-02-28 00:25:30.000000000 +0100
+++ PluginEngine-patched.h	2020-01-13 21:30:09.721051000 +0100
@@ -50,7 +50,6 @@ class PluginEngine : public QScriptEngin
 //    static QScriptValue includeScript(QScriptContext *context, QScriptEngine *engine);
     static QScriptValue vdebug(QScriptContext *context, QScriptEngine *engine);
     static QScriptValue findWidget(QScriptContext *context, QScriptEngine *engine);
-    static QScriptValue sleep(QScriptContext *context, QScriptEngine *engine);
 
     void loadAllPlugins();
     void tryLoadPlugin(QDir path);
