From 0cffc552ffea835193bf990960eb3aceb6d0ff99 Mon Sep 17 00:00:00 2001
From: Rouven Rastetter <rouven.rastetter@firaweb.de>
Date: Mon, 19 Jan 2015 21:42:25 +0100
Subject: [PATCH] busid patch

---
 kernel/nv-drm.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/kernel/nv-drm.c b/kernel/nv-drm.c
index f3b74d5..7de6040 100644
--- a/kernel/nv-drm.c
+++ b/kernel/nv-drm.c
@@ -128,6 +128,8 @@ static struct drm_driver nv_drm_driver = {
     .gem_prime_vmap = nv_gem_prime_vmap,
     .gem_prime_vunmap = nv_gem_prime_vunmap,
 
+    .set_busid = drm_pci_set_busid,
+
     .name = "nvidia-drm",
     .desc = "NVIDIA DRM driver",
     .date = "20130102",
-- 
2.2.2

