diff --git a/src/wayland/meta-wayland-transaction.c b/src/wayland/meta-wayland-transaction.c
index 4400ed1ab5..585010a642 100644
--- a/src/wayland/meta-wayland-transaction.c
+++ b/src/wayland/meta-wayland-transaction.c
@@ -500,7 +500,7 @@ meta_wayland_transaction_commit (MetaWaylandTransaction *transaction)
   /* If we have a time constraint, we always defer application until just before the
    * appropriate frame clock tick.
    */
-  if (max_time_us)
+  if (max_time_us > g_get_monotonic_time ())
     {
       MetaSurfaceActor *actor;
       ClutterFrameClock *frame_clock = NULL;
