diff --git a/CMakeLists.txt b/CMakeLists.txt
index dee3055..54b4e63 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,10 +25,10 @@ set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH} ${ECM_KDE_MODULE_DIR} "${CMAKE_CURRENT_
 
 find_package(KF6 ${KF6_MIN_VERSION} REQUIRED COMPONENTS CoreAddons KIO Bookmarks)
 
-if (REQUIRE_OWNCLOUD_RESOURCES)
-    find_package(ownCloudShellResources 1.0.0 REQUIRED)
-    set_package_properties(ownCloudShellResources PROPERTIES TYPE REQUIRED DESCRIPTION "Required ownCloud resources and icons" URL "https://github.com/owncloud/client-desktop-shell-integration-resources")
-endif()
+#if (REQUIRE_OWNCLOUD_RESOURCES)
+#    find_package(ownCloudShellResources 1.0.0 REQUIRED)
+#    set_package_properties(ownCloudShellResources PROPERTIES TYPE REQUIRED DESCRIPTION "Required ownCloud resources and icons" URL "https://github.com/owncloud/client-desktop-shell-integration-resources")
+#endif()
 
 
 include(KDEInstallDirs)
