--- modules/io/src/cpp/loadlib.cpp.orig	2017-04-25 12:11:39.191706462 +0200
+++ modules/io/src/cpp/loadlib.cpp	2017-04-25 12:10:11.554798125 +0200
@@ -38,7 +38,7 @@
 {
     types::Library* lib = NULL;
 
-    wchar_t* pwstXML = pathconvertW(_wstXML.data(), FALSE, FALSE, AUTO_STYLE);
+    wchar_t* pwstXML = pathconvertW(_wstXML.data(), (BOOL) FALSE, (BOOL) FALSE, AUTO_STYLE);
     wchar_t* pwstPathLib = expandPathVariableW(pwstXML);
 
     bool expanded = true;
