diff --unified --recursive --text SAOImageDS9-8.2.1.orig/tksao/colorbar/cbgrid.C SAOImageDS9-8.2.1.new/tksao/colorbar/cbgrid.C
--- SAOImageDS9-8.2.1.orig/tksao/colorbar/cbgrid.C	2021-01-12 19:17:40.000000000 +0100
+++ SAOImageDS9-8.2.1.new/tksao/colorbar/cbgrid.C	2021-11-09 14:07:24.833647694 +0100
@@ -197,7 +197,7 @@
     pbox[3] = 0;
   }
 
-  plot = astPlot(frameSet, gbox, pbox, option_);
+  plot = astPlot(frameSet, gbox, pbox, option_, "");
 
   // and now create astGrid
   astGrid2dPtr = this;
diff --unified --recursive --text SAOImageDS9-8.2.1.orig/tksao/frame/grid25d.C SAOImageDS9-8.2.1.new/tksao/frame/grid25d.C
--- SAOImageDS9-8.2.1.orig/tksao/frame/grid25d.C	2021-01-12 19:17:40.000000000 +0100
+++ SAOImageDS9-8.2.1.new/tksao/frame/grid25d.C	2021-11-09 14:28:49.727020650 +0100
@@ -142,7 +142,7 @@
   // and now create astGrid
   astGrid25dPtr = this;
 
-  AstPlot* plot = astPlot(frameSet, gbox, pbox, option_);
+  AstPlot* plot = astPlot(frameSet, gbox, pbox, option_, "");
   astGrid(plot);
 
   astEnd; // now, clean up memory
@@ -159,5 +159,5 @@
   AstShiftMap* sm = astShiftMap(2, tt, "");
   AstCmpMap* cmp = astCmpMap(mm, sm, 1, "");
 
-  astAddFrame((AstFrameSet*)frameSet, AST__CURRENT, cmp, astFrame(2, str));
+  astAddFrame((AstFrameSet*)frameSet, AST__CURRENT, cmp, astFrame(2, str, ""));
 }
diff --unified --recursive --text SAOImageDS9-8.2.1.orig/tksao/frame/grid2d.C SAOImageDS9-8.2.1.new/tksao/frame/grid2d.C
--- SAOImageDS9-8.2.1.orig/tksao/frame/grid2d.C	2021-01-12 19:17:40.000000000 +0100
+++ SAOImageDS9-8.2.1.new/tksao/frame/grid2d.C	2021-11-09 14:29:26.573688468 +0100
@@ -164,7 +164,7 @@
   // and now create astGrid
   astGrid2dPtr = this;
 
-  AstPlot* plot = astPlot(frameSet, gbox, pbox, option_);
+  AstPlot* plot = astPlot(frameSet, gbox, pbox, option_, "");
   astGrid(plot);
 
   astEnd; // now, clean up memory
@@ -181,5 +181,5 @@
   AstShiftMap* sm = astShiftMap(2, tt, "");
   AstCmpMap* cmp = astCmpMap(mm, sm, 1, "");
 
-  astAddFrame((AstFrameSet*)frameSet, AST__CURRENT, cmp, astFrame(2, str));
+  astAddFrame((AstFrameSet*)frameSet, AST__CURRENT, cmp, astFrame(2, str, ""));
 }
diff --unified --recursive --text SAOImageDS9-8.2.1.orig/tksao/frame/grid3d.C SAOImageDS9-8.2.1.new/tksao/frame/grid3d.C
--- SAOImageDS9-8.2.1.orig/tksao/frame/grid3d.C	2021-01-12 19:17:40.000000000 +0100
+++ SAOImageDS9-8.2.1.new/tksao/frame/grid3d.C	2021-11-09 14:29:57.260356085 +0100
@@ -163,7 +163,7 @@
   astGrid3dPtr = this;
   renderMode_ = rm;
 
-  AstPlot3D* plot = astPlot3D(frameSet, gbox, pbox, option_);
+  AstPlot3D* plot = astPlot3D(frameSet, gbox, pbox, option_, "");
   astGrid(plot);
 
   astEnd; // now, clean up memory
@@ -181,5 +181,5 @@
   AstShiftMap* sm = astShiftMap(3,tt,"");
   AstCmpMap* cmp = astCmpMap(mm,sm,1,"");
 
-  astAddFrame((AstFrameSet*)frameSet, AST__CURRENT, cmp, astFrame(3, str));
+  astAddFrame((AstFrameSet*)frameSet, AST__CURRENT, cmp, astFrame(3, str, ""));
 }
diff --unified --recursive --text SAOImageDS9-8.2.1.orig/tksao/frame/wcsast.C SAOImageDS9-8.2.1.new/tksao/frame/wcsast.C
--- SAOImageDS9-8.2.1.orig/tksao/frame/wcsast.C	2021-01-12 19:17:40.000000000 +0100
+++ SAOImageDS9-8.2.1.new/tksao/frame/wcsast.C	2021-11-09 14:32:29.410360780 +0100
@@ -63,7 +63,7 @@
   }
   ostringstream str;
   str << "Format(" << id << ")=" << format << ends;
-  astSet(ast, str.str().c_str());
+  astSet(ast, str.str().c_str(), "");
 }
 
 Vector wcsTran(Context* context, AstFrameSet* ast,
