--- a/components/plus_addresses/BUILD.gn
+++ b/components/plus_addresses/BUILD.gn
@@ -95,6 +95,7 @@ source_set("plus_addresses") {
     "//net",
     "//services/data_decoder/public/cpp",
     "//services/network/public/cpp",
+    "//third_party/re2",
     "//ui/base",
   ]
   public_deps = [
--- a/components/component_updater/installer_policies/BUILD.gn
+++ b/components/component_updater/installer_policies/BUILD.gn
@@ -45,6 +45,7 @@ static_library("installer_policies_no_co
     "//components/update_client",
     "//mojo/public/cpp/base:protobuf_support",
     "//services/network/public/cpp",
+    "//third_party/re2",
   ]
 
   # Disallow depending on content.
