diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index eb329bc88fec..7c98dfddcf8a 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -2054,7 +2054,7 @@ config("chromium_code") {
       # Non-chromium code is not guaranteed to compile cleanly with
       # _FORTIFY_SOURCE. Also, fortified build may fail when optimizations are
       # disabled, so only do that for Release build.
-      fortify_level = "2"
+      fortify_level = "3"
 
       # ChromeOS's toolchain supports a high-quality _FORTIFY_SOURCE=3
       # implementation with a few custom glibc patches. Use that if it's
