--- a/content/browser/BUILD.gn
+++ b/content/browser/BUILD.gn
@@ -73,6 +73,7 @@
     "//cc/animation",
     "//cc/mojo_embedder",
     "//cc/paint",
+    "//chrome/common:buildflags",
     "//components/attribution_reporting:mojom",
     "//components/back_forward_cache:enum",
     "//components/browsing_topics/common:common",
@@ -83,6 +84,7 @@
     "//components/download/public/common:public",
     "//components/file_access",
     "//components/filename_generation",
+    "//components/lens:buildflags",
     "//components/link_header_util",
     "//components/metrics",
     "//components/metrics:single_sample_metrics",
--- a/chrome/browser/extensions/BUILD.gn
+++ b/chrome/browser/extensions/BUILD.gn
@@ -898,6 +898,7 @@
     "//components/resources",
     "//components/safe_browsing:buildflags",
     "//components/safe_browsing/content/browser/web_ui:web_ui",
+    "//components/safe_browsing/content/common/proto:download_file_types_proto",
     "//components/safe_browsing/core/browser/db:database_manager",
     "//components/safe_browsing/core/common",
     "//components/safe_browsing/core/common:safe_browsing_prefs",
--- a/chrome/common/BUILD.gn
+++ b/chrome/common/BUILD.gn
@@ -605,6 +605,7 @@
     "//components/optimization_guide/optimization_guide_internals/webui:url_constants",
     "//components/password_manager/content/common",
     "//components/safe_browsing/core/common",
+    "//components/supervised_user/core/common:buildflags",
     "//device/vr/buildflags",
   ]
 }
