diff -upr chromium-113.0.5672.63.orig/gin/time_clamper.h chromium-113.0.5672.63/gin/time_clamper.h
--- chromium-113.0.5672.63.orig/gin/time_clamper.h	2023-04-26 21:43:06.707860200 +0000
+++ chromium-113.0.5672.63/gin/time_clamper.h	2023-05-09 00:14:09.744221610 +0000
@@ -6,6 +6,7 @@
 #define GIN_TIME_CLAMPER_H_
 
 #include <algorithm>
+#include <cstdlib>
 
 #include "base/rand_util.h"
 #include "base/time/time.h"
@@ -48,7 +49,7 @@ class GIN_EXPORT TimeClamper {
     const int64_t micros = now_micros % 1000;
     // abs() is necessary for devices with times before unix-epoch (most likely
     // configured incorrectly).
-    if (abs(micros) + kResolutionMicros < 1000) {
+    if (std::abs(micros) + kResolutionMicros < 1000) {
       return now_micros / 1000;
     }
     return ClampTimeResolution(now_micros) / 1000;
diff -upr chromium-113.0.5672.63.orig/third_party/maldoca/src/maldoca/ole/header.h chromium-113.0.5672.63/third_party/maldoca/src/maldoca/ole/header.h
--- chromium-113.0.5672.63.orig/third_party/maldoca/src/maldoca/ole/header.h	2023-04-26 21:43:58.708449600 +0000
+++ chromium-113.0.5672.63/third_party/maldoca/src/maldoca/ole/header.h	2023-05-09 00:14:09.744221610 +0000
@@ -43,6 +43,7 @@
 #ifndef MALDOCA_OLE_HEADER_H_
 #define MALDOCA_OLE_HEADER_H_
 
+#include <cstdint>
 #include "absl/strings/string_view.h"
 
 namespace maldoca {
diff -upr chromium-113.0.5672.63.orig/third_party/ruy/src/ruy/profiler/instrumentation.h chromium-113.0.5672.63/third_party/ruy/src/ruy/profiler/instrumentation.h
--- chromium-113.0.5672.63.orig/third_party/ruy/src/ruy/profiler/instrumentation.h	2023-04-26 21:47:36.527673200 +0000
+++ chromium-113.0.5672.63/third_party/ruy/src/ruy/profiler/instrumentation.h	2023-05-09 00:14:09.744221610 +0000
@@ -19,6 +19,7 @@ limitations under the License.
 #ifdef RUY_PROFILER
 #include <cstdio>
 #include <mutex>
+#include <string>
 #include <vector>
 #endif
 
diff -upr chromium-113.0.5672.63.orig/third_party/tflite/src/tensorflow/lite/kernels/internal/spectrogram.cc chromium-113.0.5672.63/third_party/tflite/src/tensorflow/lite/kernels/internal/spectrogram.cc
--- chromium-113.0.5672.63.orig/third_party/tflite/src/tensorflow/lite/kernels/internal/spectrogram.cc	2023-04-26 21:47:39.531938300 +0000
+++ chromium-113.0.5672.63/third_party/tflite/src/tensorflow/lite/kernels/internal/spectrogram.cc	2023-05-09 00:14:09.747555160 +0000
@@ -17,6 +17,7 @@ limitations under the License.
 
 #include <assert.h>
 #include <math.h>
+#include <stdint.h>
 
 #include "third_party/fft2d/fft.h"
 
diff -upr chromium-113.0.5672.63.orig/third_party/vulkan-deps/vulkan-validation-layers/src/layers/external/vma/vk_mem_alloc.h chromium-113.0.5672.63/third_party/vulkan-deps/vulkan-validation-layers/src/layers/external/vma/vk_mem_alloc.h
--- chromium-113.0.5672.63.orig/third_party/vulkan-deps/vulkan-validation-layers/src/layers/external/vma/vk_mem_alloc.h	2023-04-26 21:47:38.323831800 +0000
+++ chromium-113.0.5672.63/third_party/vulkan-deps/vulkan-validation-layers/src/layers/external/vma/vk_mem_alloc.h	2023-05-09 00:14:09.757555808 +0000
@@ -2579,6 +2579,7 @@ VMA_CALL_PRE void VMA_CALL_POST vmaFreeS
 #undef VMA_IMPLEMENTATION
 
 #include <cstdint>
+#include <cstdio>
 #include <cstdlib>
 #include <cstring>
 #include <utility>
diff -upr chromium-113.0.5672.63.orig/third_party/vulkan_memory_allocator/include/vk_mem_alloc.h chromium-113.0.5672.63/third_party/vulkan_memory_allocator/include/vk_mem_alloc.h
--- chromium-113.0.5672.63.orig/third_party/vulkan_memory_allocator/include/vk_mem_alloc.h	2023-04-26 21:47:36.679686500 +0000
+++ chromium-113.0.5672.63/third_party/vulkan_memory_allocator/include/vk_mem_alloc.h	2023-05-09 00:14:41.046248921 +0000
@@ -2389,6 +2389,7 @@ VMA_CALL_PRE void VMA_CALL_POST vmaFreeV
 #undef VMA_IMPLEMENTATION
 
 #include <cstdint>
+#include <cstdio>
 #include <cstdlib>
 #include <cstring>
 #include <utility>
