--- chromium-104.0.5112.57/build/config/compiler/BUILD.gn.orig	2022-07-26 08:11:54.192788877 +0000
+++ chromium-104.0.5112.57/build/config/compiler/BUILD.gn	2022-07-26 08:12:10.019328155 +0000
@@ -1540,15 +1540,6 @@ config("default_warnings") {
         cflags += [ "-Wno-unqualified-std-cast-call" ]
       }
 
-      if (!is_nacl && !(is_chromeos ||
-                        default_toolchain == "//build/toolchain/cros:target")) {
-        # TODO(https://crbug.com/1322823): Remove flags once potential miscompile is investigated.
-        cflags += [
-          "-Xclang",
-          "-no-opaque-pointers",
-        ]
-      }
-
       if (is_fuchsia) {
         # TODO(https://bugs.chromium.org/p/fuchsia/issues/detail?id=77383)
         cflags += [ "-Wno-deprecated-copy" ]
