diff --git a/skia/BUILD.gn b/skia/BUILD.gn
index b2c8b79..7e2ee53 100644
--- a/skia/BUILD.gn
+++ b/skia/BUILD.gn
@@ -139,7 +139,7 @@ config("skia_library_config") {
   # Settings for text blitting, chosen to approximate the system browser.
   if (is_linux || is_chromeos) {
     defines += [
-      "SK_GAMMA_EXPONENT=1.2",
+      "SK_GAMMA_EXPONENT=1.8",
       "SK_GAMMA_CONTRAST=0.2",
     ]
   } else if (is_android) {
