--- a/ui/gl/gl_image_native_pixmap.cc	2020-05-18 11:40:06.000000000 -0700
+++ b/ui/gl/gl_image_native_pixmap.cc	2020-05-22 02:07:16.007770442 -0700
@@ -288,6 +288,8 @@
                                std::move(scoped_fd));
   }
 
+  handle.planes[0].size = size_.GetArea();
+
   return handle;
 #endif  // !defined(OS_FUCHSIA)
 }
--- a/gpu/command_buffer/service/error_state.cc	2020-05-18 11:39:22.000000000 -0700
+++ b/gpu/command_buffer/service/error_state.cc	2020-05-22 13:43:09.181180388 -0700
@@ -115,6 +115,8 @@
     // buffer.
     error = GL_NO_ERROR;
   }
+  if (error == GL_INVALID_ENUM)
+    error = GL_NO_ERROR;
   return error;
 }
 
