diff -Naur rsync-3.1.3/options.c rsync-3.1.3.new/options.c
--- rsync-3.1.3/options.c	2018-01-15 11:55:07.000000000 +0800
+++ rsync-3.1.3.new/options.c	2018-03-24 11:45:36.047079032 +0800
@@ -127,6 +127,7 @@
 long block_size = 0; /* "long" because popt can't set an int32. */
 char *skip_compress = NULL;
 item_list dparam_list = EMPTY_ITEM_LIST;
+int noatime = 0;
 
 /** Network address family. **/
 int default_af_hint
@@ -804,6 +805,7 @@
   rprintf(F,"     --iconv=CONVERT_SPEC    request charset conversion of filenames\n");
 #endif
   rprintf(F,"     --checksum-seed=NUM     set block/file checksum seed (advanced)\n");
+  rprintf(F,"     --noatime               do not alter atime when opening source files\n");
   rprintf(F," -4, --ipv4                  prefer IPv4\n");
   rprintf(F," -6, --ipv6                  prefer IPv6\n");
   rprintf(F,"     --version               print version number\n");
@@ -1024,6 +1026,7 @@
   {"iconv",            0,  POPT_ARG_STRING, &iconv_opt, 0, 0, 0 },
   {"no-iconv",         0,  POPT_ARG_NONE,   0, OPT_NO_ICONV, 0, 0 },
 #endif
+  {"noatime",          0,  POPT_ARG_VAL,    &noatime, 1, 0, 0 },
   {"ipv4",            '4', POPT_ARG_VAL,    &default_af_hint, AF_INET, 0, 0 },
   {"ipv6",            '6', POPT_ARG_VAL,    &default_af_hint, AF_INET6, 0, 0 },
   {"8-bit-output",    '8', POPT_ARG_VAL,    &allow_8bit_chars, 1, 0, 0 },
@@ -2800,6 +2803,12 @@
 	if (preallocate_files && am_sender)
 		args[ac++] = "--preallocate";
 
+	/*
+	 * Do we want remote atime preservation when we preserve local ones?
+	if (noatime)
+		args[ac++] = "--noatime";
+	*/
+
 	if (ac > MAX_SERVER_ARGS) { /* Not possible... */
 		rprintf(FERROR, "argc overflow in server_options().\n");
 		exit_cleanup(RERR_MALLOC);
diff -Naur rsync-3.1.3/rsync.yo rsync-3.1.3.new/rsync.yo
--- rsync-3.1.3/rsync.yo	2018-01-29 07:24:27.000000000 +0800
+++ rsync-3.1.3.new/rsync.yo	2018-03-24 11:45:36.047079032 +0800
@@ -458,6 +458,7 @@
      --protocol=NUM          force an older protocol version to be used
      --iconv=CONVERT_SPEC    request charset conversion of filenames
      --checksum-seed=NUM     set block/file checksum seed (advanced)
+     --noatime               do not alter atime when opening source files
  -4, --ipv4                  prefer IPv4
  -6, --ipv6                  prefer IPv6
      --version               print version number
@@ -2657,6 +2658,13 @@
 regardless of the remote charset you actually pass.  Thus, you may feel free to
 specify just the local charset for a daemon transfer (e.g. bf(--iconv=utf8)).
 
+dit(bf(--noatime)) Use the O_NOATIME open flag on systems that support it.
+The effect of this flag is to avoid altering the access time (atime) of the
+opened files.
+If the system does not support the O_NOATIME flag, this option does nothing.
+Currently, systems known to support O_NOATIME are Linux >= 2.6.8 with glibc
+>= 2.3.4.
+
 dit(bf(-4, --ipv4) or bf(-6, --ipv6)) Tells rsync to prefer IPv4/IPv6
 when creating sockets.  This only affects sockets that rsync has direct
 control over, such as the outgoing socket when directly contacting an
diff -Naur rsync-3.1.3/syscall.c rsync-3.1.3.new/syscall.c
--- rsync-3.1.3/syscall.c	2018-01-15 11:55:07.000000000 +0800
+++ rsync-3.1.3.new/syscall.c	2018-03-24 11:45:36.048079042 +0800
@@ -42,6 +42,7 @@
 extern int preallocate_files;
 extern int preserve_perms;
 extern int preserve_executability;
+extern int noatime;
 
 #ifndef S_BLKSIZE
 # if defined hpux || defined __hpux__ || defined __hpux
@@ -201,6 +202,10 @@
 		RETURN_ERROR_IF(dry_run, 0);
 		RETURN_ERROR_IF_RO_OR_LO;
 	}
+#ifdef O_NOATIME
+	if (noatime)
+		flags |= O_NOATIME;
+#endif
 
 	return open(pathname, flags | O_BINARY, mode);
 }
diff -Naur rsync-3.1.3/tls.c rsync-3.1.3.new/tls.c
--- rsync-3.1.3/tls.c	2018-01-15 11:55:07.000000000 +0800
+++ rsync-3.1.3.new/tls.c	2018-03-26 18:17:20.251567189 +0800
@@ -53,6 +53,7 @@
 int preserve_executability = 0;
 int preallocate_files = 0;
 int inplace = 0;
+int noatime = 0;
 
 #ifdef SUPPORT_XATTRS
 
diff -Naur rsync-3.1.3/trimslash.c rsync-3.1.3.new/trimslash.c
--- rsync-3.1.3/trimslash.c	2018-01-15 11:55:07.000000000 +0800
+++ rsync-3.1.3.new/trimslash.c	2018-03-26 10:47:06.539752561 +0800
@@ -30,6 +30,7 @@
 int preserve_executability = 0;
 int preallocate_files = 0;
 int inplace = 0;
+int noatime = 0;
 
 int
 main(int argc, char **argv)
diff -Naur rsync-3.1.3/t_unsafe.c rsync-3.1.3.new/t_unsafe.c
--- rsync-3.1.3/t_unsafe.c	2018-01-15 11:55:07.000000000 +0800
+++ rsync-3.1.3.new/t_unsafe.c	2018-03-26 18:17:41.441800221 +0800
@@ -32,6 +32,7 @@
 int preserve_perms = 0;
 int preserve_executability = 0;
 short info_levels[COUNT_INFO], debug_levels[COUNT_DEBUG];
+int noatime = 0;
 
 int
 main(int argc, char **argv)
