diff --git a/Makefile b/Makefile
index a473d64..0e64ea6 100644
--- a/Makefile
+++ b/Makefile
@@ -11,7 +11,7 @@ OUT_DIR := output
 OUT_FILE := $(OUT_DIR)/hustmirror-cli
 OUT_MIRROR_DIR := $(OUT_DIR)/mirrors
 OUT_MIRROR_FILES := $(patsubst $(MIRROR_DIR)/%,$(OUT_MIRROR_DIR)/%,$(MIRROR_FILES))
-INSTALL_DIR := /usr/local/bin/
+INSTALL_DIR := /usr/bin/
 
 all: $(OUT_FILE)
 	@echo "Done, object script is $(OUT_FILE)."
@@ -20,7 +20,8 @@ test: $(OUT_FILE)
 	@tests/test.sh
 
 install: $(OUT_FILE)
-	@cp $(OUT_FILE) $(INSTALL_DIR)
+	@mkdir -p $(DESTDIR)usr/bin
+	@cp $(OUT_FILE) $(DESTDIR)usr/bin
 
 $(OUT_FILE): $(TEMPLATE_FILE) $(OUT_MIRROR_FILES) $(INCLUDE_FILES) $(OTHER_FILES)
 	@mkdir -p $(OUT_DIR)
