diff -rupN orig/bindings/Makefile.am patched/bindings/Makefile.am
--- orig/bindings/Makefile.am	2014-11-10 17:29:35.000000000 +0100
+++ patched/bindings/Makefile.am	2016-02-03 15:05:47.040153479 +0100
@@ -3,7 +3,6 @@ SUBDIRS =	\
 	vala
 else
 SUBDIRS =		\
-	python		\
 	vala
 endif
 
diff -rupN orig/bindings/Makefile.in patched/bindings/Makefile.in
--- orig/bindings/Makefile.in	2016-02-03 15:04:25.949772207 +0100
+++ patched/bindings/Makefile.in	2016-02-03 15:06:18.366965150 +0100
@@ -157,7 +157,7 @@ am__define_uniq_tagged_files = \
   done | $(am__uniquify_input)`
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = python vala mono
+DIST_SUBDIRS = vala mono
 am__DIST_COMMON = $(srcdir)/Makefile.in
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 am__relativize = \
@@ -368,7 +368,7 @@ target_alias = @target_alias@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-@USE_GTK3_FALSE@SUBDIRS = python vala $(am__append_1)
+@USE_GTK3_FALSE@SUBDIRS = vala $(am__append_1)
 @USE_GTK3_TRUE@SUBDIRS = vala $(am__append_1)
 all: all-recursive
 
diff -rupN orig/configure patched/configure
--- orig/configure	2016-02-03 15:04:24.733099750 +0100
+++ patched/configure	2016-02-03 15:11:15.358290879 +0100
@@ -14480,17 +14480,7 @@ save_CPPFLAGS="$CPPFLAGS"
 CPPFLAGS="$CPPFLAGS $PYTHON_INCLUDES"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-#include <Python.h>
 _ACEOF
-if ac_fn_c_try_cpp "$LINENO"; then :
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: found" >&5
-$as_echo "found" >&6; }
-
-else
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: not found" >&5
-$as_echo "not found" >&6; }
-as_fn_error $? "could not find Python headers" "$LINENO" 5
-fi
 rm -f conftest.err conftest.i conftest.$ac_ext
 CPPFLAGS="$save_CPPFLAGS"
 
@@ -14934,7 +14924,7 @@ fi
 # Files
 ###########################
 
-ac_config_files="$ac_config_files Makefile src/Makefile src/appindicator-0.1.pc src/appindicator3-0.1.pc bindings/Makefile bindings/python/Makefile bindings/python/appindicator.override bindings/vala/Makefile bindings/vala/examples/Makefile tests/Makefile example/Makefile docs/Makefile docs/reference/Makefile docs/reference/version.xml docs/reference/libappindicator-docs.sgml"
+ac_config_files="$ac_config_files Makefile src/Makefile src/appindicator-0.1.pc src/appindicator3-0.1.pc bindings/Makefile bindings/vala/Makefile bindings/vala/examples/Makefile tests/Makefile example/Makefile docs/Makefile docs/reference/Makefile docs/reference/version.xml docs/reference/libappindicator-docs.sgml"
 
 
 if test "x$has_mono" = "xtrue" ; then
@@ -16031,8 +16021,6 @@ do
     "src/appindicator-0.1.pc") CONFIG_FILES="$CONFIG_FILES src/appindicator-0.1.pc" ;;
     "src/appindicator3-0.1.pc") CONFIG_FILES="$CONFIG_FILES src/appindicator3-0.1.pc" ;;
     "bindings/Makefile") CONFIG_FILES="$CONFIG_FILES bindings/Makefile" ;;
-    "bindings/python/Makefile") CONFIG_FILES="$CONFIG_FILES bindings/python/Makefile" ;;
-    "bindings/python/appindicator.override") CONFIG_FILES="$CONFIG_FILES bindings/python/appindicator.override" ;;
     "bindings/vala/Makefile") CONFIG_FILES="$CONFIG_FILES bindings/vala/Makefile" ;;
     "bindings/vala/examples/Makefile") CONFIG_FILES="$CONFIG_FILES bindings/vala/examples/Makefile" ;;
     "tests/Makefile") CONFIG_FILES="$CONFIG_FILES tests/Makefile" ;;
