--- a/configure.ac      2016-03-22 09:36:03.000000000 +0100
+++ b/configure.ac      2016-03-26 18:35:44.697586161 +0100
@@ -946,10 +946,10 @@
        PKG_CHECK_MODULES(systemd, [libsystemd >= 209],
                [AC_SUBST(systemd_CFLAGS)
                 AC_SUBST(systemd_LIBS)],
-               [PKG_CHECK_MODULES(systemd_daemon, [libsystemd-daemon])
+               [PKG_CHECK_MODULES(systemd_daemon, [libsystemd])
                 AC_SUBST(systemd_daemon_CFLAGS)
                 AC_SUBST(systemd_daemon_LIBS)
-                PKG_CHECK_MODULES(systemd_journal, [libsystemd-journal])
+                PKG_CHECK_MODULES(systemd_journal, [libsystemd])
                 AC_SUBST(systemd_journal_CFLAGS)
                 AC_SUBST(systemd_journal_LIBS)]
        )

