--- a/conf/zabbix_agent.conf
+++ b/conf/zabbix_agent.conf
@@ -38,7 +38,7 @@
 # Include=
 
 # Include=/usr/local/etc/zabbix_agent.userparams.conf
-# Include=/usr/local/etc/zabbix_agent.conf.d/
+Include=/etc/zabbix/zabbix_agent.conf.d/
 
 ####### USER-DEFINED MONITORED PARAMETERS #######
 
--- a/conf/zabbix_agentd.conf
+++ b/conf/zabbix_agentd.conf
@@ -8,7 +8,7 @@
 #
 # Mandatory: no
 # Default:
-# PidFile=/tmp/zabbix_agentd.pid
+PidFile=/run/zabbix/zabbix_agentd.pid
 
 ### Option: LogFile
 #	Name of log file.
@@ -18,7 +18,7 @@
 # Default:
 # LogFile=
 
-LogFile=/tmp/zabbix_agentd.log
+LogFile=/var/log/zabbix/zabbix_agentd.log
 
 ### Option: LogFileSize
 #	Maximum size of log file in MB.
@@ -213,7 +213,7 @@
 # Include=
 
 # Include=/usr/local/etc/zabbix_agentd.userparams.conf
-# Include=/usr/local/etc/zabbix_agentd.conf.d/
+Include=/etc/zabbix/zabbix_agentd.conf.d/
 
 ####### USER-DEFINED MONITORED PARAMETERS #######
 
--- a/conf/zabbix_server.conf
+++ b/conf/zabbix_server.conf
@@ -36,7 +36,7 @@
 # Default:
 # LogFile=
 
-LogFile=/tmp/zabbix_server.log
+LogFile=/var/log/zabbix/zabbix_server.log
 
 ### Option: LogFileSize
 #	Maximum size of log file in MB.
@@ -65,7 +65,7 @@
 #
 # Mandatory: no
 # Default:
-# PidFile=/tmp/zabbix_server.pid
+PidFile=/run/zabbix/zabbix_server.pid
 
 ### Option: DBHost
 #	Database host name.
@@ -115,7 +115,7 @@
 #
 # Mandatory: no
 # Default:
-# DBSocket=/tmp/mysql.sock
+DBSocket=/run/mysqld/mysqld.sock
 
 ### Option: DBPort
 #	Database port when not using local socket. Ignored for SQLite.
@@ -448,7 +448,7 @@
 # Include=
 
 # Include=/usr/local/etc/zabbix_server.general.conf
-# Include=/usr/local/etc/zabbix_server.conf.d/
+Include=/etc/zabbix/zabbix_server.conf.d/
 
 ### Option: StartProxyPollers
 #	Number of pre-forked instances of pollers for passive proxies.

--- a/man/zabbix_agentd.man
+++ b/man/zabbix_agentd.man
@@ -21,7 +21,7 @@
 Output version information and exit.
 .SH FILES
 .TP
-.I /usr/local/etc/zabbix_agentd.conf
+.I /etc/zabbix/zabbix_agentd.conf
 Default location of Zabbix agent configuration file (if not modified during compile time).
 .SH "SEE ALSO"
 .BR zabbix_get (8),

--- a/man/zabbix_sender.man
+++ b/man/zabbix_sender.man
@@ -39,7 +39,7 @@
 .IP "-V, --version"
 Output version information and exit.
 .SH "EXAMPLES"
-.B zabbix_sender -c /usr/local/etc/zabbix_agentd.conf -s """Monitored Host""" -k mysql.queries -o 342.45
+.B zabbix_sender -c /etc/zabbix/zabbix_agentd.conf -s """Monitored Host""" -k mysql.queries -o 342.45
 
 Send \fB342.45\fR as the value for \fBmysql.queries\fR key in \fBMonitored Host\fR host using Zabbix server defined in agent daemon configuration file.
 
@@ -47,7 +47,7 @@
 
 Send values from file \fBdata_values.txt\fR to server with IP \fB192.168.1.113\fR. Host names and keys are defined in the file.
 
-.B echo """- hw.serial.number 1287872261 SQ4321ASDF""" | zabbix_sender -c /usr/local/etc/zabbix_agentd.conf -T -i -
+.B echo """- hw.serial.number 1287872261 SQ4321ASDF""" | zabbix_sender -c /etc/zabbix/zabbix_agentd.conf -T -i -
 
 Send a timestamped value from the commandline to Zabbix server, specified in the agent daemon configuration file. Dash in the input data indicates that hostname also should be used from the same configuration file.
 
--- a/man/zabbix_server.man
+++ b/man/zabbix_server.man
@@ -30,7 +30,7 @@
 Output version information and exit.
 .SH FILES
 .TP
-.I /usr/local/etc/zabbix_server.conf
+.I /etc/zabbix/zabbix_server.conf
 Default location of Zabbix server configuration file (if not modified during compile time).
 .SH "SEE ALSO"
 .BR zabbix_agentd (8),
