Description: Remove linking of unused libraries
Author: Nicholas Guriev <guriev-ns@ya.ru>
Last-Update: 2017-04-05

diff --git a/Telegram/gyp/Telegram.gyp b/Telegram/gyp/Telegram.gyp
--- a/Telegram/gyp/Telegram.gyp
+++ b/Telegram/gyp/Telegram.gyp
@@ -78,7 +78,6 @@
       'codegen.gyp:codegen_style',
       'codegen.gyp:codegen_numbers',
       'codegen.gyp:MetaLang',
-      'utils.gyp:Updater',
     ],
 
     'defines': [
diff --git a/Telegram/gyp/qt.gypi b/Telegram/gyp/qt.gypi
--- a/Telegram/gyp/qt.gypi
+++ b/Telegram/gyp/qt.gypi
@@ -32,13 +32,9 @@
             ],
           },
           'qt_libs': [
-            'qwebp',
-            'Qt5PrintSupport',
-            'Qt5PlatformSupport',
             'Qt5Network',
             'Qt5Widgets',
             'Qt5Gui',
-            'qtharfbuzzng',
           ],
           'qt_version%': '<(qt_version)',
           'conditions': [
@@ -89,28 +86,8 @@
             'qt_lib_debug_postfix': '',
             'qt_lib_release_postfix': '',
             'qt_libs': [
-              'qxcb',
-              'Qt5XcbQpa',
-              'qconnmanbearer',
-              'qgenericbearer',
-              'qnmbearer',
               '<@(qt_libs)',
-              'Qt5DBus',
               'Qt5Core',
-              'qtpcre',
-              'Xi',
-              'Xext',
-              'Xfixes',
-              'SM',
-              'ICE',
-              'fontconfig',
-              'expat',
-              'freetype',
-              'z',
-              'xcb-shm',
-              'xcb-xfixes',
-              'xcb-render',
-              'xcb-static',
             ],
           }],
         ],
@@ -139,11 +117,6 @@
     # '<!@(python <(DEPTH)/list_sources.py [sources] <(qt_moc_list_sources_arg))'
     # where [sources] contains all your source files
     'qt_moc_list_sources_arg': '--moc-prefix SHARED_INTERMEDIATE_DIR/<(_target_name)/moc/moc_',
-
-    'linux_path_xkbcommon%': '/usr/local',
-    'linux_lib_ssl%': 'ssl',
-    'linux_lib_crypto%': 'crypto',
-    'linux_lib_icu%': 'icutu icui18n icuuc icudata',
   },
 
   'configurations': {
@@ -221,12 +194,7 @@ icudata.a',
       'libraries': [
         '<@(qt_libs_release)',
         'crypto',
-        'xcb',
         'X11',
-        'X11-xcb',
-        'dbus-1',
-        'dl',
-        'gthread-2.0',
         'glib-2.0',
         'pthread',
       ],
diff --git a/Telegram/gyp/telegram_linux.gypi b/Telegram/gyp/telegram_linux.gypi
--- a/Telegram/gyp/telegram_linux.gypi
+++ b/Telegram/gyp/telegram_linux.gypi
@@ -50,19 +50,12 @@
       '<(linux_path_breakpad)/lib',
     ],
     'libraries': [
-      'lzma',
       'openal',
       'avformat',
       'avcodec',
       'swresample',
       'swscale',
       'avutil',
-      'opus',
-      'va-x11',
-      'va-drm',
-      'va',
-      'vdpau',
-      'drm',
       'z',
       'minizip',
 #      '<!(pkg-config 2> /dev/null --libs <@(pkgconfig_libs))',
