diff -U3 -r fvwm/conditional.c fvwm/conditional.c
--- fvwm/conditional.c	2007-10-06 11:17:09.000000000 +0200
+++ fvwm/conditional.c	2011-12-09 19:01:48.868307776 +0100
@@ -600,6 +600,36 @@
 			SET_HAS_HANDLES(mask, on);
 			SETM_HAS_HANDLES(mask, 1);
 		}
+		else if (StrEquals(cond, "HasTitle"))
+		{
+			SET_HAS_TITLE(mask, on);
+			SETM_HAS_TITLE(mask, 1);
+		}
+		else if (StrEquals(cond, "HasBorders"))
+		{
+			SET_HAS_NO_BORDER(mask, !on);
+			SETM_HAS_NO_BORDER(mask, 1);
+		}
+		else if (StrEquals(cond, "TitleAtBottom"))
+		{
+			SET_TITLE_DIR(mask, DIR_S);
+			SETM_TITLE_DIR(mask, 1);
+		}
+		else if (StrEquals(cond, "TitleAtTop"))
+		{
+			SET_TITLE_DIR(mask, DIR_N);
+			SETM_TITLE_DIR(mask, 1);
+		}
+		else if (StrEquals(cond, "TitleAtLeft"))
+		{
+			SET_TITLE_DIR(mask, DIR_W);
+			SETM_TITLE_DIR(mask, 1);
+		}
+		else if (StrEquals(cond, "TitleAtRight"))
+		{
+			SET_TITLE_DIR(mask, DIR_E);
+			SETM_TITLE_DIR(mask, 1);
+		}
 		else if (StrEquals(cond,"Iconifiable"))
 		{
 			SET_IS_UNICONIFIABLE(mask, !on);
Seulement dans libs: PictureGraphics.c.orig
