diff -U3 -r fvwm/builtins.c fvwm/builtins.c
--- fvwm/builtins.c	2011-12-09 20:53:09.537508019 +0100
+++ fvwm/builtins.c	2011-12-09 21:00:47.315799724 +0100
@@ -509,6 +509,21 @@
 			if (action)
 				action += next;
 		}
+		else if (!do_add && StrEquals(parm,"buttonwidth"))
+		{
+			int width = 0;
+			int next = 0;
+
+			sscanf(action, "%d%n", &width, &next);
+
+			if (decor->button_width != width)
+			{
+				decor->button_width = width;
+				decor->flags.has_changed = 1;
+			}
+			if (action)
+				action += next;
+		}
 		else if (!do_add && StrEquals(parm,"MinHeight"))
 		{
 			int height = 0;
