Index: Makefile
===================================================================
--- Makefile	(revision 715)
+++ Makefile	(working copy)
@@ -172,7 +172,7 @@
 	rm -f spim spim.exe *.o TAGS test.out lex.yy.cpp parser_yacc.cpp parser_yacc.h y.output
 
 install: spim
-	install spim $(BIN_DIR)/spim
+	install -D spim $(BIN_DIR)/spim
 	install -d $(EXCEPTION_DIR)
 	install -m 0444 $(CPU_DIR)/exceptions.s $(EXCEPTION_DIR)/exceptions.s
 
