--- a/scripts/setup-seafile-mysql.py
+++ b/scripts/setup-seafile-mysql.py
@@ -304,14 +304,13 @@
-        self.bin_dir = os.path.join(self.install_path, 'seafile', 'bin')
+        self.bin_dir = '/usr/bin'
         self.central_config_dir = os.path.join(self.top_dir, 'conf')
         Utils.must_mkdir(self.central_config_dir)

     def check_pre_condiction(self):
         def error_if_not_exists(path):
             if not os.path.exists(path):
                 Utils.error('"%s" not found' % path)

         paths = [
-            os.path.join(self.install_path, 'seafile'),
             os.path.join(self.install_path, 'seahub'),
             os.path.join(self.install_path, 'runtime'),
         ]
--- a/tools/seafile-admin
+++ b/tools/seafile-admin
@@ -864,3 +864,10 @@
+    # seahub.db is useless with MySQL and crash the startup!
+    with open(files[3], 'r') as f:
+        for line in f:
+            if 'DATABASES' in line:
+                del(files[2])
+                break
+
     for fpath in files:
         if not os.path.exists(fpath):
             error('%s not found' % fpath)
