From 067973c193f78729d4c3eb55d2132ce9c49f8a97 Mon Sep 17 00:00:00 2001
From: Caleb Maclennan <caleb@alerque.com>
Date: Thu, 23 Apr 2020 22:52:39 +0300
Subject: [PATCH] patch-args

Signed-off-by: Caleb Maclennan <caleb@alerque.com>
---
 src/main/cli/index.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main/cli/index.js b/src/main/cli/index.js
index d306c409..1f07087e 100644
--- a/src/main/cli/index.js
+++ b/src/main/cli/index.js
@@ -10,7 +10,7 @@ const write = s => process.stdout.write(s)
 const writeLine = s => write(s + '\n')
 
 const cli = () => {
-  let argv = process.argv.slice(1)
+  let argv = process.argv.slice(2)
   if (process.env.NODE_ENV === 'development') {
     // Don't pass electron development arguments to Mark Text and change user data path.
     argv = ['--user-data-dir', path.join(getPath('appData'), 'marktext-dev')]
-- 
2.26.2

