diff -ruN a/doc/example.conf.in b/doc/example.conf.in
--- a/doc/example.conf.in	2021-12-09 08:55:44.000000000 +0100
+++ b/doc/example.conf.in	2021-12-11 23:42:13.542310886 +0100
@@ -533,7 +533,7 @@
 	# with several entries, one file per entry.
 	# Zone file format, with DS and DNSKEY entries.
 	# Note this gets out of date, use auto-trust-anchor-file please.
-	# trust-anchor-file: ""
+	trust-anchor-file: "/etc/unbound/trusted-key.key"
 
 	# Trusted key for validation. DS or DNSKEY. specify the RR on a
 	# single line, surrounded by "". TTL is ignored. class is IN default.
