--- a/build/lib/git-environment.js
+++ b/build/lib/git-environment.js
@@ -18,7 +18,7 @@
         return path.resolve(process.env.LOCAL_GIT_DIRECTORY);
     }
     else {
-        return resolveEmbeddedGitDir();
+        return '/usr';
     }
 }
 /**
@@ -100,9 +100,9 @@
         if (!env.GIT_SSL_CAINFO && !env.LOCAL_GIT_DIRECTORY) {
             // use the SSL certificate bundle included in the distribution only
             // when using embedded Git and not providing your own bundle
-            const distDir = resolveEmbeddedGitDir();
-            const sslCABundle = `${distDir}/ssl/cacert.pem`;
-            env.GIT_SSL_CAINFO = sslCABundle;
+            // const distDir = resolveEmbeddedGitDir();
+            // const sslCABundle = `${distDir}/ssl/cacert.pem`;
+            // env.GIT_SSL_CAINFO = sslCABundle;
         }
     }
     return { env, gitLocation };
