commit dabb3237db4112abde0020cbb0b6a80f477e6aee
Author: Keeyou <keeyou-cn@outlook.com>
Date:   Sat May 11 10:55:46 2024 +0800

    gperftools: set benchmark to off (trivial)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a36248d9..98942410 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2411,12 +2411,12 @@ if (USE_SYSTEM_TCMALLOC AND USE_TCMALLOC)
 elseif (USE_TCMALLOC)
   message(STATUS "Compiling with bundled tcmalloc support")
 
-  set(GPERFTOOLS_BUILD_DEBUGALLOC OFF)
-  set(gperftools_build_minimal ON)
-  set(gperftools_build_benchmark OFF)
-  set(GPERFTOOLS_BUILD_STATIC ON)
-  set(GPERFTOOLS_ENABLE_INSTALL OFF)
-  set(GPERFTOOLS_BUILD_TESTING OFF)
+  set(GPERFTOOLS_BUILD_DEBUGALLOC OFF CACHE BOOL "")
+  set(gperftools_build_minimal ON CACHE BOOL "")
+  set(gperftools_build_benchmark OFF CACHE BOOL "")
+  set(GPERFTOOLS_BUILD_STATIC ON CACHE BOOL "")
+  set(GPERFTOOLS_ENABLE_INSTALL OFF CACHE BOOL "")
+  set(GPERFTOOLS_BUILD_TESTING OFF CACHE BOOL "")
   add_subdirectory(third_party/gperftools)
 
   set_target_properties(tcmalloc_minimal PROPERTIES EXCLUDE_FROM_ALL TRUE)
