diff --git a/third_party/libc++/trunk/include/__type_traits/remove_pointer.h b/third_party/libc++/trunk/include/__type_traits/remove_pointer.h
index 54390a193..c01eb3c37 100644
--- a/third_party/libc++/trunk/include/__type_traits/remove_pointer.h
+++ b/third_party/libc++/trunk/include/__type_traits/remove_pointer.h
@@ -17,7 +17,7 @@

 _LIBCPP_BEGIN_NAMESPACE_STD

-#if !defined(_LIBCPP_WORKAROUND_OBJCXX_COMPILER_INTRINSICS) && __has_builtin(__remove_pointer)
+#if !defined(_LIBCPP_WORKAROUND_OBJCXX_COMPILER_INTRINSICS) && __has_builtin(__remove_pointer) && !defined(_LIBCPP_GCC_VER)
 template <class _Tp>
 struct remove_pointer {
   using type _LIBCPP_NODEBUG = __remove_pointer(_Tp);
