--- VTK/ThirdParty/netcdf/vtknetcdf/CMakeLists.txt	2015-10-01 16:27:10.529402587 +0200
+++ VTK/ThirdParty/netcdf/vtknetcdf/CMakeLists.txt.new	2015-10-01 16:26:47.586013737 +0200
@@ -229,6 +229,7 @@
 SET(CXX NETCDF_ENABLE_CXX)
 
 # NetCDF4 needs HDF5 HL support. Check that's available.
+find_package(HDF5 REQUIRED)
 set (USE_NETCDF4 OFF)
 if (NOT VTK_USE_SYSTEM_HDF5)
   # using VTK's HDF5, we always build that with HL support.
