diff --git a/Applications/ParaView/paraview.desktop.in b/Applications/ParaView/paraview.desktop.in
index 8b224f5..d1da219 100644
--- a/Applications/ParaView/paraview.desktop.in
+++ b/Applications/ParaView/paraview.desktop.in
@@ -1,7 +1,8 @@
 [Desktop Entry]
-Version=@PARAVIEW_VERSION_FULL@
+Version=1.0
 Type=Application
 Name=ParaView
 Comment=Parallel visualization application
+Exec=paraview
 TryExec=paraview
 Icon=paraview
