From cc57ec29c9d0e1891036f1bc0baee71056606469 Mon Sep 17 00:00:00 2001
From: kxxt <rsworktech@outlook.com>
Date: Fri, 20 Jun 2025 07:06:39 +0800
Subject: [PATCH] Disable static linking of libstdc++

---
 Common/base.pri | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Common/base.pri b/Common/base.pri
index 8e240dfcf5..b5702a17ef 100644
--- a/Common/base.pri
+++ b/Common/base.pri
@@ -459,7 +459,7 @@ enable_cpp_17 {
 core_linux {
 core_static_link_libstd {
 	!core_linux_clang {
-		QMAKE_LFLAGS += -static-libstdc++ -static-libgcc
+		# QMAKE_LFLAGS += -static-libstdc++ -static-libgcc
 	} else {
 		# TODO: add libc++abi?
 	}
-- 
2.49.0

