From 624c69e145830d615d0fafbbaf40b7114464b233 Mon Sep 17 00:00:00 2001
From: kxxt <rsworktech@outlook.com>
Date: Fri, 26 Jul 2024 10:19:36 +0800
Subject: [PATCH 1/4] Add --update-only to avoid download and build at once

---
 make.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/make.py b/make.py
index 9c52800..1fe0cb0 100755
--- a/make.py
+++ b/make.py
@@ -23,6 +23,7 @@ base.check_python()
 
 parser = argparse.ArgumentParser(description="options")
 parser.add_argument("--build-only-branding", action="store_true")
+parser.add_argument("--update-only", action="store_true")
 args = parser.parse_args()
 
 if (args.build_only_branding):
@@ -65,9 +66,11 @@ config.parse_defaults()
 base.check_build_version(base_dir)
 
 # update
-if ("1" == config.option("update")):
+if ("1" == config.option("update")) or args.update_only:
   repositories = base.get_repositories()
   base.update_repositories(repositories)
+  if args.update_only:
+    exit(0)
 
 base.configure_common_apps()
 
-- 
2.45.2

