From 0eb78a67a79b10c7f95169e23b305161d78b5327 Mon Sep 17 00:00:00 2001
From: kxxt <rsworktech@outlook.com>
Date: Fri, 26 Jul 2024 10:46:58 +0800
Subject: [PATCH 2/4] Add --no-third-party-update and --update-third-party-only

---
 make.py | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/make.py b/make.py
index 1fe0cb0..82e62c5 100755
--- a/make.py
+++ b/make.py
@@ -24,6 +24,9 @@ base.check_python()
 parser = argparse.ArgumentParser(description="options")
 parser.add_argument("--build-only-branding", action="store_true")
 parser.add_argument("--update-only", action="store_true")
+group = parser.add_mutually_exclusive_group()
+group.add_argument("--update-third-party-only", action="store_true", help="Only update third party dependencies")
+group.add_argument("--no-third-party-update", action="store_true", help="Don't update third party dependencies")
 args = parser.parse_args()
 
 if (args.build_only_branding):
@@ -65,6 +68,11 @@ config.parse_defaults()
 
 base.check_build_version(base_dir)
 
+if args.update_third_party_only:
+  # core 3rdParty
+  make_common.make()
+  exit(0)
+
 # update
 if ("1" == config.option("update")) or args.update_only:
   repositories = base.get_repositories()
@@ -84,8 +92,9 @@ if ("1" == base.get_env("OO_ONLY_BUILD_JS")):
 
 #base.check_tools()
 
-# core 3rdParty
-make_common.make()
+if not args.no_third_party_update:
+  # core 3rdParty
+  make_common.make()
 
 # build updmodule for desktop (only for windows version)
 if config.check_option("module", "desktop"):
-- 
2.45.2

