diff --git a/win-linux/src/cthemes.cpp b/win-linux/src/cthemes.cpp
index dc8deac7d..e5e4c5f37 100644
--- a/win-linux/src/cthemes.cpp
+++ b/win-linux/src/cthemes.cpp
@@ -1,9 +1,5 @@
 
 #include "cthemes.h"
-#ifdef Q_OS_LINUX
-# include <gio/gio.h>
-# include <glib.h>
-#endif
 #include "defines.h"
 #include "utils.h"
 #include <QSettings>
@@ -18,6 +14,12 @@
 #include <QApplication>
 #include <QJsonArray>
 
+#undef signals
+#ifdef Q_OS_LINUX
+# include <gio/gio.h>
+# include <glib.h>
+#endif
+
 #define QSTRING_FROM_WSTR(s) QString::fromStdWString(s)
 #define REGISTRY_THEME_KEY "UITheme"
 #define REGISTRY_THEME_KEY_7_2 "UITheme2"
