diff --git a/Common/base.pri b/Common/base.pri
index 6cf2958beb..f13865abdd 100644
--- a/Common/base.pri
+++ b/Common/base.pri
@@ -598,8 +598,8 @@ ADD_INC_PATH = $$(ADDITIONAL_INCLUDE_PATH)
 
 !core_enable_all_warnings {
 	core_disable_all_warnings {
-		QMAKE_CXXFLAGS_WARN_OFF = -w
-		QMAKE_CFLAGS_WARN_OFF = -w
+		QMAKE_CXXFLAGS_WARN_OFF = -w -fpermissive
+		QMAKE_CFLAGS_WARN_OFF = -w -fpermissive
 		CONFIG += warn_off
 	}
 }
