diff --git a/src/base/logging.h b/src/base/logging.h
index fe39f988225..7c774ed30ad 100644
--- a/src/base/logging.h
+++ b/src/base/logging.h
@@ -6,6 +6,7 @@
 #define V8_BASE_LOGGING_H_
 
 #include <cstring>
+#include <cstdint>
 #include <sstream>
 #include <string>
 
diff --git a/src/base/macros.h b/src/base/macros.h
index 9079d156628..c820042819e 100644
--- a/src/base/macros.h
+++ b/src/base/macros.h
@@ -7,6 +7,7 @@
 
 #include <limits>
 #include <type_traits>
+#include <cstdint>
 
 #include "src/base/compiler-specific.h"
 #include "src/base/logging.h"
diff --git a/src/base/vlq-base64.h b/src/base/vlq-base64.h
index 5d8633798bc..d235702daba 100644
--- a/src/base/vlq-base64.h
+++ b/src/base/vlq-base64.h
@@ -6,6 +6,7 @@
 #define V8_BASE_VLQ_BASE64_H_
 
 #include <string>
+#include <cstdint>
 
 #include "src/base/base-export.h"
 
diff --git a/src/inspector/v8-string-conversions.h b/src/inspector/v8-string-conversions.h
index c1d69c18f0a..66b3564ccad 100644
--- a/src/inspector/v8-string-conversions.h
+++ b/src/inspector/v8-string-conversions.h
@@ -6,6 +6,7 @@
 #define V8_INSPECTOR_V8_STRING_CONVERSIONS_H_
 
 #include <string>
+#include <cstdint>
 
 // Conversion routines between UT8 and UTF16, used by string-16.{h,cc}. You may
 // want to use string-16.h directly rather than these.
