diff -r -U 0 a/libraries/javacheck/CMakeLists.txt b/libraries/javacheck/CMakeLists.txt
--- a/libraries/javacheck/CMakeLists.txt	2019-08-05 00:46:59.000000000 +0200
+++ b/libraries/javacheck/CMakeLists.txt	2019-09-27 21:47:16.033545671 +0200
@@ -7 +7 @@
-set(CMAKE_JAVA_COMPILE_FLAGS -target 1.6 -source 1.6 -Xlint:deprecation -Xlint:unchecked)
+set(CMAKE_JAVA_COMPILE_FLAGS -target 8 -source 8 -Xlint:deprecation -Xlint:unchecked)
diff -r -U 0 a/libraries/launcher/CMakeLists.txt b/libraries/launcher/CMakeLists.txt
--- a/libraries/launcher/CMakeLists.txt	2019-08-05 00:46:59.000000000 +0200
+++ b/libraries/launcher/CMakeLists.txt	2019-09-27 21:47:36.803357383 +0200
@@ -7 +7 @@
-set(CMAKE_JAVA_COMPILE_FLAGS -target 1.6 -source 1.6 -Xlint:deprecation -Xlint:unchecked)
+set(CMAKE_JAVA_COMPILE_FLAGS -target 8 -source 8 -Xlint:deprecation -Xlint:unchecked)
