--- a/src/pavucontrol.glade	2015-04-10 13:01:49.000000000 +0100
+++ b/src/pavucontrol.glade	2015-04-10 13:00:23.630554973 +0100
@@ -256,7 +256,6 @@
                             <property name="xalign">0</property>
                             <property name="label" translatable="yes">Device Title</property>
                             <property name="use_markup">True</property>
-                            <property name="ellipsize">end</property>
                           </object>
                           <packing>
                             <property name="expand">False</property>
@@ -1386,7 +1385,6 @@
                             <property name="xalign">0</property>
                             <property name="label" translatable="yes">Stream Title</property>
                             <property name="use_markup">True</property>
-                            <property name="ellipsize">end</property>
                           </object>
                           <packing>
                             <property name="expand">False</property>
