From 8eba47c1ea6058b63f03e3aaef1cc5fb8e8d8826 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Sun, 2 Oct 2016 03:19:19 +0000
Subject: [PATCH 2/4] Handle non-signal frame unwind info lookup in ARM exidx
 unwinder

---
 src/arm/Gstep.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/arm/Gstep.c b/src/arm/Gstep.c
index 2b9e903..70d7aef 100644
--- a/src/arm/Gstep.c
+++ b/src/arm/Gstep.c
@@ -44,13 +44,16 @@ arm_exidx_step (struct cursor *c)
 
   /* mark PC unsaved */
   c->dwarf.loc[UNW_ARM_R15] = DWARF_NULL_LOC;
+  unw_word_t ip = c->dwarf.ip;
+  if (c->dwarf.use_prev_instr)
+    --ip;
 
   /* check dynamic info first --- it overrides everything else */
-  ret = unwi_find_dynamic_proc_info (c->dwarf.as, c->dwarf.ip, &c->dwarf.pi, 1,
+  ret = unwi_find_dynamic_proc_info (c->dwarf.as, ip, &c->dwarf.pi, 1,
                                      c->dwarf.as_arg);
   if (ret == -UNW_ENOINFO)
     {
-      if ((ret = tdep_find_proc_info (&c->dwarf, c->dwarf.ip, 1)) < 0)
+      if ((ret = tdep_find_proc_info (&c->dwarf, ip, 1)) < 0)
         return ret;
     }
 
-- 
2.13.1

